<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-11-21">

<title>Daily-Blog-Sai - 2.1.1_prepare_predictDB_standard_inputs</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#expression-files" id="toc-expression-files" class="nav-link" data-scroll-target="#expression-files">Expression files</a>
  <ul class="collapse">
  <li><a href="#uncorrected-expression" id="toc-uncorrected-expression" class="nav-link" data-scroll-target="#uncorrected-expression">Uncorrected expression</a></li>
  <li><a href="#uncorrected-quantile-norm-expression" id="toc-uncorrected-quantile-norm-expression" class="nav-link" data-scroll-target="#uncorrected-quantile-norm-expression">Uncorrected, quantile norm expression</a></li>
  </ul></li>
  <li><a href="#gene-annotation-file" id="toc-gene-annotation-file" class="nav-link" data-scroll-target="#gene-annotation-file">Gene annotation file</a></li>
  <li><a href="#genotype-file" id="toc-genotype-file" class="nav-link" data-scroll-target="#genotype-file">Genotype file</a></li>
  <li><a href="#snp-annotation-file" id="toc-snp-annotation-file" class="nav-link" data-scroll-target="#snp-annotation-file">SNP annotation file</a></li>
  <li><a href="#check-compatibility-of-files" id="toc-check-compatibility-of-files" class="nav-link" data-scroll-target="#check-compatibility-of-files">Check compatibility of files</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">2.1.1_prepare_predictDB_standard_inputs</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 21, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="context" class="level1">
<h1>Context</h1>
<p>I need to run a standard PrediXcan TWAS analysis on the Aracena et al.&nbsp;as a standard of comparison. To do this, I need to prepare the proper input files for the PrediXcan pipeline. This takes some reformatting as shown below:</p>
<div class="cell" data-execution_count="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os,sys</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="3">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>genotype_file <span class="op">=</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/QTL_mapping/SNP_genotypes_b37.txt"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>genotype_data <span class="op">=</span> pd.read_csv(genotype_file, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>genotype_data.head()</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>genotype_samples <span class="op">=</span> genotype_data.columns</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(genotype_samples)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Index(['AF04', 'AF06', 'AF08', 'AF10', 'AF12', 'AF14', 'AF16', 'AF18', 'AF20',
       'EU22', 'AF24', 'AF26', 'AF28', 'AF30', 'AF34', 'EU36', 'EU38', 'EU03',
       'EU05', 'EU07', 'EU09', 'EU13', 'EU15', 'EU17', 'EU19', 'EU21', 'EU25',
       'EU27', 'EU29', 'EU33', 'EU37', 'EU39', 'EU41', 'EU43', 'EU47'],
      dtype='object')</code></pre>
</div>
</div>
<section id="expression-files" class="level2">
<h2 class="anchored" data-anchor-id="expression-files">Expression files</h2>
<div class="cell" data-execution_count="4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load existing expression file</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>file_tag <span class="op">=</span> <span class="st">"_onta"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>all_expression <span class="op">=</span> pd.read_csv(<span class="ss">f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression</span><span class="sc">{</span>file_tag<span class="sc">}</span><span class="ss">.txt"</span>, sep<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># all_expression = pd.read_csv("/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_expression.txt_NO_LENGTH_NORMALIZATION", sep=" ")</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>samples <span class="op">=</span> all_expression.columns</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>flu_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> samples <span class="cf">if</span> <span class="st">"Flu"</span> <span class="kw">in</span> x]</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>ni_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> samples <span class="cf">if</span> <span class="st">"NI"</span> <span class="kw">in</span> x]</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>flu_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> flu_samples <span class="cf">if</span> x.split(<span class="st">"_"</span>)[<span class="dv">0</span>] <span class="kw">in</span> genotype_samples]</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>ni_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> ni_samples <span class="cf">if</span> x.split(<span class="st">"_"</span>)[<span class="dv">0</span>] <span class="kw">in</span> genotype_samples]</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>flu_expression <span class="op">=</span> all_expression[flu_samples]</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>ni_expression <span class="op">=</span> all_expression[ni_samples]</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>flu_expression <span class="op">=</span> flu_expression.rename(columns<span class="op">=</span>{x:x.replace(<span class="st">"_Flu"</span>,<span class="st">""</span>) <span class="cf">for</span> x <span class="kw">in</span> flu_samples})</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>ni_expression <span class="op">=</span> ni_expression.rename(columns<span class="op">=</span>{x:x.replace(<span class="st">"_NI"</span>,<span class="st">""</span>) <span class="cf">for</span> x <span class="kw">in</span> ni_samples})</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Write expression files with "Name" added</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>flu_expression.to_csv(<span class="ss">f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_flu_expression_for_predictDB</span><span class="sc">{</span>file_tag<span class="sc">}</span><span class="ss">.csv"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>ni_expression.to_csv(<span class="ss">f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_NI_expression_for_predictDB</span><span class="sc">{</span>file_tag<span class="sc">}</span><span class="ss">.csv"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>)</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>flu_expression.to_csv(<span class="ss">f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_flu_expression_for_predictDB</span><span class="sc">{</span>file_tag<span class="sc">}</span><span class="ss">.txt"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>ni_expression.to_csv(<span class="ss">f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_NI_expression_for_predictDB</span><span class="sc">{</span>file_tag<span class="sc">}</span><span class="ss">.txt"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="co"># flu_expression.T.to_csv(f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_flu_expression_for_predictDB{file_tag}_transposed.txt", index=True, index_label="NAME", sep="\t")</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="co"># ni_expression.T.to_csv(f"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_NI_expression_for_predictDB{file_tag}_transposed.txt", index=True, index_label="NAME", sep="\t")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>flu_expression</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="6">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">AF04</th>
<th data-quarto-table-cell-role="th">AF06</th>
<th data-quarto-table-cell-role="th">AF08</th>
<th data-quarto-table-cell-role="th">AF10</th>
<th data-quarto-table-cell-role="th">AF12</th>
<th data-quarto-table-cell-role="th">AF14</th>
<th data-quarto-table-cell-role="th">AF16</th>
<th data-quarto-table-cell-role="th">AF18</th>
<th data-quarto-table-cell-role="th">AF20</th>
<th data-quarto-table-cell-role="th">AF24</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">EU21</th>
<th data-quarto-table-cell-role="th">EU25</th>
<th data-quarto-table-cell-role="th">EU27</th>
<th data-quarto-table-cell-role="th">EU29</th>
<th data-quarto-table-cell-role="th">EU33</th>
<th data-quarto-table-cell-role="th">EU37</th>
<th data-quarto-table-cell-role="th">EU39</th>
<th data-quarto-table-cell-role="th">EU41</th>
<th data-quarto-table-cell-role="th">EU43</th>
<th data-quarto-table-cell-role="th">EU47</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSG00000000419</td>
<td>4.247264</td>
<td>3.984541</td>
<td>3.808667</td>
<td>4.387522</td>
<td>4.280257</td>
<td>3.301860</td>
<td>4.370497</td>
<td>4.329033</td>
<td>4.142495</td>
<td>4.278582</td>
<td>...</td>
<td>4.441305</td>
<td>4.248960</td>
<td>3.825667</td>
<td>4.227968</td>
<td>3.236020</td>
<td>4.296582</td>
<td>4.040271</td>
<td>3.995014</td>
<td>4.428678</td>
<td>4.229811</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSG00000000460</td>
<td>2.084423</td>
<td>2.374117</td>
<td>2.186385</td>
<td>1.552826</td>
<td>1.766373</td>
<td>1.974538</td>
<td>2.709397</td>
<td>2.201944</td>
<td>2.116858</td>
<td>1.837106</td>
<td>...</td>
<td>2.089717</td>
<td>2.050071</td>
<td>1.986421</td>
<td>2.008548</td>
<td>1.284029</td>
<td>1.715819</td>
<td>1.959209</td>
<td>1.967797</td>
<td>2.100634</td>
<td>2.173422</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSG00000000938</td>
<td>5.927820</td>
<td>6.311732</td>
<td>6.212348</td>
<td>6.165741</td>
<td>5.387954</td>
<td>5.285123</td>
<td>5.879858</td>
<td>6.431732</td>
<td>6.056096</td>
<td>5.995297</td>
<td>...</td>
<td>5.931943</td>
<td>6.022909</td>
<td>6.031408</td>
<td>5.808432</td>
<td>6.175322</td>
<td>6.133609</td>
<td>5.839329</td>
<td>5.960366</td>
<td>6.255181</td>
<td>5.403151</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSG00000000971</td>
<td>4.857289</td>
<td>3.717234</td>
<td>4.786917</td>
<td>5.008106</td>
<td>4.887095</td>
<td>3.714513</td>
<td>4.190340</td>
<td>5.011576</td>
<td>4.841874</td>
<td>4.747534</td>
<td>...</td>
<td>5.334937</td>
<td>4.257221</td>
<td>4.152555</td>
<td>3.980055</td>
<td>3.840778</td>
<td>4.808895</td>
<td>4.575839</td>
<td>4.377932</td>
<td>5.166898</td>
<td>3.961062</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSG00000001036</td>
<td>6.455511</td>
<td>6.605715</td>
<td>6.273509</td>
<td>6.295124</td>
<td>6.144346</td>
<td>5.846182</td>
<td>6.566031</td>
<td>6.707117</td>
<td>6.429354</td>
<td>6.130178</td>
<td>...</td>
<td>6.311400</td>
<td>6.386900</td>
<td>5.959519</td>
<td>6.495524</td>
<td>5.730415</td>
<td>6.500412</td>
<td>6.144727</td>
<td>6.394556</td>
<td>6.655935</td>
<td>6.424312</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSG00000273375</td>
<td>1.189708</td>
<td>0.403898</td>
<td>0.789746</td>
<td>0.788083</td>
<td>0.926125</td>
<td>0.095197</td>
<td>0.810983</td>
<td>0.374500</td>
<td>0.717079</td>
<td>1.091195</td>
<td>...</td>
<td>-0.024988</td>
<td>1.203391</td>
<td>0.859122</td>
<td>0.918967</td>
<td>0.647828</td>
<td>0.645034</td>
<td>0.837831</td>
<td>0.641001</td>
<td>0.790858</td>
<td>0.212928</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSG00000273437</td>
<td>-0.135665</td>
<td>0.085988</td>
<td>0.357227</td>
<td>0.008651</td>
<td>-0.017337</td>
<td>-0.411125</td>
<td>-0.285801</td>
<td>-0.861348</td>
<td>-0.370453</td>
<td>0.242389</td>
<td>...</td>
<td>-0.563449</td>
<td>0.415358</td>
<td>0.300610</td>
<td>0.054347</td>
<td>0.630219</td>
<td>0.192951</td>
<td>-0.379374</td>
<td>0.015349</td>
<td>-0.244158</td>
<td>-0.086177</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSG00000273448</td>
<td>-0.292841</td>
<td>0.136501</td>
<td>0.425021</td>
<td>0.087109</td>
<td>0.356929</td>
<td>0.672980</td>
<td>0.354483</td>
<td>-0.570697</td>
<td>0.255480</td>
<td>-0.000678</td>
<td>...</td>
<td>-0.244652</td>
<td>0.282939</td>
<td>0.438417</td>
<td>0.653817</td>
<td>0.028750</td>
<td>0.000916</td>
<td>0.198524</td>
<td>0.434090</td>
<td>-0.347806</td>
<td>0.836002</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSG00000273451</td>
<td>-0.252675</td>
<td>-1.628698</td>
<td>-0.662008</td>
<td>-1.975064</td>
<td>-1.373232</td>
<td>-0.013630</td>
<td>-2.330006</td>
<td>-1.792680</td>
<td>-0.097467</td>
<td>-2.375824</td>
<td>...</td>
<td>-1.679244</td>
<td>-0.421800</td>
<td>0.548356</td>
<td>-1.164319</td>
<td>-0.552637</td>
<td>-0.445711</td>
<td>-2.078597</td>
<td>0.072709</td>
<td>-1.245223</td>
<td>-0.498993</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSG00000273486</td>
<td>0.832979</td>
<td>0.807239</td>
<td>1.212064</td>
<td>0.146193</td>
<td>0.678311</td>
<td>1.627964</td>
<td>1.017530</td>
<td>0.161301</td>
<td>1.368186</td>
<td>0.436491</td>
<td>...</td>
<td>0.175049</td>
<td>1.218379</td>
<td>1.033344</td>
<td>1.209306</td>
<td>1.180173</td>
<td>0.762603</td>
<td>1.391422</td>
<td>0.762733</td>
<td>-0.204011</td>
<td>1.239956</td>
</tr>
</tbody>
</table>

<p>14120 rows × 32 columns</p>
</div>
</div>
</div>
<section id="uncorrected-expression" class="level3">
<h3 class="anchored" data-anchor-id="uncorrected-expression">Uncorrected expression</h3>
<div class="cell" data-execution_count="4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>all_expression <span class="op">=</span> pd.read_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_expression_NO_CORRECTION.txt"</span>, sep<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>samples <span class="op">=</span> all_expression.columns</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>flu_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> samples <span class="cf">if</span> <span class="st">"Flu"</span> <span class="kw">in</span> x]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>ni_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> samples <span class="cf">if</span> <span class="st">"NI"</span> <span class="kw">in</span> x]</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>flu_expression <span class="op">=</span> all_expression[flu_samples]</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>ni_expression <span class="op">=</span> all_expression[ni_samples]</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>flu_expression <span class="op">=</span> flu_expression.rename(columns<span class="op">=</span>{x:x.replace(<span class="st">"_Flu"</span>,<span class="st">""</span>) <span class="cf">for</span> x <span class="kw">in</span> flu_samples})</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>ni_expression <span class="op">=</span> ni_expression.rename(columns<span class="op">=</span>{x:x.replace(<span class="st">"_NI"</span>,<span class="st">""</span>) <span class="cf">for</span> x <span class="kw">in</span> ni_samples})</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Write expression files with "Name" added</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>flu_expression.transpose().to_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_flu_expression_for_predictDB_NO_CORRECTION.csv"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>)</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>ni_expression.transpose().to_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_NI_expression_for_predictDB_NO_CORRECTION.csv"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="uncorrected-quantile-norm-expression" class="level3">
<h3 class="anchored" data-anchor-id="uncorrected-quantile-norm-expression">Uncorrected, quantile norm expression</h3>
<div class="cell" data-execution_count="5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>all_expression <span class="op">=</span> pd.read_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_expression_NO_CORRECTION_QUANTILE.txt"</span>, sep<span class="op">=</span><span class="st">" "</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>samples <span class="op">=</span> all_expression.columns</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>flu_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> samples <span class="cf">if</span> <span class="st">"Flu"</span> <span class="kw">in</span> x]</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>ni_samples <span class="op">=</span> [x <span class="cf">for</span> x <span class="kw">in</span> samples <span class="cf">if</span> <span class="st">"NI"</span> <span class="kw">in</span> x]</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>flu_expression <span class="op">=</span> all_expression[flu_samples]</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>ni_expression <span class="op">=</span> all_expression[ni_samples]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>flu_expression <span class="op">=</span> flu_expression.rename(columns<span class="op">=</span>{x:x.replace(<span class="st">"_Flu"</span>,<span class="st">""</span>) <span class="cf">for</span> x <span class="kw">in</span> flu_samples})</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>ni_expression <span class="op">=</span> ni_expression.rename(columns<span class="op">=</span>{x:x.replace(<span class="st">"_NI"</span>,<span class="st">""</span>) <span class="cf">for</span> x <span class="kw">in</span> ni_samples})</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Write expression files with "Name" added</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>flu_expression.to_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_flu_expression_for_predictDB_NO_CORRECTION_QUANTILE.csv"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>ni_expression.to_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_NI_expression_for_predictDB_NO_CORRECTION_QUANTILE.csv"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"NAME"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="gene-annotation-file" class="level2">
<h2 class="anchored" data-anchor-id="gene-annotation-file">Gene annotation file</h2>
<div class="cell" data-execution_count="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>gtf_file <span class="op">=</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/ref/gencode.v44lift37.annotation.gtf"</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>parse_script <span class="op">=</span> <span class="st">"/beagle3/haky/users/saideep/github_repos/PredictDB-Tutorial/code/parse_gtf.py"</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>parse_com <span class="op">=</span> [</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"python"</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    parse_script,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    gtf_file,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    gtf_file<span class="op">+</span><span class="st">".parsed.txt"</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>os.system(<span class="st">" "</span>.join(parse_com))</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Load parsed gtf file</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>parsed_gtf <span class="op">=</span> pd.read_csv(gtf_file<span class="op">+</span><span class="st">".parsed.txt"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>parsed_gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="6">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chr</th>
<th data-quarto-table-cell-role="th">gene_id</th>
<th data-quarto-table-cell-role="th">gene_name</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">gene_type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>1</td>
<td>ENSG00000290825.1_1</td>
<td>DDX11L2</td>
<td>11869</td>
<td>14409</td>
<td>lncRNA</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>1</td>
<td>ENSG00000223972.6_6</td>
<td>DDX11L1</td>
<td>12010</td>
<td>13670</td>
<td>transcribed_unprocessed_pseudogene</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>1</td>
<td>ENSG00000227232.5_5</td>
<td>WASH7P</td>
<td>14404</td>
<td>29570</td>
<td>unprocessed_pseudogene</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>1</td>
<td>ENSG00000243485.5_11</td>
<td>MIR1302-2HG</td>
<td>29554</td>
<td>31109</td>
<td>lncRNA</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>1</td>
<td>ENSG00000237613.2_6</td>
<td>FAM138A</td>
<td>34554</td>
<td>36081</td>
<td>lncRNA</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">64481</td>
<td>M</td>
<td>ENSG00000198695.2_4</td>
<td>MT-ND6</td>
<td>14149</td>
<td>14673</td>
<td>protein_coding</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">64482</td>
<td>M</td>
<td>ENSG00000210194.1</td>
<td>MT-TE</td>
<td>14674</td>
<td>14742</td>
<td>Mt_tRNA</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">64483</td>
<td>M</td>
<td>ENSG00000198727.2_2</td>
<td>MT-CYB</td>
<td>14747</td>
<td>15887</td>
<td>protein_coding</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">64484</td>
<td>M</td>
<td>ENSG00000210195.2</td>
<td>MT-TT</td>
<td>15888</td>
<td>15953</td>
<td>Mt_tRNA</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">64485</td>
<td>M</td>
<td>ENSG00000210196.2</td>
<td>MT-TP</td>
<td>15956</td>
<td>16023</td>
<td>Mt_tRNA</td>
</tr>
</tbody>
</table>

<p>64486 rows × 6 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>gene_ids_trim <span class="op">=</span> [x.split(<span class="st">"."</span>)[<span class="dv">0</span>] <span class="cf">for</span> x <span class="kw">in</span> parsed_gtf[<span class="st">"gene_id"</span>].values]</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>parsed_gtf[<span class="st">"gene_id"</span>] <span class="op">=</span> gene_ids_trim</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># parsed_gtf["chr"] = "chr" + parsed_gtf["chr"].astype(str)</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>parsed_gtf.drop_duplicates(subset<span class="op">=</span>[<span class="st">"gene_id"</span>], inplace<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>parsed_gtf.to_csv(gtf_file<span class="op">+</span><span class="st">".parsed.txt"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="genotype-file" class="level2">
<h2 class="anchored" data-anchor-id="genotype-file">Genotype file</h2>
<div class="cell" data-execution_count="8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>genotypes_source_file <span class="op">=</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/QTL_mapping/SNP_genotypes.txt"</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>unformated_genotypes <span class="op">=</span> pd.read_csv(genotypes_source_file, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="9">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>unformated_genotypes.index <span class="op">=</span> <span class="st">"chr"</span><span class="op">+</span>unformated_genotypes.index<span class="op">+</span><span class="st">"_b37"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>unformated_genotypes.to_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/QTL_mapping/SNP_genotypes_b37.txt"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, index<span class="op">=</span><span class="va">True</span>, index_label<span class="op">=</span><span class="st">"varID"</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>unformated_genotypes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="9">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">AF04</th>
<th data-quarto-table-cell-role="th">AF06</th>
<th data-quarto-table-cell-role="th">AF08</th>
<th data-quarto-table-cell-role="th">AF10</th>
<th data-quarto-table-cell-role="th">AF12</th>
<th data-quarto-table-cell-role="th">AF14</th>
<th data-quarto-table-cell-role="th">AF16</th>
<th data-quarto-table-cell-role="th">AF18</th>
<th data-quarto-table-cell-role="th">AF20</th>
<th data-quarto-table-cell-role="th">EU22</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">EU21</th>
<th data-quarto-table-cell-role="th">EU25</th>
<th data-quarto-table-cell-role="th">EU27</th>
<th data-quarto-table-cell-role="th">EU29</th>
<th data-quarto-table-cell-role="th">EU33</th>
<th data-quarto-table-cell-role="th">EU37</th>
<th data-quarto-table-cell-role="th">EU39</th>
<th data-quarto-table-cell-role="th">EU41</th>
<th data-quarto-table-cell-role="th">EU43</th>
<th data-quarto-table-cell-role="th">EU47</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">chr1_10492_C_T_b37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">chr1_10583_G_A_b37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>...</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">chr1_12783_A_G_b37</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>2</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">chr1_12807_C_T_b37</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">chr1_12839_G_C_b37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">chr22_51241882_C_T_b37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">chr22_51241883_C_A_b37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">chr22_51242271_C_G_b37</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">chr22_51242557_A_T_b37</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>...</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">chr22_51244332_A_C_b37</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>...</td>
<td>1</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>-9</td>
<td>0</td>
<td>0</td>
<td>1</td>
</tr>
</tbody>
</table>

<p>7383243 rows × 35 columns</p>
</div>
</div>
</div>
</section>
<section id="snp-annotation-file" class="level2">
<h2 class="anchored" data-anchor-id="snp-annotation-file">SNP annotation file</h2>
<div class="cell" data-execution_count="10">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>snp_annotation_source_file <span class="op">=</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/ref/rsIDs.for.matrixeqtl.txt"</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>unformatted_snp_annotation <span class="op">=</span> pd.read_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/ref/rsIDs.for.matrixeqtl.txt"</span>, sep<span class="op">=</span><span class="st">" "</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="11">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>unformatted_snp_annotation[<span class="st">"varID"</span>] <span class="op">=</span> unformatted_snp_annotation[<span class="st">"snps"</span>]<span class="op">+</span><span class="st">"_b37"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>unformatted_snp_annotation</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="11">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chr</th>
<th data-quarto-table-cell-role="th">pos</th>
<th data-quarto-table-cell-role="th">rsID</th>
<th data-quarto-table-cell-role="th">A1</th>
<th data-quarto-table-cell-role="th">A2</th>
<th data-quarto-table-cell-role="th">snps</th>
<th data-quarto-table-cell-role="th">varID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">1</td>
<td>1</td>
<td>10492</td>
<td>rs55998931</td>
<td>C</td>
<td>T</td>
<td>1_10492_C_T</td>
<td>1_10492_C_T_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2</td>
<td>1</td>
<td>10583</td>
<td>rs58108140</td>
<td>G</td>
<td>A</td>
<td>1_10583_G_A</td>
<td>1_10583_G_A_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">3</td>
<td>1</td>
<td>12783</td>
<td>rs62635284</td>
<td>A</td>
<td>G</td>
<td>1_12783_A_G</td>
<td>1_12783_A_G_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">4</td>
<td>1</td>
<td>12807</td>
<td>rs62635285</td>
<td>C</td>
<td>T</td>
<td>1_12807_C_T</td>
<td>1_12807_C_T_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5</td>
<td>1</td>
<td>12839</td>
<td>rs747124606</td>
<td>G</td>
<td>C</td>
<td>1_12839_G_C</td>
<td>1_12839_G_C_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">7383239</td>
<td>22</td>
<td>51241882</td>
<td>.</td>
<td>C</td>
<td>T</td>
<td>22_51241882_C_T</td>
<td>22_51241882_C_T_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7383240</td>
<td>22</td>
<td>51241883</td>
<td>.</td>
<td>C</td>
<td>A</td>
<td>22_51241883_C_A</td>
<td>22_51241883_C_A_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">7383241</td>
<td>22</td>
<td>51242271</td>
<td>rs782727945</td>
<td>C</td>
<td>G</td>
<td>22_51242271_C_G</td>
<td>22_51242271_C_G_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7383242</td>
<td>22</td>
<td>51242557</td>
<td>rs3888401</td>
<td>A</td>
<td>T</td>
<td>22_51242557_A_T</td>
<td>22_51242557_A_T_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">7383243</td>
<td>22</td>
<td>51244332</td>
<td>rs200908937</td>
<td>A</td>
<td>C</td>
<td>22_51244332_A_C</td>
<td>22_51244332_A_C_b37</td>
</tr>
</tbody>
</table>

<p>7383243 rows × 7 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="12">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>formatted_snp_annotation_dict <span class="op">=</span> {</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"chr"</span>:unformatted_snp_annotation[<span class="st">"chr"</span>].astype(<span class="bu">str</span>),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"pos"</span>:unformatted_snp_annotation[<span class="st">"pos"</span>],</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"varID"</span>:<span class="st">"chr"</span><span class="op">+</span>unformatted_snp_annotation[<span class="st">"varID"</span>],</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ref_vcf"</span>:unformatted_snp_annotation[<span class="st">"A1"</span>],</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"alt_vcf"</span>:unformatted_snp_annotation[<span class="st">"A2"</span>],</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"maf"</span>:<span class="fl">0.01</span>,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rsid"</span>:<span class="st">"chr"</span><span class="op">+</span>unformatted_snp_annotation[<span class="st">"varID"</span>]</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>formatted_snp_annotation <span class="op">=</span> pd.DataFrame(formatted_snp_annotation_dict)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>formatted_snp_annotation</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="12">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chr</th>
<th data-quarto-table-cell-role="th">pos</th>
<th data-quarto-table-cell-role="th">varID</th>
<th data-quarto-table-cell-role="th">ref_vcf</th>
<th data-quarto-table-cell-role="th">alt_vcf</th>
<th data-quarto-table-cell-role="th">maf</th>
<th data-quarto-table-cell-role="th">rsid</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">1</td>
<td>1</td>
<td>10492</td>
<td>chr1_10492_C_T_b37</td>
<td>C</td>
<td>T</td>
<td>0.01</td>
<td>chr1_10492_C_T_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">2</td>
<td>1</td>
<td>10583</td>
<td>chr1_10583_G_A_b37</td>
<td>G</td>
<td>A</td>
<td>0.01</td>
<td>chr1_10583_G_A_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">3</td>
<td>1</td>
<td>12783</td>
<td>chr1_12783_A_G_b37</td>
<td>A</td>
<td>G</td>
<td>0.01</td>
<td>chr1_12783_A_G_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">4</td>
<td>1</td>
<td>12807</td>
<td>chr1_12807_C_T_b37</td>
<td>C</td>
<td>T</td>
<td>0.01</td>
<td>chr1_12807_C_T_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">5</td>
<td>1</td>
<td>12839</td>
<td>chr1_12839_G_C_b37</td>
<td>G</td>
<td>C</td>
<td>0.01</td>
<td>chr1_12839_G_C_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">7383239</td>
<td>22</td>
<td>51241882</td>
<td>chr22_51241882_C_T_b37</td>
<td>C</td>
<td>T</td>
<td>0.01</td>
<td>chr22_51241882_C_T_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7383240</td>
<td>22</td>
<td>51241883</td>
<td>chr22_51241883_C_A_b37</td>
<td>C</td>
<td>A</td>
<td>0.01</td>
<td>chr22_51241883_C_A_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">7383241</td>
<td>22</td>
<td>51242271</td>
<td>chr22_51242271_C_G_b37</td>
<td>C</td>
<td>G</td>
<td>0.01</td>
<td>chr22_51242271_C_G_b37</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7383242</td>
<td>22</td>
<td>51242557</td>
<td>chr22_51242557_A_T_b37</td>
<td>A</td>
<td>T</td>
<td>0.01</td>
<td>chr22_51242557_A_T_b37</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">7383243</td>
<td>22</td>
<td>51244332</td>
<td>chr22_51244332_A_C_b37</td>
<td>A</td>
<td>C</td>
<td>0.01</td>
<td>chr22_51244332_A_C_b37</td>
</tr>
</tbody>
</table>

<p>7383243 rows × 7 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="13">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>formatted_snp_annotation.to_csv(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/ref/rsIDs.for.predixcan_b37.txt"</span>, sep<span class="op">=</span><span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="check-compatibility-of-files" class="level2">
<h2 class="anchored" data-anchor-id="check-compatibility-of-files">Check compatibility of files</h2>
<div class="cell" data-execution_count="12">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>parsed_gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="12">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">chr</th>
<th data-quarto-table-cell-role="th">gene_id</th>
<th data-quarto-table-cell-role="th">gene_name</th>
<th data-quarto-table-cell-role="th">start</th>
<th data-quarto-table-cell-role="th">end</th>
<th data-quarto-table-cell-role="th">gene_type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>chr1</td>
<td>ENSG00000290825</td>
<td>DDX11L2</td>
<td>11869</td>
<td>14409</td>
<td>lncRNA</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>chr1</td>
<td>ENSG00000223972</td>
<td>DDX11L1</td>
<td>12010</td>
<td>13670</td>
<td>transcribed_unprocessed_pseudogene</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>chr1</td>
<td>ENSG00000227232</td>
<td>WASH7P</td>
<td>14404</td>
<td>29570</td>
<td>unprocessed_pseudogene</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>chr1</td>
<td>ENSG00000243485</td>
<td>MIR1302-2HG</td>
<td>29554</td>
<td>31109</td>
<td>lncRNA</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>chr1</td>
<td>ENSG00000237613</td>
<td>FAM138A</td>
<td>34554</td>
<td>36081</td>
<td>lncRNA</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">64481</td>
<td>chrM</td>
<td>ENSG00000198695</td>
<td>MT-ND6</td>
<td>14149</td>
<td>14673</td>
<td>protein_coding</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">64482</td>
<td>chrM</td>
<td>ENSG00000210194</td>
<td>MT-TE</td>
<td>14674</td>
<td>14742</td>
<td>Mt_tRNA</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">64483</td>
<td>chrM</td>
<td>ENSG00000198727</td>
<td>MT-CYB</td>
<td>14747</td>
<td>15887</td>
<td>protein_coding</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">64484</td>
<td>chrM</td>
<td>ENSG00000210195</td>
<td>MT-TT</td>
<td>15888</td>
<td>15953</td>
<td>Mt_tRNA</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">64485</td>
<td>chrM</td>
<td>ENSG00000210196</td>
<td>MT-TP</td>
<td>15956</td>
<td>16023</td>
<td>Mt_tRNA</td>
</tr>
</tbody>
</table>

<p>64476 rows × 6 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="13">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>expression_inds <span class="op">=</span> flu_expression.columns</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>expression_genes <span class="op">=</span> flu_expression.index</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>gene_annotation_genes <span class="op">=</span> <span class="bu">list</span>(parsed_gtf[<span class="st">"gene_id"</span>].values)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>genotype_snpids <span class="op">=</span> unformated_genotypes.index</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>genotype_inds <span class="op">=</span> unformated_genotypes.columns</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>snp_annotation_snpids <span class="op">=</span> formatted_snp_annotation[<span class="st">"varID"</span>].values</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Look for duplicates</p>
<div class="cell" data-execution_count="14">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Num individuals, unique individuals in expression file"</span>, <span class="bu">len</span>(expression_inds), <span class="bu">len</span>(<span class="bu">set</span>(expression_inds)))</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Num genes, unique genes in expression file"</span>, <span class="bu">len</span>(expression_genes), <span class="bu">len</span>(<span class="bu">set</span>(expression_genes)))</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Num genes, unique genes in gene annotation file"</span>, <span class="bu">len</span>(gene_annotation_genes), <span class="bu">len</span>(<span class="bu">set</span>(gene_annotation_genes)))</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Num SNPs, unique SNPs in genotype file"</span>, <span class="bu">len</span>(genotype_snpids), <span class="bu">len</span>(<span class="bu">set</span>(genotype_snpids)))</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Num individuals, unique individuals in genotype file"</span>, <span class="bu">len</span>(genotype_inds), <span class="bu">len</span>(<span class="bu">set</span>(genotype_inds)))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Num SNPs, unique SNPs in SNP annotation file"</span>, <span class="bu">len</span>(snp_annotation_snpids), <span class="bu">len</span>(<span class="bu">set</span>(snp_annotation_snpids)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Num individuals, unique individuals in expression file 35 35
Num genes, unique genes in expression file 13716 13716
Num genes, unique genes in gene annotation file 64476 64476
Num SNPs, unique SNPs in genotype file 7383243 7383243
Num individuals, unique individuals in genotype file 35 35
Num SNPs, unique SNPs in SNP annotation file 7383243 7383243</code></pre>
</div>
</div>
<p>Check overlap between different files</p>
<div class="cell" data-execution_count="15">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Genes overlap"</span>, <span class="bu">len</span>(<span class="bu">set</span>(expression_genes).intersection(<span class="bu">set</span>(gene_annotation_genes))))</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"SNPs overlap"</span>, <span class="bu">len</span>(<span class="bu">set</span>(genotype_snpids).intersection(<span class="bu">set</span>(snp_annotation_snpids))))</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Individuals overlap"</span>, <span class="bu">len</span>(<span class="bu">set</span>(expression_inds).intersection(<span class="bu">set</span>(genotype_inds))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Genes overlap 13605
SNPs overlap 7383243
Individuals overlap 35</code></pre>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>