<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-06-03">

<title>Daily-Blog-Sai - calculating_genetic_distance_from_ref</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#computing-the-genetic-distance-bw-reference-and-genome-of-track" id="toc-computing-the-genetic-distance-bw-reference-and-genome-of-track" class="nav-link" data-scroll-target="#computing-the-genetic-distance-bw-reference-and-genome-of-track">Computing the genetic distance b/w reference and genome of track</a>
  <ul class="collapse">
  <li><a href="#load-in-the-vcf-and-gene-regions" id="toc-load-in-the-vcf-and-gene-regions" class="nav-link" data-scroll-target="#load-in-the-vcf-and-gene-regions">Load in the VCF and gene regions</a></li>
  <li><a href="#check-if-the-vcf-file-reference-is-the-reference-genome" id="toc-check-if-the-vcf-file-reference-is-the-reference-genome" class="nav-link" data-scroll-target="#check-if-the-vcf-file-reference-is-the-reference-genome">Check if the VCF file reference is the reference genome</a></li>
  <li><a href="#compute-genomic-distance-from-reference" id="toc-compute-genomic-distance-from-reference" class="nav-link" data-scroll-target="#compute-genomic-distance-from-reference">Compute genomic distance from reference</a></li>
  <li><a href="#recompute-gene-correlations" id="toc-recompute-gene-correlations" class="nav-link" data-scroll-target="#recompute-gene-correlations">Recompute gene correlations</a></li>
  <li><a href="#incorporate-population-level-information" id="toc-incorporate-population-level-information" class="nav-link" data-scroll-target="#incorporate-population-level-information">Incorporate population level information</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">calculating_genetic_distance_from_ref</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 3, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"calculating_genetic_distance_from_ref"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-06-03'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="context" class="level1">
<h1>Context</h1>
<p>Previously, I looked for an association between the local ancestry of genes in the refernce genome and their direction of personalized Enformer expression prediction correlation. This found no association. Another useful metric is to look at the genetic distance of the ENCODE tracks used in training against the reference background. This gives a sort of measure of discordance between the training input and output.</p>
<section id="computing-the-genetic-distance-bw-reference-and-genome-of-track" class="level2">
<h2 class="anchored" data-anchor-id="computing-the-genetic-distance-bw-reference-and-genome-of-track">Computing the genetic distance b/w reference and genome of track</h2>
<p>For this case, I will use the genome of the track pertaining to LCL CAGE expression. This track comes from a common cell line: GM12878 . Roughly, this is a cell line derived from a European Ancestry female. I have downloaded a VCF file from sequencing of the cell line.</p>
<p>I will start by computing the average alternative allele dosage of variants within each gene window. This will give some measure of the sequence deviation from the reference genome.</p>
<section id="load-in-the-vcf-and-gene-regions" class="level3">
<h3 class="anchored" data-anchor-id="load-in-the-vcf-and-gene-regions">Load in the VCF and gene regions</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VariantAnnotation)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>vcf_file <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai/2023-06-03-calculating_genetic_distance_from_ref/HG001_GRCh38_1_22_v4.2.1_benchmark.vcf.gz"</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># vcf &lt;- readVcf(file = vcf_file, genome = "GRCh38")</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>)))</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_metadata) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gene_metadata<span class="sc">$</span>...<span class="dv">1</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="check-if-the-vcf-file-reference-is-the-reference-genome" class="level3">
<h3 class="anchored" data-anchor-id="check-if-the-vcf-file-reference-is-the-reference-genome">Check if the VCF file reference is the reference genome</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>example_region <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="st">"chr22"</span>, <span class="at">ranges=</span><span class="fu">IRanges</span>(</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">start=</span><span class="fu">c</span>(<span class="dv">50301422</span>, <span class="dv">50989541</span>),</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">end=</span><span class="fu">c</span>(<span class="dv">50312106</span>, <span class="dv">51001328</span>),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">names=</span><span class="fu">c</span>(<span class="st">"gene_79087"</span>, <span class="st">"gene_644186"</span>)))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>vcf_example <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="at">file =</span> vcf_file, <span class="at">genome =</span> <span class="st">"GRCh38"</span>, <span class="at">param=</span>example_region)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">geno</span>(vcf_example)<span class="sc">$</span>GT</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rowRanges</span>(vcf_example)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Through manual inspection of this region, it is clear that the reference matches the reference genome, rather than being linked to relative population frequency.</p>
</section>
<section id="compute-genomic-distance-from-reference" class="level3">
<h3 class="anchored" data-anchor-id="compute-genomic-distance-from-reference">Compute genomic distance from reference</h3>
<p>We can write and test a function to compute this genetic distance metric</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>gene_row <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>example_gene <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span>gene_metadata[gene_row,<span class="dv">1</span>], <span class="at">ranges=</span><span class="fu">IRanges</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">start=</span><span class="fu">as.numeric</span>(gene_metadata[gene_row,<span class="dv">2</span>]),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">end=</span><span class="fu">as.numeric</span>(gene_metadata[gene_row,<span class="dv">3</span>]),</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">names=</span>gene_metadata[gene_row,<span class="dv">4</span>]))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>vcf_example_gene <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="at">file =</span> vcf_file, <span class="at">genome =</span> <span class="st">"GRCh38"</span>, <span class="at">param=</span>example_gene)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># rowRanges(vcf_example)</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>genos <span class="ot">=</span> <span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>total_a_count <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gt <span class="cf">in</span> genos) {</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  total_a_count <span class="ot">=</span> total_a_count <span class="sc">+</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(gt,<span class="st">"/"</span>)[[<span class="dv">1</span>]]))</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>average_a_count <span class="ot">=</span> total_a_count<span class="sc">/</span><span class="fu">width</span>(example_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>calc_alleles_single_gene <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_row, <span class="at">gene_metadata=</span>gene_metadata, <span class="at">vcf_file=</span>vcf_file) {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (gene_row[<span class="dv">1</span>] <span class="sc">==</span> <span class="st">"chrM"</span>) {</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span> (<span class="cn">NA</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  example_gene <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span>gene_row[<span class="dv">1</span>], <span class="at">ranges=</span><span class="fu">IRanges</span>(</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">start=</span><span class="fu">as.numeric</span>(gene_row[<span class="dv">2</span>]),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">end=</span><span class="fu">as.numeric</span>(gene_row[<span class="dv">3</span>]),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">names=</span>gene_row[<span class="dv">4</span>]))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  vcf_example_gene <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="at">file =</span> vcf_file, <span class="at">genome =</span> <span class="st">"GRCh38"</span>, <span class="at">param=</span>example_gene)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  genos <span class="ot">=</span> <span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  total_a_count <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  num_het <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  num_homo <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (gt <span class="cf">in</span> genos) {</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    cur_a_count <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(gt,<span class="st">"/"</span>)[[<span class="dv">1</span>]]))</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (cur_a_count<span class="sc">==</span><span class="dv">2</span>) {</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>      num_homo <span class="ot">=</span> num_homo <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>      num_het <span class="ot">=</span> num_het <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>    total_a_count <span class="ot">=</span> total_a_count <span class="sc">+</span> cur_a_count</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">c</span>(total_a_count, <span class="fu">length</span>(genos),num_het,num_homo))</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now let’s run it across all genes</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gene_metadata_trimmed <span class="ot">&lt;-</span> gene_metadata[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>total_alt_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(gene_metadata,<span class="dv">1</span>, calc_alleles_single_gene, <span class="at">gene_metadata=</span>gene_metadata, <span class="at">vcf_file=</span>vcf_file)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>ref_var_counts <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>,<span class="at">nrow=</span><span class="fu">length</span>(total_alt_counts),<span class="at">ncol =</span> <span class="dv">4</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(total_alt_counts)) {</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(total_alt_counts[[x]])<span class="sc">==</span><span class="dv">1</span>) {</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>    ref_var_counts[x,] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="cn">NA</span>,<span class="cn">NA</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    ref_var_counts[x,] <span class="ot">&lt;-</span> total_alt_counts[[x]]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>ref_var_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ref_var_counts)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ref_var_counts) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"total_alt_count_LCL"</span>,<span class="st">"num_variants_LCL"</span>,<span class="st">"num_het_LCL"</span>,<span class="st">"num_homo_LCL"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This takes a while to run, so let’s save it just in case:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(total_alt_counts, <span class="at">file=</span><span class="st">"altcounts.rda"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Reload alt allele counts here</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load("altcounts.rda")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can now plot the allelic dosage as a proportion of gene length</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>total_alt_count_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>total_alt_count_LCL</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>num_variants_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>num_variants_LCL</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>num_het_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>num_het_LCL</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>num_homo_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>num_homo_LCL</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>alt_counts_len_normalized_LCL <span class="ot">&lt;-</span> gene_metadata<span class="sc">$</span>total_alt_count_LCL<span class="sc">/</span>gene_metadata<span class="sc">$</span>length</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_metadata) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"count"</span>,<span class="fu">aes</span>(<span class="at">x=</span>best_ancestry))</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_metadata) <span class="sc">+</span> <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>best_ancestry,<span class="at">y=</span>alt_counts_len_normalized_LCL)) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.025</span>))</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_metadata) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>best_ancestry,<span class="at">y=</span>alt_counts_len_normalized_LCL)) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.025</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gene_metadata, <span class="st">"gene_metadata_w_allele_counts.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The figures above essentially show that the overall alternate allelic dosage from the cell line (normalized for gene length) is enriched among genes with African reference genome ancestry, which supports our assumptions.</p>
</section>
<section id="recompute-gene-correlations" class="level3">
<h3 class="anchored" data-anchor-id="recompute-gene-correlations">Recompute gene correlations</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_canonical_haplo1_filtered.csv"</span>)))</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(enformer_haplo1) &lt;- enformer_haplo1$X</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_haplo1$X &lt;- NULL</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>enformer_haplo2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_canonical_haplo2_filtered.csv"</span>)))</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo2) <span class="ot">&lt;-</span> enformer_haplo2<span class="sc">$</span>X</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>enformer_haplo2<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> (enformer_haplo1 <span class="sc">+</span> enformer_haplo2)</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>AFR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,AFR_ind)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>EUR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,EUR_ind)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, ge_ind]</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>geuvadis_AFR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, AFR_ge_ind]</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>geuvadis_EUR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, EUR_ge_ind]</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[ge_genes, ge_ind]</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>enformer_AFR_compare <span class="ot">&lt;-</span> enformer[ge_genes, AFR_ge_ind]</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>enformer_EUR_compare <span class="ot">&lt;-</span> enformer[ge_genes, EUR_ge_ind]</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[ge_genes,]</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals: "</span>, <span class="fu">length</span>(ge_ind)))</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals AFR: "</span>, <span class="fu">length</span>(AFR_ge_ind)))</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals EUR: "</span>, <span class="fu">length</span>(EUR_ge_ind)))</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>ge_genes_plus <span class="ot">&lt;-</span> ge_genes</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes: "</span>, <span class="fu">length</span>(gene_meta_compare<span class="sc">$</span>best_ancestry)))</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes,AFR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"AFR"</span>,]<span class="sc">$</span>best_ancestry)))</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes,EUR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"EUR"</span>,]<span class="sc">$</span>best_ancestry)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>all_geuvadis_ge_corrs <span class="ot">&lt;-</span> corrs_df[<span class="fu">rownames</span>(gene_metadata),]<span class="sc">$</span>ge</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>corrs_df_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>corrs_df_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>corrs_df_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_AFR_compare, enformer_AFR_compare)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>AFR_geuvadis_ge_corrs <span class="ot">&lt;-</span> corrs_df[<span class="fu">rownames</span>(gene_metadata),]<span class="sc">$</span>ge</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family =</span> <span class="st">"binomial"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_EUR_compare, enformer_EUR_compare)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>EUR_geuvadis_ge_corrs <span class="ot">&lt;-</span> corrs_df[<span class="fu">rownames</span>(gene_metadata),]<span class="sc">$</span>ge</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(data.frame(table(corrs_df_plus_AFR_EUR$ref_gene_ancestry, corrs_df_plus_AFR_EUR$corr_direction)), aes(x=Var1,y=Freq,fill=Var2, group=Var1)) + geom_bar(stat="identity", position="dodge")</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gene_metadata, <span class="st">"gene_metadata_w_allele_counts.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="incorporate-population-level-information" class="level3">
<h3 class="anchored" data-anchor-id="incorporate-population-level-information">Incorporate population level information</h3>
<p>A potential issue with the above metric is that we haven’t really controlled for the fact that some genes also tend to show more variation in general, so would have stronger alt allele counts by default. If this is roughly even between AFR and EUR genes then we are ok, but it may not be.</p>


<!-- -->

</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb17" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "calculating_genetic_distance_from_ref"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-06-03"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Set up box storage directory</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"calculating_genetic_distance_from_ref"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-06-03'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>Previously, I looked for an association between the local ancestry of genes in the refernce genome and their direction of personalized Enformer expression prediction correlation. This found no association. Another useful metric is to look at the genetic distance of the ENCODE tracks used in training against the reference background. This gives a sort of measure of discordance between the training input and output.</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="fu">## Computing the genetic distance b/w reference and genome of track</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>For this case, I will use the genome of the track pertaining to LCL CAGE expression. This track comes from a common cell line: GM12878 . Roughly, this is a cell line derived from a European Ancestry female. I have downloaded a VCF file from sequencing of the cell line.</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>I will start by computing the average alternative allele dosage of variants within each gene window. This will give some measure of the sequence deviation from the reference genome.</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a><span class="fu">### Load in the VCF and gene regions</span></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VariantAnnotation)</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>vcf_file <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai/2023-06-03-calculating_genetic_distance_from_ref/HG001_GRCh38_1_22_v4.2.1_benchmark.vcf.gz"</span></span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a><span class="co"># vcf &lt;- readVcf(file = vcf_file, genome = "GRCh38")</span></span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>)))</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_metadata) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gene_metadata<span class="sc">$</span>...<span class="dv">1</span>)</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a><span class="fu">### Check if the VCF file reference is the reference genome</span></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>example_region <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="st">"chr22"</span>, <span class="at">ranges=</span><span class="fu">IRanges</span>(</span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a>  <span class="at">start=</span><span class="fu">c</span>(<span class="dv">50301422</span>, <span class="dv">50989541</span>),</span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a>  <span class="at">end=</span><span class="fu">c</span>(<span class="dv">50312106</span>, <span class="dv">51001328</span>),</span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>  <span class="at">names=</span><span class="fu">c</span>(<span class="st">"gene_79087"</span>, <span class="st">"gene_644186"</span>)))</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>vcf_example <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="at">file =</span> vcf_file, <span class="at">genome =</span> <span class="st">"GRCh38"</span>, <span class="at">param=</span>example_region)</span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a><span class="fu">geno</span>(vcf_example)<span class="sc">$</span>GT</span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a><span class="fu">rowRanges</span>(vcf_example)</span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a>Through manual inspection of this region, it is clear that the reference matches the reference genome, rather than being linked to relative population frequency.</span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a><span class="fu">### Compute genomic distance from reference</span></span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a>We can write and test a function to compute this genetic distance metric</span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a>gene_row <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>example_gene <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span>gene_metadata[gene_row,<span class="dv">1</span>], <span class="at">ranges=</span><span class="fu">IRanges</span>(</span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a>  <span class="at">start=</span><span class="fu">as.numeric</span>(gene_metadata[gene_row,<span class="dv">2</span>]),</span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>  <span class="at">end=</span><span class="fu">as.numeric</span>(gene_metadata[gene_row,<span class="dv">3</span>]),</span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>  <span class="at">names=</span>gene_metadata[gene_row,<span class="dv">4</span>]))</span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>vcf_example_gene <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="at">file =</span> vcf_file, <span class="at">genome =</span> <span class="st">"GRCh38"</span>, <span class="at">param=</span>example_gene)</span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT)</span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a><span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT</span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a><span class="co"># rowRanges(vcf_example)</span></span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a>genos <span class="ot">=</span> <span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT</span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a>total_a_count <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb17-103"><a href="#cb17-103" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gt <span class="cf">in</span> genos) {</span>
<span id="cb17-104"><a href="#cb17-104" aria-hidden="true" tabindex="-1"></a>  total_a_count <span class="ot">=</span> total_a_count <span class="sc">+</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(gt,<span class="st">"/"</span>)[[<span class="dv">1</span>]]))</span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a>average_a_count <span class="ot">=</span> total_a_count<span class="sc">/</span><span class="fu">width</span>(example_gene)</span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-112"><a href="#cb17-112" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-113"><a href="#cb17-113" aria-hidden="true" tabindex="-1"></a>calc_alleles_single_gene <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_row, <span class="at">gene_metadata=</span>gene_metadata, <span class="at">vcf_file=</span>vcf_file) {</span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (gene_row[<span class="dv">1</span>] <span class="sc">==</span> <span class="st">"chrM"</span>) {</span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span> (<span class="cn">NA</span>)</span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a>  example_gene <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span>gene_row[<span class="dv">1</span>], <span class="at">ranges=</span><span class="fu">IRanges</span>(</span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a>    <span class="at">start=</span><span class="fu">as.numeric</span>(gene_row[<span class="dv">2</span>]),</span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a>    <span class="at">end=</span><span class="fu">as.numeric</span>(gene_row[<span class="dv">3</span>]),</span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a>    <span class="at">names=</span>gene_row[<span class="dv">4</span>]))</span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-124"><a href="#cb17-124" aria-hidden="true" tabindex="-1"></a>  vcf_example_gene <span class="ot">&lt;-</span> <span class="fu">readVcf</span>(<span class="at">file =</span> vcf_file, <span class="at">genome =</span> <span class="st">"GRCh38"</span>, <span class="at">param=</span>example_gene)</span>
<span id="cb17-125"><a href="#cb17-125" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a>  genos <span class="ot">=</span> <span class="fu">geno</span>(vcf_example_gene)<span class="sc">$</span>GT</span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a>  total_a_count <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a>  num_het <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a>  num_homo <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (gt <span class="cf">in</span> genos) {</span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a>    cur_a_count <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(gt,<span class="st">"/"</span>)[[<span class="dv">1</span>]]))</span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (cur_a_count<span class="sc">==</span><span class="dv">2</span>) {</span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a>      num_homo <span class="ot">=</span> num_homo <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a>      num_het <span class="ot">=</span> num_het <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a>    total_a_count <span class="ot">=</span> total_a_count <span class="sc">+</span> cur_a_count</span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">c</span>(total_a_count, <span class="fu">length</span>(genos),num_het,num_homo))</span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-144"><a href="#cb17-144" aria-hidden="true" tabindex="-1"></a>Now let's run it across all genes</span>
<span id="cb17-145"><a href="#cb17-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-148"><a href="#cb17-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-149"><a href="#cb17-149" aria-hidden="true" tabindex="-1"></a>gene_metadata_trimmed <span class="ot">&lt;-</span> gene_metadata[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span>
<span id="cb17-150"><a href="#cb17-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-151"><a href="#cb17-151" aria-hidden="true" tabindex="-1"></a>total_alt_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(gene_metadata,<span class="dv">1</span>, calc_alleles_single_gene, <span class="at">gene_metadata=</span>gene_metadata, <span class="at">vcf_file=</span>vcf_file)</span>
<span id="cb17-152"><a href="#cb17-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-153"><a href="#cb17-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-154"><a href="#cb17-154" aria-hidden="true" tabindex="-1"></a>ref_var_counts <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>,<span class="at">nrow=</span><span class="fu">length</span>(total_alt_counts),<span class="at">ncol =</span> <span class="dv">4</span>)</span>
<span id="cb17-155"><a href="#cb17-155" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(total_alt_counts)) {</span>
<span id="cb17-156"><a href="#cb17-156" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(total_alt_counts[[x]])<span class="sc">==</span><span class="dv">1</span>) {</span>
<span id="cb17-157"><a href="#cb17-157" aria-hidden="true" tabindex="-1"></a>    ref_var_counts[x,] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="cn">NA</span>,<span class="cn">NA</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb17-158"><a href="#cb17-158" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb17-159"><a href="#cb17-159" aria-hidden="true" tabindex="-1"></a>    ref_var_counts[x,] <span class="ot">&lt;-</span> total_alt_counts[[x]]</span>
<span id="cb17-160"><a href="#cb17-160" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-161"><a href="#cb17-161" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-162"><a href="#cb17-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-163"><a href="#cb17-163" aria-hidden="true" tabindex="-1"></a>ref_var_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ref_var_counts)</span>
<span id="cb17-164"><a href="#cb17-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-165"><a href="#cb17-165" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ref_var_counts) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"total_alt_count_LCL"</span>,<span class="st">"num_variants_LCL"</span>,<span class="st">"num_het_LCL"</span>,<span class="st">"num_homo_LCL"</span>)</span>
<span id="cb17-166"><a href="#cb17-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-167"><a href="#cb17-167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-168"><a href="#cb17-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-169"><a href="#cb17-169" aria-hidden="true" tabindex="-1"></a>This takes a while to run, so let's save it just in case:</span>
<span id="cb17-170"><a href="#cb17-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-173"><a href="#cb17-173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-174"><a href="#cb17-174" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(total_alt_counts, <span class="at">file=</span><span class="st">"altcounts.rda"</span>)</span>
<span id="cb17-175"><a href="#cb17-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-176"><a href="#cb17-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-177"><a href="#cb17-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-178"><a href="#cb17-178" aria-hidden="true" tabindex="-1"></a>Reload alt allele counts here</span>
<span id="cb17-179"><a href="#cb17-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-182"><a href="#cb17-182" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-183"><a href="#cb17-183" aria-hidden="true" tabindex="-1"></a><span class="co"># load("altcounts.rda")</span></span>
<span id="cb17-184"><a href="#cb17-184" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-185"><a href="#cb17-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-186"><a href="#cb17-186" aria-hidden="true" tabindex="-1"></a>We can now plot the allelic dosage as a proportion of gene length</span>
<span id="cb17-187"><a href="#cb17-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-190"><a href="#cb17-190" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-191"><a href="#cb17-191" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>total_alt_count_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>total_alt_count_LCL</span>
<span id="cb17-192"><a href="#cb17-192" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>num_variants_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>num_variants_LCL</span>
<span id="cb17-193"><a href="#cb17-193" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>num_het_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>num_het_LCL</span>
<span id="cb17-194"><a href="#cb17-194" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>num_homo_LCL <span class="ot">&lt;-</span> ref_var_counts<span class="sc">$</span>num_homo_LCL</span>
<span id="cb17-195"><a href="#cb17-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-196"><a href="#cb17-196" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>alt_counts_len_normalized_LCL <span class="ot">&lt;-</span> gene_metadata<span class="sc">$</span>total_alt_count_LCL<span class="sc">/</span>gene_metadata<span class="sc">$</span>length</span>
<span id="cb17-197"><a href="#cb17-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-198"><a href="#cb17-198" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_metadata) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"count"</span>,<span class="fu">aes</span>(<span class="at">x=</span>best_ancestry))</span>
<span id="cb17-199"><a href="#cb17-199" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_metadata) <span class="sc">+</span> <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>best_ancestry,<span class="at">y=</span>alt_counts_len_normalized_LCL)) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.025</span>))</span>
<span id="cb17-200"><a href="#cb17-200" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_metadata) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>best_ancestry,<span class="at">y=</span>alt_counts_len_normalized_LCL)) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.025</span>))</span>
<span id="cb17-201"><a href="#cb17-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-202"><a href="#cb17-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-203"><a href="#cb17-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-204"><a href="#cb17-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-207"><a href="#cb17-207" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-208"><a href="#cb17-208" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gene_metadata, <span class="st">"gene_metadata_w_allele_counts.csv"</span>)</span>
<span id="cb17-209"><a href="#cb17-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-210"><a href="#cb17-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-211"><a href="#cb17-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-212"><a href="#cb17-212" aria-hidden="true" tabindex="-1"></a>The figures above essentially show that the overall alternate allelic dosage from the cell line (normalized for gene length) is enriched among genes with African reference genome ancestry, which supports our assumptions.</span>
<span id="cb17-213"><a href="#cb17-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-214"><a href="#cb17-214" aria-hidden="true" tabindex="-1"></a><span class="fu">### Recompute gene correlations</span></span>
<span id="cb17-215"><a href="#cb17-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-218"><a href="#cb17-218" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-219"><a href="#cb17-219" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Load Datasets</span></span>
<span id="cb17-220"><a href="#cb17-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-221"><a href="#cb17-221" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb17-222"><a href="#cb17-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-223"><a href="#cb17-223" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb17-224"><a href="#cb17-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-225"><a href="#cb17-225" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb17-226"><a href="#cb17-226" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb17-227"><a href="#cb17-227" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb17-228"><a href="#cb17-228" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb17-229"><a href="#cb17-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-230"><a href="#cb17-230" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb17-231"><a href="#cb17-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-232"><a href="#cb17-232" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_canonical_haplo1_filtered.csv"</span>)))</span>
<span id="cb17-233"><a href="#cb17-233" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span></span>
<span id="cb17-234"><a href="#cb17-234" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-235"><a href="#cb17-235" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(enformer_haplo1) &lt;- enformer_haplo1$X</span></span>
<span id="cb17-236"><a href="#cb17-236" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_haplo1$X &lt;- NULL</span></span>
<span id="cb17-237"><a href="#cb17-237" aria-hidden="true" tabindex="-1"></a>enformer_haplo2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_canonical_haplo2_filtered.csv"</span>)))</span>
<span id="cb17-238"><a href="#cb17-238" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo2) <span class="ot">&lt;-</span> enformer_haplo2<span class="sc">$</span>X</span>
<span id="cb17-239"><a href="#cb17-239" aria-hidden="true" tabindex="-1"></a>enformer_haplo2<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-240"><a href="#cb17-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-241"><a href="#cb17-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-242"><a href="#cb17-242" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb17-243"><a href="#cb17-243" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> (enformer_haplo1 <span class="sc">+</span> enformer_haplo2)</span>
<span id="cb17-244"><a href="#cb17-244" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span>
<span id="cb17-245"><a href="#cb17-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-246"><a href="#cb17-246" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-247"><a href="#cb17-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-250"><a href="#cb17-250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-251"><a href="#cb17-251" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb17-252"><a href="#cb17-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-253"><a href="#cb17-253" aria-hidden="true" tabindex="-1"></a>AFR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,AFR_ind)</span>
<span id="cb17-254"><a href="#cb17-254" aria-hidden="true" tabindex="-1"></a>EUR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,EUR_ind)</span>
<span id="cb17-255"><a href="#cb17-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-256"><a href="#cb17-256" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb17-257"><a href="#cb17-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-258"><a href="#cb17-258" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, ge_ind]</span>
<span id="cb17-259"><a href="#cb17-259" aria-hidden="true" tabindex="-1"></a>geuvadis_AFR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, AFR_ge_ind]</span>
<span id="cb17-260"><a href="#cb17-260" aria-hidden="true" tabindex="-1"></a>geuvadis_EUR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, EUR_ge_ind]</span>
<span id="cb17-261"><a href="#cb17-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-262"><a href="#cb17-262" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[ge_genes, ge_ind]</span>
<span id="cb17-263"><a href="#cb17-263" aria-hidden="true" tabindex="-1"></a>enformer_AFR_compare <span class="ot">&lt;-</span> enformer[ge_genes, AFR_ge_ind]</span>
<span id="cb17-264"><a href="#cb17-264" aria-hidden="true" tabindex="-1"></a>enformer_EUR_compare <span class="ot">&lt;-</span> enformer[ge_genes, EUR_ge_ind]</span>
<span id="cb17-265"><a href="#cb17-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-266"><a href="#cb17-266" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[ge_genes,]</span>
<span id="cb17-267"><a href="#cb17-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-268"><a href="#cb17-268" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals: "</span>, <span class="fu">length</span>(ge_ind)))</span>
<span id="cb17-269"><a href="#cb17-269" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals AFR: "</span>, <span class="fu">length</span>(AFR_ge_ind)))</span>
<span id="cb17-270"><a href="#cb17-270" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals EUR: "</span>, <span class="fu">length</span>(EUR_ge_ind)))</span>
<span id="cb17-271"><a href="#cb17-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-272"><a href="#cb17-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-273"><a href="#cb17-273" aria-hidden="true" tabindex="-1"></a>ge_genes_plus <span class="ot">&lt;-</span> ge_genes</span>
<span id="cb17-274"><a href="#cb17-274" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes: "</span>, <span class="fu">length</span>(gene_meta_compare<span class="sc">$</span>best_ancestry)))</span>
<span id="cb17-275"><a href="#cb17-275" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes,AFR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"AFR"</span>,]<span class="sc">$</span>best_ancestry)))</span>
<span id="cb17-276"><a href="#cb17-276" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes,EUR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"EUR"</span>,]<span class="sc">$</span>best_ancestry)))</span>
<span id="cb17-277"><a href="#cb17-277" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-278"><a href="#cb17-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-281"><a href="#cb17-281" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-282"><a href="#cb17-282" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb17-283"><a href="#cb17-283" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb17-284"><a href="#cb17-284" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb17-285"><a href="#cb17-285" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb17-286"><a href="#cb17-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb17-287"><a href="#cb17-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb17-288"><a href="#cb17-288" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-289"><a href="#cb17-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-290"><a href="#cb17-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-291"><a href="#cb17-291" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb17-292"><a href="#cb17-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-293"><a href="#cb17-293" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb17-294"><a href="#cb17-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-295"><a href="#cb17-295" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>all_geuvadis_ge_corrs <span class="ot">&lt;-</span> corrs_df[<span class="fu">rownames</span>(gene_metadata),]<span class="sc">$</span>ge</span>
<span id="cb17-296"><a href="#cb17-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-297"><a href="#cb17-297" aria-hidden="true" tabindex="-1"></a>corrs_df_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb17-298"><a href="#cb17-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-299"><a href="#cb17-299" aria-hidden="true" tabindex="-1"></a>corrs_df_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb17-300"><a href="#cb17-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-301"><a href="#cb17-301" aria-hidden="true" tabindex="-1"></a>corrs_df_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb17-302"><a href="#cb17-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-303"><a href="#cb17-303" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb17-304"><a href="#cb17-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-305"><a href="#cb17-305" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb17-306"><a href="#cb17-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-307"><a href="#cb17-307" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span>
<span id="cb17-308"><a href="#cb17-308" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-309"><a href="#cb17-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-312"><a href="#cb17-312" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-313"><a href="#cb17-313" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb17-314"><a href="#cb17-314" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb17-315"><a href="#cb17-315" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb17-316"><a href="#cb17-316" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb17-317"><a href="#cb17-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb17-318"><a href="#cb17-318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb17-319"><a href="#cb17-319" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-320"><a href="#cb17-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-321"><a href="#cb17-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-322"><a href="#cb17-322" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_AFR_compare, enformer_AFR_compare)</span>
<span id="cb17-323"><a href="#cb17-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-324"><a href="#cb17-324" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb17-325"><a href="#cb17-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-326"><a href="#cb17-326" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>AFR_geuvadis_ge_corrs <span class="ot">&lt;-</span> corrs_df[<span class="fu">rownames</span>(gene_metadata),]<span class="sc">$</span>ge</span>
<span id="cb17-327"><a href="#cb17-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-328"><a href="#cb17-328" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb17-329"><a href="#cb17-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-330"><a href="#cb17-330" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb17-331"><a href="#cb17-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-332"><a href="#cb17-332" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb17-333"><a href="#cb17-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-334"><a href="#cb17-334" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb17-335"><a href="#cb17-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-336"><a href="#cb17-336" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb17-337"><a href="#cb17-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-338"><a href="#cb17-338" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb17-339"><a href="#cb17-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-340"><a href="#cb17-340" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family =</span> <span class="st">"binomial"</span>))</span>
<span id="cb17-341"><a href="#cb17-341" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-342"><a href="#cb17-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-345"><a href="#cb17-345" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-346"><a href="#cb17-346" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb17-347"><a href="#cb17-347" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb17-348"><a href="#cb17-348" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb17-349"><a href="#cb17-349" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb17-350"><a href="#cb17-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb17-351"><a href="#cb17-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb17-352"><a href="#cb17-352" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-353"><a href="#cb17-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-354"><a href="#cb17-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-355"><a href="#cb17-355" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_EUR_compare, enformer_EUR_compare)</span>
<span id="cb17-356"><a href="#cb17-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-357"><a href="#cb17-357" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb17-358"><a href="#cb17-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-359"><a href="#cb17-359" aria-hidden="true" tabindex="-1"></a>gene_metadata<span class="sc">$</span>EUR_geuvadis_ge_corrs <span class="ot">&lt;-</span> corrs_df[<span class="fu">rownames</span>(gene_metadata),]<span class="sc">$</span>ge</span>
<span id="cb17-360"><a href="#cb17-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-361"><a href="#cb17-361" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb17-362"><a href="#cb17-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-363"><a href="#cb17-363" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb17-364"><a href="#cb17-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-365"><a href="#cb17-365" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb17-366"><a href="#cb17-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-367"><a href="#cb17-367" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb17-368"><a href="#cb17-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-369"><a href="#cb17-369" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb17-370"><a href="#cb17-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-371"><a href="#cb17-371" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(data.frame(table(corrs_df_plus_AFR_EUR$ref_gene_ancestry, corrs_df_plus_AFR_EUR$corr_direction)), aes(x=Var1,y=Freq,fill=Var2, group=Var1)) + geom_bar(stat="identity", position="dodge")</span></span>
<span id="cb17-372"><a href="#cb17-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-373"><a href="#cb17-373" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb17-374"><a href="#cb17-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-375"><a href="#cb17-375" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span>
<span id="cb17-376"><a href="#cb17-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-377"><a href="#cb17-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-380"><a href="#cb17-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-381"><a href="#cb17-381" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(gene_metadata, <span class="st">"gene_metadata_w_allele_counts.csv"</span>)</span>
<span id="cb17-382"><a href="#cb17-382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-383"><a href="#cb17-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-384"><a href="#cb17-384" aria-hidden="true" tabindex="-1"></a><span class="fu">### Incorporate population level information</span></span>
<span id="cb17-385"><a href="#cb17-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-386"><a href="#cb17-386" aria-hidden="true" tabindex="-1"></a>A potential issue with the above metric is that we haven't really controlled for the fact that some genes also tend to show more variation in general, so would have stronger alt allele counts by default. If this is roughly even between AFR and EUR genes then we are ok, but it may not be.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>