<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2024-01-24">

<title>Daily-Blog-Sai - 2.1.2_check_predictDB_standard</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#load-summary-standard" id="toc-load-summary-standard" class="nav-link" data-scroll-target="#load-summary-standard">Load summary standard</a></li>
  </ul></li>
  <li><a href="#check-for-egenes-from-aracena-et-al." id="toc-check-for-egenes-from-aracena-et-al." class="nav-link" data-scroll-target="#check-for-egenes-from-aracena-et-al.">Check for eGenes from Aracena et al.</a>
  <ul class="collapse">
  <li><a href="#load-summary-enpact" id="toc-load-summary-enpact" class="nav-link" data-scroll-target="#load-summary-enpact">Load summary Enpact</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">2.1.2_check_predictDB_standard.qmd</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 24, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="context" class="level1">
<h1>Context</h1>
<p>I seem to be getting strangely low gene counts past filtering after running PredictDB. Need to investigate this.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-summary-standard" class="level3">
<h3 class="anchored" data-anchor-id="load-summary-standard">Load summary standard</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>path_pred_summ <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu/database/Model_summary.txt"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>pred_summ <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_pred_summ, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 10487    24</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>rho_avg, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Standard PredictDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-2-1.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>zscore_pval, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Standard PredictDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-2-2.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>pred_summ_filt <span class="ot">&lt;-</span> pred_summ[(pred_summ<span class="sc">$</span>rho_avg <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> (pred_summ<span class="sc">$</span>zscore_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 230  24</code></pre>
</div>
</div>
</section>
</section>
<section id="check-for-egenes-from-aracena-et-al." class="level1">
<h1>Check for eGenes from Aracena et al.</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load eGenes table</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>path_to_egenes <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/QTL_mapping/eqtls.csv"</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>egenes_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_to_egenes, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(egenes_table<span class="sc">$</span>pvalue_Flu, <span class="at">main =</span> <span class="st">"eGenes pvalues"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-3-1.svg" class="img-fluid"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(pred_summ)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 10487    24</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(egenes_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 13680    13</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>subset_to_egenes <span class="ot">&lt;-</span> <span class="fu">merge</span>(egenes_table, pred_summ, <span class="at">by.x =</span> <span class="st">"feature"</span>, <span class="at">by.y =</span> <span class="st">"gene"</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(subset_to_egenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 10487    36</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(subset_to_egenes, <span class="fu">aes</span>(<span class="at">x =</span> pvalue_Flu, <span class="at">y =</span> zscore_pval)) <span class="sc">+</span> <span class="fu">geom_point</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-5-1.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>spearcor <span class="ot">&lt;-</span> <span class="fu">cor</span>(subset_to_egenes<span class="sc">$</span>pvalue_Flu, subset_to_egenes<span class="sc">$</span>zscore_pval, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>spearcor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1050551</code></pre>
</div>
</div>
<section id="load-summary-enpact" class="level3">
<h3 class="anchored" data-anchor-id="load-summary-enpact">Load summary Enpact</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>path_pred_summ <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/predixcan_RNAseq_Flu_from_HDF5_mean_log_revised/database/Model_summary.txt"</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>pred_summ <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_pred_summ, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18714    24</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>rho_avg, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Enpact PredictDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-6-1.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>zscore_pval, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Enpact PredictDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-6-2.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>pred_summ_filt <span class="ot">&lt;-</span> pred_summ[(pred_summ<span class="sc">$</span>rho_avg <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> (pred_summ<span class="sc">$</span>zscore_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18695    24</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>path_pred_summ <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/charles/project/singleXcanDL/PredicDB/OneK1K/PrediXcan/work_dir/results/database/Model_summary.txt"</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>pred_summ <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_pred_summ, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18082    24</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>rho_avg, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Charles PredictDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-7-1.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>zscore_pval, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Charles PredictDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="2.1.2_check_predictDB_standard_files/figure-html/unnamed-chunk-7-2.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>pred_summ_filt <span class="ot">&lt;-</span> pred_summ[(pred_summ<span class="sc">$</span>rho_avg <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> (pred_summ<span class="sc">$</span>zscore_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2992   24</code></pre>
</div>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb30" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "2.1.2_check_predictDB_standard"</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2024-01-24"</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>I seem to be getting strangely low gene counts past filtering after running PredictDB. Need to investigate this.</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a><span class="fu">## Compare normalized counts from different sources</span></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a><span class="fu">### Gene expression</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>expression_files <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">onta_expression=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_onta.txt"</span>,</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_3_7_lnormTRUE_beforeTRUE_scale1.txt"</span>,</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_after_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_3_7_lnormTRUE_beforeFALSE_scale1.txt"</span>,</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">no_lnorm=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_3_7_lnormFALSE_beforeTRUE_scale1.txt"</span>,</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">scaled_lnorm_before_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_3_7_lnormTRUE_beforeTRUE_scale1e+06.txt"</span>,</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM_4_4=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_4_4_lnormTRUE_beforeTRUE_scale1.txt"</span>,</span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">kallisto_norm=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/final_testing/fully_preprocessed_expression_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto.txt"</span></span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a>expression_dfs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(expression_files, <span class="cf">function</span>(x) {</span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.table</span>(x, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a>common_colnames <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(expression_dfs, colnames))</span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a>common_flu_colnames <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"Flu"</span>, common_colnames, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a>flu_dfs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(expression_dfs, <span class="cf">function</span>(x) x[common_flu_colnames])</span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a><span class="fu">### Cross-genome mean expression correlations</span></span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(flu_dfs, rownames))</span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a>common_genes_dfs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(flu_dfs, <span class="cf">function</span>(x) x[common_genes,])</span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a>mean_expression_dfs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(common_genes_dfs, <span class="cf">function</span>(x) {</span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowMeans</span>(x)</span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a>mean_expression_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, mean_expression_dfs)</span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(mean_expression_df, <span class="at">method =</span> <span class="st">"spearman"</span>), <span class="at">main =</span> <span class="st">"Spearman correlation of mean expression across sources"</span>, <span class="at">method =</span> <span class="st">"number"</span>)</span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(mean_expression_df, <span class="at">method =</span> <span class="st">"pearson"</span>), <span class="at">main =</span> <span class="st">"Pearson correlation of mean expression across sources"</span>, <span class="at">method =</span> <span class="st">"number"</span>)</span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example cross-individual correlation for genes</span></span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a>example_genes <span class="ot">&lt;-</span> common_genes[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb30-87"><a href="#cb30-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-88"><a href="#cb30-88" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(flu_dfs, colnames))</span>
<span id="cb30-89"><a href="#cb30-89" aria-hidden="true" tabindex="-1"></a>common_sample_dfs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(flu_dfs, <span class="cf">function</span>(x) x[,common_samples])</span>
<span id="cb30-90"><a href="#cb30-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-91"><a href="#cb30-91" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> example_genes) {</span>
<span id="cb30-92"><a href="#cb30-92" aria-hidden="true" tabindex="-1"></a>  gene_dfs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(common_sample_dfs, <span class="cf">function</span>(x) x[gene,])</span>
<span id="cb30-93"><a href="#cb30-93" aria-hidden="true" tabindex="-1"></a>  gene_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, gene_dfs)</span>
<span id="cb30-94"><a href="#cb30-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(gene_df))</span>
<span id="cb30-95"><a href="#cb30-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pairs</span>(<span class="fu">as.data.frame</span>(<span class="fu">t</span>(gene_df)), <span class="at">main =</span> gene)</span>
<span id="cb30-96"><a href="#cb30-96" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-97"><a href="#cb30-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-98"><a href="#cb30-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-99"><a href="#cb30-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-100"><a href="#cb30-100" aria-hidden="true" tabindex="-1"></a><span class="fu">### Genotype</span></span>
<span id="cb30-101"><a href="#cb30-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-102"><a href="#cb30-102" aria-hidden="true" tabindex="-1"></a>One way to check the genotype data is to simply look for eQTLs from the manuscript and plot their allele dosage vs. expression.</span>
<span id="cb30-103"><a href="#cb30-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-106"><a href="#cb30-106" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-107"><a href="#cb30-107" aria-hidden="true" tabindex="-1"></a>genotypes_source_file <span class="ot">=</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/QTL_mapping/SNP_genotypes_b37.txt"</span></span>
<span id="cb30-108"><a href="#cb30-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-109"><a href="#cb30-109" aria-hidden="true" tabindex="-1"></a>genotypes <span class="ot">&lt;-</span> <span class="fu">read.table</span>(genotypes_source_file, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb30-110"><a href="#cb30-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-111"><a href="#cb30-111" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-112"><a href="#cb30-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-113"><a href="#cb30-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-116"><a href="#cb30-116" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-117"><a href="#cb30-117" aria-hidden="true" tabindex="-1"></a>top_eGenes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ENSG00000121716"</span>, <span class="st">"ENSG00000256274"</span>, <span class="st">"ENSG00000166435"</span>)</span>
<span id="cb30-118"><a href="#cb30-118" aria-hidden="true" tabindex="-1"></a>top_flu_qtls <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"chr7_99910443_C_G_b37"</span>,<span class="st">"chr12_11229016_C_G_b37"</span>,<span class="st">"chr11_74560117_T_C_b37"</span>)</span>
<span id="cb30-119"><a href="#cb30-119" aria-hidden="true" tabindex="-1"></a>qtl_betas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.895589436</span>,<span class="sc">-</span><span class="fl">0.838071979</span>,<span class="fl">0.96135383</span>)</span>
<span id="cb30-120"><a href="#cb30-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-121"><a href="#cb30-121" aria-hidden="true" tabindex="-1"></a>top_eGene_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">top_eGenes=</span>top_eGenes, <span class="at">top_flu_qtls=</span>top_flu_qtls, <span class="at">qtl_betas=</span>qtl_betas)</span>
<span id="cb30-122"><a href="#cb30-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-123"><a href="#cb30-123" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(top_eGene_df) <span class="ot">&lt;-</span> top_eGene_df<span class="sc">$</span>top_eGenes</span>
<span id="cb30-124"><a href="#cb30-124" aria-hidden="true" tabindex="-1"></a>top_eGene_df <span class="ot">&lt;-</span> top_eGene_df[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-125"><a href="#cb30-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-126"><a href="#cb30-126" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(genotypes) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(genotypes), <span class="st">"_Flu"</span>)</span>
<span id="cb30-127"><a href="#cb30-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-128"><a href="#cb30-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-129"><a href="#cb30-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-130"><a href="#cb30-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-131"><a href="#cb30-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-134"><a href="#cb30-134" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-135"><a href="#cb30-135" aria-hidden="true" tabindex="-1"></a>row_list <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow=</span>(<span class="fu">length</span>(flu_dfs)<span class="sc">*</span><span class="dv">3</span><span class="sc">*</span><span class="fu">length</span>(<span class="fu">colnames</span>(flu_dfs[[<span class="dv">1</span>]]))), <span class="at">ncol=</span><span class="dv">6</span>)</span>
<span id="cb30-136"><a href="#cb30-136" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb30-137"><a href="#cb30-137" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene_i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(top_eGenes)) {</span>
<span id="cb30-138"><a href="#cb30-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-139"><a href="#cb30-139" aria-hidden="true" tabindex="-1"></a>  cur_gene <span class="ot">&lt;-</span> top_eGenes[gene_i]</span>
<span id="cb30-140"><a href="#cb30-140" aria-hidden="true" tabindex="-1"></a>  cur_qtl <span class="ot">&lt;-</span> top_flu_qtls[gene_i]</span>
<span id="cb30-141"><a href="#cb30-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-142"><a href="#cb30-142" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (ind <span class="cf">in</span> <span class="fu">colnames</span>(flu_dfs[[<span class="fu">names</span>(flu_dfs)[<span class="dv">1</span>]]])) {</span>
<span id="cb30-143"><a href="#cb30-143" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (table <span class="cf">in</span> <span class="fu">names</span>(flu_dfs)) {</span>
<span id="cb30-144"><a href="#cb30-144" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(gene_i)</span>
<span id="cb30-145"><a href="#cb30-145" aria-hidden="true" tabindex="-1"></a>      cur_row <span class="ot">&lt;-</span> <span class="fu">c</span>(cur_gene, cur_qtl, ind, genotypes[cur_qtl, ind], flu_dfs[[table]][cur_gene, ind], table)</span>
<span id="cb30-146"><a href="#cb30-146" aria-hidden="true" tabindex="-1"></a>      row_list[c,] <span class="ot">&lt;-</span> cur_row</span>
<span id="cb30-147"><a href="#cb30-147" aria-hidden="true" tabindex="-1"></a>      c <span class="ot">&lt;-</span> c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb30-148"><a href="#cb30-148" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb30-149"><a href="#cb30-149" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-150"><a href="#cb30-150" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-151"><a href="#cb30-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-152"><a href="#cb30-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-153"><a href="#cb30-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-154"><a href="#cb30-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-157"><a href="#cb30-157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-158"><a href="#cb30-158" aria-hidden="true" tabindex="-1"></a>qtl_exp_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(row_list)</span>
<span id="cb30-159"><a href="#cb30-159" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(qtl_exp_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"qtl"</span>, <span class="st">"ind"</span>, <span class="st">"genotype"</span>, <span class="st">"expression"</span>, <span class="st">"source"</span>)</span>
<span id="cb30-160"><a href="#cb30-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-161"><a href="#cb30-161" aria-hidden="true" tabindex="-1"></a>qtl_exp_df<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">as.character</span>(qtl_exp_df<span class="sc">$</span>genotype))</span>
<span id="cb30-162"><a href="#cb30-162" aria-hidden="true" tabindex="-1"></a>qtl_exp_df<span class="sc">$</span>expression <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(qtl_exp_df<span class="sc">$</span>expression))</span>
<span id="cb30-163"><a href="#cb30-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-164"><a href="#cb30-164" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(qtl_exp_df)</span>
<span id="cb30-165"><a href="#cb30-165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-166"><a href="#cb30-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-169"><a href="#cb30-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-170"><a href="#cb30-170" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb30-171"><a href="#cb30-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-172"><a href="#cb30-172" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(qtl_exp_df, <span class="fu">aes</span>(<span class="at">x=</span>genotype, <span class="at">y=</span>expression)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(gene<span class="sc">~</span>source, <span class="at">nrow=</span><span class="dv">3</span>)</span>
<span id="cb30-173"><a href="#cb30-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-174"><a href="#cb30-174" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (table <span class="cf">in</span> <span class="fu">names</span>(flu_dfs)) {</span>
<span id="cb30-175"><a href="#cb30-175" aria-hidden="true" tabindex="-1"></a>  only_full_norm <span class="ot">&lt;-</span> qtl_exp_df[qtl_exp_df<span class="sc">$</span>source <span class="sc">==</span> table,]</span>
<span id="cb30-176"><a href="#cb30-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">ggplot</span>(only_full_norm, <span class="fu">aes</span>(<span class="at">x=</span>genotype, <span class="at">y=</span>expression)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(gene<span class="sc">~</span>source, <span class="at">nrow=</span><span class="fu">length</span>(top_eGenes)))</span>
<span id="cb30-177"><a href="#cb30-177" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-178"><a href="#cb30-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-179"><a href="#cb30-179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-180"><a href="#cb30-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-181"><a href="#cb30-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-182"><a href="#cb30-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-183"><a href="#cb30-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-184"><a href="#cb30-184" aria-hidden="true" tabindex="-1"></a><span class="fu">## Compare PredictDB results</span></span>
<span id="cb30-185"><a href="#cb30-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-188"><a href="#cb30-188" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-189"><a href="#cb30-189" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb30-190"><a href="#cb30-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-191"><a href="#cb30-191" aria-hidden="true" tabindex="-1"></a>model_summaries <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb30-192"><a href="#cb30-192" aria-hidden="true" tabindex="-1"></a>  <span class="at">onta_expression=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_NI_singlenested_onta/database/Model_summary.txt"</span>,</span>
<span id="cb30-193"><a href="#cb30-193" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_3_7_lnormTRUE_beforeTRUE_scale1/database/Model_summary.txt"</span>,</span>
<span id="cb30-194"><a href="#cb30-194" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_after_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_3_7_lnormTRUE_beforeFALSE_scale1/database/Model_summary.txt"</span>,</span>
<span id="cb30-195"><a href="#cb30-195" aria-hidden="true" tabindex="-1"></a>  <span class="at">no_lnorm=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_3_7_lnormFALSE_beforeTRUE_scale1/database/Model_summary.txt"</span>,</span>
<span id="cb30-196"><a href="#cb30-196" aria-hidden="true" tabindex="-1"></a>  <span class="at">scaled_lnorm_before_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_3_7_lnormTRUE_beforeTRUE_scale1e+06/database/Model_summary.txt"</span>,</span>
<span id="cb30-197"><a href="#cb30-197" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM_4_4=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_4_4_lnormTRUE_beforeTRUE_scale1/database/Model_summary.txt"</span>,</span>
<span id="cb30-198"><a href="#cb30-198" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM_4_4_repeat=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_4_4_lnormTRUE_beforeTRUE_scale1_REPEAT/database/Model_summary.txt"</span></span>
<span id="cb30-199"><a href="#cb30-199" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-200"><a href="#cb30-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-201"><a href="#cb30-201" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-202"><a href="#cb30-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-205"><a href="#cb30-205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-206"><a href="#cb30-206" aria-hidden="true" tabindex="-1"></a>loaded_model_summaries <span class="ot">&lt;-</span> <span class="fu">lapply</span>(model_summaries, <span class="cf">function</span>(x) {</span>
<span id="cb30-207"><a href="#cb30-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.table</span>(x, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb30-208"><a href="#cb30-208" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb30-209"><a href="#cb30-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-210"><a href="#cb30-210" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(loaded_model_summaries, <span class="cf">function</span>(x) <span class="fu">rownames</span>(x)))</span>
<span id="cb30-211"><a href="#cb30-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-212"><a href="#cb30-212" aria-hidden="true" tabindex="-1"></a>enriched_onta_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(loaded_model_summaries<span class="sc">$</span>onta_expression[loaded_model_summaries<span class="sc">$</span>onta_expression<span class="sc">$</span>zscore_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>,])</span>
<span id="cb30-213"><a href="#cb30-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-214"><a href="#cb30-214" aria-hidden="true" tabindex="-1"></a>common_genes_enriched <span class="ot">&lt;-</span> <span class="fu">intersect</span>(common_genes, enriched_onta_genes)</span>
<span id="cb30-215"><a href="#cb30-215" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-216"><a href="#cb30-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-219"><a href="#cb30-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-220"><a href="#cb30-220" aria-hidden="true" tabindex="-1"></a>zscore_pval_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">genes=</span>common_genes)</span>
<span id="cb30-221"><a href="#cb30-221" aria-hidden="true" tabindex="-1"></a>zscore_pval_df_enriched <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">genes=</span>common_genes_enriched)</span>
<span id="cb30-222"><a href="#cb30-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-223"><a href="#cb30-223" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (table <span class="cf">in</span> <span class="fu">names</span>(loaded_model_summaries)) {</span>
<span id="cb30-224"><a href="#cb30-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(table)</span>
<span id="cb30-225"><a href="#cb30-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(loaded_model_summaries[[table]]<span class="sc">$</span>rho_avg, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> table)</span>
<span id="cb30-226"><a href="#cb30-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(loaded_model_summaries[[table]]<span class="sc">$</span>zscore_pval, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> table)</span>
<span id="cb30-227"><a href="#cb30-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-228"><a href="#cb30-228" aria-hidden="true" tabindex="-1"></a>  zscore_pval_df[table] <span class="ot">&lt;-</span> loaded_model_summaries[[table]][common_genes,]<span class="sc">$</span>zscore_pval</span>
<span id="cb30-229"><a href="#cb30-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-230"><a href="#cb30-230" aria-hidden="true" tabindex="-1"></a>  zscore_pval_df_enriched[table] <span class="ot">&lt;-</span> loaded_model_summaries[[table]][common_genes_enriched,]<span class="sc">$</span>zscore_pval</span>
<span id="cb30-231"><a href="#cb30-231" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-232"><a href="#cb30-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-233"><a href="#cb30-233" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-234"><a href="#cb30-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-237"><a href="#cb30-237" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-238"><a href="#cb30-238" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(zscore_pval_df) <span class="ot">&lt;-</span> zscore_pval_df<span class="sc">$</span>genes</span>
<span id="cb30-239"><a href="#cb30-239" aria-hidden="true" tabindex="-1"></a>zscore_pval_df<span class="sc">$</span>genes <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb30-240"><a href="#cb30-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-241"><a href="#cb30-241" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(zscore_pval_df_enriched) <span class="ot">&lt;-</span> zscorxe_pval_df_enriched<span class="sc">$</span>genes</span>
<span id="cb30-242"><a href="#cb30-242" aria-hidden="true" tabindex="-1"></a>zscore_pval_df_enriched<span class="sc">$</span>genes <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb30-243"><a href="#cb30-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-244"><a href="#cb30-244" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(zscore_pval_df, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of zscore pvalues"</span>)</span>
<span id="cb30-245"><a href="#cb30-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-246"><a href="#cb30-246" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(zscore_pval_df_enriched, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of zscore pvalues (enriched genes)"</span>)</span>
<span id="cb30-247"><a href="#cb30-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-248"><a href="#cb30-248" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(zscore_pval_df_enriched)</span>
<span id="cb30-249"><a href="#cb30-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-250"><a href="#cb30-250" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-251"><a href="#cb30-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-252"><a href="#cb30-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-253"><a href="#cb30-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-256"><a href="#cb30-256" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-257"><a href="#cb30-257" aria-hidden="true" tabindex="-1"></a>filtered_dbs <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb30-258"><a href="#cb30-258" aria-hidden="true" tabindex="-1"></a>  <span class="at">onta_expression=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_onta/filtered_db/predict_db_Model_training_filtered.db"</span>,</span>
<span id="cb30-259"><a href="#cb30-259" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_3_7_lnormTRUE_beforeTRUE_scale1/filtered_db/predict_db_Model_training_filtered.db"</span>,</span>
<span id="cb30-260"><a href="#cb30-260" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lnorm_after_TMM="/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_3_7_lnormTRUE_beforeFALSE_scale1/filtered_db/predict_db_Model_training_filtered.db",</span></span>
<span id="cb30-261"><a href="#cb30-261" aria-hidden="true" tabindex="-1"></a>  <span class="at">no_lnorm=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_3_7_lnormFALSE_beforeTRUE_scale1/filtered_db/predict_db_Model_training_filtered.db"</span>,</span>
<span id="cb30-262"><a href="#cb30-262" aria-hidden="true" tabindex="-1"></a>  <span class="at">scaled_lnorm_before_TMM=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_3_7_lnormTRUE_beforeTRUE_scale1e+06/filtered_db/predict_db_Model_training_filtered.db"</span>,</span>
<span id="cb30-263"><a href="#cb30-263" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM_4_4=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_4_4_lnormTRUE_beforeTRUE_scale1/filtered_db/predict_db_Model_training_filtered.db"</span>,</span>
<span id="cb30-264"><a href="#cb30-264" aria-hidden="true" tabindex="-1"></a>  <span class="at">lnorm_before_TMM_4_4_repeat=</span><span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_singlenested_4_4_lnormTRUE_beforeTRUE_scale1_REPEAT/filtered_db/predict_db_Model_training_filtered.db"</span></span>
<span id="cb30-265"><a href="#cb30-265" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-266"><a href="#cb30-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-267"><a href="#cb30-267" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-268"><a href="#cb30-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-269"><a href="#cb30-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-272"><a href="#cb30-272" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-273"><a href="#cb30-273" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DBI)</span>
<span id="cb30-274"><a href="#cb30-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-275"><a href="#cb30-275" aria-hidden="true" tabindex="-1"></a>filtered_dbs_loaded <span class="ot">&lt;-</span> loaded_model_summaries <span class="ot">&lt;-</span> <span class="fu">lapply</span>(filtered_dbs, <span class="cf">function</span>(x) {</span>
<span id="cb30-276"><a href="#cb30-276" aria-hidden="true" tabindex="-1"></a>  mydb <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), x)</span>
<span id="cb30-277"><a href="#cb30-277" aria-hidden="true" tabindex="-1"></a>  myquery <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(mydb, <span class="st">'SELECT * FROM extra'</span>)</span>
<span id="cb30-278"><a href="#cb30-278" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb30-279"><a href="#cb30-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-280"><a href="#cb30-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-281"><a href="#cb30-281" aria-hidden="true" tabindex="-1"></a><span class="co"># onta_expression="/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/standard_predictDB_aracena_Flu_nested_onta/filtered_db/predict_db_Model_training_filtered.db"</span></span>
<span id="cb30-282"><a href="#cb30-282" aria-hidden="true" tabindex="-1"></a><span class="co"># mydb &lt;- dbConnect(RSQLite::SQLite(), onta_expression)</span></span>
<span id="cb30-283"><a href="#cb30-283" aria-hidden="true" tabindex="-1"></a><span class="co"># myquery &lt;- dbGetQuery(mydb, 'SELECT * FROM extra')</span></span>
<span id="cb30-284"><a href="#cb30-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-285"><a href="#cb30-285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-286"><a href="#cb30-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-289"><a href="#cb30-289" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-290"><a href="#cb30-290" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(filtered_dbs_loaded, <span class="cf">function</span>(x) x<span class="sc">$</span>genename))</span>
<span id="cb30-291"><a href="#cb30-291" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-292"><a href="#cb30-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-295"><a href="#cb30-295" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-296"><a href="#cb30-296" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR)</span>
<span id="cb30-297"><a href="#cb30-297" aria-hidden="true" tabindex="-1"></a>genes_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(filtered_dbs_loaded, <span class="cf">function</span>(x) x<span class="sc">$</span>genename)</span>
<span id="cb30-298"><a href="#cb30-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-299"><a href="#cb30-299" aria-hidden="true" tabindex="-1"></a><span class="fu">upset</span>(<span class="fu">fromList</span>(genes_list), <span class="at">order.by =</span> <span class="st">"freq"</span>, <span class="at">nsets =</span> <span class="fu">length</span>(genes_list))</span>
<span id="cb30-300"><a href="#cb30-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-301"><a href="#cb30-301" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-302"><a href="#cb30-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-303"><a href="#cb30-303" aria-hidden="true" tabindex="-1"></a><span class="fu"># Check for eGenes from Aracena et al.</span></span>
<span id="cb30-304"><a href="#cb30-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-305"><a href="#cb30-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-308"><a href="#cb30-308" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-309"><a href="#cb30-309" aria-hidden="true" tabindex="-1"></a><span class="co"># Load eGenes table</span></span>
<span id="cb30-310"><a href="#cb30-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-311"><a href="#cb30-311" aria-hidden="true" tabindex="-1"></a>path_to_egenes <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/QTL_mapping/eqtls.csv"</span></span>
<span id="cb30-312"><a href="#cb30-312" aria-hidden="true" tabindex="-1"></a>egenes_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_to_egenes, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb30-313"><a href="#cb30-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-314"><a href="#cb30-314" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(egenes_table<span class="sc">$</span>pvalue_Flu, <span class="at">main =</span> <span class="st">"eGenes pvalues"</span>)</span>
<span id="cb30-315"><a href="#cb30-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-316"><a href="#cb30-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-317"><a href="#cb30-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-318"><a href="#cb30-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-321"><a href="#cb30-321" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-322"><a href="#cb30-322" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(pred_summ)</span>
<span id="cb30-323"><a href="#cb30-323" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(egenes_table)</span>
<span id="cb30-324"><a href="#cb30-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-325"><a href="#cb30-325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-326"><a href="#cb30-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-327"><a href="#cb30-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-330"><a href="#cb30-330" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-331"><a href="#cb30-331" aria-hidden="true" tabindex="-1"></a>subset_to_egenes <span class="ot">&lt;-</span> <span class="fu">merge</span>(egenes_table, pred_summ, <span class="at">by.x =</span> <span class="st">"feature"</span>, <span class="at">by.y =</span> <span class="st">"gene"</span>)</span>
<span id="cb30-332"><a href="#cb30-332" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(subset_to_egenes)</span>
<span id="cb30-333"><a href="#cb30-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-334"><a href="#cb30-334" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(subset_to_egenes, <span class="fu">aes</span>(<span class="at">x =</span> pvalue_Flu, <span class="at">y =</span> zscore_pval)) <span class="sc">+</span> <span class="fu">geom_point</span>() </span>
<span id="cb30-335"><a href="#cb30-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-336"><a href="#cb30-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-337"><a href="#cb30-337" aria-hidden="true" tabindex="-1"></a>spearcor <span class="ot">&lt;-</span> <span class="fu">cor</span>(subset_to_egenes<span class="sc">$</span>pvalue_Flu, subset_to_egenes<span class="sc">$</span>zscore_pval, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb30-338"><a href="#cb30-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-339"><a href="#cb30-339" aria-hidden="true" tabindex="-1"></a>spearcor</span>
<span id="cb30-340"><a href="#cb30-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-341"><a href="#cb30-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-342"><a href="#cb30-342" aria-hidden="true" tabindex="-1"></a><span class="fu">### Load summary Enpact</span></span>
<span id="cb30-345"><a href="#cb30-345" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-346"><a href="#cb30-346" aria-hidden="true" tabindex="-1"></a>path_pred_summ <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/linearization/predixcan_RNAseq_Flu_from_HDF5_mean_log_revised/database/Model_summary.txt"</span></span>
<span id="cb30-347"><a href="#cb30-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-348"><a href="#cb30-348" aria-hidden="true" tabindex="-1"></a>pred_summ <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_pred_summ, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb30-349"><a href="#cb30-349" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ))</span>
<span id="cb30-350"><a href="#cb30-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-351"><a href="#cb30-351" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>rho_avg, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Enpact PredictDB"</span>)</span>
<span id="cb30-352"><a href="#cb30-352" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>zscore_pval, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Enpact PredictDB"</span>)</span>
<span id="cb30-353"><a href="#cb30-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-354"><a href="#cb30-354" aria-hidden="true" tabindex="-1"></a>pred_summ_filt <span class="ot">&lt;-</span> pred_summ[(pred_summ<span class="sc">$</span>rho_avg <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> (pred_summ<span class="sc">$</span>zscore_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span>
<span id="cb30-355"><a href="#cb30-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-356"><a href="#cb30-356" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ_filt))</span>
<span id="cb30-357"><a href="#cb30-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-358"><a href="#cb30-358" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-359"><a href="#cb30-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-360"><a href="#cb30-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-363"><a href="#cb30-363" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-364"><a href="#cb30-364" aria-hidden="true" tabindex="-1"></a>path_pred_summ <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/charles/project/singleXcanDL/PredicDB/OneK1K/PrediXcan/work_dir/results/database/Model_summary.txt"</span></span>
<span id="cb30-365"><a href="#cb30-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-366"><a href="#cb30-366" aria-hidden="true" tabindex="-1"></a>pred_summ <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_pred_summ, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb30-367"><a href="#cb30-367" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ))</span>
<span id="cb30-368"><a href="#cb30-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-369"><a href="#cb30-369" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>rho_avg, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Charles PredictDB"</span>)</span>
<span id="cb30-370"><a href="#cb30-370" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(pred_summ<span class="sc">$</span>zscore_pval, <span class="at">breaks =</span> <span class="dv">20</span>, <span class="at">main =</span> <span class="st">"Charles PredictDB"</span>)</span>
<span id="cb30-371"><a href="#cb30-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-372"><a href="#cb30-372" aria-hidden="true" tabindex="-1"></a>pred_summ_filt <span class="ot">&lt;-</span> pred_summ[(pred_summ<span class="sc">$</span>rho_avg <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> (pred_summ<span class="sc">$</span>zscore_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span>
<span id="cb30-373"><a href="#cb30-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-374"><a href="#cb30-374" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(pred_summ_filt))</span>
<span id="cb30-375"><a href="#cb30-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-376"><a href="#cb30-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-377"><a href="#cb30-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-378"><a href="#cb30-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-379"><a href="#cb30-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-380"><a href="#cb30-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-381"><a href="#cb30-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-382"><a href="#cb30-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-383"><a href="#cb30-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-384"><a href="#cb30-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-385"><a href="#cb30-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-386"><a href="#cb30-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-387"><a href="#cb30-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-388"><a href="#cb30-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-389"><a href="#cb30-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-390"><a href="#cb30-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-391"><a href="#cb30-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-392"><a href="#cb30-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-393"><a href="#cb30-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-394"><a href="#cb30-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-395"><a href="#cb30-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-396"><a href="#cb30-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-397"><a href="#cb30-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-398"><a href="#cb30-398" aria-hidden="true" tabindex="-1"></a><span class="fu">### Old code</span></span>
<span id="cb30-399"><a href="#cb30-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-402"><a href="#cb30-402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-403"><a href="#cb30-403" aria-hidden="true" tabindex="-1"></a><span class="co"># Create merged expression tables</span></span>
<span id="cb30-404"><a href="#cb30-404" aria-hidden="true" tabindex="-1"></a>merged_expression <span class="ot">&lt;-</span> <span class="fu">merge</span>(flu_onta_expression, flu_my_normalized_counts, <span class="at">by =</span> <span class="st">"row.names"</span>,)</span>
<span id="cb30-405"><a href="#cb30-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-406"><a href="#cb30-406" aria-hidden="true" tabindex="-1"></a>merged_expression_no_lnorm <span class="ot">&lt;-</span> <span class="fu">merge</span>(flu_onta_expression, flu_my_normalized_counts_no_lnorm, <span class="at">by =</span> <span class="st">"row.names"</span>,)</span>
<span id="cb30-407"><a href="#cb30-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-408"><a href="#cb30-408" aria-hidden="true" tabindex="-1"></a>merged_expression_no_voom <span class="ot">&lt;-</span> <span class="fu">merge</span>(flu_onta_expression, flu_my_normalized_counts_no_voom, <span class="at">by =</span> <span class="st">"row.names"</span>,)</span>
<span id="cb30-409"><a href="#cb30-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-410"><a href="#cb30-410" aria-hidden="true" tabindex="-1"></a>merged_expression_no_exp_pcs <span class="ot">&lt;-</span> <span class="fu">merge</span>(flu_onta_expression, flu_my_normalized_counts_no_exp_pcs, <span class="at">by =</span> <span class="st">"row.names"</span>,)</span>
<span id="cb30-411"><a href="#cb30-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-412"><a href="#cb30-412" aria-hidden="true" tabindex="-1"></a>merged_expression_10k <span class="ot">&lt;-</span> <span class="fu">merge</span>(flu_onta_expression, flu_my_normalized_counts_10k, <span class="at">by =</span> <span class="st">"row.names"</span>,)</span>
<span id="cb30-413"><a href="#cb30-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-414"><a href="#cb30-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-415"><a href="#cb30-415" aria-hidden="true" tabindex="-1"></a>merged_expression_full_10k <span class="ot">&lt;-</span> <span class="fu">merge</span>(flu_my_normalized_counts, flu_my_normalized_counts_10k, <span class="at">by =</span> <span class="st">"row.names"</span>,)</span>
<span id="cb30-416"><a href="#cb30-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-417"><a href="#cb30-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-418"><a href="#cb30-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-419"><a href="#cb30-419" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">colnames</span>(merged_expression))</span>
<span id="cb30-420"><a href="#cb30-420" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression))</span>
<span id="cb30-421"><a href="#cb30-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-422"><a href="#cb30-422" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_expression) <span class="ot">&lt;-</span> merged_expression<span class="sc">$</span>Row.names</span>
<span id="cb30-423"><a href="#cb30-423" aria-hidden="true" tabindex="-1"></a>merged_expression <span class="ot">&lt;-</span> merged_expression[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-424"><a href="#cb30-424" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression))</span>
<span id="cb30-425"><a href="#cb30-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-426"><a href="#cb30-426" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_expression_no_lnorm) <span class="ot">&lt;-</span> merged_expression_no_lnorm<span class="sc">$</span>Row.names</span>
<span id="cb30-427"><a href="#cb30-427" aria-hidden="true" tabindex="-1"></a>merged_expression_no_lnorm <span class="ot">&lt;-</span> merged_expression_no_lnorm[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-428"><a href="#cb30-428" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression_no_lnorm))</span>
<span id="cb30-429"><a href="#cb30-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-430"><a href="#cb30-430" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_expression_no_voom) <span class="ot">&lt;-</span> merged_expression_no_voom<span class="sc">$</span>Row.names</span>
<span id="cb30-431"><a href="#cb30-431" aria-hidden="true" tabindex="-1"></a>merged_expression_no_voom <span class="ot">&lt;-</span> merged_expression_no_voom[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-432"><a href="#cb30-432" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression_no_voom))</span>
<span id="cb30-433"><a href="#cb30-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-434"><a href="#cb30-434" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_expression_no_exp_pcs) <span class="ot">&lt;-</span> merged_expression_no_exp_pcs<span class="sc">$</span>Row.names</span>
<span id="cb30-435"><a href="#cb30-435" aria-hidden="true" tabindex="-1"></a>merged_expression_no_exp_pcs <span class="ot">&lt;-</span> merged_expression_no_exp_pcs[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-436"><a href="#cb30-436" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression_no_exp_pcs))</span>
<span id="cb30-437"><a href="#cb30-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-438"><a href="#cb30-438" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_expression_10k) <span class="ot">&lt;-</span> merged_expression_10k<span class="sc">$</span>Row.names</span>
<span id="cb30-439"><a href="#cb30-439" aria-hidden="true" tabindex="-1"></a>merged_expression_10k <span class="ot">&lt;-</span> merged_expression_10k[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-440"><a href="#cb30-440" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression_10k))</span>
<span id="cb30-441"><a href="#cb30-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-442"><a href="#cb30-442" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(merged_expression_full_10k) <span class="ot">&lt;-</span> merged_expression_full_10k<span class="sc">$</span>Row.names</span>
<span id="cb30-443"><a href="#cb30-443" aria-hidden="true" tabindex="-1"></a>merged_expression_full_10k <span class="ot">&lt;-</span> merged_expression_full_10k[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb30-444"><a href="#cb30-444" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(merged_expression_full_10k))</span>
<span id="cb30-445"><a href="#cb30-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-446"><a href="#cb30-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-447"><a href="#cb30-447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-448"><a href="#cb30-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-451"><a href="#cb30-451" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-452"><a href="#cb30-452" aria-hidden="true" tabindex="-1"></a>inds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AF04"</span>, <span class="st">"AF16"</span>)</span>
<span id="cb30-453"><a href="#cb30-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-454"><a href="#cb30-454" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ind <span class="cf">in</span> inds) {</span>
<span id="cb30-455"><a href="#cb30-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-456"><a href="#cb30-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(merged_expression[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.x"</span>)], merged_expression[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.y"</span>)], <span class="at">main =</span> <span class="fu">paste0</span>(ind, <span class="st">"Flu expression comparison, full normalization</span></span>
<span id="cb30-457"><a href="#cb30-457" aria-hidden="true" tabindex="-1"></a><span class="st">  "</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-458"><a href="#cb30-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-459"><a href="#cb30-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(merged_expression_no_lnorm[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.x"</span>)], merged_expression_no_lnorm[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.y"</span>)], <span class="at">main =</span> <span class="fu">paste0</span>(ind, <span class="st">"Flu expression comparison, no length normalization)"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-460"><a href="#cb30-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-461"><a href="#cb30-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(merged_expression_no_voom[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.x"</span>)], merged_expression_no_voom[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.y"</span>)], <span class="at">main =</span> <span class="fu">paste0</span>(ind, <span class="st">"Flu expression comparison, no voom)"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-462"><a href="#cb30-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-463"><a href="#cb30-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(merged_expression_no_exp_pcs[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.x"</span>)], merged_expression_no_exp_pcs[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.y"</span>)], <span class="at">main =</span> <span class="fu">paste0</span>(ind, <span class="st">"Flu expression comparison, no expression pcs)"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-464"><a href="#cb30-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-465"><a href="#cb30-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(merged_expression_10k[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.x"</span>)], merged_expression_10k[,<span class="fu">paste0</span>(ind,<span class="st">"_Flu.y"</span>)], <span class="at">main =</span> <span class="fu">paste0</span>(ind, <span class="st">"Flu expression comparison, 10k scaling)"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-466"><a href="#cb30-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-467"><a href="#cb30-467" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-468"><a href="#cb30-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-469"><a href="#cb30-469" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-470"><a href="#cb30-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-471"><a href="#cb30-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-474"><a href="#cb30-474" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-475"><a href="#cb30-475" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb30-476"><a href="#cb30-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-477"><a href="#cb30-477" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(merged_expression, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of expression data (full normalization)"</span>)</span>
<span id="cb30-478"><a href="#cb30-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-479"><a href="#cb30-479" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(merged_expression_no_lnorm, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of expression data (no length normalization)"</span>)</span>
<span id="cb30-480"><a href="#cb30-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-481"><a href="#cb30-481" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(merged_expression_no_voom, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of expression data (no voom)"</span>)</span>
<span id="cb30-482"><a href="#cb30-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-483"><a href="#cb30-483" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(merged_expression_no_exp_pcs, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of expression data (no expression pcs)"</span>)</span>
<span id="cb30-484"><a href="#cb30-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-485"><a href="#cb30-485" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(merged_expression_10k, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of expression data (scaled length norm 10k)"</span>)</span>
<span id="cb30-486"><a href="#cb30-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-487"><a href="#cb30-487" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(<span class="fu">cor</span>(merged_expression_full_10k, <span class="at">method=</span><span class="st">"spearman"</span>), <span class="at">method =</span> <span class="st">"color"</span>, <span class="at">main =</span> <span class="st">"Correlation of expression data b/w 10k scaling and full normalization"</span>)</span>
<span id="cb30-488"><a href="#cb30-488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-489"><a href="#cb30-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-490"><a href="#cb30-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-493"><a href="#cb30-493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-494"><a href="#cb30-494" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ENSG00000121716"</span>, <span class="st">"ENSG00000256274"</span>, <span class="st">"ENSG00000166435"</span>)</span>
<span id="cb30-495"><a href="#cb30-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-496"><a href="#cb30-496" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> genes) {</span>
<span id="cb30-497"><a href="#cb30-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-498"><a href="#cb30-498" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">as.numeric</span>(flu_onta_expression[gene,]), <span class="fu">as.numeric</span>(flu_my_normalized_counts[gene,]), <span class="at">main =</span> <span class="fu">paste0</span>(gene, <span class="st">" expression comparison"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-499"><a href="#cb30-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-500"><a href="#cb30-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">as.numeric</span>(flu_onta_expression[gene,]), <span class="fu">as.numeric</span>(flu_my_normalized_counts_no_lnorm[gene,]), <span class="at">main =</span> <span class="fu">paste0</span>(gene, <span class="st">" expression comparison, no length normalization"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-501"><a href="#cb30-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-502"><a href="#cb30-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">as.numeric</span>(flu_onta_expression[gene,]), <span class="fu">as.numeric</span>(flu_my_normalized_counts_no_voom[gene,]), <span class="at">main =</span> <span class="fu">paste0</span>(gene, <span class="st">" expression comparison, no voom"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-503"><a href="#cb30-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-504"><a href="#cb30-504" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">as.numeric</span>(flu_my_normalized_counts[gene,]), <span class="fu">as.numeric</span>(flu_my_normalized_counts_no_exp_pcs[gene,]), <span class="at">main =</span> <span class="fu">paste0</span>(gene, <span class="st">" expression comparison, no exp pcs"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-505"><a href="#cb30-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-506"><a href="#cb30-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">as.numeric</span>(flu_my_normalized_counts[gene,]), <span class="fu">as.numeric</span>(flu_my_normalized_counts_10k[gene,]), <span class="at">main =</span> <span class="fu">paste0</span>(gene, <span class="st">" expression comparison, 10k scaling"</span>), <span class="at">xlab =</span> <span class="st">"Onta"</span>, <span class="at">ylab =</span> <span class="st">"Mine"</span>)</span>
<span id="cb30-507"><a href="#cb30-507" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-508"><a href="#cb30-508" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-509"><a href="#cb30-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-510"><a href="#cb30-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-511"><a href="#cb30-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-512"><a href="#cb30-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-513"><a href="#cb30-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-514"><a href="#cb30-514" aria-hidden="true" tabindex="-1"></a><span class="fu">### Check if length norm lower than 1 is causing the issue</span></span>
<span id="cb30-515"><a href="#cb30-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-516"><a href="#cb30-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-519"><a href="#cb30-519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-520"><a href="#cb30-520" aria-hidden="true" tabindex="-1"></a><span class="co"># Load my normalized counts</span></span>
<span id="cb30-521"><a href="#cb30-521" aria-hidden="true" tabindex="-1"></a>path_to_normalized_counts <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_expression_3_7.txt"</span></span>
<span id="cb30-522"><a href="#cb30-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-523"><a href="#cb30-523" aria-hidden="true" tabindex="-1"></a>my_normalized_counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_to_normalized_counts, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb30-524"><a href="#cb30-524" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(my_normalized_counts))</span>
<span id="cb30-525"><a href="#cb30-525" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">colnames</span>(my_normalized_counts))</span>
<span id="cb30-526"><a href="#cb30-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-527"><a href="#cb30-527" aria-hidden="true" tabindex="-1"></a><span class="co"># Load my counts with scaled counts</span></span>
<span id="cb30-528"><a href="#cb30-528" aria-hidden="true" tabindex="-1"></a>path_to_normalized_counts_10k <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/fully_preprocessed_expression_3_7_1milscale.txt"</span></span>
<span id="cb30-529"><a href="#cb30-529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-530"><a href="#cb30-530" aria-hidden="true" tabindex="-1"></a>my_normalized_counts_10k <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_to_normalized_counts_10k, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb30-531"><a href="#cb30-531" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(my_normalized_counts_10k))</span>
<span id="cb30-532"><a href="#cb30-532" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">colnames</span>(my_normalized_counts_10k))</span>
<span id="cb30-533"><a href="#cb30-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-534"><a href="#cb30-534" aria-hidden="true" tabindex="-1"></a><span class="co"># Load only length normalized counts</span></span>
<span id="cb30-535"><a href="#cb30-535" aria-hidden="true" tabindex="-1"></a>path_to_normalized_counts_lnorm <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/Inputs/counts_matrices/RNAseq_filtered.counts_LENGTH_NORM.txt"</span></span>
<span id="cb30-536"><a href="#cb30-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-537"><a href="#cb30-537" aria-hidden="true" tabindex="-1"></a>my_normalized_counts_lnorm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(path_to_normalized_counts_lnorm, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb30-538"><a href="#cb30-538" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(my_normalized_counts_lnorm))</span>
<span id="cb30-539"><a href="#cb30-539" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">colnames</span>(my_normalized_counts_lnorm))</span>
<span id="cb30-540"><a href="#cb30-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-541"><a href="#cb30-541" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-542"><a href="#cb30-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-543"><a href="#cb30-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-546"><a href="#cb30-546" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-547"><a href="#cb30-547" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(my_normalized_counts_10k[,"AF04_Flu"], my_normalized_counts_lnorm[,"AF04_Flu"], main = "AF04 Flu expression comparison, 10k scaling vs. length norm", xlab = "10k scaling", ylab = "Length norm")</span></span>
<span id="cb30-548"><a href="#cb30-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-549"><a href="#cb30-549" aria-hidden="true" tabindex="-1"></a>AF04_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">onta=</span> onta_expression[,<span class="st">"AF04_Flu"</span>],<span class="at">full_norm=</span>my_normalized_counts[,<span class="st">"AF04_Flu"</span>],<span class="at">scaled=</span>my_normalized_counts_10k[,<span class="st">"AF04_Flu"</span>], <span class="at">lnorm=</span>my_normalized_counts_lnorm[,<span class="st">"AF04_Flu"</span>], <span class="at">lnorm_less_one=</span>my_normalized_counts_lnorm[,<span class="st">"AF04_Flu"</span>] <span class="sc">&lt;</span> <span class="dv">1</span>,</span>
<span id="cb30-550"><a href="#cb30-550" aria-hidden="true" tabindex="-1"></a><span class="at">lnorm_less_half=</span>my_normalized_counts_lnorm[,<span class="st">"AF04_Flu"</span>] <span class="sc">&lt;</span> <span class="fl">0.5</span></span>
<span id="cb30-551"><a href="#cb30-551" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-552"><a href="#cb30-552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-553"><a href="#cb30-553" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(AF04_df, <span class="fu">aes</span>(<span class="at">x=</span>full_norm, <span class="at">y=</span>scaled, <span class="at">color=</span>lnorm_less_one)) <span class="sc">+</span> <span class="fu">geom_point</span>()</span>
<span id="cb30-554"><a href="#cb30-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-555"><a href="#cb30-555" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(AF04_df, <span class="fu">aes</span>(<span class="at">x=</span>full_norm, <span class="at">y=</span>lnorm, <span class="at">color=</span>lnorm_less_half)) <span class="sc">+</span> <span class="fu">geom_point</span>()</span>
<span id="cb30-556"><a href="#cb30-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-557"><a href="#cb30-557" aria-hidden="true" tabindex="-1"></a>my_normalized_counts_lnorm[my_normalized_counts_lnorm[,<span class="st">"AF04_Flu"</span>] <span class="sc">&lt;</span> <span class="dv">1</span>,][,<span class="st">"AF04_Flu"</span>]</span>
<span id="cb30-558"><a href="#cb30-558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>