<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">

<title>Daily-Blog-Sai - Portability analysis of Enformer gene expression predictions between populations</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-datasets-and-metadata" id="toc-load-datasets-and-metadata" class="nav-link active" data-scroll-target="#load-datasets-and-metadata">Load datasets and metadata</a></li>
  <li><a href="#compute-gene-individual-set-intersections-between-enformer-predixcan-and-geuvadis-datasets-for-cross-comparison" id="toc-compute-gene-individual-set-intersections-between-enformer-predixcan-and-geuvadis-datasets-for-cross-comparison" class="nav-link" data-scroll-target="#compute-gene-individual-set-intersections-between-enformer-predixcan-and-geuvadis-datasets-for-cross-comparison">Compute gene, individual set intersections between Enformer, Predixcan, and Geuvadis datasets for cross-comparison</a></li>
  <li><a href="#assess-correlation-across-individuals" id="toc-assess-correlation-across-individuals" class="nav-link" data-scroll-target="#assess-correlation-across-individuals">Assess correlation across individuals</a></li>
  <li><a href="#portability" id="toc-portability" class="nav-link" data-scroll-target="#portability">Portability</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Portability analysis of Enformer gene expression predictions between populations</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpmisc)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggExtra)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(preprocessCore)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-datasets-and-metadata" class="level3">
<h3 class="anchored" data-anchor-id="load-datasets-and-metadata">Load datasets and metadata</h3>
<p>Load the raw expression datasets</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_canonical_haplo1_filtered.csv"</span>)))</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(enformer_haplo1) &lt;- enformer_haplo1$X</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_haplo1$X &lt;- NULL</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>enformer_haplo2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_canonical_haplo2_filtered.csv"</span>)))</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo2) <span class="ot">&lt;-</span> enformer_haplo2<span class="sc">$</span>X</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>enformer_haplo2<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> (enformer_haplo1 <span class="sc">+</span> enformer_haplo2)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Load corresponding metadata files to help with analysis</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compute-gene-individual-set-intersections-between-enformer-predixcan-and-geuvadis-datasets-for-cross-comparison" class="level2">
<h2 class="anchored" data-anchor-id="compute-gene-individual-set-intersections-between-enformer-predixcan-and-geuvadis-datasets-for-cross-comparison">Compute gene, individual set intersections between Enformer, Predixcan, and Geuvadis datasets for cross-comparison</h2>
<p>The given datasets may not be consistent in terms of the sets of individuals and genes they contain. We can only correlate and compare the ones in common, so create those intersections here.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Make common gene, individual subsets between datasets</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer &lt;- enformer_no_zero</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># geuvadis &lt;- geuvadis</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># predixcan &lt;- predixcan</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="co"># USING IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="co"># ge_genes_not_predixcan &lt;- ge_genes[!(ge_genes %in% rownames(predixcan))]</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a><span class="co"># geuvadis_compare &lt;- geuvadis[ge_genes_not_predixcan, all_three_ind]</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="co"># predixcan_compare &lt;- predixcan[ge_genes_not_predixcan,all_three_ind]</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="co"># predixcan_compare[is.na(predixcan_compare)] &lt;- 1e-5</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_compare &lt;- data.frame(enformer[ge_genes_not_predixcan, all_three_ind])</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co"># # </span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="co"># # </span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a><span class="co"># all_three_genes &lt;- ge_genes_not_predixcan</span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="co"># log values</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="co"># geuvadis_compare &lt;- log(geuvadis_compare+1)</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a><span class="co"># predixcan_compare &lt;- log(predixcan_compare+1)</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_compare &lt;- log(enformer_compare+1)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can explore the population breakdown of the geuvadis samples</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(individual_metadata_compare) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>Superpopulation.code, <span class="at">fill=</span>Population.code)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(individual_metadata_compare) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>Superpopulation.name, <span class="at">fill=</span>Population.name)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Lot’s plot ERAP2 Enformer vs.&nbsp;Geuvadis as a check</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># allele_count_df &lt;- as.data.frame(allele_count_df)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># splice_geno &lt;- c()</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># for (ind in all_three_ind) {</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   if (ind %in% colnames(allele_count_df)) {</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     splice_geno &lt;- c(splice_geno, allele_count_df["96900192",ind])</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   } else {</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     splice_geno &lt;- c(splice_geno, NA)</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># erap2_plot &lt;- data.frame(geuvadis=as.numeric(geuvadis_compare["ENSG00000164308",]), enformer=as.numeric(enformer_compare["ENSG00000164308",]), splice_geno=splice_geno)</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(erap2_plot) + geom_point(aes(x=geuvadis,y=enformer, col = splice_geno))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="assess-correlation-across-individuals" class="level2">
<h2 class="anchored" data-anchor-id="assess-correlation-across-individuals">Assess correlation across individuals</h2>
<p>For each gene, compute the cross-individual correlation between geuvadis and enformer, as well as between geuvadis and predixcan. Store per gene values as rows in a table.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>?@fig-prediction-heritability</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-prediction_heritability-1</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Plot of prediction correlations against heritability per gene"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-subcap: </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#| - "Predixcan predictions"</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#| - "Enformer predictions"</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 5</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>gene_h2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>gene_h2), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"heritability"</span>) <span class="sc">+</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p1, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>ge_abs,<span class="at">y=</span>gene_h2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>ge_abs,<span class="at">y=</span>gene_h2), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. enformer"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"heritability"</span>) <span class="sc">+</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p2, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>?@fig-prediction-heritability-2</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-prediction_heritability-2</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Plot of prediction correlations against heritability per gene"</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-subcap: </span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#| - "Predixcan predictions"</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#| - "Enformer predictions"</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 5</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>gene_h2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>gene_h2), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"gene heritability"</span>) <span class="sc">+</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p1, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>ge,<span class="at">y=</span>gene_h2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>ge_abs,<span class="at">y=</span>gene_h2), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"gene heritability"</span>) <span class="sc">+</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p2, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><a href="#fig-all_popcorrelations">Figure&nbsp;1</a></p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge, <span class="at">colour=</span>gene_h2), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p1, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge, <span class="at">colour=</span>strand), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p2, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>, <span class="at">groupFill =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge, <span class="at">colour=</span>ref_gene_ancestry), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p3, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>, <span class="at">groupFill =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-all_popcorrelations" class="cell quarto-layout-panel" data-layour-nrow="1">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 33.3%;justify-content: center;">
<div id="fig-all_popcorrelations" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-all_popcorrelations-1.png" class="img-fluid figure-img" data-ref-parent="fig-all_popcorrelations" width="960"></p>
<figcaption class="figure-caption">(a) Heritability</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Comparison between enformer and predixcan predictions against GEUVADIS expression. Each point corresponds to a single gene with correlations being done across individuals.</figcaption><p></p>
</figure>
</div>
</div>
<p>#TODO split by ancestry</p>
<p><a href="#fig-all_popcorrelations_absolute_val">Figure&nbsp;2</a></p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge_abs)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge_abs, <span class="at">colour=</span>gene_h2), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge_abs,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>) <span class="sc">+</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p1, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge_abs)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge_abs, <span class="at">colour=</span>strand), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>) <span class="sc">+</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(p2, <span class="at">type =</span> <span class="st">"histogram"</span>, <span class="at">binwidth=</span><span class="fl">0.025</span>, <span class="at">groupFill =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-all_popcorrelations_absolute_val" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-all_popcorrelations_absolute_val" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-all_popcorrelations_absolute_val-1.png" class="img-fluid figure-img" data-ref-parent="fig-all_popcorrelations_absolute_val" width="960"></p>
<figcaption class="figure-caption">(a) Heritability</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: Comparison between enformer and predixcan predictions against GEUVADIS expression. Each point corresponds to a single gene with correlations being done across individuals. In this case we the absolute value of the Enformer correlation is used.</figcaption><p></p>
</figure>
</div>
</div>
<p>Repeat the correlations but only for EUR individuals.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| EUR comparisons </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>EUR_inds_all <span class="ot">&lt;-</span> individual_metadata_compare[individual_metadata_compare<span class="sc">$</span>Superpopulation.code <span class="sc">==</span> <span class="st">"EUR"</span>,]<span class="sc">$</span>Sample.name</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>EUR_ind <span class="ot">&lt;-</span> EUR_inds_all[EUR_inds_all <span class="sc">%in%</span> <span class="fu">colnames</span>(geuvadis_compare)]</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>EUR_geuvadis_compare <span class="ot">&lt;-</span> geuvadis_compare[,<span class="fu">colnames</span>(geuvadis_compare) <span class="sc">%in%</span> EUR_ind]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>EUR_predixcan_compare <span class="ot">&lt;-</span> predixcan_compare[,<span class="fu">colnames</span>(predixcan_compare) <span class="sc">%in%</span> EUR_ind]</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>EUR_enformer_compare <span class="ot">&lt;-</span> enformer_compare[,<span class="fu">colnames</span>(enformer_compare) <span class="sc">%in%</span> EUR_ind]</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlation across genes</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># gp_corrs &lt;- corr_ind_across_genes(data.frame(t(EUR_geuvadis_compare)), data.frame(t(EUR_predixcan_compare)))</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(EUR_enformer_compare)) {</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  ge_corrs <span class="ot">&lt;-</span> <span class="fu">c</span>(ge_corrs, <span class="fu">cor</span>(EUR_geuvadis_compare[,i], EUR_enformer_compare[,i]))</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlation across individuals</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(EUR_geuvadis_compare, EUR_predixcan_compare)</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>gp_corrs_EUR <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(EUR_geuvadis_compare, EUR_enformer_compare)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>ge_corrs_EUR <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">genes=</span>all_three_genes , <span class="at">gene_names=</span>gene_meta_compare<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><a href="#fig-EUR_correlations">Figure&nbsp;3</a></p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df, <span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"EUR: Correlation of each gene across all individuals b/w paired datasets"</span>) <span class="sc">+</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>)<span class="sc">+</span><span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>))<span class="sc">+</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df, <span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_hex</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"EUR: Correlation of each gene across all individuals b/w paired datasets"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>)<span class="sc">+</span><span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>))<span class="sc">+</span> </span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-EUR_correlations" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-EUR_correlations-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-EUR_correlations-1.png" class="img-fluid figure-img" data-ref-parent="fig-EUR_correlations" width="480"></p>
<figcaption class="figure-caption">(a) Scatterplot</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-EUR_correlations-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-EUR_correlations-2.png" class="img-fluid figure-img" data-ref-parent="fig-EUR_correlations" width="480"></p>
<figcaption class="figure-caption">(b) Density</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: Comparison between enformer and predixcan predictions against GEUVADIS expression among EUR individuals. Each point corresponds to a single gene with correlations being done across individuals.</figcaption><p></p>
</figure>
</div>
</div>
<p>Repeat the correlations but only for AFR individuals.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>AFR_inds_all <span class="ot">&lt;-</span> individual_metadata_compare[individual_metadata_compare<span class="sc">$</span>Superpopulation.code <span class="sc">==</span> <span class="st">"AFR"</span>,]<span class="sc">$</span>Sample.name</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>AFR_ind <span class="ot">&lt;-</span> AFR_inds_all[AFR_inds_all <span class="sc">%in%</span> <span class="fu">colnames</span>(geuvadis_compare)]</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>AFR_geuvadis_compare <span class="ot">&lt;-</span> geuvadis_compare[,<span class="fu">colnames</span>(geuvadis_compare) <span class="sc">%in%</span> AFR_ind]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>AFR_predixcan_compare <span class="ot">&lt;-</span> predixcan_compare[,<span class="fu">colnames</span>(predixcan_compare) <span class="sc">%in%</span> AFR_ind]</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>AFR_enformer_compare <span class="ot">&lt;-</span> enformer_compare[,<span class="fu">colnames</span>(enformer_compare) <span class="sc">%in%</span> AFR_ind]</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlation across individuals</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(AFR_geuvadis_compare, AFR_predixcan_compare)</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>gp_corrs_AFR <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(AFR_geuvadis_compare, AFR_enformer_compare)</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>ge_corrs_AFR <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span>gene_meta_compare<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df, <span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge,<span class="at">label=</span>gene_names)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"AFR: Correlation of each gene across all individuals b/w paired datasets"</span>) <span class="sc">+</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>)<span class="sc">+</span><span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df, <span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_hex</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"AFR: Correlation of each gene across all individuals b/w paired datasets"</span>) <span class="sc">+</span> </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer"</span>)<span class="sc">+</span><span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-AFR_correlations" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-AFR_correlations-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-AFR_correlations-1.png" class="img-fluid figure-img" data-ref-parent="fig-AFR_correlations" width="480"></p>
<figcaption class="figure-caption">(a) Scatterplot</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-AFR_correlations-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-AFR_correlations-2.png" class="img-fluid figure-img" data-ref-parent="fig-AFR_correlations" width="480"></p>
<figcaption class="figure-caption">(b) Density</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;4: Comparison between enformer and predixcan predictions against GEUVADIS expression among AFR individuals. Each point corresponds to a single gene with correlations being done across individuals.</figcaption><p></p>
</figure>
</div>
</div>
<p>It seems that among population subsets, there are broadly similar correlation patterns.Let’s investigate further:</p>
</section>
<section id="portability" class="level2">
<h2 class="anchored" data-anchor-id="portability">Portability</h2>
<p>Ultimately, we would like to know how well the different prediction methods have different relative performance between population groups.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>ge_corrs_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">EUR=</span>ge_corrs_EUR, <span class="at">AFR=</span>ge_corrs_AFR)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>xlab_str <span class="ot">=</span> <span class="st">"geuvadis vs predixcan"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>ylab_str <span class="ot">=</span> <span class="st">"geuvadis vs enformer"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Comment for using null distribution</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co"># mm=2900</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co"># nn=60</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co"># gp_corrs_EUR = tanh(rnorm(mm)/sqrt(nn-3))</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co"># gp_corrs_AFR = tanh(rnorm(mm)/sqrt(nn-3))</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co"># xlab_str = ""</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(gp_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, ge_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"EUR: Predixcan vs enformer"</span>, <span class="at">xlab =</span> <span class="st">"geuvadis vs predixcan correlation across individuals"</span>, <span class="at">ylab =</span> <span class="st">"geuvadis vs. enformer correlation across individuals"</span>, <span class="at">col=</span><span class="st">"blue"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(gp_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"AFR: predixcan vs. enformer"</span>, <span class="at">xlab =</span> <span class="st">"geuvadis vs predixcan correlation across individuals"</span>, <span class="at">ylab =</span> <span class="st">"geuvadis vs. enformer correlation across individuals"</span>, <span class="at">col =</span> <span class="st">"red"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-between_predictor_qqplots" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_predictor_qqplots-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_predictor_qqplots-1.png" class="img-fluid figure-img" data-ref-parent="fig-between_predictor_qqplots" width="576"></p>
<figcaption class="figure-caption">(a) EUR</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_predictor_qqplots-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_predictor_qqplots-2.png" class="img-fluid figure-img" data-ref-parent="fig-between_predictor_qqplots" width="576"></p>
<figcaption class="figure-caption">(b) AFR</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;5: QQPlots of R squared distribution between Predixcan, Enformer correlated against Geuvadis.</figcaption><p></p>
</figure>
</div>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(ge_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"enformer: EUR vs. AFR"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>) </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ge_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>); <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ge_corrs_EUR, ge_corrs_AFR)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(gp_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, gp_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"predixcan: EUR vs. AFR"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(gp_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, gp_corrs_AFR<span class="sc">^</span><span class="dv">2</span>); <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-between_pop_qqplots" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots-1.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots" width="384"></p>
<figcaption class="figure-caption">(a) Enformer</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots-2.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots" width="384"></p>
<figcaption class="figure-caption">(b) Predixcan</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots-3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots-3.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots" width="384"></p>
<figcaption class="figure-caption">(c) Enformer</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots-4" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots-4.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots" width="384"></p>
<figcaption class="figure-caption">(d) Predixcan</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots-5" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots-5.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots" width="384"></p>
<figcaption class="figure-caption">(e) Enformer</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;6: QQPlots of R squared distribution between Predixcan and Enformer prediction for EUR and</figcaption><p></p>
</figure>
</div>
</div>
<p>We now compare the portability of Enformer vs.&nbsp;Predixcan between populations.</p>
<p>More formally, we define portability here as:</p>
<p><span class="math display">\[
P_{g} = \frac{|r_Y| + \epsilon}{|r_C| + \epsilon}
\]</span> Where <span class="math inline">\(P_{g}\)</span> is the portability for gene <span class="math inline">\(g\)</span>, <span class="math inline">\(r_Y,r_C\)</span> are the correlations of gene expression for gene <span class="math inline">\(g\)</span> across AFR, EUR individuals respectively. <span class="math inline">\(\epsilon = 0.1\)</span> here for numerical stability.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>epsi<span class="ot">=</span><span class="fl">0.1</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>portability_Enformer <span class="ot">=</span> ( <span class="fu">abs</span>(ge_corrs_AFR) <span class="sc">+</span> epsi )  <span class="sc">/</span> ( <span class="fu">abs</span>(ge_corrs_EUR)  <span class="sc">+</span> epsi )</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>portability_Predixcan <span class="ot">=</span> ( <span class="fu">abs</span>(gp_corrs_AFR) <span class="sc">+</span> epsi )<span class="sc">/</span> ( <span class="fu">abs</span>(gp_corrs_EUR) <span class="sc">+</span> epsi )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can now plot the portability statistics as a qqplot.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>portability_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">portability_Predixcan =</span> portability_Predixcan, <span class="at">portability_Enformer =</span> portability_Enformer, <span class="at">abs_log2_portability_Predixcan =</span> <span class="fu">abs</span>(<span class="fu">log2</span>(portability_Predixcan)), <span class="at">abs_log2_portability_Enformer =</span> <span class="fu">abs</span>(<span class="fu">log2</span>(portability_Enformer)), <span class="at">log2_portability_Predixcan =</span> <span class="fu">log2</span>(portability_Predixcan), <span class="at">log2_portability_Enformer =</span> <span class="fu">log2</span>(portability_Enformer), <span class="at">ge_corrs_AFR =</span> ge_corrs_AFR, <span class="at">gp_corrs_AFR=</span>gp_corrs_AFR, <span class="at">ge_corrs_EUR =</span> ge_corrs_EUR, <span class="at">gp_corrs_EUR=</span>gp_corrs_EUR, <span class="at">ge_corrs_all =</span> ge_corrs_all, gp_corrs_all)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(portability_Predixcan,portability_Enformer, <span class="at">main=</span><span class="st">"Portability Enformer vs. Predixcan"</span>,<span class="at">xlab =</span> <span class="st">"Predixcan Portability"</span>, <span class="at">ylab =</span> <span class="st">"Enformer portability"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(<span class="fu">log2</span>(portability_Predixcan),<span class="fu">log2</span>(portability_Enformer), <span class="at">main=</span><span class="st">"Portability Enformer vs. Predixcan absolute log2"</span>,<span class="at">xlab =</span> <span class="st">"Predixcan log2(Portability)"</span>, <span class="at">ylab =</span> <span class="st">"Enformer log2(portability)"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(<span class="fu">abs</span>(<span class="fu">log2</span>(portability_Predixcan)),<span class="fu">abs</span>(<span class="fu">log2</span>(portability_Enformer)), <span class="at">main=</span><span class="st">"Portability Enformer vs. Predixcan absolute log2"</span>,<span class="at">xlab =</span> <span class="st">"Predixcan abs(log2(Portability))"</span>, <span class="at">ylab =</span> <span class="st">"Enformer abs(log2(portability))"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-3.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>(portability_Predixcan),<span class="fu">log2</span>(portability_Enformer))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-4.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(<span class="fu">log2</span>(portability_Predixcan))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.04259071</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(<span class="fu">log2</span>(portability_Predixcan))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.02636951</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(<span class="fu">log2</span>(portability_Enformer), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1096916</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(<span class="fu">log2</span>(portability_Enformer), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1032937</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>portability_df_na <span class="ot">&lt;-</span> portability_df[<span class="sc">!</span><span class="fu">is.na</span>(portability_df<span class="sc">$</span>ge_corrs_all),]</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>mean_predixcan <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>mean_enformer <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>num_genes <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fl">0.5</span>,<span class="fl">0.95</span>,<span class="fl">0.05</span>)) {</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  portability_cutoff_gp <span class="ot">&lt;-</span> portability_df_na[portability_df_na<span class="sc">$</span>gp_corrs_all <span class="sc">&gt;</span> x <span class="sc">&amp;</span> portability_df_na<span class="sc">$</span>gp_corrs_all <span class="sc">&lt;</span> (x<span class="fl">+0.05</span>), ]</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  mean_predixcan <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_predixcan,<span class="fu">mean</span>(portability_cutoff_gp<span class="sc">$</span>log2_portability_Predixcan))</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  mean_enformer <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_enformer,<span class="fu">mean</span>(portability_cutoff_gp<span class="sc">$</span>log2_portability_Enformer, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  num_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes, <span class="fu">nrow</span>(portability_cutoff_gp))</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>means_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mean_predixcan=</span>mean_predixcan,<span class="at">mean_enformer=</span>mean_enformer, <span class="at">dif=</span>mean_predixcan<span class="sc">-</span>mean_enformer,<span class="at">cutoffs=</span><span class="fu">seq</span>(<span class="sc">-</span><span class="fl">0.5</span>,<span class="fl">0.95</span>,<span class="fl">0.05</span>), <span class="at">num_genes=</span>num_genes)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>means_df <span class="ot">&lt;-</span> means_df[<span class="sc">!</span><span class="fu">is.na</span>(means_df<span class="sc">$</span>mean_predixcan),]</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>means_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(means_df, <span class="at">measure.vars =</span> <span class="fu">c</span>(<span class="st">"mean_predixcan"</span>,<span class="st">"mean_enformer"</span>))</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(means_df_melt) <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>cutoffs,<span class="at">y=</span>value,<span class="at">color=</span>variable))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-5.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(means_df) <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>cutoffs,<span class="at">y=</span>num_genes))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-6.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>portability_df_na_sorted <span class="ot">&lt;-</span> portability_df_na[<span class="fu">order</span>(portability_df_na<span class="sc">$</span>gp_corrs_all),] </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>portability_df_na_sorted<span class="sc">$</span>order <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="fu">nrow</span>(portability_df_na_sorted)) </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>portability_df_na_sorted<span class="sc">$</span>quantile <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="fu">nrow</span>(portability_df_na_sorted))<span class="sc">/</span><span class="fu">nrow</span>(portability_df_na_sorted)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>portability_df_na_sorted_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(portability_df_na_sorted, <span class="at">measure.vars =</span> <span class="fu">c</span>(<span class="st">"log2_portability_Predixcan"</span>,<span class="st">"log2_portability_Enformer"</span>))</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(portability_df_na_sorted_melt, <span class="fu">aes</span>(<span class="at">x=</span>quantile,<span class="at">y=</span>value,<span class="at">color=</span>variable)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span> <span class="fu">stat_smooth</span>(<span class="at">n=</span><span class="dv">2000</span>, <span class="at">span=</span>.<span class="dv">25</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.5</span>,<span class="fl">1.5</span>)) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"prediXcan prediction accuracy quantile"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"log2 portability"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-7.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(portability_df_na_sorted_melt) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp_corrs_all,<span class="at">y=</span>value,<span class="at">color=</span>variable))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-8.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(portability_df_na_sorted_melt) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>quantile,<span class="at">x=</span>gp_corrs_all)) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"prediXcan prediction accuracy"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"prediXcan prediction accuracy quantile"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-20-9.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>middle_quantile <span class="ot">&lt;-</span> portability_df_na_sorted[portability_df_na_sorted<span class="sc">$</span>quantile <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">&amp;</span> portability_df_na_sorted<span class="sc">$</span>quantile <span class="sc">&lt;</span> <span class="fl">0.75</span>, ]</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(middle_quantile<span class="sc">$</span>log2_portability_Predixcan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.06367863</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(middle_quantile<span class="sc">$</span>log2_portability_Enformer)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1634003</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>upper_quantile <span class="ot">&lt;-</span> portability_df_na_sorted[portability_df_na_sorted<span class="sc">$</span>quantile <span class="sc">&gt;</span> <span class="fl">0.5</span>, ]</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(upper_quantile<span class="sc">$</span>log2_portability_Predixcan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.3897802</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(upper_quantile<span class="sc">$</span>log2_portability_Enformer)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.06843718</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># length(intersect(ge_025_genes, gp_025_genes))</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># portability_0.25_ge &lt;- portability_df_na[abs(portability_df_na$ge_corrs_all) &gt; 0.5, ]</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co"># mean(portability_0.25_ge$log2_portability_Predixcan)</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co"># mean(portability_0.25_ge$log2_portability_Enformer, na.rm=TRUE)</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="co"># portability_0.25_gp &lt;- portability_df_na[portability_df_na$gp_corrs_all &gt; 0.1, ]</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="co"># mean(portability_0.25_gp$log2_portability_Predixcan)</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="co"># mean(portability_0.25_gp$log2_portability_Enformer, na.rm=TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>portability_p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(portability_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>log2_portability_Predixcan,<span class="at">y=</span>log2_portability_Enformer,<span class="at">color=</span>gp_corrs_AFR))</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(portability_p1, <span class="at">type=</span><span class="st">"densigram"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(portability_df) <span class="sc">+</span> <span class="fu">geom_hex</span>(<span class="fu">aes</span>(<span class="at">x=</span>portability_Predixcan,<span class="at">y=</span>portability_Enformer))</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>portability_p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(portability_df, <span class="fu">aes</span>(<span class="at">x=</span>gp_corrs_AFR<span class="sc">^</span><span class="dv">2</span>,<span class="at">y=</span>gp_corrs_EUR<span class="sc">^</span><span class="dv">2</span>)) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(portability_p2, <span class="at">type=</span><span class="st">"densigram"</span>)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>portability_p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(portability_df, <span class="fu">aes</span>(<span class="at">x=</span>log2_portability_Enformer,<span class="at">y=</span>gp_corrs_EUR)) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(portability_p3, <span class="at">type=</span><span class="st">"densigram"</span>)</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>portability_p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(portability_df, <span class="fu">aes</span>(<span class="at">x=</span>log2_portability_Predixcan,<span class="at">y=</span>gp_corrs_EUR)) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>ggExtra<span class="sc">::</span><span class="fu">ggMarginal</span>(portability_p4, <span class="at">type=</span><span class="st">"densigram"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="enformer_portability_analysis_files/figure-html/unnamed-chunk-21-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>What if we stratify the portability into genes depending on their reference genome ancestral origin?</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>AFR_genes <span class="ot">&lt;-</span> gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,]</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>EUR_genes <span class="ot">&lt;-</span> gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry <span class="sc">==</span> <span class="st">"EUR"</span>,]</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>AFR_geuvadis_compare_AFR <span class="ot">&lt;-</span> AFR_geuvadis_compare[<span class="fu">rownames</span>(AFR_genes),]</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>AFR_geuvadis_compare_EUR <span class="ot">&lt;-</span> AFR_geuvadis_compare[<span class="fu">rownames</span>(EUR_genes),]</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>AFR_predixcan_compare_AFR <span class="ot">&lt;-</span> AFR_predixcan_compare[<span class="fu">rownames</span>(AFR_genes),]</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>AFR_predixcan_compare_EUR <span class="ot">&lt;-</span> AFR_predixcan_compare[<span class="fu">rownames</span>(EUR_genes),]</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>AFR_enformer_compare_AFR <span class="ot">&lt;-</span> AFR_enformer_compare[<span class="fu">rownames</span>(AFR_genes),]</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>AFR_enformer_compare_EUR <span class="ot">&lt;-</span> AFR_enformer_compare[<span class="fu">rownames</span>(EUR_genes),]</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>EUR_geuvadis_compare_AFR <span class="ot">&lt;-</span> EUR_geuvadis_compare[<span class="fu">rownames</span>(AFR_genes),]</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>EUR_geuvadis_compare_EUR <span class="ot">&lt;-</span> EUR_geuvadis_compare[<span class="fu">rownames</span>(EUR_genes),]</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>EUR_predixcan_compare_AFR <span class="ot">&lt;-</span> EUR_predixcan_compare[<span class="fu">rownames</span>(AFR_genes),]</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>EUR_predixcan_compare_EUR <span class="ot">&lt;-</span> EUR_predixcan_compare[<span class="fu">rownames</span>(EUR_genes),]</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>EUR_enformer_compare_AFR <span class="ot">&lt;-</span> EUR_enformer_compare[<span class="fu">rownames</span>(AFR_genes),]</span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>EUR_enformer_compare_EUR <span class="ot">&lt;-</span> EUR_enformer_compare[<span class="fu">rownames</span>(EUR_genes),]</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlations between enformer and geuvadis subsets</span></span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>AFR_ge_corrs_AFR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(AFR_geuvadis_compare_AFR, AFR_enformer_compare_AFR)</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>AFR_ge_corrs_EUR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(AFR_geuvadis_compare_EUR, AFR_enformer_compare_EUR)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>EUR_ge_corrs_AFR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(EUR_geuvadis_compare_AFR, EUR_enformer_compare_AFR)</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>EUR_ge_corrs_EUR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(EUR_geuvadis_compare_EUR, EUR_enformer_compare_EUR)</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlations between predixcan and geuvadis subsets</span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>AFR_gp_corrs_AFR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(AFR_geuvadis_compare_AFR, AFR_predixcan_compare_AFR)</span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>AFR_gp_corrs_EUR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(AFR_geuvadis_compare_EUR, AFR_predixcan_compare_EUR)</span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>EUR_gp_corrs_AFR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(EUR_geuvadis_compare_AFR, EUR_predixcan_compare_AFR)</span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>EUR_gp_corrs_EUR <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(EUR_geuvadis_compare_EUR, EUR_predixcan_compare_EUR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>epsi<span class="ot">=</span><span class="fl">0.1</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>portability_Enformer_AFR <span class="ot">=</span> ( <span class="fu">abs</span>(AFR_ge_corrs_AFR) <span class="sc">+</span> epsi )  <span class="sc">/</span> ( <span class="fu">abs</span>(EUR_ge_corrs_AFR)  <span class="sc">+</span> epsi )</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>portability_Predixcan_AFR <span class="ot">=</span> ( <span class="fu">abs</span>(AFR_gp_corrs_AFR) <span class="sc">+</span> epsi ) <span class="sc">/</span> ( <span class="fu">abs</span>(EUR_gp_corrs_AFR) <span class="sc">+</span> epsi )</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>portability_Enformer_EUR <span class="ot">=</span> ( <span class="fu">abs</span>(AFR_ge_corrs_EUR) <span class="sc">+</span> epsi )  <span class="sc">/</span> ( <span class="fu">abs</span>(EUR_ge_corrs_EUR)  <span class="sc">+</span> epsi )</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>portability_Predixcan_EUR <span class="ot">=</span> ( <span class="fu">abs</span>(AFR_gp_corrs_EUR) <span class="sc">+</span> epsi ) <span class="sc">/</span> ( <span class="fu">abs</span>(EUR_gp_corrs_EUR) <span class="sc">+</span> epsi )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(EUR_ge_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, AFR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"enformer: EUR vs. AFR"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(EUR_ge_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, EUR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"enformer: EUR vs. AFR"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(AFR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, EUR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"enformer: EUR vs. AFR"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(AFR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, EUR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"enformer: EUR vs. AFR"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-between_pop_qqplots_subset" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots_subset-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots_subset-1.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots_subset" width="384"></p>
<figcaption class="figure-caption">(a) Enformer</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots_subset-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots_subset-2.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots_subset" width="384"></p>
<figcaption class="figure-caption">(b) Predixcan</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots_subset-3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots_subset-3.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots_subset" width="384"></p>
<figcaption class="figure-caption">(c) Enformer</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_pop_qqplots_subset-4" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_pop_qqplots_subset-4.png" class="img-fluid figure-img" data-ref-parent="fig-between_pop_qqplots_subset" width="384"></p>
<figcaption class="figure-caption">(d) Predixcan</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;7: QQPlots of R squared distribution between Predixcan and Enformer prediction for EUR and</figcaption><p></p>
</figure>
</div>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(EUR_gp_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, EUR_ge_corrs_EUR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"European individuals predixcan vs enformer for European reference genes"</span>, <span class="at">xlab =</span> <span class="st">"geuvadis vs predixcan"</span>, <span class="at">ylab =</span> <span class="st">"geuvadis vs. enformer"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(AFR_gp_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, AFR_ge_corrs_AFR<span class="sc">^</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">"African individuals predixcan vs enformer for African reference genes"</span>, <span class="at">xlab =</span> <span class="st">"geuvadis vs predixcan"</span>, <span class="at">ylab =</span> <span class="st">"geuvadis vs. enformer"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-between_predictor_qqplots_subset" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_predictor_qqplots_subset-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_predictor_qqplots_subset-1.png" class="img-fluid figure-img" data-ref-parent="fig-between_predictor_qqplots_subset" width="384"></p>
<figcaption class="figure-caption">(a) EUR</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-between_predictor_qqplots_subset-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-between_predictor_qqplots_subset-2.png" class="img-fluid figure-img" data-ref-parent="fig-between_predictor_qqplots_subset" width="384"></p>
<figcaption class="figure-caption">(b) AFR</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;8: QQPlots of R squared distribution between Predixcan, Enformer correlated against Geuvadis.</figcaption><p></p>
</figure>
</div>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(portability_Predixcan_AFR,portability_Enformer_AFR, <span class="at">main=</span><span class="st">"Portability Enformer vs. Predixcan in AFR reference genes"</span>,<span class="at">xlab =</span> <span class="st">"Predixcan Portability"</span>, <span class="at">ylab =</span> <span class="st">"Enformer portability"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-portability-AFR-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-portability-AFR-genes-1.png" class="img-fluid figure-img" width="480"></p>
<figcaption class="figure-caption">Figure&nbsp;9: QQPlots of R squared distribution between Predixcan and Enformer prediction for EUR and</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(portability_Predixcan_EUR,portability_Enformer_EUR, <span class="at">main=</span><span class="st">"Portability Enformer vs. Predixcan in EUR reference genes"</span>,<span class="at">xlab =</span> <span class="st">"Predixcan Portability"</span>, <span class="at">ylab =</span> <span class="st">"Enformer portability"</span>) ; <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div id="fig-portability-EUR-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="enformer_portability_analysis_files/figure-html/fig-portability-EUR-genes-1.png" class="img-fluid figure-img" width="480"></p>
<figcaption class="figure-caption">Figure&nbsp;10: QQPlots of R squared distribution between Predixcan and Enformer prediction for EUR and</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>