<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-10-04">

<title>Daily-Blog-Sai - train_aracena_tss_predictors_evaluation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">train_aracena_tss_predictors_evaluation</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 4, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="context" class="level1">
<h1>Context</h1>
<p>Here I trained an elastic net and simple MLP mode to predict TSS RNAseq expression from Flu condition (aracena dataset) from Enformer predictions. Training was done using corpus of TSS sites from a ranmdom mixture of individuals. Here I am evaluating the performance of the trained models on a held out test set.</p>
<p>TODO: Average of individuals TODO: Collect RNAseq from Katie, and use this for DL-EN training AND training PrediXcan TODO: Try different MLP params</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"train_aracena_tss_predictor"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-10-04'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(neuralnet)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>predict_on_data_subset <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, glmnet_model){</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load trained models</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp.rds"</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/glmnet_lambdas.png"</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_model)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load dataset to be predicted on</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  data_X <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_inputs.csv"</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  data_y <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_targets.csv"</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  X_train <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_X))</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  y_train <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_y))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  y_train_for_cv <span class="ot">&lt;-</span> y_train[,<span class="dv">1</span>]</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make predictions and associated plots</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X_train, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_corr_en_pred.png"</span>))</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_predictions, y_train[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"glmnet predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Glmnet predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  pred_corr_en <span class="ot">&lt;-</span> <span class="fu">cor</span>(glmnet_predictions, y_train[,<span class="dv">1</span>])</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_en} for {data_subset} data subset with glmnet predictions"</span>))</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X_train, y_train_for_cv))</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_corr_nn_pred.png"</span>))</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(predictions_nn, y_train[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"NN predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"NN predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  pred_corr_nn <span class="ot">&lt;-</span> <span class="fu">cor</span>(predictions_nn, y_train[,<span class="dv">1</span>])</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_nn} for {data_subset} data subset with NN predictions"</span>))</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>analyze_input_data <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset) {</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_inputs.csv"</span>)))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_targets.csv"</span>)))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_flu_vs_NI_rnaseq.png"</span>))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>],<span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>),<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"ground truth NI rnaseq"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Flu vs. NI in ground truth for {data_subset} data subset"</span>))</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  corr_rnaseq <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>])</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_rnaseq} for {data_subset} data subset between flu/NI rnaseq"</span>))</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_flu_vs_epi_rnaseq.png"</span>))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena flu rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_NI_vs_epi_rnaseq.png"</span>))</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena NI rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s first take a look at some properties of the data itself</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"train"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.73904726191164 for train data subset between flu/NI rnaseq</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.228283741744927 for train data subset between aracena flu rnaseq and corresponding enformer prediction</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.214077042278839 for train data subset between aracena NI rnaseq and corresponding enformer prediction</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"valid"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.989184938027612 for valid data subset between flu/NI rnaseq</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0643572220801176 for valid data subset between aracena flu rnaseq and corresponding enformer prediction</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0819605957767975 for valid data subset between aracena NI rnaseq and corresponding enformer prediction</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"test"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.362012765998215 for test data subset between flu/NI rnaseq</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.126590285627353 for test data subset between aracena flu rnaseq and corresponding enformer prediction</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.334457542541329 for test data subset between aracena NI rnaseq and corresponding enformer prediction</code></pre>
</div>
</div>
<p>Now we can use the trained models to make predictions and assess model performance.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"train"</span>, <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.rds.linear.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.437281473980008 for train data subset with glmnet predictions</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.483994813636335 for train data subset with NN predictions</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"valid"</span>, <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.rds.linear.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.11744649852639 for valid data subset with glmnet predictions</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0682833668657665 for valid data subset with NN predictions</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"test"</span>, <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.rds.linear.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.396900747294329 for test data subset with glmnet predictions</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.539515763787496 for test data subset with NN predictions</code></pre>
</div>
</div>
<p>It seems like for some reason this randomly selected validation subset is just tougher to predict on, showing a reduced correlation betweeen the CAGE epigenomic track in the input as well as poorer prediction performance.</p>
<p>Why are the predictions so poor on the validation dataset?</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>remapped_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/remapped_table_filt.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>))</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(remapped_table) <span class="ot">&lt;-</span> remapped_table<span class="sc">$</span>V1</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>remapped_table<span class="sc">$</span>V1 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>remapped_table_train <span class="ot">&lt;-</span> remapped_table[,remapped_table[<span class="st">"group"</span>,]<span class="sc">==</span><span class="st">"train"</span>]</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>regions_train <span class="ot">&lt;-</span> <span class="fu">colnames</span>(remapped_table_train)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>chroms_train <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions_train, <span class="dv">2</span>, <span class="dv">2</span>)</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(chroms_train)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>chroms_train
   1    2    3    4    9 
7687 1976 1059  748  762 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>remapped_table_test <span class="ot">&lt;-</span> remapped_table[,remapped_table[<span class="st">"group"</span>,]<span class="sc">==</span><span class="st">"test"</span>]</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(remapped_table_test)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1]   29 2041</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(remapped_table_test)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>chroms <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions, <span class="dv">2</span>, <span class="dv">2</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>half_test <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">length</span>(regions)<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>chroms_valid <span class="ot">&lt;-</span> chroms[<span class="dv">1</span><span class="sc">:</span>half_test]</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(chroms_valid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>chroms_valid
  1   2 
919 101 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>chroms_test <span class="ot">&lt;-</span> chroms[(half_test<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(regions)]</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(chroms_test)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>chroms_test
  2   8 
330 691 </code></pre>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb34" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "train_aracena_tss_predictors_evaluation"</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-10-04"</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>Here I trained an elastic net and simple MLP mode to predict TSS RNAseq expression from Flu condition (aracena dataset) from Enformer predictions. Training was done using corpus of TSS sites from a ranmdom mixture of individuals. Here I am evaluating the performance of the trained models on a held out test set.</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>TODO: Average of individuals</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>TODO: Collect RNAseq from Katie, and use this for DL-EN training AND training PrediXcan</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>TODO: Try different MLP params</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Set up box storage directory</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"train_aracena_tss_predictor"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-10-04'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(neuralnet)</span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a>predict_on_data_subset <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, glmnet_model){</span>
<span id="cb34-56"><a href="#cb34-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-57"><a href="#cb34-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load trained models</span></span>
<span id="cb34-58"><a href="#cb34-58" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model)</span>
<span id="cb34-59"><a href="#cb34-59" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp.rds"</span>)</span>
<span id="cb34-60"><a href="#cb34-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-61"><a href="#cb34-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/glmnet_lambdas.png"</span>)</span>
<span id="cb34-62"><a href="#cb34-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_model)</span>
<span id="cb34-63"><a href="#cb34-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-64"><a href="#cb34-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-65"><a href="#cb34-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load dataset to be predicted on</span></span>
<span id="cb34-66"><a href="#cb34-66" aria-hidden="true" tabindex="-1"></a>  data_X <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_inputs.csv"</span>)</span>
<span id="cb34-67"><a href="#cb34-67" aria-hidden="true" tabindex="-1"></a>  data_y <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_targets.csv"</span>)</span>
<span id="cb34-68"><a href="#cb34-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-69"><a href="#cb34-69" aria-hidden="true" tabindex="-1"></a>  X_train <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_X))</span>
<span id="cb34-70"><a href="#cb34-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-71"><a href="#cb34-71" aria-hidden="true" tabindex="-1"></a>  y_train <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_y))</span>
<span id="cb34-72"><a href="#cb34-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-73"><a href="#cb34-73" aria-hidden="true" tabindex="-1"></a>  y_train_for_cv <span class="ot">&lt;-</span> y_train[,<span class="dv">1</span>]</span>
<span id="cb34-74"><a href="#cb34-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-75"><a href="#cb34-75" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make predictions and associated plots</span></span>
<span id="cb34-76"><a href="#cb34-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-77"><a href="#cb34-77" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X_train, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb34-78"><a href="#cb34-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-79"><a href="#cb34-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_corr_en_pred.png"</span>))</span>
<span id="cb34-80"><a href="#cb34-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_predictions, y_train[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"glmnet predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Glmnet predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb34-81"><a href="#cb34-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-82"><a href="#cb34-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-83"><a href="#cb34-83" aria-hidden="true" tabindex="-1"></a>  pred_corr_en <span class="ot">&lt;-</span> <span class="fu">cor</span>(glmnet_predictions, y_train[,<span class="dv">1</span>])</span>
<span id="cb34-84"><a href="#cb34-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_en} for {data_subset} data subset with glmnet predictions"</span>))</span>
<span id="cb34-85"><a href="#cb34-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-86"><a href="#cb34-86" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X_train, y_train_for_cv))</span>
<span id="cb34-87"><a href="#cb34-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-88"><a href="#cb34-88" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb34-89"><a href="#cb34-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-90"><a href="#cb34-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_corr_nn_pred.png"</span>))</span>
<span id="cb34-91"><a href="#cb34-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(predictions_nn, y_train[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"NN predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"NN predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb34-92"><a href="#cb34-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-93"><a href="#cb34-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-94"><a href="#cb34-94" aria-hidden="true" tabindex="-1"></a>  pred_corr_nn <span class="ot">&lt;-</span> <span class="fu">cor</span>(predictions_nn, y_train[,<span class="dv">1</span>])</span>
<span id="cb34-95"><a href="#cb34-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_nn} for {data_subset} data subset with NN predictions"</span>))</span>
<span id="cb34-96"><a href="#cb34-96" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-97"><a href="#cb34-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-98"><a href="#cb34-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-99"><a href="#cb34-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-100"><a href="#cb34-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-103"><a href="#cb34-103" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-104"><a href="#cb34-104" aria-hidden="true" tabindex="-1"></a>analyze_input_data <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset) {</span>
<span id="cb34-105"><a href="#cb34-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-106"><a href="#cb34-106" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_inputs.csv"</span>)))</span>
<span id="cb34-107"><a href="#cb34-107" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/new_{data_subset}_targets.csv"</span>)))</span>
<span id="cb34-108"><a href="#cb34-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-109"><a href="#cb34-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_flu_vs_NI_rnaseq.png"</span>))</span>
<span id="cb34-110"><a href="#cb34-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>],<span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>),<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"ground truth NI rnaseq"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Flu vs. NI in ground truth for {data_subset} data subset"</span>))</span>
<span id="cb34-111"><a href="#cb34-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-112"><a href="#cb34-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-113"><a href="#cb34-113" aria-hidden="true" tabindex="-1"></a>  corr_rnaseq <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>])</span>
<span id="cb34-114"><a href="#cb34-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_rnaseq} for {data_subset} data subset between flu/NI rnaseq"</span>))</span>
<span id="cb34-115"><a href="#cb34-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-116"><a href="#cb34-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_flu_vs_epi_rnaseq.png"</span>))</span>
<span id="cb34-117"><a href="#cb34-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb34-118"><a href="#cb34-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-119"><a href="#cb34-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-120"><a href="#cb34-120" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb34-121"><a href="#cb34-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena flu rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb34-122"><a href="#cb34-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-123"><a href="#cb34-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-04-train_aracena_tss_predictor/figures/{data_subset}_NI_vs_epi_rnaseq.png"</span>))</span>
<span id="cb34-124"><a href="#cb34-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb34-125"><a href="#cb34-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-126"><a href="#cb34-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-127"><a href="#cb34-127" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb34-128"><a href="#cb34-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena NI rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb34-129"><a href="#cb34-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-130"><a href="#cb34-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-131"><a href="#cb34-131" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-132"><a href="#cb34-132" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-133"><a href="#cb34-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-134"><a href="#cb34-134" aria-hidden="true" tabindex="-1"></a>Let's first take a look at some properties of the data itself</span>
<span id="cb34-135"><a href="#cb34-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-138"><a href="#cb34-138" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-139"><a href="#cb34-139" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"train"</span>)</span>
<span id="cb34-140"><a href="#cb34-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-141"><a href="#cb34-141" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"valid"</span>)</span>
<span id="cb34-142"><a href="#cb34-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-143"><a href="#cb34-143" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"test"</span>)</span>
<span id="cb34-144"><a href="#cb34-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-145"><a href="#cb34-145" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-146"><a href="#cb34-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-147"><a href="#cb34-147" aria-hidden="true" tabindex="-1"></a>Now we can use the trained models to make predictions and assess model performance.</span>
<span id="cb34-148"><a href="#cb34-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-151"><a href="#cb34-151" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-152"><a href="#cb34-152" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"train"</span>, <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.rds.linear.rds"</span>)</span>
<span id="cb34-153"><a href="#cb34-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-154"><a href="#cb34-154" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"valid"</span>, <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.rds.linear.rds"</span>)</span>
<span id="cb34-155"><a href="#cb34-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-156"><a href="#cb34-156" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"test"</span>, <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.rds.linear.rds"</span>)</span>
<span id="cb34-157"><a href="#cb34-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-158"><a href="#cb34-158" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-159"><a href="#cb34-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-160"><a href="#cb34-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-161"><a href="#cb34-161" aria-hidden="true" tabindex="-1"></a>It seems like for some reason this randomly selected validation subset is just tougher to predict on, showing a reduced correlation betweeen the CAGE epigenomic track in the input as well as poorer prediction performance. </span>
<span id="cb34-162"><a href="#cb34-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-163"><a href="#cb34-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-164"><a href="#cb34-164" aria-hidden="true" tabindex="-1"></a>Why are the predictions so poor on the validation dataset? </span>
<span id="cb34-165"><a href="#cb34-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-168"><a href="#cb34-168" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-169"><a href="#cb34-169" aria-hidden="true" tabindex="-1"></a>remapped_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/remapped_table_filt.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>))</span>
<span id="cb34-170"><a href="#cb34-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-171"><a href="#cb34-171" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(remapped_table) <span class="ot">&lt;-</span> remapped_table<span class="sc">$</span>V1</span>
<span id="cb34-172"><a href="#cb34-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-173"><a href="#cb34-173" aria-hidden="true" tabindex="-1"></a>remapped_table<span class="sc">$</span>V1 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb34-174"><a href="#cb34-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-175"><a href="#cb34-175" aria-hidden="true" tabindex="-1"></a>remapped_table_train <span class="ot">&lt;-</span> remapped_table[,remapped_table[<span class="st">"group"</span>,]<span class="sc">==</span><span class="st">"train"</span>]</span>
<span id="cb34-176"><a href="#cb34-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-177"><a href="#cb34-177" aria-hidden="true" tabindex="-1"></a>regions_train <span class="ot">&lt;-</span> <span class="fu">colnames</span>(remapped_table_train)</span>
<span id="cb34-178"><a href="#cb34-178" aria-hidden="true" tabindex="-1"></a>chroms_train <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions_train, <span class="dv">2</span>, <span class="dv">2</span>)</span>
<span id="cb34-179"><a href="#cb34-179" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(chroms_train)</span>
<span id="cb34-180"><a href="#cb34-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-181"><a href="#cb34-181" aria-hidden="true" tabindex="-1"></a>remapped_table_test <span class="ot">&lt;-</span> remapped_table[,remapped_table[<span class="st">"group"</span>,]<span class="sc">==</span><span class="st">"test"</span>]</span>
<span id="cb34-182"><a href="#cb34-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-183"><a href="#cb34-183" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(remapped_table_test)</span>
<span id="cb34-184"><a href="#cb34-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-185"><a href="#cb34-185" aria-hidden="true" tabindex="-1"></a>regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(remapped_table_test)</span>
<span id="cb34-186"><a href="#cb34-186" aria-hidden="true" tabindex="-1"></a>chroms <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions, <span class="dv">2</span>, <span class="dv">2</span>)</span>
<span id="cb34-187"><a href="#cb34-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-188"><a href="#cb34-188" aria-hidden="true" tabindex="-1"></a>half_test <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">length</span>(regions)<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb34-189"><a href="#cb34-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-190"><a href="#cb34-190" aria-hidden="true" tabindex="-1"></a>chroms_valid <span class="ot">&lt;-</span> chroms[<span class="dv">1</span><span class="sc">:</span>half_test]</span>
<span id="cb34-191"><a href="#cb34-191" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(chroms_valid)</span>
<span id="cb34-192"><a href="#cb34-192" aria-hidden="true" tabindex="-1"></a>chroms_test <span class="ot">&lt;-</span> chroms[(half_test<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(regions)]</span>
<span id="cb34-193"><a href="#cb34-193" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(chroms_test)</span>
<span id="cb34-194"><a href="#cb34-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-195"><a href="#cb34-195" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>