<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-11-22">

<title>Daily-Blog-Sai - 1.3_evaluate_fine_tuned_model</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#evaluate-model-performance-on-log-transformed-data" id="toc-evaluate-model-performance-on-log-transformed-data" class="nav-link" data-scroll-target="#evaluate-model-performance-on-log-transformed-data">Evaluate model performance on log-transformed data</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">1.3_evaluate_fine_tuned_model</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 22, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="context" class="level1">
<h1>Context</h1>
<section id="evaluate-model-performance-on-log-transformed-data" class="level2">
<h2 class="anchored" data-anchor-id="evaluate-model-performance-on-log-transformed-data">Evaluate model performance on log-transformed data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>collect_all_records <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, gene_metadata, X_path, y_path_flu, y_path_ni, glmnet_model_path_flu, glmnet_model_path_ni) {</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Collecting data records for data subset: {data_subset}"</span>))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.table</span>(X_path, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> X[<span class="fu">order</span>(<span class="fu">row.names</span>(X)),]</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  y_flu <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read.table</span>(y_path_flu, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  y_flu<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  y_flu <span class="ot">&lt;-</span> y_flu[<span class="fu">rownames</span>(X),]</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  y_flu<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  y_ni <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read.table</span>(y_path_ni, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  y_ni<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  y_ni <span class="ot">&lt;-</span> y_ni[<span class="fu">rownames</span>(X),]</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  y_ni<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(X))</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  glmnet_model_flu <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path_flu)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions_flu <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model_flu, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  glmnet_model_ni <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path_ni)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions_ni <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model_ni, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>  cur_metadata <span class="ot">&lt;-</span> gene_metadata[<span class="fu">rownames</span>(y_flu),]</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  return_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">data_subset=</span> data_subset, <span class="at">glmnet_pred_flu=</span>glmnet_predictions_flu, <span class="at">glmnet_pred_ni=</span>glmnet_predictions_ni, <span class="at">true_exp_flu =</span> <span class="fu">as.numeric</span>(y_flu[,<span class="dv">1</span>]),<span class="at">true_exp_ni =</span> <span class="fu">as.numeric</span>(y_ni[,<span class="dv">1</span>]), <span class="at">CAGE_epi =</span> <span class="fu">as.numeric</span>(X[,<span class="dv">4766</span>]), <span class="at">ensembl_id =</span> cur_metadata[[<span class="st">"ensembl_gene_id"</span>]], <span class="at">external_gene_name =</span> cur_metadata[[<span class="st">"external_gene_name"</span>]], <span class="at">region =</span> cur_metadata[[<span class="st">"target_regions"</span>]], <span class="at">chroms =</span> cur_metadata[[<span class="st">"chromosome_name"</span>]])</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_df) </span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>gene_annotation_file <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-11-21-aracena_predixcan/canonical_TSS_full_metadata.txt"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>gene_annotation_table <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(gene_annotation_file, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">","</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_annotation_table) <span class="ot">&lt;-</span> gene_annotation_table<span class="sc">$</span>ensembl_gene_id</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>glmnet_model_path_flu <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_Flu_normalized_covariates.linear.rds"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>glmnet_model_path_ni <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_NI_normalized_covariates.linear.rds"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>all_records_df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"train"</span>,<span class="st">"test"</span>, <span class="st">"valid"</span>)){</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  X_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/aracena_predixcan/{data_subset}_epigenome.txt"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  y_path_flu <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/aracena_predixcan/{data_subset}_Flu_aracena.txt"</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  y_path_ni <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/aracena_predixcan/{data_subset}_NI_aracena.txt"</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  all_records_df_store[[c]] <span class="ot">&lt;-</span> <span class="fu">collect_all_records</span>(data_subset, gene_annotation_table, X_path, y_path_flu, y_path_ni, glmnet_model_path_flu, glmnet_model_path_ni)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  c<span class="ot">=</span>c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: train
[1] 7855 5313
********* Collecting data records for data subset: test
[1] 1167 5313
********* Collecting data records for data subset: valid
[1] 1789 5313</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, all_records_df_store)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">is.na</span>(full_df)) <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>chroms <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>chroms, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>log_CAGE_epi <span class="ot">&lt;-</span> <span class="fu">log2</span>(full_df<span class="sc">$</span>CAGE_epi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>full_df_train <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> <span class="st">"train"</span>,]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>full_df_valid <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> <span class="st">"valid"</span>,]</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(full_df_train))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7855   11</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(full_df_valid))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1424   11</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">cor</span>(full_df_train<span class="sc">$</span>glmnet_pred_flu, full_df_train<span class="sc">$</span>true_exp_flu, <span class="at">method =</span> <span class="st">"spearman"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.640633</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">cor</span>(full_df_valid<span class="sc">$</span>glmnet_pred_flu, full_df_valid<span class="sc">$</span>true_exp_flu, <span class="at">method =</span> <span class="st">"spearman"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.6220389</code></pre>
</div>
</div>
<p>Set color palette and output dir</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>out_folder <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-11-22-retrain_expression_predictor/evaluation_plots"</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>, <span class="st">"all"</span> <span class="ot">=</span> <span class="st">"grey"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Plot number of genes per chrom and data subset</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>gene_counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">as.matrix</span>(<span class="fu">table</span>(full_df<span class="sc">$</span>data_subset)))</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>gene_counts_df[<span class="st">"all"</span>,<span class="st">"V1"</span>] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(full_df)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>gene_counts_df<span class="sc">$</span>grouping <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gene_counts_df)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_counts_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene_count"</span>, <span class="st">"data_subset"</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>gene_counts_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(gene_counts_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{out_folder}/counts_data_subsets.png"</span>), <span class="at">width=</span><span class="dv">800</span>, <span class="at">height=</span><span class="dv">400</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="1.3.1_evaluate_fine_tuned_model_files/figure-html/unnamed-chunk-8-1.svg" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>gene_counts_chrom_df <span class="ot">&lt;-</span> <span class="fu">distinct</span>(full_df[,<span class="fu">c</span>(<span class="st">"data_subset"</span>, <span class="st">"chroms"</span>)])</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>gene_counts_chrom_df<span class="sc">$</span>gene_count <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">as.matrix</span>(<span class="fu">table</span>(full_df<span class="sc">$</span>chroms)))[gene_counts_chrom_df<span class="sc">$</span>chroms,]</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{out_folder}/counts_all_chroms_both.png"</span>), <span class="at">width=</span><span class="dv">800</span>, <span class="at">height=</span><span class="dv">400</span>)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_chrom_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chroms, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_chrom_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chroms, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="1.3.1_evaluate_fine_tuned_model_files/figure-html/unnamed-chunk-8-2.svg" class="img-fluid"></p>
</div>
</div>
<p>Plot correlation patterns within data subsets</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>data_subsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>cor_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet_pred_flu_FLU"</span>, <span class="st">"glmnet_pred_ni_FLU"</span>, <span class="st">"glmnet_pred_ni_NI"</span>, <span class="st">"log_CAGE_epi_FLU"</span>, <span class="st">"CAGE_epi_FLU"</span>, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>cor_vecs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet_pred_flu"</span>, <span class="st">"glmnet_pred_ni"</span>, <span class="st">"glmnet_pred_ni"</span>, <span class="st">"log_CAGE_epi"</span>, <span class="st">"CAGE_epi"</span>, <span class="st">"true_exp_flu"</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>cor_against <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_ni"</span>, <span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_ni"</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>cor_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"spearman"</span>,<span class="st">"pearson"</span>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(data_subsets)<span class="sc">*</span><span class="fu">length</span>(cor_vecs)<span class="sc">*</span><span class="fu">length</span>(cor_types), <span class="at">ncol =</span> <span class="dv">4</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>i <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> data_subsets) {</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  cur_subset_df <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> data_subset,]</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  cur_subset_df <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> data_subset,]</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  cur_subset_df <span class="ot">&lt;-</span> cur_subset_df[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">is.na</span>(cur_subset_df)) <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (cor_type <span class="cf">in</span> cor_types) {</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (c <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(cor_names)) {</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>      cur_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(cur_subset_df[,cor_vecs[c]], cur_subset_df[,cor_against[c]], <span class="at">method =</span> cor_type)</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>      corrs_df_for_subsets[i,] <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset, cor_names[c], cor_type, cur_cor)</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>      i <span class="ot">=</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(corrs_df_for_subsets)</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(corrs_df_for_subsets) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"data_subset"</span>, <span class="st">"cor_id"</span>, <span class="st">"cor_type"</span>, <span class="st">"corrs"</span>)</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets<span class="sc">$</span>cor_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(corrs_df_for_subsets<span class="sc">$</span>cor_id, <span class="at">levels =</span> cor_names)</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets<span class="sc">$</span>corrs <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(corrs_df_for_subsets<span class="sc">$</span>corrs))</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{out_folder}/all_corrs_subsets.png"</span>), <span class="at">width=</span><span class="dv">600</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_for_subsets) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>corrs, <span class="at">fill=</span>cor_id), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>cor_type, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_colour_brewer</span>(<span class="at">palette =</span> <span class="st">"Pastel1"</span>)</span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_for_subsets) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>corrs, <span class="at">fill=</span>cor_id), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>cor_type, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_colour_brewer</span>(<span class="at">palette =</span> <span class="st">"Pastel1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="1.3.1_evaluate_fine_tuned_model_files/figure-html/unnamed-chunk-9-1.svg" class="img-fluid"></p>
</div>
</div>
<p>Plot correlation patterns within chroms</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># chroms &lt;- c("all", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y")</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_vecs &lt;- c("glmnet_pred", "CAGE_epi", "log_CAGE_epi")</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_types &lt;- c("spearman","pearson")</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df &lt;- matrix(nrow = length(chroms)*length(cor_vecs)*length(cor_types), ncol = 4)</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co"># i = 1</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co"># for (chrom in chroms) {</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   cur_subset_df &lt;- full_df[full_df$chroms == chrom,]</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   cur_subset_df &lt;- cur_subset_df[!rowSums(is.na(cur_subset_df)) &gt; 0,]</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   for (cor_type in cor_types) {</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     for (cor_vec in cor_vecs) {</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co">#       cur_cor &lt;- cor(cur_subset_df[,cor_vec], cur_subset_df$true_exp, method = cor_type)</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="co">#       cor_subset_df[i,] &lt;- c(chrom, cor_vec, cor_type, cur_cor)</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co">#       i = i + 1</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co">#     }</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df &lt;- as.data.frame(cor_subset_df)</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(cor_subset_df) &lt;- c("chrom", "cor_id", "cor_type", "corrs")</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df$corrs &lt;- as.numeric(as.character(cor_subset_df$corrs))</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df$cor_id &lt;- factor(cor_subset_df$cor_id, levels = names(corr_colors))</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="co"># png(glue("{out_folder}/all_corrs_chroms.png"), width=1000, height=500)</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(cor_subset_df) + geom_bar(aes(x=chrom, y=corrs, fill=cor_id), stat="identity", position="dodge") + xlab("Data subset") + ylab("Correlation") + facet_wrap(~cor_type, ncol=1) + scale_fill_manual(values = corr_colors) +ylim(c(0,1))</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(cor_subset_df) + geom_bar(aes(x=chrom, y=corrs, fill=cor_id), stat="identity", position="dodge") + xlab("Data subset") + ylab("Correlation") + facet_wrap(~cor_type, ncol=1) + scale_fill_manual(values = corr_colors) +ylim(c(0,1))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Plot scatterplots of predictions vs true expression</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># full_df_plot &lt;- full_df[!rowSums(is.na(full_df)) &gt; 0,]</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># png(glue("{out_folder}/all_records_scatterplots.png"), width =900, height=900)</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=glmnet_pred, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("glmnet_pred")+ylab("Ground truth")+theme(panel.background = element_rect(fill = alpha(corr_colors["glmnet_pred"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["CAGE_epi"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=log_CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("log_CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["log_CAGE_epi"],alpha=0.4)))+scale_color_manual(values=subset_colors))</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=glmnet_pred, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("glmnet_pred")+ylab("Ground truth")+theme(panel.background = element_rect(fill = alpha(corr_colors["glmnet_pred"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["CAGE_epi"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=log_CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("log_CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["log_CAGE_epi"],alpha=0.4)))+scale_color_manual(values=subset_colors))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb27" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "1.3_evaluate_fine_tuned_model"</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-11-22"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="fu">## Evaluate model performance on log-transformed data</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>collect_all_records <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, gene_metadata, X_path, y_path_flu, y_path_ni, glmnet_model_path_flu, glmnet_model_path_ni) {</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Collecting data records for data subset: {data_subset}"</span>))</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.table</span>(X_path, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> X[<span class="fu">order</span>(<span class="fu">row.names</span>(X)),]</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>  y_flu <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read.table</span>(y_path_flu, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>  y_flu<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>  y_flu <span class="ot">&lt;-</span> y_flu[<span class="fu">rownames</span>(X),]</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>  y_flu<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>  y_ni <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read.table</span>(y_path_ni, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>  y_ni<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>  y_ni <span class="ot">&lt;-</span> y_ni[<span class="fu">rownames</span>(X),]</span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>  y_ni<span class="sc">$</span>temp <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(X))</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>  glmnet_model_flu <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path_flu)</span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions_flu <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model_flu, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>  glmnet_model_ni <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path_ni)</span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions_ni <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model_ni, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a>  cur_metadata <span class="ot">&lt;-</span> gene_metadata[<span class="fu">rownames</span>(y_flu),]</span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>  return_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">data_subset=</span> data_subset, <span class="at">glmnet_pred_flu=</span>glmnet_predictions_flu, <span class="at">glmnet_pred_ni=</span>glmnet_predictions_ni, <span class="at">true_exp_flu =</span> <span class="fu">as.numeric</span>(y_flu[,<span class="dv">1</span>]),<span class="at">true_exp_ni =</span> <span class="fu">as.numeric</span>(y_ni[,<span class="dv">1</span>]), <span class="at">CAGE_epi =</span> <span class="fu">as.numeric</span>(X[,<span class="dv">4766</span>]), <span class="at">ensembl_id =</span> cur_metadata[[<span class="st">"ensembl_gene_id"</span>]], <span class="at">external_gene_name =</span> cur_metadata[[<span class="st">"external_gene_name"</span>]], <span class="at">region =</span> cur_metadata[[<span class="st">"target_regions"</span>]], <span class="at">chroms =</span> cur_metadata[[<span class="st">"chromosome_name"</span>]])</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_df) </span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a>gene_annotation_file <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-11-21-aracena_predixcan/canonical_TSS_full_metadata.txt"</span></span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a>gene_annotation_table <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(gene_annotation_file, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">","</span>)</span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_annotation_table) <span class="ot">&lt;-</span> gene_annotation_table<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a>glmnet_model_path_flu <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_Flu_normalized_covariates.linear.rds"</span></span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a>glmnet_model_path_ni <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_NI_normalized_covariates.linear.rds"</span></span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a>all_records_df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"train"</span>,<span class="st">"test"</span>, <span class="st">"valid"</span>)){</span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a>  X_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/aracena_predixcan/{data_subset}_epigenome.txt"</span>)</span>
<span id="cb27-96"><a href="#cb27-96" aria-hidden="true" tabindex="-1"></a>  y_path_flu <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/aracena_predixcan/{data_subset}_Flu_aracena.txt"</span>)</span>
<span id="cb27-97"><a href="#cb27-97" aria-hidden="true" tabindex="-1"></a>  y_path_ni <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/aracena_predixcan/{data_subset}_NI_aracena.txt"</span>)</span>
<span id="cb27-98"><a href="#cb27-98" aria-hidden="true" tabindex="-1"></a>  all_records_df_store[[c]] <span class="ot">&lt;-</span> <span class="fu">collect_all_records</span>(data_subset, gene_annotation_table, X_path, y_path_flu, y_path_ni, glmnet_model_path_flu, glmnet_model_path_ni)</span>
<span id="cb27-99"><a href="#cb27-99" aria-hidden="true" tabindex="-1"></a>  c<span class="ot">=</span>c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb27-100"><a href="#cb27-100" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-101"><a href="#cb27-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-102"><a href="#cb27-102" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, all_records_df_store)</span>
<span id="cb27-103"><a href="#cb27-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-104"><a href="#cb27-104" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">is.na</span>(full_df)) <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb27-105"><a href="#cb27-105" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb27-106"><a href="#cb27-106" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>chroms <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>chroms, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb27-107"><a href="#cb27-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-108"><a href="#cb27-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-109"><a href="#cb27-109" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>log_CAGE_epi <span class="ot">&lt;-</span> <span class="fu">log2</span>(full_df<span class="sc">$</span>CAGE_epi)</span>
<span id="cb27-110"><a href="#cb27-110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-111"><a href="#cb27-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-114"><a href="#cb27-114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-115"><a href="#cb27-115" aria-hidden="true" tabindex="-1"></a>full_df_train <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> <span class="st">"train"</span>,]</span>
<span id="cb27-116"><a href="#cb27-116" aria-hidden="true" tabindex="-1"></a>full_df_valid <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> <span class="st">"valid"</span>,]</span>
<span id="cb27-117"><a href="#cb27-117" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(full_df_train))</span>
<span id="cb27-118"><a href="#cb27-118" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(full_df_valid))</span>
<span id="cb27-119"><a href="#cb27-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-120"><a href="#cb27-120" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">cor</span>(full_df_train<span class="sc">$</span>glmnet_pred_flu, full_df_train<span class="sc">$</span>true_exp_flu, <span class="at">method =</span> <span class="st">"spearman"</span>))</span>
<span id="cb27-121"><a href="#cb27-121" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">cor</span>(full_df_valid<span class="sc">$</span>glmnet_pred_flu, full_df_valid<span class="sc">$</span>true_exp_flu, <span class="at">method =</span> <span class="st">"spearman"</span>))</span>
<span id="cb27-122"><a href="#cb27-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-123"><a href="#cb27-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-126"><a href="#cb27-126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-127"><a href="#cb27-127" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-128"><a href="#cb27-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-129"><a href="#cb27-129" aria-hidden="true" tabindex="-1"></a>Set color palette and output dir</span>
<span id="cb27-132"><a href="#cb27-132" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-133"><a href="#cb27-133" aria-hidden="true" tabindex="-1"></a>out_folder <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-11-22-retrain_expression_predictor/evaluation_plots"</span></span>
<span id="cb27-134"><a href="#cb27-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-135"><a href="#cb27-135" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>, <span class="st">"all"</span> <span class="ot">=</span> <span class="st">"grey"</span>)</span>
<span id="cb27-136"><a href="#cb27-136" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-137"><a href="#cb27-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-138"><a href="#cb27-138" aria-hidden="true" tabindex="-1"></a>Plot number of genes per chrom and data subset</span>
<span id="cb27-141"><a href="#cb27-141" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-142"><a href="#cb27-142" aria-hidden="true" tabindex="-1"></a>gene_counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">as.matrix</span>(<span class="fu">table</span>(full_df<span class="sc">$</span>data_subset)))</span>
<span id="cb27-143"><a href="#cb27-143" aria-hidden="true" tabindex="-1"></a>gene_counts_df[<span class="st">"all"</span>,<span class="st">"V1"</span>] <span class="ot">&lt;-</span> <span class="fu">nrow</span>(full_df)</span>
<span id="cb27-144"><a href="#cb27-144" aria-hidden="true" tabindex="-1"></a>gene_counts_df<span class="sc">$</span>grouping <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gene_counts_df)</span>
<span id="cb27-145"><a href="#cb27-145" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_counts_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene_count"</span>, <span class="st">"data_subset"</span>)</span>
<span id="cb27-146"><a href="#cb27-146" aria-hidden="true" tabindex="-1"></a>gene_counts_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(gene_counts_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb27-147"><a href="#cb27-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-148"><a href="#cb27-148" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{out_folder}/counts_data_subsets.png"</span>), <span class="at">width=</span><span class="dv">800</span>, <span class="at">height=</span><span class="dv">400</span>)</span>
<span id="cb27-149"><a href="#cb27-149" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors) <span class="sc">+</span> <span class="fu">theme_minimal</span>(<span class="at">base_size=</span><span class="dv">15</span>)</span>
<span id="cb27-150"><a href="#cb27-150" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb27-151"><a href="#cb27-151" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)<span class="sc">+</span> <span class="fu">theme_minimal</span>(<span class="at">base_size=</span><span class="dv">15</span>)</span>
<span id="cb27-152"><a href="#cb27-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-153"><a href="#cb27-153" aria-hidden="true" tabindex="-1"></a>gene_counts_chrom_df <span class="ot">&lt;-</span> <span class="fu">distinct</span>(full_df[,<span class="fu">c</span>(<span class="st">"data_subset"</span>, <span class="st">"chroms"</span>)])</span>
<span id="cb27-154"><a href="#cb27-154" aria-hidden="true" tabindex="-1"></a>gene_counts_chrom_df<span class="sc">$</span>gene_count <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">as.matrix</span>(<span class="fu">table</span>(full_df<span class="sc">$</span>chroms)))[gene_counts_chrom_df<span class="sc">$</span>chroms,]</span>
<span id="cb27-155"><a href="#cb27-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-156"><a href="#cb27-156" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{out_folder}/counts_all_chroms_both.png"</span>), <span class="at">width=</span><span class="dv">800</span>, <span class="at">height=</span><span class="dv">400</span>)</span>
<span id="cb27-157"><a href="#cb27-157" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_chrom_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chroms, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)<span class="sc">+</span> <span class="fu">theme_minimal</span>(<span class="at">base_size=</span><span class="dv">15</span>)</span>
<span id="cb27-158"><a href="#cb27-158" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb27-159"><a href="#cb27-159" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_counts_chrom_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chroms, <span class="at">y=</span>gene_count, <span class="at">fill=</span>data_subset), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset or chromosome"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Gene count"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)<span class="sc">+</span> <span class="fu">theme_minimal</span>(<span class="at">base_size=</span><span class="dv">15</span>)</span>
<span id="cb27-160"><a href="#cb27-160" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-161"><a href="#cb27-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-162"><a href="#cb27-162" aria-hidden="true" tabindex="-1"></a>Plot correlation patterns within data subsets</span>
<span id="cb27-163"><a href="#cb27-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-166"><a href="#cb27-166" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-167"><a href="#cb27-167" aria-hidden="true" tabindex="-1"></a>data_subsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>)</span>
<span id="cb27-168"><a href="#cb27-168" aria-hidden="true" tabindex="-1"></a>cor_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet_pred_flu_FLU"</span>, <span class="st">"glmnet_pred_ni_FLU"</span>, <span class="st">"glmnet_pred_ni_NI"</span>, <span class="st">"log_CAGE_epi_FLU"</span>, <span class="st">"CAGE_epi_FLU"</span>, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb27-169"><a href="#cb27-169" aria-hidden="true" tabindex="-1"></a>cor_vecs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet_pred_flu"</span>, <span class="st">"glmnet_pred_ni"</span>, <span class="st">"glmnet_pred_ni"</span>, <span class="st">"log_CAGE_epi"</span>, <span class="st">"CAGE_epi"</span>, <span class="st">"true_exp_flu"</span>)</span>
<span id="cb27-170"><a href="#cb27-170" aria-hidden="true" tabindex="-1"></a>cor_against <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_ni"</span>, <span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_flu"</span>, <span class="st">"true_exp_ni"</span>)</span>
<span id="cb27-171"><a href="#cb27-171" aria-hidden="true" tabindex="-1"></a>cor_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"spearman"</span>,<span class="st">"pearson"</span>)</span>
<span id="cb27-172"><a href="#cb27-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-173"><a href="#cb27-173" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow =</span> <span class="fu">length</span>(data_subsets)<span class="sc">*</span><span class="fu">length</span>(cor_vecs)<span class="sc">*</span><span class="fu">length</span>(cor_types), <span class="at">ncol =</span> <span class="dv">4</span>)</span>
<span id="cb27-174"><a href="#cb27-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-175"><a href="#cb27-175" aria-hidden="true" tabindex="-1"></a>i <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb27-176"><a href="#cb27-176" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> data_subsets) {</span>
<span id="cb27-177"><a href="#cb27-177" aria-hidden="true" tabindex="-1"></a>  cur_subset_df <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> data_subset,]</span>
<span id="cb27-178"><a href="#cb27-178" aria-hidden="true" tabindex="-1"></a>  cur_subset_df <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>data_subset <span class="sc">==</span> data_subset,]</span>
<span id="cb27-179"><a href="#cb27-179" aria-hidden="true" tabindex="-1"></a>  cur_subset_df <span class="ot">&lt;-</span> cur_subset_df[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">is.na</span>(cur_subset_df)) <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb27-180"><a href="#cb27-180" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (cor_type <span class="cf">in</span> cor_types) {</span>
<span id="cb27-181"><a href="#cb27-181" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (c <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(cor_names)) {</span>
<span id="cb27-182"><a href="#cb27-182" aria-hidden="true" tabindex="-1"></a>      cur_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(cur_subset_df[,cor_vecs[c]], cur_subset_df[,cor_against[c]], <span class="at">method =</span> cor_type)</span>
<span id="cb27-183"><a href="#cb27-183" aria-hidden="true" tabindex="-1"></a>      corrs_df_for_subsets[i,] <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset, cor_names[c], cor_type, cur_cor)</span>
<span id="cb27-184"><a href="#cb27-184" aria-hidden="true" tabindex="-1"></a>      i <span class="ot">=</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb27-185"><a href="#cb27-185" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb27-186"><a href="#cb27-186" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-187"><a href="#cb27-187" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-188"><a href="#cb27-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-189"><a href="#cb27-189" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(corrs_df_for_subsets)</span>
<span id="cb27-190"><a href="#cb27-190" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(corrs_df_for_subsets) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"data_subset"</span>, <span class="st">"cor_id"</span>, <span class="st">"cor_type"</span>, <span class="st">"corrs"</span>)</span>
<span id="cb27-191"><a href="#cb27-191" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets<span class="sc">$</span>cor_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(corrs_df_for_subsets<span class="sc">$</span>cor_id, <span class="at">levels =</span> cor_names)</span>
<span id="cb27-192"><a href="#cb27-192" aria-hidden="true" tabindex="-1"></a>corrs_df_for_subsets<span class="sc">$</span>corrs <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(corrs_df_for_subsets<span class="sc">$</span>corrs))</span>
<span id="cb27-193"><a href="#cb27-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-194"><a href="#cb27-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-195"><a href="#cb27-195" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{out_folder}/all_corrs_subsets.png"</span>), <span class="at">width=</span><span class="dv">600</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb27-196"><a href="#cb27-196" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_for_subsets) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>corrs, <span class="at">fill=</span>cor_id), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>cor_type, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_colour_brewer</span>(<span class="at">palette =</span> <span class="st">"Pastel1"</span>)<span class="sc">+</span> <span class="fu">theme_minimal</span>(<span class="at">base_size=</span><span class="dv">15</span>)</span>
<span id="cb27-197"><a href="#cb27-197" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb27-198"><a href="#cb27-198" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_for_subsets) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset, <span class="at">y=</span>corrs, <span class="at">fill=</span>cor_id), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>cor_type, <span class="at">ncol=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_colour_brewer</span>(<span class="at">palette =</span> <span class="st">"Pastel1"</span>)<span class="sc">+</span> <span class="fu">theme_minimal</span>(<span class="at">base_size=</span><span class="dv">15</span>)</span>
<span id="cb27-199"><a href="#cb27-199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-200"><a href="#cb27-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-201"><a href="#cb27-201" aria-hidden="true" tabindex="-1"></a>Plot correlation patterns within chroms</span>
<span id="cb27-202"><a href="#cb27-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-205"><a href="#cb27-205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-206"><a href="#cb27-206" aria-hidden="true" tabindex="-1"></a><span class="co"># chroms &lt;- c("all", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y")</span></span>
<span id="cb27-207"><a href="#cb27-207" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_vecs &lt;- c("glmnet_pred", "CAGE_epi", "log_CAGE_epi")</span></span>
<span id="cb27-208"><a href="#cb27-208" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_types &lt;- c("spearman","pearson")</span></span>
<span id="cb27-209"><a href="#cb27-209" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df &lt;- matrix(nrow = length(chroms)*length(cor_vecs)*length(cor_types), ncol = 4)</span></span>
<span id="cb27-210"><a href="#cb27-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-211"><a href="#cb27-211" aria-hidden="true" tabindex="-1"></a><span class="co"># i = 1</span></span>
<span id="cb27-212"><a href="#cb27-212" aria-hidden="true" tabindex="-1"></a><span class="co"># for (chrom in chroms) {</span></span>
<span id="cb27-213"><a href="#cb27-213" aria-hidden="true" tabindex="-1"></a><span class="co">#   cur_subset_df &lt;- full_df[full_df$chroms == chrom,]</span></span>
<span id="cb27-214"><a href="#cb27-214" aria-hidden="true" tabindex="-1"></a><span class="co">#   cur_subset_df &lt;- cur_subset_df[!rowSums(is.na(cur_subset_df)) &gt; 0,]</span></span>
<span id="cb27-215"><a href="#cb27-215" aria-hidden="true" tabindex="-1"></a><span class="co">#   for (cor_type in cor_types) {</span></span>
<span id="cb27-216"><a href="#cb27-216" aria-hidden="true" tabindex="-1"></a><span class="co">#     for (cor_vec in cor_vecs) {</span></span>
<span id="cb27-217"><a href="#cb27-217" aria-hidden="true" tabindex="-1"></a><span class="co">#       cur_cor &lt;- cor(cur_subset_df[,cor_vec], cur_subset_df$true_exp, method = cor_type)</span></span>
<span id="cb27-218"><a href="#cb27-218" aria-hidden="true" tabindex="-1"></a><span class="co">#       cor_subset_df[i,] &lt;- c(chrom, cor_vec, cor_type, cur_cor)</span></span>
<span id="cb27-219"><a href="#cb27-219" aria-hidden="true" tabindex="-1"></a><span class="co">#       i = i + 1</span></span>
<span id="cb27-220"><a href="#cb27-220" aria-hidden="true" tabindex="-1"></a><span class="co">#     }</span></span>
<span id="cb27-221"><a href="#cb27-221" aria-hidden="true" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb27-222"><a href="#cb27-222" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb27-223"><a href="#cb27-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-224"><a href="#cb27-224" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df &lt;- as.data.frame(cor_subset_df)</span></span>
<span id="cb27-225"><a href="#cb27-225" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(cor_subset_df) &lt;- c("chrom", "cor_id", "cor_type", "corrs")</span></span>
<span id="cb27-226"><a href="#cb27-226" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df$corrs &lt;- as.numeric(as.character(cor_subset_df$corrs))</span></span>
<span id="cb27-227"><a href="#cb27-227" aria-hidden="true" tabindex="-1"></a><span class="co"># cor_subset_df$cor_id &lt;- factor(cor_subset_df$cor_id, levels = names(corr_colors))</span></span>
<span id="cb27-228"><a href="#cb27-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-229"><a href="#cb27-229" aria-hidden="true" tabindex="-1"></a><span class="co"># png(glue("{out_folder}/all_corrs_chroms.png"), width=1000, height=500)</span></span>
<span id="cb27-230"><a href="#cb27-230" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(cor_subset_df) + geom_bar(aes(x=chrom, y=corrs, fill=cor_id), stat="identity", position="dodge") + xlab("Data subset") + ylab("Correlation") + facet_wrap(~cor_type, ncol=1) + scale_fill_manual(values = corr_colors) +ylim(c(0,1))</span></span>
<span id="cb27-231"><a href="#cb27-231" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb27-232"><a href="#cb27-232" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(cor_subset_df) + geom_bar(aes(x=chrom, y=corrs, fill=cor_id), stat="identity", position="dodge") + xlab("Data subset") + ylab("Correlation") + facet_wrap(~cor_type, ncol=1) + scale_fill_manual(values = corr_colors) +ylim(c(0,1))</span></span>
<span id="cb27-233"><a href="#cb27-233" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-234"><a href="#cb27-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-235"><a href="#cb27-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-236"><a href="#cb27-236" aria-hidden="true" tabindex="-1"></a>Plot scatterplots of predictions vs true expression</span>
<span id="cb27-237"><a href="#cb27-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-240"><a href="#cb27-240" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-241"><a href="#cb27-241" aria-hidden="true" tabindex="-1"></a><span class="co"># full_df_plot &lt;- full_df[!rowSums(is.na(full_df)) &gt; 0,]</span></span>
<span id="cb27-242"><a href="#cb27-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-243"><a href="#cb27-243" aria-hidden="true" tabindex="-1"></a><span class="co"># png(glue("{out_folder}/all_records_scatterplots.png"), width =900, height=900)</span></span>
<span id="cb27-244"><a href="#cb27-244" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=glmnet_pred, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("glmnet_pred")+ylab("Ground truth")+theme(panel.background = element_rect(fill = alpha(corr_colors["glmnet_pred"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb27-245"><a href="#cb27-245" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["CAGE_epi"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb27-246"><a href="#cb27-246" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=log_CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("log_CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["log_CAGE_epi"],alpha=0.4)))+scale_color_manual(values=subset_colors))</span></span>
<span id="cb27-247"><a href="#cb27-247" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb27-248"><a href="#cb27-248" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=glmnet_pred, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("glmnet_pred")+ylab("Ground truth")+theme(panel.background = element_rect(fill = alpha(corr_colors["glmnet_pred"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb27-249"><a href="#cb27-249" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["CAGE_epi"],0.4)))+scale_color_manual(values=subset_colors)) /</span></span>
<span id="cb27-250"><a href="#cb27-250" aria-hidden="true" tabindex="-1"></a><span class="co"># (ggplot(full_df_plot) + geom_point(aes(x=log_CAGE_epi, y=true_exp, color=data_subset)) + facet_wrap(~data_subset, nrow=1, scales="free")+xlab("log_CAGE_epi")+theme(panel.background = element_rect(fill = alpha(corr_colors["log_CAGE_epi"],alpha=0.4)))+scale_color_manual(values=subset_colors))</span></span>
<span id="cb27-251"><a href="#cb27-251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>