<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-10-22">

<title>Daily-Blog-Sai - evaluate_tss_center_mean_pred</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">evaluate_tss_center_mean_pred</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 22, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"evaluate_tss_center_mean_pred"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-10-22'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="context" class="level1">
<h1>Context</h1>
<p>Here I trained an elastic net and simple MLP mode to predict TSS RNAseq expression from Flu condition (aracena dataset) from Enformer predictions. Training was done using corpus of TSS sites from a ranmdom mixture of individuals. Here I am evaluating the performance of the trained models on a held out test set.</p>
<p>TODO: Average of individuals TODO: Collect RNAseq from Katie, and use this for DL-EN training AND training PrediXcan TODO: Try different MLP params</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(neuralnet)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>predict_on_data_subset <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, glmnet_model_path, nnmodel_path){</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Predicting for data subset: {data_subset}"</span>))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load trained models</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/glmnet_lambdas.png"</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_model)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load dataset to be predicted on</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  data_X <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  data_y <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv"</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_X)))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_y)))</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make predictions and associated plots</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_corr_en_pred.png"</span>))</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_predictions, y[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"glmnet predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Glmnet predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  pred_corr_en <span class="ot">&lt;-</span> <span class="fu">cor</span>(glmnet_predictions, y[,<span class="dv">1</span>])</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_en} for {data_subset} data subset with glmnet predictions"</span>))</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_corr_nn_pred.png"</span>))</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(predictions_nn, y[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"NN predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"NN predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  pred_corr_nn <span class="ot">&lt;-</span> <span class="fu">cor</span>(predictions_nn, y[,<span class="dv">1</span>])</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_nn} for {data_subset} data subset with NN predictions"</span>))</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>analyze_input_data <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset) {</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for data subset: {data_subset}"</span>))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>))))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv"</span>))))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(X))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(y))</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_flu_vs_NI_rnaseq.png"</span>))</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>],<span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>),<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"ground truth NI rnaseq"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Flu vs. NI in ground truth for {data_subset} data subset"</span>))</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  corr_rnaseq <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>])</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_rnaseq} for {data_subset} data subset between flu/NI rnaseq"</span>))</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_flu_vs_epi_rnaseq.png"</span>))</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena flu rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_NI_vs_epi_rnaseq.png"</span>))</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena NI rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we can use the trained models to make predictions and assess model performance.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"train"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for data subset: train
[1] 12232  5313
[1] 12232    12</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.720936539936496 for train data subset between flu/NI rnaseq</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.197374238365873 for train data subset between aracena flu rnaseq and corresponding enformer prediction</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.220588426286466 for train data subset between aracena NI rnaseq and corresponding enformer prediction</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"valid"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for data subset: valid
[1] 1577 5313
[1] 1577   12</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.700955195863934 for valid data subset between flu/NI rnaseq</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.385249011868238 for valid data subset between aracena flu rnaseq and corresponding enformer prediction</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.40371591624657 for valid data subset between aracena NI rnaseq and corresponding enformer prediction</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"test"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for data subset: test
[1] 2041 5313
[1] 2041   12</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.998305261247481 for test data subset between flu/NI rnaseq</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0465746240713448 for test data subset between aracena flu rnaseq and corresponding enformer prediction</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0563540836639804 for test data subset between aracena NI rnaseq and corresponding enformer prediction</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>glmnet_model_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.linear.rds"</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>nnmodel_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp_RNAseq_from_HDF5_mean"</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"train"</span>, glmnet_model_path, nnmodel_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Predicting for data subset: train</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.564373045433215 for train data subset with glmnet predictions</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.461306680081697 for train data subset with NN predictions</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"valid"</span>, glmnet_model_path, nnmodel_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Predicting for data subset: valid</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.72381811933344 for valid data subset with glmnet predictions</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.380274064824189 for valid data subset with NN predictions</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"test"</span>, glmnet_model_path, nnmodel_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Predicting for data subset: test</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0824962133319529 for test data subset with glmnet predictions</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Correlation of 0.0202890063481298 for test data subset with NN predictions</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>convert_region_to_gene <span class="ot">&lt;-</span> <span class="cf">function</span>(region, gene_metadata){</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> <span class="st">"1_1471765_1471767"</span>,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> <span class="st">"1_1471765_1471767"</span>,]<span class="sc">$</span>ensembl_gene_id</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENSG00000160072"</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>split_datasets_chroms <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(chroms)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>plot_boxplot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, path) {</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>is_CAGE <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"CAGE_epi"</span>, <span class="st">"CAGE"</span>, <span class="st">"RNAseq"</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(path)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>variable, <span class="at">y=</span>value)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_CAGE, <span class="at">scales=</span><span class="st">"free"</span>))</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>run_cor<span class="ot">&lt;-</span> <span class="cf">function</span>(x,y, <span class="at">method=</span><span class="st">"pearson"</span>) {</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">cor</span>(x,y, <span class="at">method=</span>method))</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>per_chrom_metrics <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path, <span class="at">cor_method=</span><span class="st">"spearman"</span>) {</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for data subset: {data_subset}"</span>))</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions, y[,<span class="dv">1</span>]))</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions, y[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage nn corr</span></span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn, y[,<span class="dv">1</span>]))</span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn, y[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>]))</span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>]))</span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Done recording</span></span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a>  chroms_all <span class="ot">&lt;-</span> <span class="fu">split_datasets_chroms</span>(data_subset, mapping_table)</span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(chroms_all)</span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a>  uniq_chroms <span class="ot">&lt;-</span> <span class="fu">unique</span>(chroms)</span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (chrom <span class="cf">in</span> uniq_chroms) {</span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a>    index_of_chrom <span class="ot">&lt;-</span> chroms <span class="sc">==</span> chrom</span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a>    X_chrom <span class="ot">&lt;-</span> X[index_of_chrom,]</span>
<span id="cb38-64"><a href="#cb38-64" aria-hidden="true" tabindex="-1"></a>    y_chrom <span class="ot">&lt;-</span> y[index_of_chrom,]</span>
<span id="cb38-65"><a href="#cb38-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-66"><a href="#cb38-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for chrom: {chrom}"</span>))</span>
<span id="cb38-67"><a href="#cb38-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(X_chrom))</span>
<span id="cb38-68"><a href="#cb38-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(y_chrom))</span>
<span id="cb38-69"><a href="#cb38-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-70"><a href="#cb38-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>]))</span>
<span id="cb38-71"><a href="#cb38-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">1</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb38-72"><a href="#cb38-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">7</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb38-73"><a href="#cb38-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-74"><a href="#cb38-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb38-75"><a href="#cb38-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(predictions_nn[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb38-76"><a href="#cb38-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-77"><a href="#cb38-77" aria-hidden="true" tabindex="-1"></a>    chrom_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">glmnet_pred=</span>glmnet_predictions[index_of_chrom], <span class="at">flu_rnaseq =</span> y_chrom[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y_chrom[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X_chrom[,<span class="dv">4766</span>])</span>
<span id="cb38-78"><a href="#cb38-78" aria-hidden="true" tabindex="-1"></a>    chrom_df_for_plotting <span class="ot">&lt;-</span> <span class="fu">melt</span>(chrom_df, <span class="at">measure_vars =</span> <span class="fu">colnames</span>(chrom_df))</span>
<span id="cb38-79"><a href="#cb38-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_boxplot</span>(chrom_df_for_plotting, <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_chrom_{chrom}_boxplot.png"</span>))</span>
<span id="cb38-80"><a href="#cb38-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-81"><a href="#cb38-81" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb38-82"><a href="#cb38-82" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-83"><a href="#cb38-83" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb38-84"><a href="#cb38-84" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-85"><a href="#cb38-85" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb38-86"><a href="#cb38-86" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb38-87"><a href="#cb38-87" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb38-88"><a href="#cb38-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-89"><a href="#cb38-89" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage nn corr</span></span>
<span id="cb38-90"><a href="#cb38-90" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-91"><a href="#cb38-91" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb38-92"><a href="#cb38-92" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-93"><a href="#cb38-93" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb38-94"><a href="#cb38-94" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb38-95"><a href="#cb38-95" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb38-96"><a href="#cb38-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-97"><a href="#cb38-97" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb38-98"><a href="#cb38-98" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-99"><a href="#cb38-99" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>]))</span>
<span id="cb38-100"><a href="#cb38-100" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-101"><a href="#cb38-101" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb38-102"><a href="#cb38-102" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb38-103"><a href="#cb38-103" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb38-104"><a href="#cb38-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-105"><a href="#cb38-105" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb38-106"><a href="#cb38-106" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb38-107"><a href="#cb38-107" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb38-108"><a href="#cb38-108" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], X_chrom[,<span class="dv">4766</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb38-109"><a href="#cb38-109" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb38-110"><a href="#cb38-110" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb38-111"><a href="#cb38-111" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb38-112"><a href="#cb38-112" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb38-113"><a href="#cb38-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-114"><a href="#cb38-114" aria-hidden="true" tabindex="-1"></a>  storage_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(corrs_v, corrs_v_spear, data_subset_v, chrom_v, num_genes_v, cor_id_v)</span>
<span id="cb38-115"><a href="#cb38-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-116"><a href="#cb38-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(storage_df)</span>
<span id="cb38-117"><a href="#cb38-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-118"><a href="#cb38-118" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>split_datasets <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  ensembl_ids <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>  external_gene_names <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(regions)) {</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>    region <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions[x],<span class="dv">2</span>,<span class="fu">str_length</span>(regions[x]))</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>    ensembl_id <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>    external_gene_name <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>external_gene_name</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>    ensembl_ids[x]  <span class="ot">&lt;-</span> ensembl_id</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>    external_gene_names[x] <span class="ot">&lt;-</span> external_gene_name</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(ensembl_ids))</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(external_gene_names))</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(chroms))</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(regions))</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>  return_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"chroms"</span>]] <span class="ot">&lt;-</span> chroms</span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"regions"</span>]] <span class="ot">&lt;-</span> regions</span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"ensembl_ids"</span>]] <span class="ot">&lt;-</span> ensembl_ids</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"external_gene_names"</span>]] <span class="ot">&lt;-</span> external_gene_names</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_list)</span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>collect_all_records <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path) {</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Collecting data records for data subset: {data_subset}"</span>))</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>  gene_metadata <span class="ot">&lt;-</span> <span class="fu">split_datasets</span>(data_subset, mapping_table)</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  return_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">data_subset=</span> data_subset, <span class="at">glmnet_pred=</span>glmnet_predictions, <span class="at">nn_pred=</span> predictions_nn, <span class="at">flu_rnaseq =</span> y[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X[,<span class="dv">4766</span>], <span class="at">ensembl_id =</span> gene_metadata[[<span class="st">"ensembl_ids"</span>]], <span class="at">external_gene_name =</span> gene_metadata[[<span class="st">"external_gene_names"</span>]], <span class="at">region =</span> gene_metadata[[<span class="st">"regions"</span>]], <span class="at">chroms =</span> gene_metadata[[<span class="st">"chroms"</span>]])</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_df)</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>remapped_table_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/remapped_table_filt.csv"</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>remapped_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(remapped_table_path, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(remapped_table))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1]    29 15851</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">as.character</span>(remapped_table[,<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "EU39"          "EU27"          "AF26"          "EU33"         
 [5] "AF18"          "AF38"          "AF20"          "AF22"         
 [9] "AF08"          "AF28"          "EU25"          "AF24"         
[13] "EU05"          "AF34"          "EU47"          "EU13"         
[17] "AF36"          "EU03"          "AF30"          "AF12"         
[21] "EU09"          "EU15"          "EU07"          "EU21"         
[25] "AF16"          "EU19"          "EU41"          "ref-epigenome"
[29] "group"        </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(remapped_table) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(remapped_table[,<span class="dv">1</span>])</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>remapped_table[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>glmnet_model_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.linear.rds"</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>nnmodel_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp_RNAseq_from_HDF5_mean"</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>cor_method <span class="ot">=</span> <span class="st">"spearman"</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>all_records_df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"train"</span>,<span class="st">"test"</span>,<span class="st">"valid"</span>)){</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  X_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  y_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv"</span>)</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  cur_out_df_perchrom_metrics <span class="ot">&lt;-</span> <span class="fu">per_chrom_metrics</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path, cor_method)</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  df_store[[c]] <span class="ot">&lt;-</span> cur_out_df_perchrom_metrics</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  all_records_df_store[[c]] <span class="ot">&lt;-</span> <span class="fu">collect_all_records</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>  c<span class="ot">=</span>c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for data subset: train
********* Analyzing data for chrom: 1
[1] 2014 5313
[1] 2014   12
[1] 0.585159
[1] 0.3320535
[1] 0.3200134
[1] 0.7007275
[1] 0.4931243</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 10
[1]  710 5313
[1] 710  12
[1] 0.4914441
[1] 0.2711813
[1] 0.4177587
[1] 0.7732099
[1] 0.6417866</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 11
[1] 1287 5313
[1] 1287   12
[1] 0.4685465
[1] 0.3471125
[1] 0.4118005
[1] 0.773881
[1] 0.5169719</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 16
[1]  839 5313
[1] 839  12
[1] 0.4881721
[1] 0.275333
[1] 0.4377035
[1] 0.5745219
[1] 0.4885602</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 17
[1] 1152 5313
[1] 1152   12
[1] 0.4668504
[1] 0.2194275
[1] 0.3007733
[1] 0.6878922
[1] 0.5306309</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 18
[1]  262 5313
[1] 262  12
[1] 0.7290399
[1] 0.3850517
[1] 0.4533551
[1] 0.6865047
[1] 0.2268427</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 19
[1] 1423 5313
[1] 1423   12
[1] 0.9739646
[1] 0.1549743
[1] 0.1543157
[1] 0.5526522
[1] 0.3117509</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 2
[1] 1225 5313
[1] 1225   12
[1] 0.2905684
[1] 0.1839524
[1] 0.3956614
[1] 0.6216482
[1] 0.6836889</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 20
[1]  533 5313
[1] 533  12
[1] 0.8520808
[1] 0.3117639
[1] 0.3367319
[1] 0.5004507
[1] 0.2347415</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 21
[1]  218 5313
[1] 218  12
[1] 0.6917517
[1] 0.4573482
[1] 0.6141122
[1] 0.7530417
[1] 0.3123373</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 3
[1] 1059 5313
[1] 1059   12
[1] 0.4420113
[1] 0.3056034
[1] 0.413794
[1] 0.6469695
[1] 0.3702832</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 4
[1]  748 5313
[1] 748  12
[1] 0.1806992
[1] 0.05613264
[1] 0.397568
[1] 0.5375329
[1] 0.8132907</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 9
[1]  762 5313
[1] 762  12
[1] 0.7291291
[1] 0.2504148
[1] 0.4520028
[1] 0.5974853
[1] 0.3584159</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: train
[1] 12232
[1] 12232
[1] 12232
[1] 12232
********* Analyzing data for data subset: test
********* Analyzing data for chrom: 13
[1]  319 5313
[1] 319  12
[1] 0.7429337
[1] 0.430552
[1] 0.4427821
[1] 0.7805843
[1] 0.6003512</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 14
[1]  600 5313
[1] 600  12
[1] 0.9994539
[1] 0.072076
[1] 0.07470177
[1] 0.1531511
[1] 0.05186287</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 22
[1]  431 5313
[1] 431  12
[1] 0.3050559
[1] 0.2059432
[1] 0.484827
[1] 0.4394194
[1] 0.2096601</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 8
[1]  691 5313
[1] 691  12
[1] 0.6345597
[1] 0.2186081
[1] 0.4057245
[1] 0.5265987
[1] 0.2321117</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: test
[1] 2041
[1] 2041
[1] 2041
[1] 2041
********* Analyzing data for data subset: valid
********* Analyzing data for chrom: 12
[1]  990 5313
[1] 990  12
[1] 0.6603037
[1] 0.333304
[1] 0.3614893
[1] 0.6966347
[1] 0.2454961</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 15
[1]  587 5313
[1] 587  12
[1] 0.8018562
[1] 0.4498818
[1] 0.4859656
[1] 0.8063956
[1] 0.4831483</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: valid
[1] 1577
[1] 1577
[1] 1577
[1] 1577</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df_store)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>data_subset_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>data_subset_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>cor_id_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>cor_id_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>))</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>chrom_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>chrom_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>full_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>      corrs_v corrs_v_spear data_subset_v chrom_v num_genes_v   cor_id_v
1  0.56437305     0.4400607         train     all       12232     glmnet
2  0.46130668     0.6149674         train     all       12232         nn
3  0.72093654     0.8981904         train     all       12232  flu_vs_NI
4  0.19737424     0.7190657         train     all       12232 flu_vs_epi
5  0.70072752     0.4759208         train       1        2014     glmnet
6  0.49312427     0.6343898         train       1        2014         nn
7  0.58515902     0.9016380         train       1        2014  flu_vs_NI
8  0.33205353     0.7298823         train       1        2014 flu_vs_epi
9  0.77320992     0.4772985         train      10         710     glmnet
10 0.64178663     0.6487913         train      10         710         nn
11 0.49144409     0.8878139         train      10         710  flu_vs_NI
12 0.27118134     0.7187106         train      10         710 flu_vs_epi
13 0.77388098     0.3879118         train      11        1287     glmnet
14 0.51697192     0.6607209         train      11        1287         nn
15 0.46854652     0.9059549         train      11        1287  flu_vs_NI
16 0.34711246     0.7530519         train      11        1287 flu_vs_epi
17 0.57452192     0.4828096         train      16         839     glmnet
18 0.48856024     0.5234316         train      16         839         nn
19 0.48817208     0.8933155         train      16         839  flu_vs_NI
20 0.27533304     0.6388009         train      16         839 flu_vs_epi
21 0.68789220     0.4667182         train      17        1152     glmnet
22 0.53063095     0.5859384         train      17        1152         nn
23 0.46685038     0.8930906         train      17        1152  flu_vs_NI
24 0.21942752     0.7038228         train      17        1152 flu_vs_epi
25 0.68650467     0.4426711         train      18         262     glmnet
26 0.22684268     0.6387822         train      18         262         nn
27 0.72903990     0.9154724         train      18         262  flu_vs_NI
28 0.38505169     0.7902356         train      18         262 flu_vs_epi
29 0.55265220     0.4056606         train      19        1423     glmnet
30 0.31175090     0.5672843         train      19        1423         nn
31 0.97396459     0.8988924         train      19        1423  flu_vs_NI
32 0.15497429     0.6557346         train      19        1423 flu_vs_epi
33 0.62164822     0.4493787         train       2        1225     glmnet
34 0.68368886     0.6006122         train       2        1225         nn
35 0.29056839     0.9029122         train       2        1225  flu_vs_NI
36 0.18395238     0.7360532         train       2        1225 flu_vs_epi
37 0.50045073     0.4321967         train      20         533     glmnet
38 0.23474151     0.6430574         train      20         533         nn
39 0.85208081     0.9036955         train      20         533  flu_vs_NI
40 0.31176386     0.7297021         train      20         533 flu_vs_epi
41 0.75304174     0.5479164         train      21         218     glmnet
42 0.31233726     0.7711796         train      21         218         nn
43 0.69175174     0.9274325         train      21         218  flu_vs_NI
44 0.45734825     0.8176841         train      21         218 flu_vs_epi
45 0.64696953     0.4364245         train       3        1059     glmnet
46 0.37028322     0.6337275         train       3        1059         nn
47 0.44201128     0.8951994         train       3        1059  flu_vs_NI
48 0.30560340     0.7297525         train       3        1059 flu_vs_epi
49 0.53753285     0.3664512         train       4         748     glmnet
50 0.81329066     0.6168320         train       4         748         nn
51 0.18069922     0.9006577         train       4         748  flu_vs_NI
52 0.05613264     0.7691620         train       4         748 flu_vs_epi
53 0.59748531     0.4263626         train       9         762     glmnet
54 0.35841591     0.5390046         train       9         762         nn
55 0.72912914     0.8530804         train       9         762  flu_vs_NI
56 0.25041484     0.6488148         train       9         762 flu_vs_epi
57 0.08249621     0.4208230          test     all        2041     glmnet
58 0.02028901     0.6142361          test     all        2041         nn
59 0.99830526     0.8965072          test     all        2041  flu_vs_NI
60 0.04657462     0.7382084          test     all        2041 flu_vs_epi
61 0.78058434     0.3894792          test      13         319     glmnet
62 0.60035120     0.6189132          test      13         319         nn
63 0.74293368     0.8793152          test      13         319  flu_vs_NI
64 0.43055203     0.7464936          test      13         319 flu_vs_epi
65 0.15315112     0.4327574          test      14         600     glmnet
66 0.05186287     0.5917317          test      14         600         nn
67 0.99945392     0.9133753          test      14         600  flu_vs_NI
68 0.07207600     0.6967590          test      14         600 flu_vs_epi
69 0.43941944     0.4661211          test      22         431     glmnet
70 0.20966013     0.6194520          test      22         431         nn
71 0.30505592     0.8901027          test      22         431  flu_vs_NI
72 0.20594316     0.7080015          test      22         431 flu_vs_epi
73 0.52659868     0.3980502          test       8         691     glmnet
74 0.23211174     0.6376234          test       8         691         nn
75 0.63455974     0.8866055          test       8         691  flu_vs_NI
76 0.21860805     0.7755985          test       8         691 flu_vs_epi
77 0.72381812     0.4566016         valid     all        1577     glmnet
78 0.38027406     0.6083679         valid     all        1577         nn
79 0.70095520     0.8977642         valid     all        1577  flu_vs_NI
80 0.38524901     0.7205278         valid     all        1577 flu_vs_epi
81 0.69663472     0.4578672         valid      12         990     glmnet
82 0.24549606     0.5868101         valid      12         990         nn
83 0.66030373     0.8924773         valid      12         990  flu_vs_NI
84 0.33330399     0.7015403         valid      12         990 flu_vs_epi
85 0.80639559     0.4517969         valid      15         587     glmnet
86 0.48314833     0.6449347         valid      15         587         nn
87 0.80185617     0.9048721         valid      15         587  flu_vs_NI
88 0.44988177     0.7534411         valid      15         587 flu_vs_epi</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>num_genes_v <span class="ot">&lt;-</span> <span class="fu">as.character</span>(full_df<span class="sc">$</span>num_genes_v)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(full_df, <span class="at">measure_vars=</span><span class="fu">c</span>(<span class="st">"corrs_v"</span>,<span class="st">"corrs_v_spear"</span>), <span class="at">id_vars=</span><span class="fu">c</span>(<span class="st">"data_subset_v"</span>, <span class="st">"chrom_v"</span>, <span class="st">"cor_id_v"</span>, <span class="st">"num_genes_v"</span>))</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>full_df_melt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>    data_subset_v chrom_v num_genes_v   cor_id_v      variable      value
1           train     all       12232     glmnet       corrs_v 0.56437305
2           train     all       12232         nn       corrs_v 0.46130668
3           train     all       12232  flu_vs_NI       corrs_v 0.72093654
4           train     all       12232 flu_vs_epi       corrs_v 0.19737424
5           train       1        2014     glmnet       corrs_v 0.70072752
6           train       1        2014         nn       corrs_v 0.49312427
7           train       1        2014  flu_vs_NI       corrs_v 0.58515902
8           train       1        2014 flu_vs_epi       corrs_v 0.33205353
9           train      10         710     glmnet       corrs_v 0.77320992
10          train      10         710         nn       corrs_v 0.64178663
11          train      10         710  flu_vs_NI       corrs_v 0.49144409
12          train      10         710 flu_vs_epi       corrs_v 0.27118134
13          train      11        1287     glmnet       corrs_v 0.77388098
14          train      11        1287         nn       corrs_v 0.51697192
15          train      11        1287  flu_vs_NI       corrs_v 0.46854652
16          train      11        1287 flu_vs_epi       corrs_v 0.34711246
17          train      16         839     glmnet       corrs_v 0.57452192
18          train      16         839         nn       corrs_v 0.48856024
19          train      16         839  flu_vs_NI       corrs_v 0.48817208
20          train      16         839 flu_vs_epi       corrs_v 0.27533304
21          train      17        1152     glmnet       corrs_v 0.68789220
22          train      17        1152         nn       corrs_v 0.53063095
23          train      17        1152  flu_vs_NI       corrs_v 0.46685038
24          train      17        1152 flu_vs_epi       corrs_v 0.21942752
25          train      18         262     glmnet       corrs_v 0.68650467
26          train      18         262         nn       corrs_v 0.22684268
27          train      18         262  flu_vs_NI       corrs_v 0.72903990
28          train      18         262 flu_vs_epi       corrs_v 0.38505169
29          train      19        1423     glmnet       corrs_v 0.55265220
30          train      19        1423         nn       corrs_v 0.31175090
31          train      19        1423  flu_vs_NI       corrs_v 0.97396459
32          train      19        1423 flu_vs_epi       corrs_v 0.15497429
33          train       2        1225     glmnet       corrs_v 0.62164822
34          train       2        1225         nn       corrs_v 0.68368886
35          train       2        1225  flu_vs_NI       corrs_v 0.29056839
36          train       2        1225 flu_vs_epi       corrs_v 0.18395238
37          train      20         533     glmnet       corrs_v 0.50045073
38          train      20         533         nn       corrs_v 0.23474151
39          train      20         533  flu_vs_NI       corrs_v 0.85208081
40          train      20         533 flu_vs_epi       corrs_v 0.31176386
41          train      21         218     glmnet       corrs_v 0.75304174
42          train      21         218         nn       corrs_v 0.31233726
43          train      21         218  flu_vs_NI       corrs_v 0.69175174
44          train      21         218 flu_vs_epi       corrs_v 0.45734825
45          train       3        1059     glmnet       corrs_v 0.64696953
46          train       3        1059         nn       corrs_v 0.37028322
47          train       3        1059  flu_vs_NI       corrs_v 0.44201128
48          train       3        1059 flu_vs_epi       corrs_v 0.30560340
49          train       4         748     glmnet       corrs_v 0.53753285
50          train       4         748         nn       corrs_v 0.81329066
51          train       4         748  flu_vs_NI       corrs_v 0.18069922
52          train       4         748 flu_vs_epi       corrs_v 0.05613264
53          train       9         762     glmnet       corrs_v 0.59748531
54          train       9         762         nn       corrs_v 0.35841591
55          train       9         762  flu_vs_NI       corrs_v 0.72912914
56          train       9         762 flu_vs_epi       corrs_v 0.25041484
57           test     all        2041     glmnet       corrs_v 0.08249621
58           test     all        2041         nn       corrs_v 0.02028901
59           test     all        2041  flu_vs_NI       corrs_v 0.99830526
60           test     all        2041 flu_vs_epi       corrs_v 0.04657462
61           test      13         319     glmnet       corrs_v 0.78058434
62           test      13         319         nn       corrs_v 0.60035120
63           test      13         319  flu_vs_NI       corrs_v 0.74293368
64           test      13         319 flu_vs_epi       corrs_v 0.43055203
65           test      14         600     glmnet       corrs_v 0.15315112
66           test      14         600         nn       corrs_v 0.05186287
67           test      14         600  flu_vs_NI       corrs_v 0.99945392
68           test      14         600 flu_vs_epi       corrs_v 0.07207600
69           test      22         431     glmnet       corrs_v 0.43941944
70           test      22         431         nn       corrs_v 0.20966013
71           test      22         431  flu_vs_NI       corrs_v 0.30505592
72           test      22         431 flu_vs_epi       corrs_v 0.20594316
73           test       8         691     glmnet       corrs_v 0.52659868
74           test       8         691         nn       corrs_v 0.23211174
75           test       8         691  flu_vs_NI       corrs_v 0.63455974
76           test       8         691 flu_vs_epi       corrs_v 0.21860805
77          valid     all        1577     glmnet       corrs_v 0.72381812
78          valid     all        1577         nn       corrs_v 0.38027406
79          valid     all        1577  flu_vs_NI       corrs_v 0.70095520
80          valid     all        1577 flu_vs_epi       corrs_v 0.38524901
81          valid      12         990     glmnet       corrs_v 0.69663472
82          valid      12         990         nn       corrs_v 0.24549606
83          valid      12         990  flu_vs_NI       corrs_v 0.66030373
84          valid      12         990 flu_vs_epi       corrs_v 0.33330399
85          valid      15         587     glmnet       corrs_v 0.80639559
86          valid      15         587         nn       corrs_v 0.48314833
87          valid      15         587  flu_vs_NI       corrs_v 0.80185617
88          valid      15         587 flu_vs_epi       corrs_v 0.44988177
89          train     all       12232     glmnet corrs_v_spear 0.44006067
90          train     all       12232         nn corrs_v_spear 0.61496743
91          train     all       12232  flu_vs_NI corrs_v_spear 0.89819038
92          train     all       12232 flu_vs_epi corrs_v_spear 0.71906572
93          train       1        2014     glmnet corrs_v_spear 0.47592078
94          train       1        2014         nn corrs_v_spear 0.63438978
95          train       1        2014  flu_vs_NI corrs_v_spear 0.90163796
96          train       1        2014 flu_vs_epi corrs_v_spear 0.72988230
97          train      10         710     glmnet corrs_v_spear 0.47729851
98          train      10         710         nn corrs_v_spear 0.64879133
99          train      10         710  flu_vs_NI corrs_v_spear 0.88781386
100         train      10         710 flu_vs_epi corrs_v_spear 0.71871063
101         train      11        1287     glmnet corrs_v_spear 0.38791178
102         train      11        1287         nn corrs_v_spear 0.66072093
103         train      11        1287  flu_vs_NI corrs_v_spear 0.90595495
104         train      11        1287 flu_vs_epi corrs_v_spear 0.75305191
105         train      16         839     glmnet corrs_v_spear 0.48280962
106         train      16         839         nn corrs_v_spear 0.52343163
107         train      16         839  flu_vs_NI corrs_v_spear 0.89331548
108         train      16         839 flu_vs_epi corrs_v_spear 0.63880091
109         train      17        1152     glmnet corrs_v_spear 0.46671819
110         train      17        1152         nn corrs_v_spear 0.58593836
111         train      17        1152  flu_vs_NI corrs_v_spear 0.89309058
112         train      17        1152 flu_vs_epi corrs_v_spear 0.70382284
113         train      18         262     glmnet corrs_v_spear 0.44267108
114         train      18         262         nn corrs_v_spear 0.63878222
115         train      18         262  flu_vs_NI corrs_v_spear 0.91547235
116         train      18         262 flu_vs_epi corrs_v_spear 0.79023561
117         train      19        1423     glmnet corrs_v_spear 0.40566056
118         train      19        1423         nn corrs_v_spear 0.56728434
119         train      19        1423  flu_vs_NI corrs_v_spear 0.89889239
120         train      19        1423 flu_vs_epi corrs_v_spear 0.65573461
121         train       2        1225     glmnet corrs_v_spear 0.44937869
122         train       2        1225         nn corrs_v_spear 0.60061216
123         train       2        1225  flu_vs_NI corrs_v_spear 0.90291218
124         train       2        1225 flu_vs_epi corrs_v_spear 0.73605316
125         train      20         533     glmnet corrs_v_spear 0.43219673
126         train      20         533         nn corrs_v_spear 0.64305741
127         train      20         533  flu_vs_NI corrs_v_spear 0.90369551
128         train      20         533 flu_vs_epi corrs_v_spear 0.72970210
129         train      21         218     glmnet corrs_v_spear 0.54791637
130         train      21         218         nn corrs_v_spear 0.77117959
131         train      21         218  flu_vs_NI corrs_v_spear 0.92743255
132         train      21         218 flu_vs_epi corrs_v_spear 0.81768413
133         train       3        1059     glmnet corrs_v_spear 0.43642452
134         train       3        1059         nn corrs_v_spear 0.63372748
135         train       3        1059  flu_vs_NI corrs_v_spear 0.89519940
136         train       3        1059 flu_vs_epi corrs_v_spear 0.72975246
137         train       4         748     glmnet corrs_v_spear 0.36645123
138         train       4         748         nn corrs_v_spear 0.61683195
139         train       4         748  flu_vs_NI corrs_v_spear 0.90065772
140         train       4         748 flu_vs_epi corrs_v_spear 0.76916201
141         train       9         762     glmnet corrs_v_spear 0.42636257
142         train       9         762         nn corrs_v_spear 0.53900459
143         train       9         762  flu_vs_NI corrs_v_spear 0.85308044
144         train       9         762 flu_vs_epi corrs_v_spear 0.64881475
145          test     all        2041     glmnet corrs_v_spear 0.42082300
146          test     all        2041         nn corrs_v_spear 0.61423606
147          test     all        2041  flu_vs_NI corrs_v_spear 0.89650717
148          test     all        2041 flu_vs_epi corrs_v_spear 0.73820835
149          test      13         319     glmnet corrs_v_spear 0.38947919
150          test      13         319         nn corrs_v_spear 0.61891315
151          test      13         319  flu_vs_NI corrs_v_spear 0.87931524
152          test      13         319 flu_vs_epi corrs_v_spear 0.74649363
153          test      14         600     glmnet corrs_v_spear 0.43275744
154          test      14         600         nn corrs_v_spear 0.59173172
155          test      14         600  flu_vs_NI corrs_v_spear 0.91337526
156          test      14         600 flu_vs_epi corrs_v_spear 0.69675904
157          test      22         431     glmnet corrs_v_spear 0.46612115
158          test      22         431         nn corrs_v_spear 0.61945201
159          test      22         431  flu_vs_NI corrs_v_spear 0.89010271
160          test      22         431 flu_vs_epi corrs_v_spear 0.70800148
161          test       8         691     glmnet corrs_v_spear 0.39805017
162          test       8         691         nn corrs_v_spear 0.63762339
163          test       8         691  flu_vs_NI corrs_v_spear 0.88660553
164          test       8         691 flu_vs_epi corrs_v_spear 0.77559845
165         valid     all        1577     glmnet corrs_v_spear 0.45660163
166         valid     all        1577         nn corrs_v_spear 0.60836790
167         valid     all        1577  flu_vs_NI corrs_v_spear 0.89776421
168         valid     all        1577 flu_vs_epi corrs_v_spear 0.72052783
169         valid      12         990     glmnet corrs_v_spear 0.45786716
170         valid      12         990         nn corrs_v_spear 0.58681013
171         valid      12         990  flu_vs_NI corrs_v_spear 0.89247731
172         valid      12         990 flu_vs_epi corrs_v_spear 0.70154032
173         valid      15         587     glmnet corrs_v_spear 0.45179692
174         valid      15         587         nn corrs_v_spear 0.64493467
175         valid      15         587  flu_vs_NI corrs_v_spear 0.90487214
176         valid      15         587 flu_vs_epi corrs_v_spear 0.75344108</code></pre>
</div>
</div>
<p>Set color palette</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>full_df_melt_all_chroms <span class="ot">&lt;-</span> full_df_melt[full_df_melt<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_all_chroms_both.png"</span>))</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_melt_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>value, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">nrow=</span><span class="dv">1</span>)</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>full_df_all_chroms <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_all_chroms.png"</span>))</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_nonfaceted.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>)</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">fill=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>)</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_numgenes_long.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">fill=</span>data_subset_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> id,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>  plot_list[[id]] <span class="ot">&lt;-</span> g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_{id}.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot_list[[id]])</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>all_records_full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, all_records_df_store)</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>chroms<span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>chroms, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "data_subset"        "glmnet_pred"        "nn_pred"           
 [4] "flu_rnaseq"         "NI_rnaseq"          "CAGE_epi"          
 [7] "ensembl_id"         "external_gene_name" "region"            
[10] "chroms"            </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>all_records_full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(all_records_full_df, <span class="at">measure_vars =</span> <span class="fu">c</span>(<span class="st">"glmnet_pred"</span>, <span class="st">"nn_pred"</span>, <span class="st">"flu_rnaseq"</span>, <span class="st">"NI_rnaseq"</span>, <span class="st">"CAGE_epi"</span>))</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df_melt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "data_subset"        "ensembl_id"         "external_gene_name"
[4] "region"             "chroms"             "variable"          
[7] "value"             </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/all_records_scatterplots.png"</span>), <span class="at">width =</span><span class="dv">900</span>, <span class="at">height=</span><span class="dv">1200</span>)</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>glmnet_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Elastic net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"glmnet"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>nn_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Neural net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"nn"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>CAGE_epi, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Enformer CAGE pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_epi"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>NI_rnaseq, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"NI RNAseq"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_NI"</span>],<span class="at">alpha=</span><span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors))</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># data_subset &lt;- "test"</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="co"># X_path &lt;- glue("/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv")</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="co"># y_path &lt;- glue("/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv")</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # glmnet_model &lt;- readRDS(glmnet_model_path)</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a><span class="co"># # nnmodel &lt;- readRDS(nnmodel_path)</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a><span class="co"># X &lt;- as.matrix(t(data.table::fread(X_path)))</span></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a><span class="co"># y &lt;- as.data.frame(t(data.table::fread(y_path)))</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a><span class="co"># y_for_cv &lt;- y[,1]</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a><span class="co"># data_frame_for_training &lt;- data.frame(cbind(X, y_for_cv))</span></span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a><span class="co"># predictions_nn &lt;- predict(nnmodel, data_frame_for_training)</span></span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a><span class="co"># glmnet_predictions &lt;- predict(glmnet_model, X, s = "lambda.min", type="response")</span></span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a><span class="co"># cor(glmnet_predictions, y[,1])</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb95" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "evaluate_tss_center_mean_pred"</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-10-22"</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Set up box storage directory</span></span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"evaluate_tss_center_mean_pred"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-10-22'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>Here I trained an elastic net and simple MLP mode to predict TSS RNAseq expression from Flu condition (aracena dataset) from Enformer predictions. Training was done using corpus of TSS sites from a ranmdom mixture of individuals. Here I am evaluating the performance of the trained models on a held out test set.</span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>TODO: Average of individuals</span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>TODO: Collect RNAseq from Katie, and use this for DL-EN training AND training PrediXcan</span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>TODO: Try different MLP params</span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-43"><a href="#cb95-43" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-44"><a href="#cb95-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb95-45"><a href="#cb95-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb95-46"><a href="#cb95-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb95-47"><a href="#cb95-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb95-48"><a href="#cb95-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(neuralnet)</span>
<span id="cb95-49"><a href="#cb95-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb95-50"><a href="#cb95-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb95-51"><a href="#cb95-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb95-52"><a href="#cb95-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb95-53"><a href="#cb95-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb95-54"><a href="#cb95-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-55"><a href="#cb95-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-56"><a href="#cb95-56" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-57"><a href="#cb95-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-60"><a href="#cb95-60" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-61"><a href="#cb95-61" aria-hidden="true" tabindex="-1"></a>predict_on_data_subset <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, glmnet_model_path, nnmodel_path){</span>
<span id="cb95-62"><a href="#cb95-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-63"><a href="#cb95-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Predicting for data subset: {data_subset}"</span>))</span>
<span id="cb95-64"><a href="#cb95-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-65"><a href="#cb95-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load trained models</span></span>
<span id="cb95-66"><a href="#cb95-66" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb95-67"><a href="#cb95-67" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb95-68"><a href="#cb95-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-69"><a href="#cb95-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/glmnet_lambdas.png"</span>)</span>
<span id="cb95-70"><a href="#cb95-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_model)</span>
<span id="cb95-71"><a href="#cb95-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-72"><a href="#cb95-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-73"><a href="#cb95-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load dataset to be predicted on</span></span>
<span id="cb95-74"><a href="#cb95-74" aria-hidden="true" tabindex="-1"></a>  data_X <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>)</span>
<span id="cb95-75"><a href="#cb95-75" aria-hidden="true" tabindex="-1"></a>  data_y <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv"</span>)</span>
<span id="cb95-76"><a href="#cb95-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-77"><a href="#cb95-77" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_X)))</span>
<span id="cb95-78"><a href="#cb95-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-79"><a href="#cb95-79" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(data_y)))</span>
<span id="cb95-80"><a href="#cb95-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-81"><a href="#cb95-81" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb95-82"><a href="#cb95-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-83"><a href="#cb95-83" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make predictions and associated plots</span></span>
<span id="cb95-84"><a href="#cb95-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-85"><a href="#cb95-85" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb95-86"><a href="#cb95-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-87"><a href="#cb95-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_corr_en_pred.png"</span>))</span>
<span id="cb95-88"><a href="#cb95-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(glmnet_predictions, y[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"glmnet predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Glmnet predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb95-89"><a href="#cb95-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-90"><a href="#cb95-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-91"><a href="#cb95-91" aria-hidden="true" tabindex="-1"></a>  pred_corr_en <span class="ot">&lt;-</span> <span class="fu">cor</span>(glmnet_predictions, y[,<span class="dv">1</span>])</span>
<span id="cb95-92"><a href="#cb95-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_en} for {data_subset} data subset with glmnet predictions"</span>))</span>
<span id="cb95-93"><a href="#cb95-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-94"><a href="#cb95-94" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb95-95"><a href="#cb95-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-96"><a href="#cb95-96" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb95-97"><a href="#cb95-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-98"><a href="#cb95-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_corr_nn_pred.png"</span>))</span>
<span id="cb95-99"><a href="#cb95-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(predictions_nn, y[,<span class="dv">1</span>], <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">ylab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">xlab=</span><span class="st">"NN predictions"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"NN predictions vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb95-100"><a href="#cb95-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-101"><a href="#cb95-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-102"><a href="#cb95-102" aria-hidden="true" tabindex="-1"></a>  pred_corr_nn <span class="ot">&lt;-</span> <span class="fu">cor</span>(predictions_nn, y[,<span class="dv">1</span>])</span>
<span id="cb95-103"><a href="#cb95-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {pred_corr_nn} for {data_subset} data subset with NN predictions"</span>))</span>
<span id="cb95-104"><a href="#cb95-104" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-105"><a href="#cb95-105" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-106"><a href="#cb95-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-109"><a href="#cb95-109" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-110"><a href="#cb95-110" aria-hidden="true" tabindex="-1"></a>analyze_input_data <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset) {</span>
<span id="cb95-111"><a href="#cb95-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-112"><a href="#cb95-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for data subset: {data_subset}"</span>))</span>
<span id="cb95-113"><a href="#cb95-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-114"><a href="#cb95-114" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>))))</span>
<span id="cb95-115"><a href="#cb95-115" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv"</span>))))</span>
<span id="cb95-116"><a href="#cb95-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-117"><a href="#cb95-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(X))</span>
<span id="cb95-118"><a href="#cb95-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(y))</span>
<span id="cb95-119"><a href="#cb95-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-120"><a href="#cb95-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_flu_vs_NI_rnaseq.png"</span>))</span>
<span id="cb95-121"><a href="#cb95-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>],<span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>),<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"ground truth NI rnaseq"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Flu vs. NI in ground truth for {data_subset} data subset"</span>))</span>
<span id="cb95-122"><a href="#cb95-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-123"><a href="#cb95-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-124"><a href="#cb95-124" aria-hidden="true" tabindex="-1"></a>  corr_rnaseq <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>])</span>
<span id="cb95-125"><a href="#cb95-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_rnaseq} for {data_subset} data subset between flu/NI rnaseq"</span>))</span>
<span id="cb95-126"><a href="#cb95-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-127"><a href="#cb95-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_flu_vs_epi_rnaseq.png"</span>))</span>
<span id="cb95-128"><a href="#cb95-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb95-129"><a href="#cb95-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-130"><a href="#cb95-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-131"><a href="#cb95-131" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb95-132"><a href="#cb95-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena flu rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb95-133"><a href="#cb95-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-134"><a href="#cb95-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_NI_vs_epi_rnaseq.png"</span>))</span>
<span id="cb95-135"><a href="#cb95-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>],<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">20</span>), <span class="at">xlab=</span><span class="st">"ground truth flu rnaseq"</span>, <span class="at">ylab=</span><span class="st">"Reference CAGE epigenome monocyte derived macrophages"</span>, <span class="at">main =</span> <span class="fu">glue</span>(<span class="st">"Reference epi vs. ground truth for {data_subset} data subset"</span>))</span>
<span id="cb95-136"><a href="#cb95-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-137"><a href="#cb95-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-138"><a href="#cb95-138" aria-hidden="true" tabindex="-1"></a>  corr_io <span class="ot">&lt;-</span> <span class="fu">cor</span>(y[,<span class="dv">7</span>], X[,<span class="dv">4766</span>])</span>
<span id="cb95-139"><a href="#cb95-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"Correlation of {corr_io} for {data_subset} data subset between aracena NI rnaseq and corresponding enformer prediction"</span>))</span>
<span id="cb95-140"><a href="#cb95-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-141"><a href="#cb95-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-142"><a href="#cb95-142" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-143"><a href="#cb95-143" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-144"><a href="#cb95-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-145"><a href="#cb95-145" aria-hidden="true" tabindex="-1"></a>Now we can use the trained models to make predictions and assess model performance.</span>
<span id="cb95-146"><a href="#cb95-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-149"><a href="#cb95-149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-150"><a href="#cb95-150" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"train"</span>)</span>
<span id="cb95-151"><a href="#cb95-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-152"><a href="#cb95-152" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"valid"</span>)</span>
<span id="cb95-153"><a href="#cb95-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-154"><a href="#cb95-154" aria-hidden="true" tabindex="-1"></a><span class="fu">analyze_input_data</span>(<span class="st">"test"</span>)</span>
<span id="cb95-155"><a href="#cb95-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-156"><a href="#cb95-156" aria-hidden="true" tabindex="-1"></a>glmnet_model_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.linear.rds"</span></span>
<span id="cb95-157"><a href="#cb95-157" aria-hidden="true" tabindex="-1"></a>nnmodel_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp_RNAseq_from_HDF5_mean"</span></span>
<span id="cb95-158"><a href="#cb95-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-159"><a href="#cb95-159" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"train"</span>, glmnet_model_path, nnmodel_path)</span>
<span id="cb95-160"><a href="#cb95-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-161"><a href="#cb95-161" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"valid"</span>, glmnet_model_path, nnmodel_path)</span>
<span id="cb95-162"><a href="#cb95-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-163"><a href="#cb95-163" aria-hidden="true" tabindex="-1"></a><span class="fu">predict_on_data_subset</span>(<span class="st">"test"</span>, glmnet_model_path, nnmodel_path)</span>
<span id="cb95-164"><a href="#cb95-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-165"><a href="#cb95-165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-166"><a href="#cb95-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-169"><a href="#cb95-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-170"><a href="#cb95-170" aria-hidden="true" tabindex="-1"></a>convert_region_to_gene <span class="ot">&lt;-</span> <span class="cf">function</span>(region, gene_metadata){</span>
<span id="cb95-171"><a href="#cb95-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-172"><a href="#cb95-172" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb95-173"><a href="#cb95-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-174"><a href="#cb95-174" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb95-175"><a href="#cb95-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-176"><a href="#cb95-176" aria-hidden="true" tabindex="-1"></a>  tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> <span class="st">"1_1471765_1471767"</span>,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb95-177"><a href="#cb95-177" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-178"><a href="#cb95-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-179"><a href="#cb95-179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-180"><a href="#cb95-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-183"><a href="#cb95-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-184"><a href="#cb95-184" aria-hidden="true" tabindex="-1"></a>tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb95-185"><a href="#cb95-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-186"><a href="#cb95-186" aria-hidden="true" tabindex="-1"></a>tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb95-187"><a href="#cb95-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-188"><a href="#cb95-188" aria-hidden="true" tabindex="-1"></a>tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> <span class="st">"1_1471765_1471767"</span>,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb95-189"><a href="#cb95-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-190"><a href="#cb95-190" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-191"><a href="#cb95-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-194"><a href="#cb95-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-195"><a href="#cb95-195" aria-hidden="true" tabindex="-1"></a>split_datasets_chroms <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb95-196"><a href="#cb95-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-197"><a href="#cb95-197" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb95-198"><a href="#cb95-198" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb95-199"><a href="#cb95-199" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb95-200"><a href="#cb95-200" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb95-201"><a href="#cb95-201" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb95-202"><a href="#cb95-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-203"><a href="#cb95-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-204"><a href="#cb95-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(chroms)</span>
<span id="cb95-205"><a href="#cb95-205" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-206"><a href="#cb95-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-207"><a href="#cb95-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-208"><a href="#cb95-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-209"><a href="#cb95-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-210"><a href="#cb95-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-213"><a href="#cb95-213" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-214"><a href="#cb95-214" aria-hidden="true" tabindex="-1"></a>plot_boxplot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, path) {</span>
<span id="cb95-215"><a href="#cb95-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-216"><a href="#cb95-216" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>is_CAGE <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"CAGE_epi"</span>, <span class="st">"CAGE"</span>, <span class="st">"RNAseq"</span>)</span>
<span id="cb95-217"><a href="#cb95-217" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(path)</span>
<span id="cb95-218"><a href="#cb95-218" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>variable, <span class="at">y=</span>value)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() </span>
<span id="cb95-219"><a href="#cb95-219" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_CAGE, <span class="at">scales=</span><span class="st">"free"</span>))</span>
<span id="cb95-220"><a href="#cb95-220" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb95-221"><a href="#cb95-221" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-222"><a href="#cb95-222" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-223"><a href="#cb95-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-226"><a href="#cb95-226" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-227"><a href="#cb95-227" aria-hidden="true" tabindex="-1"></a>run_cor<span class="ot">&lt;-</span> <span class="cf">function</span>(x,y, <span class="at">method=</span><span class="st">"pearson"</span>) {</span>
<span id="cb95-228"><a href="#cb95-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">cor</span>(x,y, <span class="at">method=</span>method))</span>
<span id="cb95-229"><a href="#cb95-229" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-230"><a href="#cb95-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-231"><a href="#cb95-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-232"><a href="#cb95-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-233"><a href="#cb95-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-236"><a href="#cb95-236" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-237"><a href="#cb95-237" aria-hidden="true" tabindex="-1"></a>per_chrom_metrics <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path, <span class="at">cor_method=</span><span class="st">"spearman"</span>) {</span>
<span id="cb95-238"><a href="#cb95-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-239"><a href="#cb95-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for data subset: {data_subset}"</span>))</span>
<span id="cb95-240"><a href="#cb95-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-241"><a href="#cb95-241" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb95-242"><a href="#cb95-242" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb95-243"><a href="#cb95-243" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb95-244"><a href="#cb95-244" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb95-245"><a href="#cb95-245" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb95-246"><a href="#cb95-246" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb95-247"><a href="#cb95-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-248"><a href="#cb95-248" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb95-249"><a href="#cb95-249" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb95-250"><a href="#cb95-250" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb95-251"><a href="#cb95-251" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb95-252"><a href="#cb95-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-253"><a href="#cb95-253" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb95-254"><a href="#cb95-254" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb95-255"><a href="#cb95-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-256"><a href="#cb95-256" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb95-257"><a href="#cb95-257" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb95-258"><a href="#cb95-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-259"><a href="#cb95-259" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb95-260"><a href="#cb95-260" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-261"><a href="#cb95-261" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions, y[,<span class="dv">1</span>]))</span>
<span id="cb95-262"><a href="#cb95-262" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions, y[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-263"><a href="#cb95-263" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb95-264"><a href="#cb95-264" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb95-265"><a href="#cb95-265" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb95-266"><a href="#cb95-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-267"><a href="#cb95-267" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage nn corr</span></span>
<span id="cb95-268"><a href="#cb95-268" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-269"><a href="#cb95-269" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn, y[,<span class="dv">1</span>]))</span>
<span id="cb95-270"><a href="#cb95-270" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn, y[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-271"><a href="#cb95-271" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb95-272"><a href="#cb95-272" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb95-273"><a href="#cb95-273" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb95-274"><a href="#cb95-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-275"><a href="#cb95-275" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb95-276"><a href="#cb95-276" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-277"><a href="#cb95-277" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>]))</span>
<span id="cb95-278"><a href="#cb95-278" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-279"><a href="#cb95-279" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb95-280"><a href="#cb95-280" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb95-281"><a href="#cb95-281" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb95-282"><a href="#cb95-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-283"><a href="#cb95-283" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb95-284"><a href="#cb95-284" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-285"><a href="#cb95-285" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>]))</span>
<span id="cb95-286"><a href="#cb95-286" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], X[,<span class="dv">4766</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-287"><a href="#cb95-287" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb95-288"><a href="#cb95-288" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb95-289"><a href="#cb95-289" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb95-290"><a href="#cb95-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-291"><a href="#cb95-291" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Done recording</span></span>
<span id="cb95-292"><a href="#cb95-292" aria-hidden="true" tabindex="-1"></a>  chroms_all <span class="ot">&lt;-</span> <span class="fu">split_datasets_chroms</span>(data_subset, mapping_table)</span>
<span id="cb95-293"><a href="#cb95-293" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(chroms_all)</span>
<span id="cb95-294"><a href="#cb95-294" aria-hidden="true" tabindex="-1"></a>  uniq_chroms <span class="ot">&lt;-</span> <span class="fu">unique</span>(chroms)</span>
<span id="cb95-295"><a href="#cb95-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-296"><a href="#cb95-296" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (chrom <span class="cf">in</span> uniq_chroms) {</span>
<span id="cb95-297"><a href="#cb95-297" aria-hidden="true" tabindex="-1"></a>    index_of_chrom <span class="ot">&lt;-</span> chroms <span class="sc">==</span> chrom</span>
<span id="cb95-298"><a href="#cb95-298" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb95-299"><a href="#cb95-299" aria-hidden="true" tabindex="-1"></a>    X_chrom <span class="ot">&lt;-</span> X[index_of_chrom,]</span>
<span id="cb95-300"><a href="#cb95-300" aria-hidden="true" tabindex="-1"></a>    y_chrom <span class="ot">&lt;-</span> y[index_of_chrom,]</span>
<span id="cb95-301"><a href="#cb95-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-302"><a href="#cb95-302" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for chrom: {chrom}"</span>))</span>
<span id="cb95-303"><a href="#cb95-303" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(X_chrom))</span>
<span id="cb95-304"><a href="#cb95-304" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(y_chrom))</span>
<span id="cb95-305"><a href="#cb95-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-306"><a href="#cb95-306" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>]))</span>
<span id="cb95-307"><a href="#cb95-307" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">1</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb95-308"><a href="#cb95-308" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">7</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb95-309"><a href="#cb95-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-310"><a href="#cb95-310" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb95-311"><a href="#cb95-311" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(predictions_nn[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb95-312"><a href="#cb95-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-313"><a href="#cb95-313" aria-hidden="true" tabindex="-1"></a>    chrom_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">glmnet_pred=</span>glmnet_predictions[index_of_chrom], <span class="at">flu_rnaseq =</span> y_chrom[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y_chrom[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X_chrom[,<span class="dv">4766</span>])</span>
<span id="cb95-314"><a href="#cb95-314" aria-hidden="true" tabindex="-1"></a>    chrom_df_for_plotting <span class="ot">&lt;-</span> <span class="fu">melt</span>(chrom_df, <span class="at">measure_vars =</span> <span class="fu">colnames</span>(chrom_df))</span>
<span id="cb95-315"><a href="#cb95-315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_boxplot</span>(chrom_df_for_plotting, <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/{data_subset}_chrom_{chrom}_boxplot.png"</span>))</span>
<span id="cb95-316"><a href="#cb95-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-317"><a href="#cb95-317" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb95-318"><a href="#cb95-318" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-319"><a href="#cb95-319" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb95-320"><a href="#cb95-320" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-321"><a href="#cb95-321" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb95-322"><a href="#cb95-322" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb95-323"><a href="#cb95-323" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb95-324"><a href="#cb95-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-325"><a href="#cb95-325" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage nn corr</span></span>
<span id="cb95-326"><a href="#cb95-326" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-327"><a href="#cb95-327" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,<span class="dv">1</span>]))</span>
<span id="cb95-328"><a href="#cb95-328" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,<span class="dv">1</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-329"><a href="#cb95-329" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb95-330"><a href="#cb95-330" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb95-331"><a href="#cb95-331" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb95-332"><a href="#cb95-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-333"><a href="#cb95-333" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb95-334"><a href="#cb95-334" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-335"><a href="#cb95-335" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>]))</span>
<span id="cb95-336"><a href="#cb95-336" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-337"><a href="#cb95-337" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb95-338"><a href="#cb95-338" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb95-339"><a href="#cb95-339" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb95-340"><a href="#cb95-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-341"><a href="#cb95-341" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb95-342"><a href="#cb95-342" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb95-343"><a href="#cb95-343" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb95-344"><a href="#cb95-344" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], X_chrom[,<span class="dv">4766</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb95-345"><a href="#cb95-345" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb95-346"><a href="#cb95-346" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb95-347"><a href="#cb95-347" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb95-348"><a href="#cb95-348" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb95-349"><a href="#cb95-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-350"><a href="#cb95-350" aria-hidden="true" tabindex="-1"></a>  storage_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(corrs_v, corrs_v_spear, data_subset_v, chrom_v, num_genes_v, cor_id_v)</span>
<span id="cb95-351"><a href="#cb95-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-352"><a href="#cb95-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(storage_df)</span>
<span id="cb95-353"><a href="#cb95-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-354"><a href="#cb95-354" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-355"><a href="#cb95-355" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-356"><a href="#cb95-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-359"><a href="#cb95-359" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-360"><a href="#cb95-360" aria-hidden="true" tabindex="-1"></a>split_datasets <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb95-361"><a href="#cb95-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-362"><a href="#cb95-362" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb95-363"><a href="#cb95-363" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb95-364"><a href="#cb95-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-365"><a href="#cb95-365" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb95-366"><a href="#cb95-366" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb95-367"><a href="#cb95-367" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb95-368"><a href="#cb95-368" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb95-369"><a href="#cb95-369" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb95-370"><a href="#cb95-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-371"><a href="#cb95-371" aria-hidden="true" tabindex="-1"></a>  ensembl_ids <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb95-372"><a href="#cb95-372" aria-hidden="true" tabindex="-1"></a>  external_gene_names <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb95-373"><a href="#cb95-373" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(regions)) {</span>
<span id="cb95-374"><a href="#cb95-374" aria-hidden="true" tabindex="-1"></a>    region <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions[x],<span class="dv">2</span>,<span class="fu">str_length</span>(regions[x]))</span>
<span id="cb95-375"><a href="#cb95-375" aria-hidden="true" tabindex="-1"></a>    ensembl_id <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb95-376"><a href="#cb95-376" aria-hidden="true" tabindex="-1"></a>    external_gene_name <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>external_gene_name</span>
<span id="cb95-377"><a href="#cb95-377" aria-hidden="true" tabindex="-1"></a>    ensembl_ids[x]  <span class="ot">&lt;-</span> ensembl_id</span>
<span id="cb95-378"><a href="#cb95-378" aria-hidden="true" tabindex="-1"></a>    external_gene_names[x] <span class="ot">&lt;-</span> external_gene_name</span>
<span id="cb95-379"><a href="#cb95-379" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb95-380"><a href="#cb95-380" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(ensembl_ids))</span>
<span id="cb95-381"><a href="#cb95-381" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(external_gene_names))</span>
<span id="cb95-382"><a href="#cb95-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(chroms))</span>
<span id="cb95-383"><a href="#cb95-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(regions))</span>
<span id="cb95-384"><a href="#cb95-384" aria-hidden="true" tabindex="-1"></a>  return_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb95-385"><a href="#cb95-385" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"chroms"</span>]] <span class="ot">&lt;-</span> chroms</span>
<span id="cb95-386"><a href="#cb95-386" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"regions"</span>]] <span class="ot">&lt;-</span> regions</span>
<span id="cb95-387"><a href="#cb95-387" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"ensembl_ids"</span>]] <span class="ot">&lt;-</span> ensembl_ids</span>
<span id="cb95-388"><a href="#cb95-388" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"external_gene_names"</span>]] <span class="ot">&lt;-</span> external_gene_names</span>
<span id="cb95-389"><a href="#cb95-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-390"><a href="#cb95-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_list)</span>
<span id="cb95-391"><a href="#cb95-391" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-392"><a href="#cb95-392" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-393"><a href="#cb95-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-396"><a href="#cb95-396" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-397"><a href="#cb95-397" aria-hidden="true" tabindex="-1"></a>collect_all_records <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path) {</span>
<span id="cb95-398"><a href="#cb95-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Collecting data records for data subset: {data_subset}"</span>))</span>
<span id="cb95-399"><a href="#cb95-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-400"><a href="#cb95-400" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb95-401"><a href="#cb95-401" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb95-402"><a href="#cb95-402" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb95-403"><a href="#cb95-403" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb95-404"><a href="#cb95-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-405"><a href="#cb95-405" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb95-406"><a href="#cb95-406" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb95-407"><a href="#cb95-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-408"><a href="#cb95-408" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb95-409"><a href="#cb95-409" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb95-410"><a href="#cb95-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-411"><a href="#cb95-411" aria-hidden="true" tabindex="-1"></a>  gene_metadata <span class="ot">&lt;-</span> <span class="fu">split_datasets</span>(data_subset, mapping_table)</span>
<span id="cb95-412"><a href="#cb95-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-413"><a href="#cb95-413" aria-hidden="true" tabindex="-1"></a>  return_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">data_subset=</span> data_subset, <span class="at">glmnet_pred=</span>glmnet_predictions, <span class="at">nn_pred=</span> predictions_nn, <span class="at">flu_rnaseq =</span> y[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X[,<span class="dv">4766</span>], <span class="at">ensembl_id =</span> gene_metadata[[<span class="st">"ensembl_ids"</span>]], <span class="at">external_gene_name =</span> gene_metadata[[<span class="st">"external_gene_names"</span>]], <span class="at">region =</span> gene_metadata[[<span class="st">"regions"</span>]], <span class="at">chroms =</span> gene_metadata[[<span class="st">"chroms"</span>]])</span>
<span id="cb95-414"><a href="#cb95-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-415"><a href="#cb95-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_df)</span>
<span id="cb95-416"><a href="#cb95-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-417"><a href="#cb95-417" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-418"><a href="#cb95-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-419"><a href="#cb95-419" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-420"><a href="#cb95-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-423"><a href="#cb95-423" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-424"><a href="#cb95-424" aria-hidden="true" tabindex="-1"></a>remapped_table_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/remapped_table_filt.csv"</span></span>
<span id="cb95-425"><a href="#cb95-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-426"><a href="#cb95-426" aria-hidden="true" tabindex="-1"></a>remapped_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(remapped_table_path, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb95-427"><a href="#cb95-427" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(remapped_table))</span>
<span id="cb95-428"><a href="#cb95-428" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">as.character</span>(remapped_table[,<span class="dv">1</span>]))</span>
<span id="cb95-429"><a href="#cb95-429" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(remapped_table) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(remapped_table[,<span class="dv">1</span>])</span>
<span id="cb95-430"><a href="#cb95-430" aria-hidden="true" tabindex="-1"></a>remapped_table[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb95-431"><a href="#cb95-431" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-432"><a href="#cb95-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-435"><a href="#cb95-435" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-436"><a href="#cb95-436" aria-hidden="true" tabindex="-1"></a>glmnet_model_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_from_HDF5_mean.linear.rds"</span></span>
<span id="cb95-437"><a href="#cb95-437" aria-hidden="true" tabindex="-1"></a>nnmodel_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp_RNAseq_from_HDF5_mean"</span></span>
<span id="cb95-438"><a href="#cb95-438" aria-hidden="true" tabindex="-1"></a>cor_method <span class="ot">=</span> <span class="st">"spearman"</span></span>
<span id="cb95-439"><a href="#cb95-439" aria-hidden="true" tabindex="-1"></a>df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb95-440"><a href="#cb95-440" aria-hidden="true" tabindex="-1"></a>all_records_df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb95-441"><a href="#cb95-441" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb95-442"><a href="#cb95-442" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"train"</span>,<span class="st">"test"</span>,<span class="st">"valid"</span>)){</span>
<span id="cb95-443"><a href="#cb95-443" aria-hidden="true" tabindex="-1"></a>  X_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>)</span>
<span id="cb95-444"><a href="#cb95-444" aria-hidden="true" tabindex="-1"></a>  y_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv"</span>)</span>
<span id="cb95-445"><a href="#cb95-445" aria-hidden="true" tabindex="-1"></a>  cur_out_df_perchrom_metrics <span class="ot">&lt;-</span> <span class="fu">per_chrom_metrics</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path, cor_method)</span>
<span id="cb95-446"><a href="#cb95-446" aria-hidden="true" tabindex="-1"></a>  df_store[[c]] <span class="ot">&lt;-</span> cur_out_df_perchrom_metrics</span>
<span id="cb95-447"><a href="#cb95-447" aria-hidden="true" tabindex="-1"></a>  all_records_df_store[[c]] <span class="ot">&lt;-</span> <span class="fu">collect_all_records</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path)</span>
<span id="cb95-448"><a href="#cb95-448" aria-hidden="true" tabindex="-1"></a>  c<span class="ot">=</span>c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb95-449"><a href="#cb95-449" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-450"><a href="#cb95-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-451"><a href="#cb95-451" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df_store)</span>
<span id="cb95-452"><a href="#cb95-452" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>data_subset_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>data_subset_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb95-453"><a href="#cb95-453" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>cor_id_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>cor_id_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>))</span>
<span id="cb95-454"><a href="#cb95-454" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>chrom_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>chrom_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb95-455"><a href="#cb95-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-456"><a href="#cb95-456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-457"><a href="#cb95-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-460"><a href="#cb95-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-461"><a href="#cb95-461" aria-hidden="true" tabindex="-1"></a>full_df</span>
<span id="cb95-462"><a href="#cb95-462" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>num_genes_v <span class="ot">&lt;-</span> <span class="fu">as.character</span>(full_df<span class="sc">$</span>num_genes_v)</span>
<span id="cb95-463"><a href="#cb95-463" aria-hidden="true" tabindex="-1"></a>full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(full_df, <span class="at">measure_vars=</span><span class="fu">c</span>(<span class="st">"corrs_v"</span>,<span class="st">"corrs_v_spear"</span>), <span class="at">id_vars=</span><span class="fu">c</span>(<span class="st">"data_subset_v"</span>, <span class="st">"chrom_v"</span>, <span class="st">"cor_id_v"</span>, <span class="st">"num_genes_v"</span>))</span>
<span id="cb95-464"><a href="#cb95-464" aria-hidden="true" tabindex="-1"></a>full_df_melt</span>
<span id="cb95-465"><a href="#cb95-465" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-466"><a href="#cb95-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-467"><a href="#cb95-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-468"><a href="#cb95-468" aria-hidden="true" tabindex="-1"></a>Set color palette</span>
<span id="cb95-471"><a href="#cb95-471" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-472"><a href="#cb95-472" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb95-473"><a href="#cb95-473" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span>
<span id="cb95-474"><a href="#cb95-474" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-475"><a href="#cb95-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-478"><a href="#cb95-478" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-479"><a href="#cb95-479" aria-hidden="true" tabindex="-1"></a>full_df_melt_all_chroms <span class="ot">&lt;-</span> full_df_melt[full_df_melt<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb95-480"><a href="#cb95-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-481"><a href="#cb95-481" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_all_chroms_both.png"</span>))</span>
<span id="cb95-482"><a href="#cb95-482" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_melt_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>value, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">nrow=</span><span class="dv">1</span>)</span>
<span id="cb95-483"><a href="#cb95-483" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-484"><a href="#cb95-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-485"><a href="#cb95-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-486"><a href="#cb95-486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-487"><a href="#cb95-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-490"><a href="#cb95-490" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-491"><a href="#cb95-491" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb95-492"><a href="#cb95-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-493"><a href="#cb95-493" aria-hidden="true" tabindex="-1"></a>full_df_all_chroms <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb95-494"><a href="#cb95-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-495"><a href="#cb95-495" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_all_chroms.png"</span>))</span>
<span id="cb95-496"><a href="#cb95-496" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb95-497"><a href="#cb95-497" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-498"><a href="#cb95-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-499"><a href="#cb95-499" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb95-500"><a href="#cb95-500" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb95-501"><a href="#cb95-501" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb95-502"><a href="#cb95-502" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-503"><a href="#cb95-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-504"><a href="#cb95-504" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_nonfaceted.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb95-505"><a href="#cb95-505" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb95-506"><a href="#cb95-506" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-507"><a href="#cb95-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-508"><a href="#cb95-508" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb95-509"><a href="#cb95-509" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>)</span>
<span id="cb95-510"><a href="#cb95-510" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-511"><a href="#cb95-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-512"><a href="#cb95-512" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb95-513"><a href="#cb95-513" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">fill=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>)</span>
<span id="cb95-514"><a href="#cb95-514" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-515"><a href="#cb95-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-516"><a href="#cb95-516" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_numgenes_long.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb95-517"><a href="#cb95-517" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">fill=</span>data_subset_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span>
<span id="cb95-518"><a href="#cb95-518" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-519"><a href="#cb95-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-520"><a href="#cb95-520" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb95-521"><a href="#cb95-521" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb95-522"><a href="#cb95-522" aria-hidden="true" tabindex="-1"></a>  g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> id,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb95-523"><a href="#cb95-523" aria-hidden="true" tabindex="-1"></a>  plot_list[[id]] <span class="ot">&lt;-</span> g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb95-524"><a href="#cb95-524" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-525"><a href="#cb95-525" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb95-526"><a href="#cb95-526" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/corrs_each_chrom_{id}.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb95-527"><a href="#cb95-527" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot_list[[id]])</span>
<span id="cb95-528"><a href="#cb95-528" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb95-529"><a href="#cb95-529" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-530"><a href="#cb95-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-531"><a href="#cb95-531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-532"><a href="#cb95-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-535"><a href="#cb95-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-536"><a href="#cb95-536" aria-hidden="true" tabindex="-1"></a>all_records_full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, all_records_df_store)</span>
<span id="cb95-537"><a href="#cb95-537" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb95-538"><a href="#cb95-538" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>chroms<span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>chroms, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb95-539"><a href="#cb95-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-540"><a href="#cb95-540" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df)</span>
<span id="cb95-541"><a href="#cb95-541" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-542"><a href="#cb95-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-545"><a href="#cb95-545" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-546"><a href="#cb95-546" aria-hidden="true" tabindex="-1"></a>all_records_full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(all_records_full_df, <span class="at">measure_vars =</span> <span class="fu">c</span>(<span class="st">"glmnet_pred"</span>, <span class="st">"nn_pred"</span>, <span class="st">"flu_rnaseq"</span>, <span class="st">"NI_rnaseq"</span>, <span class="st">"CAGE_epi"</span>))</span>
<span id="cb95-547"><a href="#cb95-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-548"><a href="#cb95-548" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df_melt)</span>
<span id="cb95-549"><a href="#cb95-549" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-550"><a href="#cb95-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-553"><a href="#cb95-553" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-554"><a href="#cb95-554" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb95-555"><a href="#cb95-555" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span>
<span id="cb95-556"><a href="#cb95-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-557"><a href="#cb95-557" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-22-evaluate_tss_center_mean_pred/figures/all_records_scatterplots.png"</span>), <span class="at">width =</span><span class="dv">900</span>, <span class="at">height=</span><span class="dv">1200</span>)</span>
<span id="cb95-558"><a href="#cb95-558" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>glmnet_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Elastic net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"glmnet"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb95-559"><a href="#cb95-559" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>nn_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Neural net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"nn"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb95-560"><a href="#cb95-560" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>CAGE_epi, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Enformer CAGE pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_epi"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb95-561"><a href="#cb95-561" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>NI_rnaseq, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"NI RNAseq"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_NI"</span>],<span class="at">alpha=</span><span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors))</span>
<span id="cb95-562"><a href="#cb95-562" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb95-563"><a href="#cb95-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-564"><a href="#cb95-564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-565"><a href="#cb95-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-566"><a href="#cb95-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-569"><a href="#cb95-569" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-570"><a href="#cb95-570" aria-hidden="true" tabindex="-1"></a><span class="co"># data_subset &lt;- "test"</span></span>
<span id="cb95-571"><a href="#cb95-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-572"><a href="#cb95-572" aria-hidden="true" tabindex="-1"></a><span class="co"># X_path &lt;- glue("/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv")</span></span>
<span id="cb95-573"><a href="#cb95-573" aria-hidden="true" tabindex="-1"></a><span class="co"># y_path &lt;- glue("/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena.csv")</span></span>
<span id="cb95-574"><a href="#cb95-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-575"><a href="#cb95-575" aria-hidden="true" tabindex="-1"></a><span class="co"># # glmnet_model &lt;- readRDS(glmnet_model_path)</span></span>
<span id="cb95-576"><a href="#cb95-576" aria-hidden="true" tabindex="-1"></a><span class="co"># # nnmodel &lt;- readRDS(nnmodel_path)</span></span>
<span id="cb95-577"><a href="#cb95-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-578"><a href="#cb95-578" aria-hidden="true" tabindex="-1"></a><span class="co"># X &lt;- as.matrix(t(data.table::fread(X_path)))</span></span>
<span id="cb95-579"><a href="#cb95-579" aria-hidden="true" tabindex="-1"></a><span class="co"># y &lt;- as.data.frame(t(data.table::fread(y_path)))</span></span>
<span id="cb95-580"><a href="#cb95-580" aria-hidden="true" tabindex="-1"></a><span class="co"># y_for_cv &lt;- y[,1]</span></span>
<span id="cb95-581"><a href="#cb95-581" aria-hidden="true" tabindex="-1"></a><span class="co"># data_frame_for_training &lt;- data.frame(cbind(X, y_for_cv))</span></span>
<span id="cb95-582"><a href="#cb95-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-583"><a href="#cb95-583" aria-hidden="true" tabindex="-1"></a><span class="co"># predictions_nn &lt;- predict(nnmodel, data_frame_for_training)</span></span>
<span id="cb95-584"><a href="#cb95-584" aria-hidden="true" tabindex="-1"></a><span class="co"># glmnet_predictions &lt;- predict(glmnet_model, X, s = "lambda.min", type="response")</span></span>
<span id="cb95-585"><a href="#cb95-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-586"><a href="#cb95-586" aria-hidden="true" tabindex="-1"></a><span class="co"># cor(glmnet_predictions, y[,1])</span></span>
<span id="cb95-587"><a href="#cb95-587" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb95-588"><a href="#cb95-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-591"><a href="#cb95-591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb95-592"><a href="#cb95-592" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>