<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-07-14">

<title>Daily-Blog-Sai - testing_pop_seq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#create-input-datasets-for-testing" id="toc-create-input-datasets-for-testing" class="nav-link" data-scroll-target="#create-input-datasets-for-testing">Create input datasets for testing</a>
  <ul class="collapse">
  <li><a href="#individuals" id="toc-individuals" class="nav-link" data-scroll-target="#individuals">Individuals</a></li>
  <li><a href="#genes" id="toc-genes" class="nav-link" data-scroll-target="#genes">Genes</a></li>
  </ul></li>
  <li><a href="#checking-correctness-of-pipeline" id="toc-checking-correctness-of-pipeline" class="nav-link" data-scroll-target="#checking-correctness-of-pipeline">Checking correctness of pipeline</a>
  <ul class="collapse">
  <li><a href="#calculate-correlations-with-ground-truth-from-standard-trained-model" id="toc-calculate-correlations-with-ground-truth-from-standard-trained-model" class="nav-link" data-scroll-target="#calculate-correlations-with-ground-truth-from-standard-trained-model">Calculate correlations with ground truth from standard trained model</a></li>
  <li><a href="#calculate-correlations-with-ground-truth-from-popseq-trained-model" id="toc-calculate-correlations-with-ground-truth-from-popseq-trained-model" class="nav-link" data-scroll-target="#calculate-correlations-with-ground-truth-from-popseq-trained-model">Calculate correlations with ground truth from popseq trained model</a></li>
  <li><a href="#compare-the-two-training-predictions" id="toc-compare-the-two-training-predictions" class="nav-link" data-scroll-target="#compare-the-two-training-predictions">Compare the two training predictions</a></li>
  </ul></li>
  <li><a href="#test-8-nodes-and-20-epochs" id="toc-test-8-nodes-and-20-epochs" class="nav-link" data-scroll-target="#test-8-nodes-and-20-epochs">Test 8 nodes and 20 epochs</a>
  <ul class="collapse">
  <li><a href="#calculate-correlations-with-ground-truth-from-popseq-trained-model-1" id="toc-calculate-correlations-with-ground-truth-from-popseq-trained-model-1" class="nav-link" data-scroll-target="#calculate-correlations-with-ground-truth-from-popseq-trained-model-1">Calculate correlations with ground truth from popseq trained model</a></li>
  <li><a href="#compare-the-two-training-predictions-1" id="toc-compare-the-two-training-predictions-1" class="nav-link" data-scroll-target="#compare-the-two-training-predictions-1">Compare the two training predictions</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">testing_pop_seq</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 14, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"testing_pop_seq"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-07-14'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="context" class="level1">
<h1>Context</h1>
<p>I have verified that the pop-seq training datasets are intact (they got out of order, but the training process is intact). Now I can test if the pop-seq training does have an impact on personalized prediction.</p>
<section id="create-input-datasets-for-testing" class="level2">
<h2 class="anchored" data-anchor-id="create-input-datasets-for-testing">Create input datasets for testing</h2>
<p>Testing the personalized prediction doesn’t require too many training examples, so I will start with testing 40 individuals for 100 genes. I’ll begin by assembling the input files for this inference pass.</p>
<section id="individuals" class="level3">
<h3 class="anchored" data-anchor-id="individuals">Individuals</h3>
<p>I will use 20 EUR and 20 AFR individuals to maintain some haplotypic diversity</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># meta_table &lt;- read_table(meta_path)</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>meta_path_rwp <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/relationships_w_pops.txt"</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>meta_table <span class="ot">&lt;-</span> <span class="fu">read_table</span>(meta_path_rwp)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>meta_table <span class="ot">&lt;-</span> meta_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(IID <span class="sc">%in%</span> <span class="fu">colnames</span>(geuvadis_full))</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>CEU <span class="ot">&lt;-</span> meta_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(population <span class="sc">!=</span> <span class="st">"YRI"</span>) </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>CEU_20 <span class="ot">&lt;-</span> CEU<span class="sc">$</span>IID[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>YRI <span class="ot">&lt;-</span> meta_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(population <span class="sc">==</span> <span class="st">"YRI"</span>) </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>YRI_20 <span class="ot">&lt;-</span> YRI<span class="sc">$</span>IID[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>all_inds <span class="ot">&lt;-</span> <span class="fu">c</span>(CEU_20,YRI_20)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_inds, <span class="st">"40_inds.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="genes" class="level3">
<h3 class="anchored" data-anchor-id="genes">Genes</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>gene_meta_path <span class="ot">&lt;-</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/gene_metadata_full.csv"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>gene_meta_path <span class="ot">&lt;-</span> <span class="st">"/Users/saideepgona/Documents/PhD/imlab/Github_Repos/Daily-Blog-Sai/posts/2023-06-03-calculating_genetic_distance_from_ref/gene_metadata_w_allele_counts.csv"</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(gene_meta_path )</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>pos_corr <span class="ot">&lt;-</span> gene_metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(all_geuvadis_ge_corrs <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(all_geuvadis_ge_corrs))</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>pos_corr_50 <span class="ot">&lt;-</span> pos_corr[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,]<span class="sc">$</span>gene_name</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>pos_corr_50_intervals <span class="ot">&lt;-</span> <span class="fu">paste0</span>(pos_corr<span class="sc">$</span>chrom,<span class="st">"_"</span>,pos_corr<span class="sc">$</span>start,<span class="st">"_"</span>,pos_corr<span class="sc">$</span>end)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>neg_corr <span class="ot">&lt;-</span> gene_metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(all_geuvadis_ge_corrs <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(all_geuvadis_ge_corrs)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>neg_corr_50 <span class="ot">&lt;-</span> neg_corr[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,]<span class="sc">$</span>gene_name</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>neg_corr_50_intervals <span class="ot">&lt;-</span> <span class="fu">paste0</span>(neg_corr<span class="sc">$</span>chrom,<span class="st">"_"</span>,neg_corr<span class="sc">$</span>start,<span class="st">"_"</span>,neg_corr<span class="sc">$</span>end)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(pos_corr_50,neg_corr_50)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_genes, <span class="st">"100_corr_genes.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>all_genes_intervals <span class="ot">&lt;-</span> <span class="fu">c</span>(pos_corr_50_intervals,neg_corr_50_intervals)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_genes_intervals, <span class="st">"100_corr_intervals.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>canonical_tss_path <span class="ot">&lt;-</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/canonical_TSS_full_metadata.txt"</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>canonical_tss_metadata <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(canonical_tss_path)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>canonical_tss_metadata <span class="ot">&lt;-</span> canonical_tss_metadata <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(external_gene_name, <span class="at">.keep_all=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">drop_na</span>()</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(canonical_tss_metadata) <span class="ot">&lt;-</span> canonical_tss_metadata<span class="sc">$</span>external_gene_name</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>all_canonical_tss <span class="ot">&lt;-</span> canonical_tss_metadata[all_genes,]<span class="sc">$</span>target_regions</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_canonical_tss, <span class="st">"100_corr_canonical_tss_intervals.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="checking-correctness-of-pipeline" class="level2">
<h2 class="anchored" data-anchor-id="checking-correctness-of-pipeline">Checking correctness of pipeline</h2>
<p>Using the ERAP2 window on a single individual,</p>
<section id="calculate-correlations-with-ground-truth-from-standard-trained-model" class="level3">
<h3 class="anchored" data-anchor-id="calculate-correlations-with-ground-truth-from-standard-trained-model">Calculate correlations with ground truth from standard trained model</h3>
<p>The standard trained model was trained to 80 epochs using reference genome sequence and 240 gpus (batch size 480)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpmisc)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggExtra)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(preprocessCore)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification.csv"</span>)))</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>corrs_df_standard <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>enformer_compare_standard <span class="ot">&lt;-</span> enformer_compare</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calculate-correlations-with-ground-truth-from-popseq-trained-model" class="level3">
<h3 class="anchored" data-anchor-id="calculate-correlations-with-ground-truth-from-popseq-trained-model">Calculate correlations with ground truth from popseq trained model</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification_popseq.csv"</span>)))</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>corrs_df_popseq <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>enformer_compare_popseq <span class="ot">&lt;-</span> enformer_compare</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compare-the-two-training-predictions" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-two-training-predictions">Compare the two training predictions</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(corrs_df_standard<span class="sc">$</span>ge,corrs_df_popseq<span class="sc">$</span>ge) <span class="sc">+</span> <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(<span class="fu">as.numeric</span>(corrs_df_standard<span class="sc">$</span>ge),<span class="fu">as.numeric</span>(corrs_df_popseq<span class="sc">$</span>ge))</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000129055"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000129055"</span>,]))</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000100321"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000100321"</span>,]))</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164978"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000164978"</span>,]))</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df_standard<span class="sc">$</span>ge, corrs_df_popseq<span class="sc">$</span>ge)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="test-8-nodes-and-20-epochs" class="level2">
<h2 class="anchored" data-anchor-id="test-8-nodes-and-20-epochs">Test 8 nodes and 20 epochs</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification_standard_8.csv"</span>)))</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>corrs_df_standard <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>enformer_compare_standard <span class="ot">&lt;-</span> enformer_compare</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="calculate-correlations-with-ground-truth-from-popseq-trained-model-1" class="level3">
<h3 class="anchored" data-anchor-id="calculate-correlations-with-ground-truth-from-popseq-trained-model-1">Calculate correlations with ground truth from popseq trained model</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification_popseq_8.csv"</span>)))</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>corrs_df_popseq <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>enformer_compare_popseq <span class="ot">&lt;-</span> enformer_compare</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compare-the-two-training-predictions-1" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-two-training-predictions-1">Compare the two training predictions</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(corrs_df_standard<span class="sc">$</span>ge,corrs_df_popseq<span class="sc">$</span>ge)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(<span class="fu">as.numeric</span>(corrs_df_standard<span class="sc">$</span>ge),<span class="fu">as.numeric</span>(corrs_df_popseq<span class="sc">$</span>ge))</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000129055"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000129055"</span>,]))</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000100321"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000100321"</span>,]))</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164978"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000164978"</span>,]))</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df_standard<span class="sc">$</span>ge, corrs_df_popseq<span class="sc">$</span>ge)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb27" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "testing_pop_seq"</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-07-14"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Set up box storage directory</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"testing_pop_seq"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-07-14'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>I have verified that the pop-seq training datasets are intact (they got out of order, but the training process is intact). Now I can test if the pop-seq training does have an impact on personalized prediction.</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create input datasets for testing</span></span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>Testing the personalized prediction doesn't require too many training examples, so I will start with testing 40 individuals for 100 genes. I'll begin by assembling the input files for this inference pass.</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a><span class="fu">### Individuals</span></span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>I will use 20 EUR and 20 AFR individuals to maintain some haplotypic diversity</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a><span class="co"># meta_table &lt;- read_table(meta_path)</span></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>meta_path_rwp <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/relationships_w_pops.txt"</span></span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>meta_table <span class="ot">&lt;-</span> <span class="fu">read_table</span>(meta_path_rwp)</span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>meta_table <span class="ot">&lt;-</span> meta_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(IID <span class="sc">%in%</span> <span class="fu">colnames</span>(geuvadis_full))</span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>CEU <span class="ot">&lt;-</span> meta_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(population <span class="sc">!=</span> <span class="st">"YRI"</span>) </span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>CEU_20 <span class="ot">&lt;-</span> CEU<span class="sc">$</span>IID[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>YRI <span class="ot">&lt;-</span> meta_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(population <span class="sc">==</span> <span class="st">"YRI"</span>) </span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>YRI_20 <span class="ot">&lt;-</span> YRI<span class="sc">$</span>IID[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a>all_inds <span class="ot">&lt;-</span> <span class="fu">c</span>(CEU_20,YRI_20)</span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_inds, <span class="st">"40_inds.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a><span class="fu">### Genes</span></span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a>gene_meta_path <span class="ot">&lt;-</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/gene_metadata_full.csv"</span></span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a>gene_meta_path <span class="ot">&lt;-</span> <span class="st">"/Users/saideepgona/Documents/PhD/imlab/Github_Repos/Daily-Blog-Sai/posts/2023-06-03-calculating_genetic_distance_from_ref/gene_metadata_w_allele_counts.csv"</span></span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(gene_meta_path )</span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a>pos_corr <span class="ot">&lt;-</span> gene_metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(all_geuvadis_ge_corrs <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(all_geuvadis_ge_corrs))</span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a>pos_corr_50 <span class="ot">&lt;-</span> pos_corr[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,]<span class="sc">$</span>gene_name</span>
<span id="cb27-86"><a href="#cb27-86" aria-hidden="true" tabindex="-1"></a>pos_corr_50_intervals <span class="ot">&lt;-</span> <span class="fu">paste0</span>(pos_corr<span class="sc">$</span>chrom,<span class="st">"_"</span>,pos_corr<span class="sc">$</span>start,<span class="st">"_"</span>,pos_corr<span class="sc">$</span>end)</span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a>neg_corr <span class="ot">&lt;-</span> gene_metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(all_geuvadis_ge_corrs <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(all_geuvadis_ge_corrs)</span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a>neg_corr_50 <span class="ot">&lt;-</span> neg_corr[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,]<span class="sc">$</span>gene_name</span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a>neg_corr_50_intervals <span class="ot">&lt;-</span> <span class="fu">paste0</span>(neg_corr<span class="sc">$</span>chrom,<span class="st">"_"</span>,neg_corr<span class="sc">$</span>start,<span class="st">"_"</span>,neg_corr<span class="sc">$</span>end)</span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(pos_corr_50,neg_corr_50)</span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_genes, <span class="st">"100_corr_genes.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-96"><a href="#cb27-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-97"><a href="#cb27-97" aria-hidden="true" tabindex="-1"></a>all_genes_intervals <span class="ot">&lt;-</span> <span class="fu">c</span>(pos_corr_50_intervals,neg_corr_50_intervals)</span>
<span id="cb27-98"><a href="#cb27-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-99"><a href="#cb27-99" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_genes_intervals, <span class="st">"100_corr_intervals.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-100"><a href="#cb27-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-101"><a href="#cb27-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-102"><a href="#cb27-102" aria-hidden="true" tabindex="-1"></a>canonical_tss_path <span class="ot">&lt;-</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/canonical_TSS_full_metadata.txt"</span></span>
<span id="cb27-103"><a href="#cb27-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-104"><a href="#cb27-104" aria-hidden="true" tabindex="-1"></a>canonical_tss_metadata <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(canonical_tss_path)</span>
<span id="cb27-105"><a href="#cb27-105" aria-hidden="true" tabindex="-1"></a>canonical_tss_metadata <span class="ot">&lt;-</span> canonical_tss_metadata <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(external_gene_name, <span class="at">.keep_all=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">drop_na</span>()</span>
<span id="cb27-106"><a href="#cb27-106" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(canonical_tss_metadata) <span class="ot">&lt;-</span> canonical_tss_metadata<span class="sc">$</span>external_gene_name</span>
<span id="cb27-107"><a href="#cb27-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-108"><a href="#cb27-108" aria-hidden="true" tabindex="-1"></a>all_canonical_tss <span class="ot">&lt;-</span> canonical_tss_metadata[all_genes,]<span class="sc">$</span>target_regions</span>
<span id="cb27-109"><a href="#cb27-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-110"><a href="#cb27-110" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(all_canonical_tss, <span class="st">"100_corr_canonical_tss_intervals.txt"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-111"><a href="#cb27-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-112"><a href="#cb27-112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-113"><a href="#cb27-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-114"><a href="#cb27-114" aria-hidden="true" tabindex="-1"></a><span class="fu">## Checking correctness of pipeline</span></span>
<span id="cb27-115"><a href="#cb27-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-116"><a href="#cb27-116" aria-hidden="true" tabindex="-1"></a>Using the ERAP2 window on a single individual,</span>
<span id="cb27-117"><a href="#cb27-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-118"><a href="#cb27-118" aria-hidden="true" tabindex="-1"></a><span class="fu">### Calculate correlations with ground truth from standard trained model</span></span>
<span id="cb27-119"><a href="#cb27-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-120"><a href="#cb27-120" aria-hidden="true" tabindex="-1"></a>The standard trained model was trained to 80 epochs using reference genome sequence and 240 gpus (batch size 480)</span>
<span id="cb27-121"><a href="#cb27-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-124"><a href="#cb27-124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-125"><a href="#cb27-125" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb27-126"><a href="#cb27-126" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb27-127"><a href="#cb27-127" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb27-128"><a href="#cb27-128" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpmisc)</span>
<span id="cb27-129"><a href="#cb27-129" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb27-130"><a href="#cb27-130" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb27-131"><a href="#cb27-131" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggExtra)</span>
<span id="cb27-132"><a href="#cb27-132" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(preprocessCore)</span>
<span id="cb27-133"><a href="#cb27-133" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb27-134"><a href="#cb27-134" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb27-135"><a href="#cb27-135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-136"><a href="#cb27-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-139"><a href="#cb27-139" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-140"><a href="#cb27-140" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb27-141"><a href="#cb27-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-142"><a href="#cb27-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb27-143"><a href="#cb27-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-144"><a href="#cb27-144" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb27-145"><a href="#cb27-145" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-146"><a href="#cb27-146" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb27-147"><a href="#cb27-147" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb27-148"><a href="#cb27-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-149"><a href="#cb27-149" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb27-150"><a href="#cb27-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-151"><a href="#cb27-151" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb27-152"><a href="#cb27-152" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb27-153"><a href="#cb27-153" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb27-154"><a href="#cb27-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-155"><a href="#cb27-155" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb27-156"><a href="#cb27-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-157"><a href="#cb27-157" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification.csv"</span>)))</span>
<span id="cb27-158"><a href="#cb27-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-159"><a href="#cb27-159" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb27-160"><a href="#cb27-160" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-161"><a href="#cb27-161" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-162"><a href="#cb27-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-163"><a href="#cb27-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-164"><a href="#cb27-164" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-165"><a href="#cb27-165" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb27-166"><a href="#cb27-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-167"><a href="#cb27-167" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb27-168"><a href="#cb27-168" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb27-169"><a href="#cb27-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-170"><a href="#cb27-170" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span>
<span id="cb27-171"><a href="#cb27-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-172"><a href="#cb27-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-173"><a href="#cb27-173" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-174"><a href="#cb27-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-177"><a href="#cb27-177" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-178"><a href="#cb27-178" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb27-179"><a href="#cb27-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-180"><a href="#cb27-180" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb27-181"><a href="#cb27-181" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb27-182"><a href="#cb27-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-183"><a href="#cb27-183" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb27-184"><a href="#cb27-184" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-185"><a href="#cb27-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-188"><a href="#cb27-188" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-189"><a href="#cb27-189" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb27-190"><a href="#cb27-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-191"><a href="#cb27-191" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb27-192"><a href="#cb27-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-193"><a href="#cb27-193" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb27-194"><a href="#cb27-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-195"><a href="#cb27-195" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-196"><a href="#cb27-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-197"><a href="#cb27-197" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb27-198"><a href="#cb27-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-199"><a href="#cb27-199" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-200"><a href="#cb27-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-201"><a href="#cb27-201" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb27-202"><a href="#cb27-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-203"><a href="#cb27-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-204"><a href="#cb27-204" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb27-205"><a href="#cb27-205" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-206"><a href="#cb27-206" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb27-207"><a href="#cb27-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-208"><a href="#cb27-208" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb27-209"><a href="#cb27-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-210"><a href="#cb27-210" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb27-211"><a href="#cb27-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-212"><a href="#cb27-212" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb27-213"><a href="#cb27-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-214"><a href="#cb27-214" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span>
<span id="cb27-215"><a href="#cb27-215" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-216"><a href="#cb27-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-219"><a href="#cb27-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-220"><a href="#cb27-220" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb27-221"><a href="#cb27-221" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb27-222"><a href="#cb27-222" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb27-223"><a href="#cb27-223" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb27-224"><a href="#cb27-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb27-225"><a href="#cb27-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb27-226"><a href="#cb27-226" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-227"><a href="#cb27-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-228"><a href="#cb27-228" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb27-229"><a href="#cb27-229" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb27-230"><a href="#cb27-230" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb27-231"><a href="#cb27-231" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb27-232"><a href="#cb27-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-233"><a href="#cb27-233" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-234"><a href="#cb27-234" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-235"><a href="#cb27-235" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb27-236"><a href="#cb27-236" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb27-237"><a href="#cb27-237" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-238"><a href="#cb27-238" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb27-239"><a href="#cb27-239" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb27-240"><a href="#cb27-240" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb27-241"><a href="#cb27-241" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-242"><a href="#cb27-242" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-243"><a href="#cb27-243" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span>
<span id="cb27-244"><a href="#cb27-244" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-245"><a href="#cb27-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-248"><a href="#cb27-248" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-249"><a href="#cb27-249" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb27-250"><a href="#cb27-250" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb27-251"><a href="#cb27-251" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb27-252"><a href="#cb27-252" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-253"><a href="#cb27-253" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb27-254"><a href="#cb27-254" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb27-255"><a href="#cb27-255" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb27-256"><a href="#cb27-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-257"><a href="#cb27-257" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb27-258"><a href="#cb27-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-259"><a href="#cb27-259" aria-hidden="true" tabindex="-1"></a>corrs_df_standard <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb27-260"><a href="#cb27-260" aria-hidden="true" tabindex="-1"></a>enformer_compare_standard <span class="ot">&lt;-</span> enformer_compare</span>
<span id="cb27-261"><a href="#cb27-261" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-262"><a href="#cb27-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-263"><a href="#cb27-263" aria-hidden="true" tabindex="-1"></a><span class="fu">### Calculate correlations with ground truth from popseq trained model</span></span>
<span id="cb27-264"><a href="#cb27-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-267"><a href="#cb27-267" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-268"><a href="#cb27-268" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb27-269"><a href="#cb27-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-270"><a href="#cb27-270" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb27-271"><a href="#cb27-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-272"><a href="#cb27-272" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb27-273"><a href="#cb27-273" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-274"><a href="#cb27-274" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb27-275"><a href="#cb27-275" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb27-276"><a href="#cb27-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-277"><a href="#cb27-277" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb27-278"><a href="#cb27-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-279"><a href="#cb27-279" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb27-280"><a href="#cb27-280" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb27-281"><a href="#cb27-281" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb27-282"><a href="#cb27-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-283"><a href="#cb27-283" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb27-284"><a href="#cb27-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-285"><a href="#cb27-285" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification_popseq.csv"</span>)))</span>
<span id="cb27-286"><a href="#cb27-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-287"><a href="#cb27-287" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb27-288"><a href="#cb27-288" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-289"><a href="#cb27-289" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-290"><a href="#cb27-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-291"><a href="#cb27-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-292"><a href="#cb27-292" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-293"><a href="#cb27-293" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb27-294"><a href="#cb27-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-295"><a href="#cb27-295" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb27-296"><a href="#cb27-296" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb27-297"><a href="#cb27-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-298"><a href="#cb27-298" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span>
<span id="cb27-299"><a href="#cb27-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-300"><a href="#cb27-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-301"><a href="#cb27-301" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-302"><a href="#cb27-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-305"><a href="#cb27-305" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-306"><a href="#cb27-306" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb27-307"><a href="#cb27-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-308"><a href="#cb27-308" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb27-309"><a href="#cb27-309" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb27-310"><a href="#cb27-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-311"><a href="#cb27-311" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb27-312"><a href="#cb27-312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-313"><a href="#cb27-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-316"><a href="#cb27-316" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-317"><a href="#cb27-317" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb27-318"><a href="#cb27-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-319"><a href="#cb27-319" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb27-320"><a href="#cb27-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-321"><a href="#cb27-321" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb27-322"><a href="#cb27-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-323"><a href="#cb27-323" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-324"><a href="#cb27-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-325"><a href="#cb27-325" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb27-326"><a href="#cb27-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-327"><a href="#cb27-327" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-328"><a href="#cb27-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-329"><a href="#cb27-329" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb27-330"><a href="#cb27-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-331"><a href="#cb27-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-332"><a href="#cb27-332" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb27-333"><a href="#cb27-333" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-334"><a href="#cb27-334" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb27-335"><a href="#cb27-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-336"><a href="#cb27-336" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb27-337"><a href="#cb27-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-338"><a href="#cb27-338" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb27-339"><a href="#cb27-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-340"><a href="#cb27-340" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb27-341"><a href="#cb27-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-342"><a href="#cb27-342" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span>
<span id="cb27-343"><a href="#cb27-343" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-344"><a href="#cb27-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-347"><a href="#cb27-347" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-348"><a href="#cb27-348" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb27-349"><a href="#cb27-349" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb27-350"><a href="#cb27-350" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb27-351"><a href="#cb27-351" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb27-352"><a href="#cb27-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb27-353"><a href="#cb27-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb27-354"><a href="#cb27-354" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-355"><a href="#cb27-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-356"><a href="#cb27-356" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb27-357"><a href="#cb27-357" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb27-358"><a href="#cb27-358" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb27-359"><a href="#cb27-359" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb27-360"><a href="#cb27-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-361"><a href="#cb27-361" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-362"><a href="#cb27-362" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-363"><a href="#cb27-363" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb27-364"><a href="#cb27-364" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb27-365"><a href="#cb27-365" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-366"><a href="#cb27-366" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb27-367"><a href="#cb27-367" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb27-368"><a href="#cb27-368" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb27-369"><a href="#cb27-369" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-370"><a href="#cb27-370" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-371"><a href="#cb27-371" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span>
<span id="cb27-372"><a href="#cb27-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-373"><a href="#cb27-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-376"><a href="#cb27-376" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-377"><a href="#cb27-377" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb27-378"><a href="#cb27-378" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb27-379"><a href="#cb27-379" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb27-380"><a href="#cb27-380" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-381"><a href="#cb27-381" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb27-382"><a href="#cb27-382" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb27-383"><a href="#cb27-383" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb27-384"><a href="#cb27-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-385"><a href="#cb27-385" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb27-386"><a href="#cb27-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-387"><a href="#cb27-387" aria-hidden="true" tabindex="-1"></a>corrs_df_popseq <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb27-388"><a href="#cb27-388" aria-hidden="true" tabindex="-1"></a>enformer_compare_popseq <span class="ot">&lt;-</span> enformer_compare</span>
<span id="cb27-389"><a href="#cb27-389" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-390"><a href="#cb27-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-391"><a href="#cb27-391" aria-hidden="true" tabindex="-1"></a><span class="fu">### Compare the two training predictions</span></span>
<span id="cb27-392"><a href="#cb27-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-395"><a href="#cb27-395" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-396"><a href="#cb27-396" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(corrs_df_standard<span class="sc">$</span>ge,corrs_df_popseq<span class="sc">$</span>ge) <span class="sc">+</span> <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb27-397"><a href="#cb27-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-398"><a href="#cb27-398" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(<span class="fu">as.numeric</span>(corrs_df_standard<span class="sc">$</span>ge),<span class="fu">as.numeric</span>(corrs_df_popseq<span class="sc">$</span>ge))</span>
<span id="cb27-399"><a href="#cb27-399" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb27-400"><a href="#cb27-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-401"><a href="#cb27-401" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000129055"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000129055"</span>,]))</span>
<span id="cb27-402"><a href="#cb27-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-403"><a href="#cb27-403" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000100321"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000100321"</span>,]))</span>
<span id="cb27-404"><a href="#cb27-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-405"><a href="#cb27-405" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164978"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000164978"</span>,]))</span>
<span id="cb27-406"><a href="#cb27-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-407"><a href="#cb27-407" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df_standard<span class="sc">$</span>ge, corrs_df_popseq<span class="sc">$</span>ge)</span>
<span id="cb27-408"><a href="#cb27-408" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-409"><a href="#cb27-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-410"><a href="#cb27-410" aria-hidden="true" tabindex="-1"></a><span class="fu">## Test 8 nodes and 20 epochs</span></span>
<span id="cb27-411"><a href="#cb27-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-414"><a href="#cb27-414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-415"><a href="#cb27-415" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb27-416"><a href="#cb27-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-417"><a href="#cb27-417" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb27-418"><a href="#cb27-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-419"><a href="#cb27-419" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb27-420"><a href="#cb27-420" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-421"><a href="#cb27-421" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb27-422"><a href="#cb27-422" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb27-423"><a href="#cb27-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-424"><a href="#cb27-424" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb27-425"><a href="#cb27-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-426"><a href="#cb27-426" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb27-427"><a href="#cb27-427" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb27-428"><a href="#cb27-428" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb27-429"><a href="#cb27-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-430"><a href="#cb27-430" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb27-431"><a href="#cb27-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-432"><a href="#cb27-432" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification_standard_8.csv"</span>)))</span>
<span id="cb27-433"><a href="#cb27-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-434"><a href="#cb27-434" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb27-435"><a href="#cb27-435" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-436"><a href="#cb27-436" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-437"><a href="#cb27-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-438"><a href="#cb27-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-439"><a href="#cb27-439" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-440"><a href="#cb27-440" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb27-441"><a href="#cb27-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-442"><a href="#cb27-442" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb27-443"><a href="#cb27-443" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb27-444"><a href="#cb27-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-445"><a href="#cb27-445" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span>
<span id="cb27-446"><a href="#cb27-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-447"><a href="#cb27-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-448"><a href="#cb27-448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-449"><a href="#cb27-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-452"><a href="#cb27-452" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-453"><a href="#cb27-453" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb27-454"><a href="#cb27-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-455"><a href="#cb27-455" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb27-456"><a href="#cb27-456" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb27-457"><a href="#cb27-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-458"><a href="#cb27-458" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb27-459"><a href="#cb27-459" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-460"><a href="#cb27-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-463"><a href="#cb27-463" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-464"><a href="#cb27-464" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb27-465"><a href="#cb27-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-466"><a href="#cb27-466" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb27-467"><a href="#cb27-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-468"><a href="#cb27-468" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb27-469"><a href="#cb27-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-470"><a href="#cb27-470" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-471"><a href="#cb27-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-472"><a href="#cb27-472" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb27-473"><a href="#cb27-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-474"><a href="#cb27-474" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-475"><a href="#cb27-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-476"><a href="#cb27-476" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb27-477"><a href="#cb27-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-478"><a href="#cb27-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-479"><a href="#cb27-479" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb27-480"><a href="#cb27-480" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-481"><a href="#cb27-481" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb27-482"><a href="#cb27-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-483"><a href="#cb27-483" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb27-484"><a href="#cb27-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-485"><a href="#cb27-485" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb27-486"><a href="#cb27-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-487"><a href="#cb27-487" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb27-488"><a href="#cb27-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-489"><a href="#cb27-489" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span>
<span id="cb27-490"><a href="#cb27-490" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-491"><a href="#cb27-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-494"><a href="#cb27-494" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-495"><a href="#cb27-495" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb27-496"><a href="#cb27-496" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb27-497"><a href="#cb27-497" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb27-498"><a href="#cb27-498" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb27-499"><a href="#cb27-499" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb27-500"><a href="#cb27-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb27-501"><a href="#cb27-501" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-502"><a href="#cb27-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-503"><a href="#cb27-503" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb27-504"><a href="#cb27-504" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb27-505"><a href="#cb27-505" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb27-506"><a href="#cb27-506" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb27-507"><a href="#cb27-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-508"><a href="#cb27-508" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-509"><a href="#cb27-509" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-510"><a href="#cb27-510" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb27-511"><a href="#cb27-511" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb27-512"><a href="#cb27-512" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-513"><a href="#cb27-513" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb27-514"><a href="#cb27-514" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb27-515"><a href="#cb27-515" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb27-516"><a href="#cb27-516" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-517"><a href="#cb27-517" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-518"><a href="#cb27-518" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span>
<span id="cb27-519"><a href="#cb27-519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-520"><a href="#cb27-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-523"><a href="#cb27-523" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-524"><a href="#cb27-524" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb27-525"><a href="#cb27-525" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb27-526"><a href="#cb27-526" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb27-527"><a href="#cb27-527" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-528"><a href="#cb27-528" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb27-529"><a href="#cb27-529" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb27-530"><a href="#cb27-530" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb27-531"><a href="#cb27-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-532"><a href="#cb27-532" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb27-533"><a href="#cb27-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-534"><a href="#cb27-534" aria-hidden="true" tabindex="-1"></a>corrs_df_standard <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb27-535"><a href="#cb27-535" aria-hidden="true" tabindex="-1"></a>enformer_compare_standard <span class="ot">&lt;-</span> enformer_compare</span>
<span id="cb27-536"><a href="#cb27-536" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-537"><a href="#cb27-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-538"><a href="#cb27-538" aria-hidden="true" tabindex="-1"></a><span class="fu">### Calculate correlations with ground truth from popseq trained model</span></span>
<span id="cb27-539"><a href="#cb27-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-542"><a href="#cb27-542" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-543"><a href="#cb27-543" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb27-544"><a href="#cb27-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-545"><a href="#cb27-545" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb27-546"><a href="#cb27-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-547"><a href="#cb27-547" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb27-548"><a href="#cb27-548" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-549"><a href="#cb27-549" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb27-550"><a href="#cb27-550" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb27-551"><a href="#cb27-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-552"><a href="#cb27-552" aria-hidden="true" tabindex="-1"></a><span class="co"># Load predixcan expression</span></span>
<span id="cb27-553"><a href="#cb27-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-554"><a href="#cb27-554" aria-hidden="true" tabindex="-1"></a>predixcan_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"predixcan_expression.txt"</span>))</span>
<span id="cb27-555"><a href="#cb27-555" aria-hidden="true" tabindex="-1"></a>predixcan <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(predixcan_full[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(predixcan_full)])</span>
<span id="cb27-556"><a href="#cb27-556" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(predixcan) <span class="ot">&lt;-</span> predixcan_full<span class="sc">$</span>gene_names_proper</span>
<span id="cb27-557"><a href="#cb27-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-558"><a href="#cb27-558" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb27-559"><a href="#cb27-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-560"><a href="#cb27-560" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(<span class="st">"CAGE_summed_haplotype_CAGE:LCL_quantification_popseq_8.csv"</span>)))</span>
<span id="cb27-561"><a href="#cb27-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-562"><a href="#cb27-562" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span></span>
<span id="cb27-563"><a href="#cb27-563" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span><span class="st">`</span><span class="at">Unnamed: 0</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-564"><a href="#cb27-564" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-565"><a href="#cb27-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-566"><a href="#cb27-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-567"><a href="#cb27-567" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb27-568"><a href="#cb27-568" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_haplo1</span>
<span id="cb27-569"><a href="#cb27-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-570"><a href="#cb27-570" aria-hidden="true" tabindex="-1"></a>enformer_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(enformer))))</span>
<span id="cb27-571"><a href="#cb27-571" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> enformer_norm</span>
<span id="cb27-572"><a href="#cb27-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-573"><a href="#cb27-573" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span>
<span id="cb27-574"><a href="#cb27-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-575"><a href="#cb27-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-576"><a href="#cb27-576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-577"><a href="#cb27-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-580"><a href="#cb27-580" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-581"><a href="#cb27-581" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb27-582"><a href="#cb27-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-583"><a href="#cb27-583" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb27-584"><a href="#cb27-584" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb27-585"><a href="#cb27-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-586"><a href="#cb27-586" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb27-587"><a href="#cb27-587" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-588"><a href="#cb27-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-591"><a href="#cb27-591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-592"><a href="#cb27-592" aria-hidden="true" tabindex="-1"></a>gp_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(predixcan))</span>
<span id="cb27-593"><a href="#cb27-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-594"><a href="#cb27-594" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb27-595"><a href="#cb27-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-596"><a href="#cb27-596" aria-hidden="true" tabindex="-1"></a>all_three_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_ind,ge_ind)</span>
<span id="cb27-597"><a href="#cb27-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-598"><a href="#cb27-598" aria-hidden="true" tabindex="-1"></a>gp_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-599"><a href="#cb27-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-600"><a href="#cb27-600" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb27-601"><a href="#cb27-601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-602"><a href="#cb27-602" aria-hidden="true" tabindex="-1"></a>ep_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(enformer), <span class="fu">rownames</span>(predixcan))</span>
<span id="cb27-603"><a href="#cb27-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-604"><a href="#cb27-604" aria-hidden="true" tabindex="-1"></a>all_three_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gp_genes, ge_genes)</span>
<span id="cb27-605"><a href="#cb27-605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-606"><a href="#cb27-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-607"><a href="#cb27-607" aria-hidden="true" tabindex="-1"></a><span class="co"># USING NON-IMPUTED VALUES FOR PREDIXCAN</span></span>
<span id="cb27-608"><a href="#cb27-608" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-609"><a href="#cb27-609" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[all_three_genes, all_three_ind]</span>
<span id="cb27-610"><a href="#cb27-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-611"><a href="#cb27-611" aria-hidden="true" tabindex="-1"></a>predixcan_compare <span class="ot">&lt;-</span> predixcan[all_three_genes,all_three_ind]</span>
<span id="cb27-612"><a href="#cb27-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-613"><a href="#cb27-613" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[all_three_genes, all_three_ind]</span>
<span id="cb27-614"><a href="#cb27-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-615"><a href="#cb27-615" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[all_three_genes,]</span>
<span id="cb27-616"><a href="#cb27-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-617"><a href="#cb27-617" aria-hidden="true" tabindex="-1"></a>individual_metadata_compare <span class="ot">&lt;-</span> individual_metadata[<span class="fu">rownames</span>(individual_metadata) <span class="sc">%in%</span> all_three_ind,]</span>
<span id="cb27-618"><a href="#cb27-618" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-619"><a href="#cb27-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-622"><a href="#cb27-622" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-623"><a href="#cb27-623" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb27-624"><a href="#cb27-624" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb27-625"><a href="#cb27-625" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb27-626"><a href="#cb27-626" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb27-627"><a href="#cb27-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb27-628"><a href="#cb27-628" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb27-629"><a href="#cb27-629" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-630"><a href="#cb27-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-631"><a href="#cb27-631" aria-hidden="true" tabindex="-1"></a>gp_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, predixcan_compare)</span>
<span id="cb27-632"><a href="#cb27-632" aria-hidden="true" tabindex="-1"></a>gp_corrs_all <span class="ot">&lt;-</span> gp_corrs</span>
<span id="cb27-633"><a href="#cb27-633" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb27-634"><a href="#cb27-634" aria-hidden="true" tabindex="-1"></a>ge_corrs_all <span class="ot">&lt;-</span> ge_corrs</span>
<span id="cb27-635"><a href="#cb27-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-636"><a href="#cb27-636" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-637"><a href="#cb27-637" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164308"</span>,]), <span class="fu">as.numeric</span>(enformer_compare[<span class="st">"ENSG00000164308"</span>,]))</span>
<span id="cb27-638"><a href="#cb27-638" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(predixcan_compare$HG00096, predixcan_compare$HG00097)</span></span>
<span id="cb27-639"><a href="#cb27-639" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot(as.numeric(predixcan_compare[1,]), as.numeric(predixcan_compare[2,]))</span></span>
<span id="cb27-640"><a href="#cb27-640" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-641"><a href="#cb27-641" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gp=</span>gp_corrs, <span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs), <span class="at">gp_abs =</span> <span class="fu">abs</span>(gp_corrs), <span class="at">genes=</span>all_three_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry, <span class="at">pred_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(predixcan_compare, <span class="dv">1</span>, var)), <span class="at">enf_vars =</span> <span class="fu">as.numeric</span>(<span class="fu">apply</span>(enformer_compare, <span class="dv">1</span>, var)))</span>
<span id="cb27-642"><a href="#cb27-642" aria-hidden="true" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb27-643"><a href="#cb27-643" aria-hidden="true" tabindex="-1"></a><span class="co"># corrs_df$var_ratio &lt;- (corrs_df$enf_vars + 0.001)/(corrs_df$pred_vars + 0.001)</span></span>
<span id="cb27-644"><a href="#cb27-644" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-645"><a href="#cb27-645" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-646"><a href="#cb27-646" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(lm(pred_vars ~ ge_corrs, corrs_df))</span></span>
<span id="cb27-647"><a href="#cb27-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-648"><a href="#cb27-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-651"><a href="#cb27-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-652"><a href="#cb27-652" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df,<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>gp,<span class="at">y=</span>ge), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.7</span>,<span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb27-653"><a href="#cb27-653" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(xintercept=0) + geom_vline(yintercept=0)</span></span>
<span id="cb27-654"><a href="#cb27-654" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">"geuvadis vs. predixcan correlation across individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"geuvadis vs. enformer correlation across individuals"</span>) <span class="sc">+</span></span>
<span id="cb27-655"><a href="#cb27-655" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-656"><a href="#cb27-656" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb27-657"><a href="#cb27-657" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb27-658"><a href="#cb27-658" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x, <span class="at">se =</span> F, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb27-659"><a href="#cb27-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-660"><a href="#cb27-660" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df<span class="sc">$</span>ge)</span>
<span id="cb27-661"><a href="#cb27-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-662"><a href="#cb27-662" aria-hidden="true" tabindex="-1"></a>corrs_df_popseq <span class="ot">&lt;-</span> corrs_df</span>
<span id="cb27-663"><a href="#cb27-663" aria-hidden="true" tabindex="-1"></a>enformer_compare_popseq <span class="ot">&lt;-</span> enformer_compare</span>
<span id="cb27-664"><a href="#cb27-664" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-665"><a href="#cb27-665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-666"><a href="#cb27-666" aria-hidden="true" tabindex="-1"></a><span class="fu">### Compare the two training predictions</span></span>
<span id="cb27-667"><a href="#cb27-667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-670"><a href="#cb27-670" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-671"><a href="#cb27-671" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(corrs_df_standard<span class="sc">$</span>ge,corrs_df_popseq<span class="sc">$</span>ge)</span>
<span id="cb27-672"><a href="#cb27-672" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb27-673"><a href="#cb27-673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-674"><a href="#cb27-674" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(<span class="fu">as.numeric</span>(corrs_df_standard<span class="sc">$</span>ge),<span class="fu">as.numeric</span>(corrs_df_popseq<span class="sc">$</span>ge))</span>
<span id="cb27-675"><a href="#cb27-675" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb27-676"><a href="#cb27-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-677"><a href="#cb27-677" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000129055"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000129055"</span>,]))</span>
<span id="cb27-678"><a href="#cb27-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-679"><a href="#cb27-679" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000100321"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000100321"</span>,]))</span>
<span id="cb27-680"><a href="#cb27-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-681"><a href="#cb27-681" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">as.numeric</span>(geuvadis_compare[<span class="st">"ENSG00000164978"</span>,]), <span class="fu">as.numeric</span>(enformer_compare_standard[<span class="st">"ENSG00000164978"</span>,]))</span>
<span id="cb27-682"><a href="#cb27-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-683"><a href="#cb27-683" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(corrs_df_standard<span class="sc">$</span>ge, corrs_df_popseq<span class="sc">$</span>ge)</span>
<span id="cb27-684"><a href="#cb27-684" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>