<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-10-25">

<title>Daily-Blog-Sai - log_transform_training_tss_predictors</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#create-log-transformed-training-data" id="toc-create-log-transformed-training-data" class="nav-link" data-scroll-target="#create-log-transformed-training-data">Create log-transformed training data</a></li>
  <li><a href="#evaluate-model-performance-on-log-transformed-data" id="toc-evaluate-model-performance-on-log-transformed-data" class="nav-link" data-scroll-target="#evaluate-model-performance-on-log-transformed-data">Evaluate model performance on log-transformed data</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">log_transform_training_tss_predictors</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 25, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="context" class="level1">
<h1>Context</h1>
<p>After evaluating the performance of my trained models, it seems clear that the spearman correlations are very stable genome-wide, but the pearson correlations are not. This is likely due to the fact that the pearson correlations are sensitive to outliers, of which I have observed many. In particular, chromosome 14 has 2 extreme outliers, and in general there are many outliers in the tails of the distributions. I think the model will perform better on log-normalized data, so I will try that next.</p>
<p>NN training: 8800 min thresh: 0.311676707121647 8817 error: 273.14126 time: 48.73 mins</p>
<section id="create-log-transformed-training-data" class="level2">
<h2 class="anchored" data-anchor-id="create-log-transformed-training-data">Create log-transformed training data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># path_to_data = "/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># for (data_subset in c("train", "valid", "test")) {</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   inputs_path &lt;- glue("{path_to_data}/{data_subset}_inputs_mean_aracena.csv")</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   inputs &lt;- fread(inputs_path)</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_inputs &lt;- log(inputs+1)</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_inputs_path &lt;- glue("{path_to_data}/{data_subset}_inputs_mean_aracena_log.csv")</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   write_delim(log_inputs, log_inputs_path, delim = "\t")</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   targets_path &lt;- glue("{path_to_data}/{data_subset}_targets_mean_aracena.csv")</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   targets &lt;- fread(targets_path)</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_targets &lt;- log(targets+1)</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_targets_path &lt;- glue("{path_to_data}/{data_subset}_targets_mean_aracena_log.csv")</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   write_delim(log_targets, log_targets_path, delim = "\t")</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="evaluate-model-performance-on-log-transformed-data" class="level2">
<h2 class="anchored" data-anchor-id="evaluate-model-performance-on-log-transformed-data">Evaluate model performance on log-transformed data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(neuralnet)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>convert_region_to_gene <span class="ot">&lt;-</span> <span class="cf">function</span>(region, gene_metadata){</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>split_datasets_chroms <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(chroms)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>plot_boxplot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, path) {</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>is_CAGE <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"CAGE_epi"</span>, <span class="st">"CAGE"</span>, <span class="st">"RNAseq"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(path)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>variable, <span class="at">y=</span>value)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_CAGE, <span class="at">scales=</span><span class="st">"free"</span>))</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>run_cor <span class="ot">&lt;-</span> <span class="cf">function</span>(x,y, <span class="at">method=</span><span class="st">"pearson"</span>, <span class="at">log_bool=</span><span class="cn">FALSE</span>) {</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (log_bool) {</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">cor</span>(<span class="fu">log</span>(x<span class="sc">+</span><span class="dv">1</span>),y, <span class="at">method=</span>method))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">cor</span>(x,y, <span class="at">method=</span>method))</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>per_chrom_metrics <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path, <span class="at">cor_method=</span><span class="st">"pearson"</span>, <span class="at">primary_ind=</span><span class="dv">1</span>) {</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for data subset: {data_subset}"</span>))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,primary_ind]</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions, y[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions, y[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage nn corr</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn, y[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn, y[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>], <span class="at">method=</span>cor_method))</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>  cur_cor <span class="ot">&lt;-</span> <span class="fu">run_cor</span>(y[,primary_ind], X[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span>cor_method)</span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, cur_cor)</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,primary_ind], X[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Done recording</span></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>  chroms_all <span class="ot">&lt;-</span> <span class="fu">split_datasets_chroms</span>(data_subset, mapping_table)</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(chroms_all)</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>  uniq_chroms <span class="ot">&lt;-</span> <span class="fu">unique</span>(chroms)</span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (chrom <span class="cf">in</span> uniq_chroms) {</span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>    index_of_chrom <span class="ot">&lt;-</span> chroms <span class="sc">==</span> chrom</span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>    X_chrom <span class="ot">&lt;-</span> X[index_of_chrom,]</span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>    y_chrom <span class="ot">&lt;-</span> y[index_of_chrom,]</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for chrom: {chrom}"</span>))</span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(X_chrom))</span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(y_chrom))</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,primary_ind], y_chrom[,<span class="dv">7</span>]))</span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,primary_ind], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">7</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,primary_ind]))</span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(predictions_nn[index_of_chrom], y_chrom[,primary_ind]))</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>    chrom_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">glmnet_pred=</span>glmnet_predictions[index_of_chrom], <span class="at">flu_rnaseq =</span> y_chrom[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y_chrom[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X_chrom[,<span class="dv">4766</span>])</span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>    chrom_df_for_plotting <span class="ot">&lt;-</span> <span class="fu">melt</span>(chrom_df, <span class="at">measure_vars =</span> <span class="fu">colnames</span>(chrom_df))</span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_boxplot</span>(chrom_df_for_plotting, <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/{data_subset}_chrom_{chrom}_boxplot.png"</span>))</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage nn corr</span></span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>], <span class="at">method=</span>cor_method))</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,primary_ind], X_chrom[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span>cor_method))</span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,primary_ind], X_chrom[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>  storage_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(corrs_v, corrs_v_spear,data_subset_v, chrom_v, num_genes_v, cor_id_v)</span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(storage_df)</span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>split_datasets <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  ensembl_ids <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  external_gene_names <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(regions)) {</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    region <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions[x],<span class="dv">2</span>,<span class="fu">str_length</span>(regions[x]))</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    ensembl_id <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    external_gene_name <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>external_gene_name</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    ensembl_ids[x]  <span class="ot">&lt;-</span> ensembl_id</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    external_gene_names[x] <span class="ot">&lt;-</span> external_gene_name</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(ensembl_ids))</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(external_gene_names))</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(chroms))</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(regions))</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  return_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"chroms"</span>]] <span class="ot">&lt;-</span> chroms</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"regions"</span>]] <span class="ot">&lt;-</span> regions</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"ensembl_ids"</span>]] <span class="ot">&lt;-</span> ensembl_ids</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"external_gene_names"</span>]] <span class="ot">&lt;-</span> external_gene_names</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_list)</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>collect_all_records <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path) {</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Collecting data records for data subset: {data_subset}"</span>))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  gene_metadata <span class="ot">&lt;-</span> <span class="fu">split_datasets</span>(data_subset, mapping_table)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  return_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">data_subset=</span> data_subset, <span class="at">glmnet_pred=</span>glmnet_predictions, <span class="at">nn_pred=</span> predictions_nn, <span class="at">flu_rnaseq =</span> y[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X[,<span class="dv">4766</span>], <span class="at">ensembl_id =</span> gene_metadata[[<span class="st">"ensembl_ids"</span>]], <span class="at">external_gene_name =</span> gene_metadata[[<span class="st">"external_gene_names"</span>]], <span class="at">region =</span> gene_metadata[[<span class="st">"regions"</span>]], <span class="at">chroms =</span> gene_metadata[[<span class="st">"chroms"</span>]])</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_df)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>remapped_table_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/remapped_table_filt.csv"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>remapped_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(remapped_table_path, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(remapped_table))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1]    29 15851</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">as.character</span>(remapped_table[,<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "EU39"          "EU27"          "AF26"          "EU33"         
 [5] "AF18"          "AF38"          "AF20"          "AF22"         
 [9] "AF08"          "AF28"          "EU25"          "AF24"         
[13] "EU05"          "AF34"          "EU47"          "EU13"         
[17] "AF36"          "EU03"          "AF30"          "AF12"         
[21] "EU09"          "EU15"          "EU07"          "EU21"         
[25] "AF16"          "EU19"          "EU41"          "ref-epigenome"
[29] "group"        </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(remapped_table) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(remapped_table[,<span class="dv">1</span>])</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>remapped_table[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>primary_ind <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>glmnet_model_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_NI_from_HDF5_mean_log.linear.rds"</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>nnmodel_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp_RNAseq_NI_from_HDF5_mean_log.rds"</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>cor_method <span class="ot">=</span> <span class="st">"pearson"</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>all_records_df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"train"</span>,<span class="st">"test"</span>,<span class="st">"valid"</span>)){</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  X_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  y_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena_log.csv"</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  cur_out_df_perchrom_metrics <span class="ot">&lt;-</span> <span class="fu">per_chrom_metrics</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path, cor_method, primary_ind)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  df_store[[c]] <span class="ot">&lt;-</span> cur_out_df_perchrom_metrics</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  all_records_df_store[[c]] <span class="ot">&lt;-</span> <span class="fu">collect_all_records</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path)</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  c<span class="ot">=</span>c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for data subset: train
********* Analyzing data for chrom: 1
[1] 2014 5313
[1] 2014   12
[1] 0.7738281
[1] 0.4906751
[1] 0.5128315
[1] 0.7047689
[1] 0.5813407</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 10
[1]  710 5313
[1] 710  12
[1] 0.6472775
[1] 0.447759
[1] 0.5692331
[1] 0.5909975
[1] 0.5339694</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 11
[1] 1287 5313
[1] 1287   12
[1] 0.7712791
[1] 0.5456487
[1] 0.5931671
[1] 0.7538166
[1] 0.6369364</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 16
[1]  839 5313
[1] 839  12
[1] 0.6938366
[1] 0.4392715
[1] 0.5129959
[1] 0.63134
[1] 0.5500776</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 17
[1] 1152 5313
[1] 1152   12
[1] 0.6449451
[1] 0.394819
[1] 0.5282789
[1] 0.6056413
[1] 0.5063302</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 18
[1]  262 5313
[1] 262  12
[1] 0.7886703
[1] 0.4710867
[1] 0.5742142
[1] 0.7016001
[1] 0.5409966</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 19
[1] 1423 5313
[1] 1423   12
[1] 0.8067522
[1] 0.3972782
[1] 0.4672016
[1] 0.6908282
[1] 0.4961597</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 2
[1] 1225 5313
[1] 1225   12
[1] 0.6746361
[1] 0.4599844
[1] 0.5473104
[1] 0.6263492
[1] 0.62898</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 20
[1]  533 5313
[1] 533  12
[1] 0.836731
[1] 0.4295909
[1] 0.5013971
[1] 0.6731075
[1] 0.556115</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 21
[1]  218 5313
[1] 218  12
[1] 0.7940787
[1] 0.5481793
[1] 0.6873015
[1] 0.7568527
[1] 0.7324491</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 3
[1] 1059 5313
[1] 1059   12
[1] 0.743747
[1] 0.4979988
[1] 0.5679769
[1] 0.6932378
[1] 0.5565457</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 4
[1]  748 5313
[1] 748  12
[1] 0.5374208
[1] 0.3372035
[1] 0.5504596
[1] 0.4418816
[1] 0.4109451</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 9
[1]  762 5313
[1] 762  12
[1] 0.7774666
[1] 0.4870424
[1] 0.5928558
[1] 0.7243833
[1] 0.5766746</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: train
[1] 12232
[1] 12232
[1] 12232
[1] 12232
********* Analyzing data for data subset: test
********* Analyzing data for chrom: 13
[1]  319 5313
[1] 319  12
[1] 0.7747543
[1] 0.5376015
[1] 0.5963206
[1] 0.6713264
[1] 0.6136282</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 14
[1]  600 5313
[1] 600  12
[1] 0.8725071
[1] 0.3379276
[1] 0.3741996
[1] 0.540512
[1] 0.4616933</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 22
[1]  431 5313
[1] 431  12
[1] 0.5932224
[1] 0.3775029
[1] 0.5739591
[1] 0.5127326
[1] 0.4755188</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 8
[1]  691 5313
[1] 691  12
[1] 0.7137821
[1] 0.460958
[1] 0.6211134
[1] 0.6131504
[1] 0.5130661</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: test
[1] 2041
[1] 2041
[1] 2041
[1] 2041
********* Analyzing data for data subset: valid
********* Analyzing data for chrom: 12
[1]  990 5313
[1] 990  12
[1] 0.7726486
[1] 0.4867567
[1] 0.5453881
[1] 0.6965704
[1] 0.5466499</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Analyzing data for chrom: 15
[1]  587 5313
[1] 587  12
[1] 0.7970598
[1] 0.5935388
[1] 0.6312264
[1] 0.7513436
[1] 0.6344261</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>********* Collecting data records for data subset: valid
[1] 1577
[1] 1577
[1] 1577
[1] 1577</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df_store)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>data_subset_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>data_subset_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>cor_id_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>cor_id_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>))</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>chrom_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>chrom_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Set color palette</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>num_genes_v <span class="ot">&lt;-</span> <span class="fu">as.character</span>(full_df<span class="sc">$</span>num_genes_v)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(full_df, <span class="at">measure_vars=</span><span class="fu">c</span>(<span class="st">"corrs_v"</span>,<span class="st">"corrs_v_spear"</span>), <span class="at">id_vars=</span><span class="fu">c</span>(<span class="st">"data_subset_v"</span>, <span class="st">"chrom_v"</span>, <span class="st">"cor_id_v"</span>, <span class="st">"num_genes_v"</span>))</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>full_df_melt<span class="sc">$</span>corr_type <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(full_df_melt<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"corrs_v"</span>, <span class="st">"pearson"</span>, <span class="st">"spearman"</span>)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>full_df_melt_all_chroms <span class="ot">&lt;-</span> full_df_melt[full_df_melt<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_all_chroms_both.png"</span>), <span class="at">width=</span><span class="dv">800</span>, <span class="at">height=</span><span class="dv">400</span>)</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_melt_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>value, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>corr_type, <span class="at">nrow=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation"</span>)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>full_df_all_chroms <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_all_chroms.png"</span>))</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_nonfaceted.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">fill=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_numgenes_long.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">fill=</span>data_subset_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> id,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  plot_list[[id]] <span class="ot">&lt;-</span> g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_{id}.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot_list[[id]])</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>all_records_full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, all_records_df_store)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>chroms<span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>chroms, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "data_subset"        "glmnet_pred"        "nn_pred"           
 [4] "flu_rnaseq"         "NI_rnaseq"          "CAGE_epi"          
 [7] "ensembl_id"         "external_gene_name" "region"            
[10] "chroms"            </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>all_records_full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(all_records_full_df, <span class="at">measure_vars =</span> <span class="fu">c</span>(<span class="st">"glmnet_pred"</span>, <span class="st">"nn_pred"</span>, <span class="st">"flu_rnaseq"</span>, <span class="st">"NI_rnaseq"</span>, <span class="st">"CAGE_epi"</span>))</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df_melt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "data_subset"        "ensembl_id"         "external_gene_name"
[4] "region"             "chroms"             "variable"          
[7] "value"             </code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/all_records_scatterplots.png"</span>), <span class="at">width =</span><span class="dv">900</span>, <span class="at">height=</span><span class="dv">1200</span>)</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>glmnet_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Elastic net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"glmnet"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>nn_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Neural net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"nn"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>CAGE_epi, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Enformer CAGE pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_epi"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>NI_rnaseq, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"NI RNAseq"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_NI"</span>],<span class="at">alpha=</span><span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors))</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb60" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "log_transform_training_tss_predictors"</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-10-25"</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>After evaluating the performance of my trained models, it seems clear that the spearman correlations are very stable genome-wide, but the pearson correlations are not. This is likely due to the fact that the pearson correlations are sensitive to outliers, of which I have observed many. In particular, chromosome 14 has 2 extreme outliers, and in  general there are many outliers in the tails of the distributions. I think the model will perform better on log-normalized data, so I will try that next.</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a>NN training: 8800   min thresh: 0.311676707121647 8817   error: 273.14126        time: 48.73 mins</span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create log-transformed training data</span></span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-34"><a href="#cb60-34" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-35"><a href="#cb60-35" aria-hidden="true" tabindex="-1"></a><span class="co"># path_to_data = "/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered"</span></span>
<span id="cb60-36"><a href="#cb60-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-37"><a href="#cb60-37" aria-hidden="true" tabindex="-1"></a><span class="co"># for (data_subset in c("train", "valid", "test")) {</span></span>
<span id="cb60-38"><a href="#cb60-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-39"><a href="#cb60-39" aria-hidden="true" tabindex="-1"></a><span class="co">#   inputs_path &lt;- glue("{path_to_data}/{data_subset}_inputs_mean_aracena.csv")</span></span>
<span id="cb60-40"><a href="#cb60-40" aria-hidden="true" tabindex="-1"></a><span class="co">#   inputs &lt;- fread(inputs_path)</span></span>
<span id="cb60-41"><a href="#cb60-41" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_inputs &lt;- log(inputs+1)</span></span>
<span id="cb60-42"><a href="#cb60-42" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_inputs_path &lt;- glue("{path_to_data}/{data_subset}_inputs_mean_aracena_log.csv")</span></span>
<span id="cb60-43"><a href="#cb60-43" aria-hidden="true" tabindex="-1"></a><span class="co">#   write_delim(log_inputs, log_inputs_path, delim = "\t")</span></span>
<span id="cb60-44"><a href="#cb60-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-45"><a href="#cb60-45" aria-hidden="true" tabindex="-1"></a><span class="co">#   targets_path &lt;- glue("{path_to_data}/{data_subset}_targets_mean_aracena.csv")</span></span>
<span id="cb60-46"><a href="#cb60-46" aria-hidden="true" tabindex="-1"></a><span class="co">#   targets &lt;- fread(targets_path)</span></span>
<span id="cb60-47"><a href="#cb60-47" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_targets &lt;- log(targets+1)</span></span>
<span id="cb60-48"><a href="#cb60-48" aria-hidden="true" tabindex="-1"></a><span class="co">#   log_targets_path &lt;- glue("{path_to_data}/{data_subset}_targets_mean_aracena_log.csv")</span></span>
<span id="cb60-49"><a href="#cb60-49" aria-hidden="true" tabindex="-1"></a><span class="co">#   write_delim(log_targets, log_targets_path, delim = "\t")</span></span>
<span id="cb60-50"><a href="#cb60-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-51"><a href="#cb60-51" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb60-52"><a href="#cb60-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-53"><a href="#cb60-53" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-54"><a href="#cb60-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-55"><a href="#cb60-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-56"><a href="#cb60-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-57"><a href="#cb60-57" aria-hidden="true" tabindex="-1"></a><span class="fu">## Evaluate model performance on log-transformed data</span></span>
<span id="cb60-58"><a href="#cb60-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-61"><a href="#cb60-61" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-62"><a href="#cb60-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httpgd)</span>
<span id="cb60-63"><a href="#cb60-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb60-64"><a href="#cb60-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb60-65"><a href="#cb60-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb60-66"><a href="#cb60-66" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(neuralnet)</span>
<span id="cb60-67"><a href="#cb60-67" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb60-68"><a href="#cb60-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb60-69"><a href="#cb60-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb60-70"><a href="#cb60-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb60-71"><a href="#cb60-71" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb60-72"><a href="#cb60-72" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-73"><a href="#cb60-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-76"><a href="#cb60-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-77"><a href="#cb60-77" aria-hidden="true" tabindex="-1"></a>convert_region_to_gene <span class="ot">&lt;-</span> <span class="cf">function</span>(region, gene_metadata){</span>
<span id="cb60-78"><a href="#cb60-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-79"><a href="#cb60-79" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb60-80"><a href="#cb60-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-81"><a href="#cb60-81" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb60-82"><a href="#cb60-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-83"><a href="#cb60-83" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-84"><a href="#cb60-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-85"><a href="#cb60-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-86"><a href="#cb60-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-89"><a href="#cb60-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-90"><a href="#cb60-90" aria-hidden="true" tabindex="-1"></a>split_datasets_chroms <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb60-91"><a href="#cb60-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-92"><a href="#cb60-92" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb60-93"><a href="#cb60-93" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb60-94"><a href="#cb60-94" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb60-95"><a href="#cb60-95" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb60-96"><a href="#cb60-96" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb60-97"><a href="#cb60-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-98"><a href="#cb60-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-99"><a href="#cb60-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(chroms)</span>
<span id="cb60-100"><a href="#cb60-100" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-101"><a href="#cb60-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-102"><a href="#cb60-102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-103"><a href="#cb60-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-104"><a href="#cb60-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-105"><a href="#cb60-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-108"><a href="#cb60-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-109"><a href="#cb60-109" aria-hidden="true" tabindex="-1"></a>plot_boxplot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, path) {</span>
<span id="cb60-110"><a href="#cb60-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-111"><a href="#cb60-111" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>is_CAGE <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"CAGE_epi"</span>, <span class="st">"CAGE"</span>, <span class="st">"RNAseq"</span>)</span>
<span id="cb60-112"><a href="#cb60-112" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(path)</span>
<span id="cb60-113"><a href="#cb60-113" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>variable, <span class="at">y=</span>value)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() </span>
<span id="cb60-114"><a href="#cb60-114" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_CAGE, <span class="at">scales=</span><span class="st">"free"</span>))</span>
<span id="cb60-115"><a href="#cb60-115" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb60-116"><a href="#cb60-116" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-117"><a href="#cb60-117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-118"><a href="#cb60-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-121"><a href="#cb60-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-122"><a href="#cb60-122" aria-hidden="true" tabindex="-1"></a>run_cor <span class="ot">&lt;-</span> <span class="cf">function</span>(x,y, <span class="at">method=</span><span class="st">"pearson"</span>, <span class="at">log_bool=</span><span class="cn">FALSE</span>) {</span>
<span id="cb60-123"><a href="#cb60-123" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (log_bool) {</span>
<span id="cb60-124"><a href="#cb60-124" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">cor</span>(<span class="fu">log</span>(x<span class="sc">+</span><span class="dv">1</span>),y, <span class="at">method=</span>method))</span>
<span id="cb60-125"><a href="#cb60-125" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb60-126"><a href="#cb60-126" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">cor</span>(x,y, <span class="at">method=</span>method))</span>
<span id="cb60-127"><a href="#cb60-127" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb60-128"><a href="#cb60-128" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-129"><a href="#cb60-129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-130"><a href="#cb60-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-131"><a href="#cb60-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-132"><a href="#cb60-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-135"><a href="#cb60-135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-136"><a href="#cb60-136" aria-hidden="true" tabindex="-1"></a>per_chrom_metrics <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path, <span class="at">cor_method=</span><span class="st">"pearson"</span>, <span class="at">primary_ind=</span><span class="dv">1</span>) {</span>
<span id="cb60-137"><a href="#cb60-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-138"><a href="#cb60-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for data subset: {data_subset}"</span>))</span>
<span id="cb60-139"><a href="#cb60-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-140"><a href="#cb60-140" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb60-141"><a href="#cb60-141" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb60-142"><a href="#cb60-142" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb60-143"><a href="#cb60-143" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb60-144"><a href="#cb60-144" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb60-145"><a href="#cb60-145" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb60-146"><a href="#cb60-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-147"><a href="#cb60-147" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb60-148"><a href="#cb60-148" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb60-149"><a href="#cb60-149" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,primary_ind]</span>
<span id="cb60-150"><a href="#cb60-150" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb60-151"><a href="#cb60-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-152"><a href="#cb60-152" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb60-153"><a href="#cb60-153" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb60-154"><a href="#cb60-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-155"><a href="#cb60-155" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>)</span>
<span id="cb60-156"><a href="#cb60-156" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb60-157"><a href="#cb60-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-158"><a href="#cb60-158" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb60-159"><a href="#cb60-159" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-160"><a href="#cb60-160" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions, y[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb60-161"><a href="#cb60-161" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions, y[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-162"><a href="#cb60-162" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb60-163"><a href="#cb60-163" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb60-164"><a href="#cb60-164" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb60-165"><a href="#cb60-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-166"><a href="#cb60-166" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage nn corr</span></span>
<span id="cb60-167"><a href="#cb60-167" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-168"><a href="#cb60-168" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn, y[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb60-169"><a href="#cb60-169" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn, y[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-170"><a href="#cb60-170" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb60-171"><a href="#cb60-171" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb60-172"><a href="#cb60-172" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb60-173"><a href="#cb60-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-174"><a href="#cb60-174" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb60-175"><a href="#cb60-175" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-176"><a href="#cb60-176" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>], <span class="at">method=</span>cor_method))</span>
<span id="cb60-177"><a href="#cb60-177" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,<span class="dv">1</span>], y[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-178"><a href="#cb60-178" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb60-179"><a href="#cb60-179" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb60-180"><a href="#cb60-180" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb60-181"><a href="#cb60-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-182"><a href="#cb60-182" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb60-183"><a href="#cb60-183" aria-hidden="true" tabindex="-1"></a>  cur_cor <span class="ot">&lt;-</span> <span class="fu">run_cor</span>(y[,primary_ind], X[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span>cor_method)</span>
<span id="cb60-184"><a href="#cb60-184" aria-hidden="true" tabindex="-1"></a>  data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-185"><a href="#cb60-185" aria-hidden="true" tabindex="-1"></a>  corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, cur_cor)</span>
<span id="cb60-186"><a href="#cb60-186" aria-hidden="true" tabindex="-1"></a>  corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y[,primary_ind], X[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-187"><a href="#cb60-187" aria-hidden="true" tabindex="-1"></a>  chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, <span class="st">"all"</span>)</span>
<span id="cb60-188"><a href="#cb60-188" aria-hidden="true" tabindex="-1"></a>  num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X)[<span class="dv">1</span>])</span>
<span id="cb60-189"><a href="#cb60-189" aria-hidden="true" tabindex="-1"></a>  cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb60-190"><a href="#cb60-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-191"><a href="#cb60-191" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Done recording</span></span>
<span id="cb60-192"><a href="#cb60-192" aria-hidden="true" tabindex="-1"></a>  chroms_all <span class="ot">&lt;-</span> <span class="fu">split_datasets_chroms</span>(data_subset, mapping_table)</span>
<span id="cb60-193"><a href="#cb60-193" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(chroms_all)</span>
<span id="cb60-194"><a href="#cb60-194" aria-hidden="true" tabindex="-1"></a>  uniq_chroms <span class="ot">&lt;-</span> <span class="fu">unique</span>(chroms)</span>
<span id="cb60-195"><a href="#cb60-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-196"><a href="#cb60-196" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (chrom <span class="cf">in</span> uniq_chroms) {</span>
<span id="cb60-197"><a href="#cb60-197" aria-hidden="true" tabindex="-1"></a>    index_of_chrom <span class="ot">&lt;-</span> chroms <span class="sc">==</span> chrom</span>
<span id="cb60-198"><a href="#cb60-198" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb60-199"><a href="#cb60-199" aria-hidden="true" tabindex="-1"></a>    X_chrom <span class="ot">&lt;-</span> X[index_of_chrom,]</span>
<span id="cb60-200"><a href="#cb60-200" aria-hidden="true" tabindex="-1"></a>    y_chrom <span class="ot">&lt;-</span> y[index_of_chrom,]</span>
<span id="cb60-201"><a href="#cb60-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-202"><a href="#cb60-202" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Analyzing data for chrom: {chrom}"</span>))</span>
<span id="cb60-203"><a href="#cb60-203" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(X_chrom))</span>
<span id="cb60-204"><a href="#cb60-204" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(y_chrom))</span>
<span id="cb60-205"><a href="#cb60-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-206"><a href="#cb60-206" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,primary_ind], y_chrom[,<span class="dv">7</span>]))</span>
<span id="cb60-207"><a href="#cb60-207" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,primary_ind], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb60-208"><a href="#cb60-208" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(y_chrom[,<span class="dv">7</span>], X_chrom[,<span class="dv">4766</span>]))</span>
<span id="cb60-209"><a href="#cb60-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-210"><a href="#cb60-210" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,primary_ind]))</span>
<span id="cb60-211"><a href="#cb60-211" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">cor</span>(predictions_nn[index_of_chrom], y_chrom[,primary_ind]))</span>
<span id="cb60-212"><a href="#cb60-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-213"><a href="#cb60-213" aria-hidden="true" tabindex="-1"></a>    chrom_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">glmnet_pred=</span>glmnet_predictions[index_of_chrom], <span class="at">flu_rnaseq =</span> y_chrom[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y_chrom[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X_chrom[,<span class="dv">4766</span>])</span>
<span id="cb60-214"><a href="#cb60-214" aria-hidden="true" tabindex="-1"></a>    chrom_df_for_plotting <span class="ot">&lt;-</span> <span class="fu">melt</span>(chrom_df, <span class="at">measure_vars =</span> <span class="fu">colnames</span>(chrom_df))</span>
<span id="cb60-215"><a href="#cb60-215" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_boxplot</span>(chrom_df_for_plotting, <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/{data_subset}_chrom_{chrom}_boxplot.png"</span>))</span>
<span id="cb60-216"><a href="#cb60-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-217"><a href="#cb60-217" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage glmnet corr</span></span>
<span id="cb60-218"><a href="#cb60-218" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-219"><a href="#cb60-219" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb60-220"><a href="#cb60-220" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(glmnet_predictions[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-221"><a href="#cb60-221" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb60-222"><a href="#cb60-222" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb60-223"><a href="#cb60-223" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"glmnet"</span>)</span>
<span id="cb60-224"><a href="#cb60-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-225"><a href="#cb60-225" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage nn corr</span></span>
<span id="cb60-226"><a href="#cb60-226" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-227"><a href="#cb60-227" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span>cor_method))</span>
<span id="cb60-228"><a href="#cb60-228" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(predictions_nn[index_of_chrom], y_chrom[,primary_ind], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-229"><a href="#cb60-229" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb60-230"><a href="#cb60-230" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb60-231"><a href="#cb60-231" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"nn"</span>)</span>
<span id="cb60-232"><a href="#cb60-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-233"><a href="#cb60-233" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and NI</span></span>
<span id="cb60-234"><a href="#cb60-234" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-235"><a href="#cb60-235" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>], <span class="at">method=</span>cor_method))</span>
<span id="cb60-236"><a href="#cb60-236" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,<span class="dv">1</span>], y_chrom[,<span class="dv">7</span>], <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-237"><a href="#cb60-237" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb60-238"><a href="#cb60-238" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb60-239"><a href="#cb60-239" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_NI"</span>)</span>
<span id="cb60-240"><a href="#cb60-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-241"><a href="#cb60-241" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Record in storage corr between flu and epi</span></span>
<span id="cb60-242"><a href="#cb60-242" aria-hidden="true" tabindex="-1"></a>    data_subset_v <span class="ot">&lt;-</span> <span class="fu">c</span>(data_subset_v, data_subset)</span>
<span id="cb60-243"><a href="#cb60-243" aria-hidden="true" tabindex="-1"></a>    corrs_v <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v, <span class="fu">run_cor</span>(y_chrom[,primary_ind], X_chrom[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span>cor_method))</span>
<span id="cb60-244"><a href="#cb60-244" aria-hidden="true" tabindex="-1"></a>    corrs_v_spear <span class="ot">&lt;-</span> <span class="fu">c</span>(corrs_v_spear, <span class="fu">run_cor</span>(y_chrom[,primary_ind], X_chrom[,<span class="dv">4766</span>], <span class="at">log=</span><span class="cn">TRUE</span>, <span class="at">method=</span><span class="st">"spearman"</span>))</span>
<span id="cb60-245"><a href="#cb60-245" aria-hidden="true" tabindex="-1"></a>    chrom_v <span class="ot">&lt;-</span> <span class="fu">c</span>(chrom_v, chrom)</span>
<span id="cb60-246"><a href="#cb60-246" aria-hidden="true" tabindex="-1"></a>    num_genes_v <span class="ot">&lt;-</span> <span class="fu">c</span>(num_genes_v, <span class="fu">dim</span>(X_chrom)[<span class="dv">1</span>])</span>
<span id="cb60-247"><a href="#cb60-247" aria-hidden="true" tabindex="-1"></a>    cor_id_v <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_id_v, <span class="st">"flu_vs_epi"</span>)</span>
<span id="cb60-248"><a href="#cb60-248" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb60-249"><a href="#cb60-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-250"><a href="#cb60-250" aria-hidden="true" tabindex="-1"></a>  storage_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(corrs_v, corrs_v_spear,data_subset_v, chrom_v, num_genes_v, cor_id_v)</span>
<span id="cb60-251"><a href="#cb60-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-252"><a href="#cb60-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(storage_df)</span>
<span id="cb60-253"><a href="#cb60-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-254"><a href="#cb60-254" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-255"><a href="#cb60-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-256"><a href="#cb60-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-259"><a href="#cb60-259" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-260"><a href="#cb60-260" aria-hidden="true" tabindex="-1"></a>split_datasets <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table) {</span>
<span id="cb60-261"><a href="#cb60-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-262"><a href="#cb60-262" aria-hidden="true" tabindex="-1"></a>  tss_metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-09-20-diagnose_training_tracks_issue/canonical_TSS_full_metadata.txt"</span>)</span>
<span id="cb60-263"><a href="#cb60-263" aria-hidden="true" tabindex="-1"></a>  tss_metadata<span class="sc">$</span>joined_region <span class="ot">&lt;-</span> <span class="fu">paste</span>(tss_metadata<span class="sc">$</span>chromosome_name, tss_metadata<span class="sc">$</span>transcription_start_site, tss_metadata<span class="sc">$</span>transcription_start_site<span class="sc">+</span><span class="dv">2</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb60-264"><a href="#cb60-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-265"><a href="#cb60-265" aria-hidden="true" tabindex="-1"></a>  data_subset_mapping_table <span class="ot">&lt;-</span> mapping_table[,mapping_table[<span class="st">"group"</span>,]<span class="sc">==</span>data_subset]</span>
<span id="cb60-266"><a href="#cb60-266" aria-hidden="true" tabindex="-1"></a>  regions <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_subset_mapping_table)</span>
<span id="cb60-267"><a href="#cb60-267" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(regions, <span class="at">split =</span> <span class="st">"_"</span>)</span>
<span id="cb60-268"><a href="#cb60-268" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb60-269"><a href="#cb60-269" aria-hidden="true" tabindex="-1"></a>  chroms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chroms, <span class="cf">function</span>(x) <span class="fu">substr</span>(x,<span class="dv">2</span>,<span class="fu">str_length</span>(x)))</span>
<span id="cb60-270"><a href="#cb60-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-271"><a href="#cb60-271" aria-hidden="true" tabindex="-1"></a>  ensembl_ids <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb60-272"><a href="#cb60-272" aria-hidden="true" tabindex="-1"></a>  external_gene_names <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(regions))</span>
<span id="cb60-273"><a href="#cb60-273" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(regions)) {</span>
<span id="cb60-274"><a href="#cb60-274" aria-hidden="true" tabindex="-1"></a>    region <span class="ot">&lt;-</span> <span class="fu">substr</span>(regions[x],<span class="dv">2</span>,<span class="fu">str_length</span>(regions[x]))</span>
<span id="cb60-275"><a href="#cb60-275" aria-hidden="true" tabindex="-1"></a>    ensembl_id <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb60-276"><a href="#cb60-276" aria-hidden="true" tabindex="-1"></a>    external_gene_name <span class="ot">&lt;-</span> tss_metadata[tss_metadata<span class="sc">$</span>joined_region <span class="sc">==</span> region,]<span class="sc">$</span>external_gene_name</span>
<span id="cb60-277"><a href="#cb60-277" aria-hidden="true" tabindex="-1"></a>    ensembl_ids[x]  <span class="ot">&lt;-</span> ensembl_id</span>
<span id="cb60-278"><a href="#cb60-278" aria-hidden="true" tabindex="-1"></a>    external_gene_names[x] <span class="ot">&lt;-</span> external_gene_name</span>
<span id="cb60-279"><a href="#cb60-279" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb60-280"><a href="#cb60-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(ensembl_ids))</span>
<span id="cb60-281"><a href="#cb60-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(external_gene_names))</span>
<span id="cb60-282"><a href="#cb60-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(chroms))</span>
<span id="cb60-283"><a href="#cb60-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(regions))</span>
<span id="cb60-284"><a href="#cb60-284" aria-hidden="true" tabindex="-1"></a>  return_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb60-285"><a href="#cb60-285" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"chroms"</span>]] <span class="ot">&lt;-</span> chroms</span>
<span id="cb60-286"><a href="#cb60-286" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"regions"</span>]] <span class="ot">&lt;-</span> regions</span>
<span id="cb60-287"><a href="#cb60-287" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"ensembl_ids"</span>]] <span class="ot">&lt;-</span> ensembl_ids</span>
<span id="cb60-288"><a href="#cb60-288" aria-hidden="true" tabindex="-1"></a>  return_list[[<span class="st">"external_gene_names"</span>]] <span class="ot">&lt;-</span> external_gene_names</span>
<span id="cb60-289"><a href="#cb60-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-290"><a href="#cb60-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_list)</span>
<span id="cb60-291"><a href="#cb60-291" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-292"><a href="#cb60-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-293"><a href="#cb60-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-296"><a href="#cb60-296" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-297"><a href="#cb60-297" aria-hidden="true" tabindex="-1"></a>collect_all_records <span class="ot">&lt;-</span> <span class="cf">function</span>(data_subset, mapping_table, X_path, y_path, glmnet_model_path, nnmodel_path) {</span>
<span id="cb60-298"><a href="#cb60-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">glue</span>(<span class="st">"********* Collecting data records for data subset: {data_subset}"</span>))</span>
<span id="cb60-299"><a href="#cb60-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-300"><a href="#cb60-300" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(X_path)))</span>
<span id="cb60-301"><a href="#cb60-301" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(y_path)))</span>
<span id="cb60-302"><a href="#cb60-302" aria-hidden="true" tabindex="-1"></a>  y_for_cv <span class="ot">&lt;-</span> y[,<span class="dv">1</span>]</span>
<span id="cb60-303"><a href="#cb60-303" aria-hidden="true" tabindex="-1"></a>  data_frame_for_training <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(X, y_for_cv))</span>
<span id="cb60-304"><a href="#cb60-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-305"><a href="#cb60-305" aria-hidden="true" tabindex="-1"></a>  glmnet_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(glmnet_model_path)</span>
<span id="cb60-306"><a href="#cb60-306" aria-hidden="true" tabindex="-1"></a>  nnmodel <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(nnmodel_path)</span>
<span id="cb60-307"><a href="#cb60-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-308"><a href="#cb60-308" aria-hidden="true" tabindex="-1"></a>  glmnet_predictions <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">predict</span>(glmnet_model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>, <span class="at">type=</span><span class="st">"response"</span>))</span>
<span id="cb60-309"><a href="#cb60-309" aria-hidden="true" tabindex="-1"></a>  predictions_nn <span class="ot">&lt;-</span> <span class="fu">predict</span>(nnmodel, data_frame_for_training)</span>
<span id="cb60-310"><a href="#cb60-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-311"><a href="#cb60-311" aria-hidden="true" tabindex="-1"></a>  gene_metadata <span class="ot">&lt;-</span> <span class="fu">split_datasets</span>(data_subset, mapping_table)</span>
<span id="cb60-312"><a href="#cb60-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-313"><a href="#cb60-313" aria-hidden="true" tabindex="-1"></a>  return_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">data_subset=</span> data_subset, <span class="at">glmnet_pred=</span>glmnet_predictions, <span class="at">nn_pred=</span> predictions_nn, <span class="at">flu_rnaseq =</span> y[,<span class="dv">1</span>], <span class="at">NI_rnaseq =</span> y[,<span class="dv">7</span>], <span class="at">CAGE_epi =</span>X[,<span class="dv">4766</span>], <span class="at">ensembl_id =</span> gene_metadata[[<span class="st">"ensembl_ids"</span>]], <span class="at">external_gene_name =</span> gene_metadata[[<span class="st">"external_gene_names"</span>]], <span class="at">region =</span> gene_metadata[[<span class="st">"regions"</span>]], <span class="at">chroms =</span> gene_metadata[[<span class="st">"chroms"</span>]])</span>
<span id="cb60-314"><a href="#cb60-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-315"><a href="#cb60-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(return_df)</span>
<span id="cb60-316"><a href="#cb60-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-317"><a href="#cb60-317" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-318"><a href="#cb60-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-319"><a href="#cb60-319" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-320"><a href="#cb60-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-323"><a href="#cb60-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-324"><a href="#cb60-324" aria-hidden="true" tabindex="-1"></a>remapped_table_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/remapped_table_filt.csv"</span></span>
<span id="cb60-325"><a href="#cb60-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-326"><a href="#cb60-326" aria-hidden="true" tabindex="-1"></a>remapped_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(remapped_table_path, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb60-327"><a href="#cb60-327" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(remapped_table))</span>
<span id="cb60-328"><a href="#cb60-328" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">as.character</span>(remapped_table[,<span class="dv">1</span>]))</span>
<span id="cb60-329"><a href="#cb60-329" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(remapped_table) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(remapped_table[,<span class="dv">1</span>])</span>
<span id="cb60-330"><a href="#cb60-330" aria-hidden="true" tabindex="-1"></a>remapped_table[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb60-331"><a href="#cb60-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-332"><a href="#cb60-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-335"><a href="#cb60-335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-336"><a href="#cb60-336" aria-hidden="true" tabindex="-1"></a>primary_ind <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb60-337"><a href="#cb60-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-338"><a href="#cb60-338" aria-hidden="true" tabindex="-1"></a>glmnet_model_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_eln_RNAseq_NI_from_HDF5_mean_log.linear.rds"</span></span>
<span id="cb60-339"><a href="#cb60-339" aria-hidden="true" tabindex="-1"></a>nnmodel_path <span class="ot">&lt;-</span> <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/elastic_net/trained_mlp_RNAseq_NI_from_HDF5_mean_log.rds"</span></span>
<span id="cb60-340"><a href="#cb60-340" aria-hidden="true" tabindex="-1"></a>cor_method <span class="ot">=</span> <span class="st">"pearson"</span></span>
<span id="cb60-341"><a href="#cb60-341" aria-hidden="true" tabindex="-1"></a>df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb60-342"><a href="#cb60-342" aria-hidden="true" tabindex="-1"></a>all_records_df_store <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb60-343"><a href="#cb60-343" aria-hidden="true" tabindex="-1"></a>c<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb60-344"><a href="#cb60-344" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (data_subset <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"train"</span>,<span class="st">"test"</span>,<span class="st">"valid"</span>)){</span>
<span id="cb60-345"><a href="#cb60-345" aria-hidden="true" tabindex="-1"></a>  X_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_inputs_mean_aracena.csv"</span>)</span>
<span id="cb60-346"><a href="#cb60-346" aria-hidden="true" tabindex="-1"></a>  y_path <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/hdf5_training_tss_centered/{data_subset}_targets_mean_aracena_log.csv"</span>)</span>
<span id="cb60-347"><a href="#cb60-347" aria-hidden="true" tabindex="-1"></a>  cur_out_df_perchrom_metrics <span class="ot">&lt;-</span> <span class="fu">per_chrom_metrics</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path, cor_method, primary_ind)</span>
<span id="cb60-348"><a href="#cb60-348" aria-hidden="true" tabindex="-1"></a>  df_store[[c]] <span class="ot">&lt;-</span> cur_out_df_perchrom_metrics</span>
<span id="cb60-349"><a href="#cb60-349" aria-hidden="true" tabindex="-1"></a>  all_records_df_store[[c]] <span class="ot">&lt;-</span> <span class="fu">collect_all_records</span>(data_subset, remapped_table, X_path, y_path, glmnet_model_path, nnmodel_path)</span>
<span id="cb60-350"><a href="#cb60-350" aria-hidden="true" tabindex="-1"></a>  c<span class="ot">=</span>c<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb60-351"><a href="#cb60-351" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-352"><a href="#cb60-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-353"><a href="#cb60-353" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df_store)</span>
<span id="cb60-354"><a href="#cb60-354" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>data_subset_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>data_subset_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb60-355"><a href="#cb60-355" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>cor_id_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>cor_id_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>))</span>
<span id="cb60-356"><a href="#cb60-356" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>chrom_v <span class="ot">&lt;-</span> <span class="fu">factor</span>(full_df<span class="sc">$</span>chrom_v, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb60-357"><a href="#cb60-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-358"><a href="#cb60-358" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-359"><a href="#cb60-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-360"><a href="#cb60-360" aria-hidden="true" tabindex="-1"></a>Set color palette</span>
<span id="cb60-363"><a href="#cb60-363" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-364"><a href="#cb60-364" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb60-365"><a href="#cb60-365" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span>
<span id="cb60-366"><a href="#cb60-366" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-367"><a href="#cb60-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-370"><a href="#cb60-370" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-371"><a href="#cb60-371" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>num_genes_v <span class="ot">&lt;-</span> <span class="fu">as.character</span>(full_df<span class="sc">$</span>num_genes_v)</span>
<span id="cb60-372"><a href="#cb60-372" aria-hidden="true" tabindex="-1"></a>full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(full_df, <span class="at">measure_vars=</span><span class="fu">c</span>(<span class="st">"corrs_v"</span>,<span class="st">"corrs_v_spear"</span>), <span class="at">id_vars=</span><span class="fu">c</span>(<span class="st">"data_subset_v"</span>, <span class="st">"chrom_v"</span>, <span class="st">"cor_id_v"</span>, <span class="st">"num_genes_v"</span>))</span>
<span id="cb60-373"><a href="#cb60-373" aria-hidden="true" tabindex="-1"></a>full_df_melt<span class="sc">$</span>corr_type <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(full_df_melt<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"corrs_v"</span>, <span class="st">"pearson"</span>, <span class="st">"spearman"</span>)</span>
<span id="cb60-374"><a href="#cb60-374" aria-hidden="true" tabindex="-1"></a>full_df_melt_all_chroms <span class="ot">&lt;-</span> full_df_melt[full_df_melt<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb60-375"><a href="#cb60-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-376"><a href="#cb60-376" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_all_chroms_both.png"</span>), <span class="at">width=</span><span class="dv">800</span>, <span class="at">height=</span><span class="dv">400</span>)</span>
<span id="cb60-377"><a href="#cb60-377" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_melt_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>value, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>corr_type, <span class="at">nrow=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Data subset"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation"</span>)</span>
<span id="cb60-378"><a href="#cb60-378" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-379"><a href="#cb60-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-380"><a href="#cb60-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-381"><a href="#cb60-381" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-384"><a href="#cb60-384" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-385"><a href="#cb60-385" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb60-386"><a href="#cb60-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-387"><a href="#cb60-387" aria-hidden="true" tabindex="-1"></a>full_df_all_chroms <span class="ot">&lt;-</span> full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">==</span> <span class="st">"all"</span>,]</span>
<span id="cb60-388"><a href="#cb60-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-389"><a href="#cb60-389" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_all_chroms.png"</span>))</span>
<span id="cb60-390"><a href="#cb60-390" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df_all_chroms) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb60-391"><a href="#cb60-391" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-392"><a href="#cb60-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-393"><a href="#cb60-393" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb60-394"><a href="#cb60-394" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb60-395"><a href="#cb60-395" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb60-396"><a href="#cb60-396" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-397"><a href="#cb60-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-398"><a href="#cb60-398" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_nonfaceted.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb60-399"><a href="#cb60-399" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v, <span class="at">fill=</span>cor_id_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> corr_colors)</span>
<span id="cb60-400"><a href="#cb60-400" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-401"><a href="#cb60-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-402"><a href="#cb60-402" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb60-403"><a href="#cb60-403" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>)</span>
<span id="cb60-404"><a href="#cb60-404" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-405"><a href="#cb60-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-406"><a href="#cb60-406" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_numgenes.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb60-407"><a href="#cb60-407" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>data_subset_v, <span class="at">fill=</span>chrom_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>)</span>
<span id="cb60-408"><a href="#cb60-408" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-409"><a href="#cb60-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-410"><a href="#cb60-410" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_numgenes_long.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb60-411"><a href="#cb60-411" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>chrom_v <span class="sc">!=</span> <span class="st">"all"</span> <span class="sc">&amp;</span> full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> <span class="st">"glmnet"</span>,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">fill=</span>data_subset_v, <span class="at">y=</span>num_genes_v), <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> subset_colors)</span>
<span id="cb60-412"><a href="#cb60-412" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-413"><a href="#cb60-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-414"><a href="#cb60-414" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb60-415"><a href="#cb60-415" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb60-416"><a href="#cb60-416" aria-hidden="true" tabindex="-1"></a>  g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(full_df[full_df<span class="sc">$</span>cor_id_v <span class="sc">==</span> id,]) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>chrom_v, <span class="at">y=</span>corrs_v), <span class="at">stat=</span><span class="st">"identity"</span>, <span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb60-417"><a href="#cb60-417" aria-hidden="true" tabindex="-1"></a>  plot_list[[id]] <span class="ot">&lt;-</span> g<span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset_v, <span class="at">ncol=</span><span class="dv">1</span>)</span>
<span id="cb60-418"><a href="#cb60-418" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-419"><a href="#cb60-419" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (id <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glmnet"</span>, <span class="st">"nn"</span>, <span class="st">"flu_vs_epi"</span>, <span class="st">"flu_vs_NI"</span>)) {</span>
<span id="cb60-420"><a href="#cb60-420" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/corrs_each_chrom_{id}.png"</span>), <span class="at">width =</span><span class="dv">1000</span>, <span class="at">height=</span><span class="dv">500</span>)</span>
<span id="cb60-421"><a href="#cb60-421" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot_list[[id]])</span>
<span id="cb60-422"><a href="#cb60-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb60-423"><a href="#cb60-423" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-424"><a href="#cb60-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-425"><a href="#cb60-425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-426"><a href="#cb60-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-429"><a href="#cb60-429" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-430"><a href="#cb60-430" aria-hidden="true" tabindex="-1"></a>all_records_full_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, all_records_df_store)</span>
<span id="cb60-431"><a href="#cb60-431" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>data_subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>data_subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"train"</span>, <span class="st">"valid"</span>, <span class="st">"test"</span>))</span>
<span id="cb60-432"><a href="#cb60-432" aria-hidden="true" tabindex="-1"></a>all_records_full_df<span class="sc">$</span>chroms<span class="ot">&lt;-</span> <span class="fu">factor</span>(all_records_full_df<span class="sc">$</span>chroms, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"all"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span>, <span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span>, <span class="st">"10"</span>, <span class="st">"11"</span>, <span class="st">"12"</span>, <span class="st">"13"</span>, <span class="st">"14"</span>, <span class="st">"15"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"18"</span>, <span class="st">"19"</span>, <span class="st">"20"</span>, <span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>))</span>
<span id="cb60-433"><a href="#cb60-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-434"><a href="#cb60-434" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df)</span>
<span id="cb60-435"><a href="#cb60-435" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-436"><a href="#cb60-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-439"><a href="#cb60-439" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-440"><a href="#cb60-440" aria-hidden="true" tabindex="-1"></a>all_records_full_df_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(all_records_full_df, <span class="at">measure_vars =</span> <span class="fu">c</span>(<span class="st">"glmnet_pred"</span>, <span class="st">"nn_pred"</span>, <span class="st">"flu_rnaseq"</span>, <span class="st">"NI_rnaseq"</span>, <span class="st">"CAGE_epi"</span>))</span>
<span id="cb60-441"><a href="#cb60-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-442"><a href="#cb60-442" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(all_records_full_df_melt)</span>
<span id="cb60-443"><a href="#cb60-443" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb60-444"><a href="#cb60-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-447"><a href="#cb60-447" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb60-448"><a href="#cb60-448" aria-hidden="true" tabindex="-1"></a>corr_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"glmnet"</span> <span class="ot">=</span> <span class="st">"#F8766D"</span>, <span class="st">"nn"</span> <span class="ot">=</span> <span class="st">"#00BA38"</span>, <span class="st">"flu_vs_epi"</span> <span class="ot">=</span> <span class="st">"#619CFF"</span>, <span class="st">"flu_vs_NI"</span> <span class="ot">=</span> <span class="st">"#F564E3"</span>)</span>
<span id="cb60-449"><a href="#cb60-449" aria-hidden="true" tabindex="-1"></a>subset_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"train"</span> <span class="ot">=</span> <span class="st">"orange"</span>, <span class="st">"valid"</span> <span class="ot">=</span> <span class="st">"yellow"</span>, <span class="st">"test"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span>
<span id="cb60-450"><a href="#cb60-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-451"><a href="#cb60-451" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-10-25-log_transform_training_tss_predictors/figures/all_records_scatterplots.png"</span>), <span class="at">width =</span><span class="dv">900</span>, <span class="at">height=</span><span class="dv">1200</span>)</span>
<span id="cb60-452"><a href="#cb60-452" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>glmnet_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Elastic net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"glmnet"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb60-453"><a href="#cb60-453" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>nn_pred, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Neural net pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"nn"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb60-454"><a href="#cb60-454" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>CAGE_epi, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"Enformer CAGE pred"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_epi"</span>],<span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors)) <span class="sc">/</span></span>
<span id="cb60-455"><a href="#cb60-455" aria-hidden="true" tabindex="-1"></a>(<span class="fu">ggplot</span>(all_records_full_df) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>NI_rnaseq, <span class="at">y=</span>flu_rnaseq, <span class="at">color=</span>data_subset)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>data_subset, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">"flu RNAseq"</span>)<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">"NI RNAseq"</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">alpha</span>(corr_colors[<span class="st">"flu_vs_NI"</span>],<span class="at">alpha=</span><span class="fl">0.4</span>)))<span class="sc">+</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>subset_colors))</span>
<span id="cb60-456"><a href="#cb60-456" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb60-457"><a href="#cb60-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-458"><a href="#cb60-458" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>