<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-05-30">

<title>Daily-Blog-Sai - testing_ancestry_differences</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#computing-metrics-of-local-ancestry" id="toc-computing-metrics-of-local-ancestry" class="nav-link" data-scroll-target="#computing-metrics-of-local-ancestry">Computing metrics of local ancestry</a></li>
  <li><a href="#calculate-correlation-across-individuals" id="toc-calculate-correlation-across-individuals" class="nav-link" data-scroll-target="#calculate-correlation-across-individuals">Calculate correlation across individuals</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">testing_ancestry_differences</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 30, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"testing_ancestry_differences"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-05-30'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="context" class="level1">
<h1>Context</h1>
<p>Previously I have collected local ancestry information for different genes (03-27). The hypothesis here is that the mismatches observed between local ancestry in the training reference genome, local ancestry in the Basenji targets (for example the lymphoblastoid cell line gm12878), and local ancestry in the inference sequence all contribute to issues observed with personalized prediction.</p>
<p>I didn’t fully test this hypothesis in enformer portability analysis notebook. Here I am going to more formally test it by collecting the following per-gene metrics:</p>
<p>1.) Local reference genome ancestry 2.) Average alternate (non-reference) allelic density per SNP in gene window from GM12878 lymphoblastoid cell line 3.) Direction of expression prediction correlation across individuals 4.) Potentially whether there is differential expression prediction between ancestry for the gene</p>
<p>We can update our gene metadata table with this added information once we have it.</p>
<section id="computing-metrics-of-local-ancestry" class="level3">
<h3 class="anchored" data-anchor-id="computing-metrics-of-local-ancestry">Computing metrics of local ancestry</h3>
<p>Let’s load our gene metadata table:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/ref_local_ancestry/gene_annotations.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calculate-correlation-across-individuals" class="level3">
<h3 class="anchored" data-anchor-id="calculate-correlation-across-individuals">Calculate correlation across individuals</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_summed_haplo1_filtered.csv"</span>)))</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(enformer_haplo1) &lt;- enformer_haplo1$X</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_haplo1$X &lt;- NULL</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>enformer_haplo2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_summed_haplo2_filtered.csv"</span>)))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo2) <span class="ot">&lt;-</span> enformer_haplo2<span class="sc">$</span>X</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>enformer_haplo2<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> (enformer_haplo1 <span class="sc">+</span> enformer_haplo2)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>AFR_ind <span class="ot">&lt;-</span> individual_metadata[individual_metadata<span class="sc">$</span>Superpopulation.code <span class="sc">==</span> <span class="st">"AFR"</span>,]<span class="sc">$</span>Sample.name</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>EUR_ind <span class="ot">&lt;-</span> individual_metadata[individual_metadata<span class="sc">$</span>Superpopulation.code <span class="sc">==</span> <span class="st">"EUR"</span>,]<span class="sc">$</span>Sample.name</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>AFR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,AFR_ind)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>EUR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,EUR_ind)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, ge_ind]</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>geuvadis_AFR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, AFR_ge_ind]</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>geuvadis_EUR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, EUR_ge_ind]</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[ge_genes, ge_ind]</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>enformer_AFR_compare <span class="ot">&lt;-</span> enformer[ge_genes, AFR_ge_ind]</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>enformer_EUR_compare <span class="ot">&lt;-</span> enformer[ge_genes, EUR_ge_ind]</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[ge_genes,]</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals: "</span>, <span class="fu">length</span>(ge_ind)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Number of individuals: 446"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals AFR: "</span>, <span class="fu">length</span>(AFR_ge_ind)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Number of individuals AFR: 81"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals EUR: "</span>, <span class="fu">length</span>(EUR_ge_ind)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Number of individuals EUR: 365"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ge_genes_plus <span class="ot">&lt;-</span> ge_genes</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes +: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>strand<span class="sc">==</span><span class="st">"+"</span>,]<span class="sc">$</span>best_ancestry)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Number of genes +: 5997"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes +,AFR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"AFR"</span> <span class="sc">&amp;</span>gene_meta_compare<span class="sc">$</span>strand<span class="sc">==</span><span class="st">"+"</span>,]<span class="sc">$</span>best_ancestry)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Number of genes +,AFR: 1446"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes -,EUR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"EUR"</span> <span class="sc">&amp;</span>gene_meta_compare<span class="sc">$</span>strand<span class="sc">==</span><span class="st">"+"</span>,]<span class="sc">$</span>best_ancestry)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Number of genes -,EUR: 1529"</code></pre>
</div>
</div>
<p>Correlate across all individuals per gene between Geuvadis, Enformer</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = AFR ~ ge, family = "binomial", data = corrs_df_plus_AFR_EUR)

Deviance Residuals: 
   Min      1Q  Median      3Q     Max  
-1.306  -1.168  -1.121   1.187   1.265  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)
(Intercept) -0.02543    0.02720  -0.935    0.350
ge           0.53248    0.32628   1.632    0.103

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 7671.3  on 5533  degrees of freedom
Residual deviance: 7668.6  on 5532  degrees of freedom
AIC: 7672.6

Number of Fisher Scoring iterations: 3</code></pre>
</div>
</div>
<p>Correlate across only AFR individuals</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_AFR_compare, enformer_AFR_compare)</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-7-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family =</span> <span class="st">"binomial"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = AFR ~ ge, family = "binomial", data = corrs_df_plus_AFR_EUR)

Deviance Residuals: 
   Min      1Q  Median      3Q     Max  
-1.196  -1.169  -1.157   1.186   1.202  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)
(Intercept) -0.01934    0.02691  -0.718    0.473
ge           0.09539    0.21383   0.446    0.656

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 7671.3  on 5533  degrees of freedom
Residual deviance: 7671.1  on 5532  degrees of freedom
AIC: 7675.1

Number of Fisher Scoring iterations: 3</code></pre>
</div>
</div>
<p>Correlate across only EUR individuals</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_EUR_compare, enformer_EUR_compare)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(data.frame(table(corrs_df_plus_AFR_EUR$ref_gene_ancestry, corrs_df_plus_AFR_EUR$corr_direction)), aes(x=Var1,y=Freq,fill=Var2, group=Var1)) + geom_bar(stat="identity", position="dodge")</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = AFR ~ ge, family = "binomial", data = corrs_df_plus_AFR_EUR)

Deviance Residuals: 
   Min      1Q  Median      3Q     Max  
-1.301  -1.168  -1.122   1.186   1.276  

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)
(Intercept) -0.02588    0.02730  -0.948    0.343
ge           0.45381    0.30051   1.510    0.131

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 7671.3  on 5533  degrees of freedom
Residual deviance: 7669.0  on 5532  degrees of freedom
AIC: 7673

Number of Fisher Scoring iterations: 3</code></pre>
</div>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb29" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "testing_ancestry_differences"</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Saideep Gona"</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-05-30"</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: true</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Set up box storage directory</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>PRE <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/Daily-Blog-Sai"</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="do">## COPY THE DATE AND SLUG fields FROM THE HEADER</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>SLUG<span class="ot">=</span><span class="st">"testing_ancestry_differences"</span> <span class="do">## copy the slug from the header</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>bDATE<span class="ot">=</span><span class="st">'2023-05-30'</span> <span class="do">## copy the date from the blog's header here</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{PRE}/{bDATE}-{SLUG}"</span>)</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(DATA)) <span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"mkdir {DATA}"</span>))</span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>WORK<span class="ot">=</span>DATA</span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="fu"># Context</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>Previously I have collected local ancestry information for different genes (03-27). The hypothesis here is that the mismatches observed between local ancestry in the training reference genome, local ancestry in the Basenji targets (for example the lymphoblastoid cell line gm12878), and local ancestry in the inference sequence all contribute to issues observed with personalized prediction.</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a>I didn't fully test this hypothesis in enformer portability analysis notebook. Here I am going to more formally test it by collecting the following per-gene metrics:</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a>1.) Local reference genome ancestry 2.) Average alternate (non-reference) allelic density per SNP in gene window from GM12878 lymphoblastoid cell line 3.) Direction of expression prediction correlation across individuals 4.) Potentially whether there is differential expression prediction between ancestry for the gene</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a>We can update our gene metadata table with this added information once we have it.</span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="fu">### Computing metrics of local ancestry</span></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a>Let's load our gene metadata table:</span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata/ref_local_ancestry/gene_annotations.csv"</span>)</span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a><span class="fu">### Calculate correlation across individuals</span></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Load Datasets</span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a>dset_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/enformer_geuvadis/enformer_portability_analysis"</span>)</span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Load GEUVADIS</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a>geuvadis_full <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"GD462.GeneQuantRPKM.50FN.samplename.resk10.txt"</span>))</span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a>geuvadis_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(geuvadis_full<span class="sc">$</span>Gene_Symbol, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a>geuvadis <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(geuvadis_full[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(geuvadis_full)])</span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geuvadis) <span class="ot">&lt;-</span> geuvadis_genes_symbols</span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Load enformer</span></span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a>enformer_haplo1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_summed_haplo1_filtered.csv"</span>)))</span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo1) <span class="ot">&lt;-</span> enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a>enformer_haplo1<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(enformer_haplo1) &lt;- enformer_haplo1$X</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a><span class="co"># enformer_haplo1$X &lt;- NULL</span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a>enformer_haplo2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="fu">file.path</span>(dset_dir,<span class="st">"CAGE_lcl_enformer_geuvadis_summed_haplo2_filtered.csv"</span>)))</span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer_haplo2) <span class="ot">&lt;-</span> enformer_haplo2<span class="sc">$</span>X</span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a>enformer_haplo2<span class="sc">$</span>...<span class="dv">1</span> <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a>enf_genes_symbols <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_split</span>(<span class="fu">rownames</span>(enformer_haplo1), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>),<span class="st">`</span><span class="at">[</span><span class="st">`</span>,<span class="dv">1</span>)</span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a>enformer <span class="ot">&lt;-</span> (enformer_haplo1 <span class="sc">+</span> enformer_haplo2)</span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(enformer) <span class="ot">&lt;-</span> enf_genes_symbols</span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-91"><a href="#cb29-91" aria-hidden="true" tabindex="-1"></a>metadata_dir <span class="ot">=</span> <span class="st">"/Users/saideepgona/Library/CloudStorage/Box-Box/imlab-data/data-Github/analysis-Sai/metadata"</span></span>
<span id="cb29-92"><a href="#cb29-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a>individual_metadata <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"igsr_samples.tsv"</span>))</span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(individual_metadata) <span class="ot">&lt;-</span> individual_metadata<span class="sc">$</span>Sample.name</span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a>AFR_ind <span class="ot">&lt;-</span> individual_metadata[individual_metadata<span class="sc">$</span>Superpopulation.code <span class="sc">==</span> <span class="st">"AFR"</span>,]<span class="sc">$</span>Sample.name</span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a>EUR_ind <span class="ot">&lt;-</span> individual_metadata[individual_metadata<span class="sc">$</span>Superpopulation.code <span class="sc">==</span> <span class="st">"EUR"</span>,]<span class="sc">$</span>Sample.name</span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv2</span>(<span class="fu">file.path</span>(metadata_dir,<span class="st">"gene_metadata_full.csv"</span>),<span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-106"><a href="#cb29-106" aria-hidden="true" tabindex="-1"></a>ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(geuvadis), <span class="fu">colnames</span>(enformer))</span>
<span id="cb29-107"><a href="#cb29-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a>AFR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,AFR_ind)</span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a>EUR_ge_ind <span class="ot">&lt;-</span> <span class="fu">intersect</span>(ge_ind,EUR_ind)</span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-111"><a href="#cb29-111" aria-hidden="true" tabindex="-1"></a>ge_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(geuvadis), <span class="fu">rownames</span>(enformer))</span>
<span id="cb29-112"><a href="#cb29-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a>geuvadis_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, ge_ind]</span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a>geuvadis_AFR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, AFR_ge_ind]</span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a>geuvadis_EUR_compare <span class="ot">&lt;-</span> geuvadis[ge_genes, EUR_ge_ind]</span>
<span id="cb29-116"><a href="#cb29-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-117"><a href="#cb29-117" aria-hidden="true" tabindex="-1"></a>enformer_compare <span class="ot">&lt;-</span> enformer[ge_genes, ge_ind]</span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a>enformer_AFR_compare <span class="ot">&lt;-</span> enformer[ge_genes, AFR_ge_ind]</span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a>enformer_EUR_compare <span class="ot">&lt;-</span> enformer[ge_genes, EUR_ge_ind]</span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a>gene_meta_compare <span class="ot">&lt;-</span> gene_metadata[ge_genes,]</span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals: "</span>, <span class="fu">length</span>(ge_ind)))</span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals AFR: "</span>, <span class="fu">length</span>(AFR_ge_ind)))</span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of individuals EUR: "</span>, <span class="fu">length</span>(EUR_ge_ind)))</span>
<span id="cb29-126"><a href="#cb29-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-127"><a href="#cb29-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a>ge_genes_plus <span class="ot">&lt;-</span> ge_genes</span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes +: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>strand<span class="sc">==</span><span class="st">"+"</span>,]<span class="sc">$</span>best_ancestry)))</span>
<span id="cb29-130"><a href="#cb29-130" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes +,AFR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"AFR"</span> <span class="sc">&amp;</span>gene_meta_compare<span class="sc">$</span>strand<span class="sc">==</span><span class="st">"+"</span>,]<span class="sc">$</span>best_ancestry)))</span>
<span id="cb29-131"><a href="#cb29-131" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Number of genes -,EUR: "</span>, <span class="fu">length</span>(gene_meta_compare[gene_meta_compare<span class="sc">$</span>best_ancestry<span class="sc">==</span><span class="st">"EUR"</span> <span class="sc">&amp;</span>gene_meta_compare<span class="sc">$</span>strand<span class="sc">==</span><span class="st">"+"</span>,]<span class="sc">$</span>best_ancestry)))</span>
<span id="cb29-132"><a href="#cb29-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-133"><a href="#cb29-133" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-134"><a href="#cb29-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-135"><a href="#cb29-135" aria-hidden="true" tabindex="-1"></a>Correlate across all individuals per gene between Geuvadis, Enformer</span>
<span id="cb29-136"><a href="#cb29-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-139"><a href="#cb29-139" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-140"><a href="#cb29-140" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb29-141"><a href="#cb29-141" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb29-142"><a href="#cb29-142" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb29-143"><a href="#cb29-143" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb29-144"><a href="#cb29-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb29-145"><a href="#cb29-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb29-146"><a href="#cb29-146" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-147"><a href="#cb29-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-148"><a href="#cb29-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-149"><a href="#cb29-149" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_compare, enformer_compare)</span>
<span id="cb29-150"><a href="#cb29-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-151"><a href="#cb29-151" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb29-152"><a href="#cb29-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-153"><a href="#cb29-153" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb29-154"><a href="#cb29-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-155"><a href="#cb29-155" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb29-156"><a href="#cb29-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-157"><a href="#cb29-157" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb29-158"><a href="#cb29-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-159"><a href="#cb29-159" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb29-160"><a href="#cb29-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-161"><a href="#cb29-161" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb29-162"><a href="#cb29-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-163"><a href="#cb29-163" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb29-164"><a href="#cb29-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-165"><a href="#cb29-165" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span>
<span id="cb29-166"><a href="#cb29-166" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-167"><a href="#cb29-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-168"><a href="#cb29-168" aria-hidden="true" tabindex="-1"></a>Correlate across only AFR individuals</span>
<span id="cb29-169"><a href="#cb29-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-172"><a href="#cb29-172" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-173"><a href="#cb29-173" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb29-174"><a href="#cb29-174" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb29-175"><a href="#cb29-175" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb29-176"><a href="#cb29-176" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb29-177"><a href="#cb29-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb29-178"><a href="#cb29-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb29-179"><a href="#cb29-179" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-180"><a href="#cb29-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-181"><a href="#cb29-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-182"><a href="#cb29-182" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_AFR_compare, enformer_AFR_compare)</span>
<span id="cb29-183"><a href="#cb29-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-184"><a href="#cb29-184" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb29-185"><a href="#cb29-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-186"><a href="#cb29-186" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb29-187"><a href="#cb29-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-188"><a href="#cb29-188" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb29-189"><a href="#cb29-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-190"><a href="#cb29-190" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb29-191"><a href="#cb29-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-192"><a href="#cb29-192" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb29-193"><a href="#cb29-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-194"><a href="#cb29-194" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb29-195"><a href="#cb29-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-196"><a href="#cb29-196" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb29-197"><a href="#cb29-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-198"><a href="#cb29-198" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family =</span> <span class="st">"binomial"</span>))</span>
<span id="cb29-199"><a href="#cb29-199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-200"><a href="#cb29-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-201"><a href="#cb29-201" aria-hidden="true" tabindex="-1"></a>Correlate across only EUR individuals</span>
<span id="cb29-202"><a href="#cb29-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-205"><a href="#cb29-205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-206"><a href="#cb29-206" aria-hidden="true" tabindex="-1"></a>corr_gene_across_inds <span class="ot">&lt;-</span> <span class="cf">function</span>(df_1,df_2) {</span>
<span id="cb29-207"><a href="#cb29-207" aria-hidden="true" tabindex="-1"></a>  df_1 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_1)</span>
<span id="cb29-208"><a href="#cb29-208" aria-hidden="true" tabindex="-1"></a>  df_2 <span class="ot">&lt;-</span> <span class="fu">scale</span>(df_2)</span>
<span id="cb29-209"><a href="#cb29-209" aria-hidden="true" tabindex="-1"></a>  corrs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">seq.int</span>(<span class="fu">dim</span>(df_1)[<span class="dv">1</span>]), <span class="cf">function</span>(i) <span class="fu">cor</span>(<span class="fu">as.numeric</span>(df_1[i,]), <span class="fu">as.numeric</span>(df_2[i,])))</span>
<span id="cb29-210"><a href="#cb29-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(corrs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(df_1)</span>
<span id="cb29-211"><a href="#cb29-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(corrs)</span>
<span id="cb29-212"><a href="#cb29-212" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-213"><a href="#cb29-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-214"><a href="#cb29-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-215"><a href="#cb29-215" aria-hidden="true" tabindex="-1"></a>ge_corrs <span class="ot">&lt;-</span> <span class="fu">corr_gene_across_inds</span>(geuvadis_EUR_compare, enformer_EUR_compare)</span>
<span id="cb29-216"><a href="#cb29-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-217"><a href="#cb29-217" aria-hidden="true" tabindex="-1"></a>corrs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ge=</span>ge_corrs, <span class="at">ge_abs =</span> <span class="fu">abs</span>(ge_corrs),<span class="at">genes=</span>ge_genes, <span class="at">gene_names=</span><span class="fu">rownames</span>(gene_meta_compare), <span class="at">gene_h2=</span>gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS, <span class="at">high_h2 =</span> gene_meta_compare<span class="sc">$</span>h2_WholeBlood_TS <span class="sc">&gt;</span> <span class="fl">0.1</span>, <span class="at">strand =</span> gene_meta_compare<span class="sc">$</span>strand, <span class="at">ref_gene_ancestry=</span>gene_meta_compare<span class="sc">$</span>best_ancestry)</span>
<span id="cb29-218"><a href="#cb29-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-219"><a href="#cb29-219" aria-hidden="true" tabindex="-1"></a>corrs_df_plus <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>strand <span class="sc">==</span> <span class="st">"+"</span>,]</span>
<span id="cb29-220"><a href="#cb29-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-221"><a href="#cb29-221" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR <span class="ot">&lt;-</span> corrs_df[corrs_df<span class="sc">$</span>ref_gene_ancestry <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"AFR"</span>,<span class="st">"EUR"</span>),]</span>
<span id="cb29-222"><a href="#cb29-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-223"><a href="#cb29-223" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>AFR <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry <span class="sc">==</span> <span class="st">"AFR"</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb29-224"><a href="#cb29-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-225"><a href="#cb29-225" aria-hidden="true" tabindex="-1"></a>corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ge <span class="sc">&gt;</span> <span class="dv">0</span>,<span class="st">"+"</span>,<span class="st">"-"</span>)</span>
<span id="cb29-226"><a href="#cb29-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-227"><a href="#cb29-227" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(corrs_df_plus_AFR_EUR) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ref_gene_ancestry,<span class="at">y=</span>ge))</span>
<span id="cb29-228"><a href="#cb29-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-229"><a href="#cb29-229" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(data.frame(table(corrs_df_plus_AFR_EUR$ref_gene_ancestry, corrs_df_plus_AFR_EUR$corr_direction)), aes(x=Var1,y=Freq,fill=Var2, group=Var1)) + geom_bar(stat="identity", position="dodge")</span></span>
<span id="cb29-230"><a href="#cb29-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-231"><a href="#cb29-231" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">table</span>(corrs_df_plus_AFR_EUR<span class="sc">$</span>ref_gene_ancestry, corrs_df_plus_AFR_EUR<span class="sc">$</span>corr_direction))</span>
<span id="cb29-232"><a href="#cb29-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-233"><a href="#cb29-233" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(AFR<span class="sc">~</span>ge, <span class="at">data=</span>corrs_df_plus_AFR_EUR, <span class="at">family=</span><span class="st">"binomial"</span>))</span>
<span id="cb29-234"><a href="#cb29-234" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>