<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Saideep Gona">
<meta name="dcterms.date" content="2023-12-06">

<title>Daily-Blog-Sai - 3.1.1_analyze_enpact_SPrediXcan_results</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Daily-Blog-Sai</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#context" id="toc-context" class="nav-link active" data-scroll-target="#context">Context</a>
  <ul class="collapse">
  <li><a href="#specify-inputs" id="toc-specify-inputs" class="nav-link" data-scroll-target="#specify-inputs">Specify inputs</a></li>
  <li><a href="#plotting-functions" id="toc-plotting-functions" class="nav-link" data-scroll-target="#plotting-functions">Plotting functions</a></li>
  <li><a href="#plots" id="toc-plots" class="nav-link" data-scroll-target="#plots">Plots</a>
  <ul class="collapse">
  <li><a href="#generate-histograms" id="toc-generate-histograms" class="nav-link" data-scroll-target="#generate-histograms">Generate Histograms</a></li>
  <li><a href="#generate-qqplots" id="toc-generate-qqplots" class="nav-link" data-scroll-target="#generate-qqplots">Generate qqplots</a></li>
  <li><a href="#investigating-immune-related-genes" id="toc-investigating-immune-related-genes" class="nav-link" data-scroll-target="#investigating-immune-related-genes">Investigating immune-related genes</a></li>
  <li><a href="#how-does-the-new-normalization-with-isoform-quantification-compare-with-the-old-in-flu-condition-immune-gwas" id="toc-how-does-the-new-normalization-with-isoform-quantification-compare-with-the-old-in-flu-condition-immune-gwas" class="nav-link" data-scroll-target="#how-does-the-new-normalization-with-isoform-quantification-compare-with-the-old-in-flu-condition-immune-gwas">How does the new normalization (with isoform quantification) compare with the old in Flu Condition, Immune GWAS?</a></li>
  <li><a href="#how-does-enpact-compare-between-flu-ni-in-allergy-eczema-gwas" id="toc-how-does-enpact-compare-between-flu-ni-in-allergy-eczema-gwas" class="nav-link" data-scroll-target="#how-does-enpact-compare-between-flu-ni-in-allergy-eczema-gwas">How does EnPACT compare between Flu, NI in Allergy Eczema GWAS?</a></li>
  <li><a href="#how-does-enpact-compare-between-flu-ni-in-t2d-gwas" id="toc-how-does-enpact-compare-between-flu-ni-in-t2d-gwas" class="nav-link" data-scroll-target="#how-does-enpact-compare-between-flu-ni-in-t2d-gwas">How does EnPACT compare between Flu, NI in T2D GWAS?</a></li>
  <li><a href="#how-does-enpact-compare-with-gtex" id="toc-how-does-enpact-compare-with-gtex" class="nav-link" data-scroll-target="#how-does-enpact-compare-with-gtex">How does EnPACT compare with GTEx?</a></li>
  <li><a href="#how-does-standard-predixcan-flu-compare-with-standard-predixcan-ni-in-allergy-eczema-gwas" id="toc-how-does-standard-predixcan-flu-compare-with-standard-predixcan-ni-in-allergy-eczema-gwas" class="nav-link" data-scroll-target="#how-does-standard-predixcan-flu-compare-with-standard-predixcan-ni-in-allergy-eczema-gwas">How does standard PrediXcan Flu compare with standard PrediXcan NI in Allergy Eczema GWAS?</a></li>
  <li><a href="#how-does-enpact-compare-with-standard-predixcan-in-flu-condition" id="toc-how-does-enpact-compare-with-standard-predixcan-in-flu-condition" class="nav-link" data-scroll-target="#how-does-enpact-compare-with-standard-predixcan-in-flu-condition">How does EnPACT compare with Standard PrediXcan in Flu condition?</a></li>
  <li><a href="#how-does-enpact-compare-with-standard-predixcan-in-ni-condition" id="toc-how-does-enpact-compare-with-standard-predixcan-in-ni-condition" class="nav-link" data-scroll-target="#how-does-enpact-compare-with-standard-predixcan-in-ni-condition">How does EnPACT compare with Standard PrediXcan in NI condition?</a></li>
  <li><a href="#we-can-try-subsetting-to-genes-which-pass-the-predictdb-threshold-in-the-standard-case" id="toc-we-can-try-subsetting-to-genes-which-pass-the-predictdb-threshold-in-the-standard-case" class="nav-link" data-scroll-target="#we-can-try-subsetting-to-genes-which-pass-the-predictdb-threshold-in-the-standard-case">We can try subsetting to genes which pass the PredictDB threshold in the Standard case</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">3.1.1_analyze_enpact_SPrediXcan_results</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Saideep Gona </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 6, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="context" class="level1">
<h1>Context</h1>
<p>Need some unified functions for analyzing SPrediXcan results.</p>
<p>Format should be a dictionary with the following tiers:</p>
<ol type="1">
<li>GWAS trait</li>
<li>Model type</li>
<li>Dataframe with SPrediXcan results</li>
</ol>
<section id="specify-inputs" class="level3">
<h3 class="anchored" data-anchor-id="specify-inputs">Specify inputs</h3>
<div class="cell" data-execution_count="25">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os,sys</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> mixbox <span class="im">as</span> mx</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>color_pal <span class="op">=</span> {</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"conditions"</span>: {</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu"</span>: (<span class="dv">255</span>, <span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI"</span>: (<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">255</span>),</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"models"</span>: {</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"EnPACT"</span>:(<span class="dv">255</span>, <span class="dv">255</span>, <span class="dv">255</span>),</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Standard"</span>:(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"GTeX_WB"</span>:(<span class="dv">0</span>,<span class="dv">64</span>,<span class="dv">0</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"subsets"</span>: {</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"immune"</span>: (<span class="dv">255</span>, <span class="dv">165</span>, <span class="dv">0</span>) </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>input_files <span class="op">=</span> {</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T2D"</span>:{</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/Flu_T2D_filtered.txt"</span>,</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/NI_T2D_filtered.txt"</span>, </span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>        <span class="st">"GTeX_WB"</span>: <span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/GTex_WB/T2D_SPrediXcan_results_GTex_WB_trimmed.txt"</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"AE"</span>: {</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_4_4_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/Flu_AE_Farreira_filtered.txt"</span>,</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/NI_AE_Farreira_filtered.txt"</span>,</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>        <span class="st">"GTeX_WB"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/GTex_WB/AE_SPrediXcan_results_GTex_WB_trimmed.txt"</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co"># "GTeX_WB", </span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="co"># gwas_traits = ["T2D", "AE"]</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="co"># expression_models = ["Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto", </span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="co">#                      "NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto", "GTeX_WB"]</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="co"># path_to_results = "/beagle3/haky/users/saideep/projects/Con_EnPACT/SPrediXcan_results"</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="co"># for trait in gwas_traits:</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="co">#     for model in expression_models:</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="co">#         input_file = os.path.join(path_to_results, trait, model, f"{trait}_SPrediXcan_results_{model}.csv")</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="co">#         if not os.path.exists(input_file):</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">#             print(f"File {input_file} does not exist")</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="co">#             sys.exit(1)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plotting-functions" class="level3">
<h3 class="anchored" data-anchor-id="plotting-functions">Plotting functions</h3>
<div class="cell" data-execution_count="49">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scipy.stats <span class="im">as</span> stats</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> to_latent(rgb_non_latent, alpha<span class="op">=</span><span class="fl">0.5</span>):</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> [x<span class="op">/</span><span class="fl">255.0</span> <span class="cf">for</span> x <span class="kw">in</span> rgb_non_latent] <span class="op">+</span> [alpha]</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> load_results(input_files, tag<span class="op">=</span><span class="va">False</span>, only_tag<span class="op">=</span><span class="va">False</span>, model_contains_string<span class="op">=</span>[<span class="st">""</span>], model_excludes_string<span class="op">=</span>[<span class="st">""</span>]):</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    output_dict <span class="op">=</span> {}</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> gwas_type <span class="kw">in</span> input_files.keys():</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>        output_dict[gwas_type] <span class="op">=</span> {}</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model_type <span class="kw">in</span> input_files[gwas_type].keys():</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Check if model name contains all substrings</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>            does_mcs_all <span class="op">=</span> <span class="va">True</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="bu">type</span>(model_contains_string) <span class="op">==</span> <span class="bu">str</span>:</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>                model_contains_string <span class="op">=</span> [model_contains_string]</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> mcs <span class="kw">in</span> model_contains_string:</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> mcs <span class="kw">not</span> <span class="kw">in</span> model_type:</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>                    does_mcs_all <span class="op">=</span> <span class="va">False</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">break</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">not</span> does_mcs_all:</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="st">"Does not contain all substrings"</span>)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>                <span class="cf">continue</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Check if model name contains any exclusion substrings</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>            does_mes_any <span class="op">=</span> <span class="va">False</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="bu">type</span>(model_excludes_string) <span class="op">==</span> <span class="bu">str</span>:</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>                model_excludes_string <span class="op">=</span> [model_excludes_string]</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> mes <span class="kw">in</span> model_excludes_string:</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> mes <span class="op">==</span> <span class="st">""</span>:</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">continue</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> mes <span class="kw">in</span> model_type:</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>                    does_mes_any <span class="op">=</span> <span class="va">True</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">break</span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> does_mes_any:</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="st">"Contains exclusion substring"</span>)</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>                <span class="cf">continue</span></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Check if tagged datasets should be loaded</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> only_tag:</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>                <span class="cf">pass</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>                cur_df <span class="op">=</span> pd.read_csv(input_files[gwas_type][model_type], sep<span class="op">=</span><span class="st">","</span>)</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>                output_dict[gwas_type][model_type] <span class="op">=</span> cur_df</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> tag:</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> <span class="bu">type</span>(tag) <span class="op">==</span> <span class="bu">str</span>:</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>                    cur_df <span class="op">=</span> pd.read_csv(input_files[gwas_type][model_type].replace(<span class="st">".txt"</span>, <span class="st">"_"</span><span class="op">+</span>tag<span class="op">+</span><span class="st">".txt"</span>), sep<span class="op">=</span><span class="st">","</span>)</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>                    cur_df_no_na <span class="op">=</span> cur_df.dropna(subset<span class="op">=</span>[<span class="st">"pvalue"</span>])</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>                    output_dict[gwas_type][model_type<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>tag] <span class="op">=</span> cur_df_no_na</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>                <span class="cf">else</span>:</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> t <span class="kw">in</span> tag:</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>                        cur_df <span class="op">=</span> pd.read_csv(input_files[gwas_type][model_type].replace(<span class="st">".txt"</span>, <span class="st">"_"</span><span class="op">+</span>t<span class="op">+</span><span class="st">".txt"</span>), sep<span class="op">=</span><span class="st">","</span>)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>                        cur_df_no_na <span class="op">=</span> cur_df.dropna(subset<span class="op">=</span>[<span class="st">"pvalue"</span>])</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>                        output_dict[gwas_type][model_type<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>t] <span class="op">=</span> cur_df_no_na</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> output_dict</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Create function for creating subsetted tables </span></span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> create_gene_subsets_SPrediXcan_results(gene_table, tag, SPrediXcan_results_files, gene_table_col<span class="op">=</span><span class="st">"ensembl"</span>):</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>    loaded_results <span class="op">=</span> load_results(SPrediXcan_results_files)</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> gwas <span class="kw">in</span> SPrediXcan_results_files.keys():</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model <span class="kw">in</span> SPrediXcan_results_files[gwas].keys():</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>            cur_df <span class="op">=</span> loaded_results[gwas][model]</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>            cur_df_subset <span class="op">=</span> cur_df[cur_df[<span class="st">'gene'</span>].isin(gene_table[gene_table_col])]</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>            <span class="co"># print("subsetting")</span></span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>            <span class="co"># print(cur_df_subset.shape)</span></span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>            <span class="co"># print(sum(cur_df_subset['gene'].isin(gene_table[gene_table_col])))</span></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">not</span> SPrediXcan_results_files[gwas][model].endswith(<span class="st">".txt"</span>):</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="st">"Error: File does not end with .txt"</span>)</span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a>                <span class="cf">return</span></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>            cur_df_subset.to_csv(SPrediXcan_results_files[gwas][model].replace(<span class="st">".txt"</span>, <span class="st">"_"</span><span class="op">+</span>tag<span class="op">+</span><span class="st">".txt"</span>), sep<span class="op">=</span><span class="st">","</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="73">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.stats <span class="im">import</span> beta</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> upsetplot</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> qqunif_maxp(p, label, BF<span class="op">=</span><span class="va">False</span>, CI<span class="op">=</span><span class="va">False</span>, mlog10_p_thres<span class="op">=</span><span class="dv">30</span>, maxp<span class="op">=</span><span class="dv">1</span>, color<span class="op">=</span><span class="va">False</span>, lw<span class="op">=</span><span class="fl">1.0</span>):</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>    nn <span class="op">=</span> <span class="bu">len</span>(p)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    xx <span class="op">=</span> <span class="op">-</span>np.log10((np.arange(<span class="dv">1</span>, nn <span class="op">+</span> <span class="dv">1</span>) <span class="op">*</span> maxp) <span class="op">/</span> (nn <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">not</span> color:</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>        plt.scatter(xx, <span class="op">-</span>np.sort(np.log10(p)), s <span class="op">=</span> <span class="dv">8</span>, label<span class="op">=</span>label)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>        plt.plot(xx, <span class="op">-</span>np.sort(np.log10(p)), lw<span class="op">=</span>lw)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="st">"secondary"</span> <span class="kw">in</span> color:</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>            plt.plot(xx, <span class="op">-</span>np.sort(np.log10(p)), lw<span class="op">=</span>lw, label<span class="op">=</span>label, color<span class="op">=</span>to_latent(color[<span class="st">"secondary"</span>]))</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>            plt.plot(xx, <span class="op">-</span>np.sort(np.log10(p)), lw<span class="op">=</span>lw, color<span class="op">=</span>to_latent(color[<span class="st">"primary"</span>]))</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>        plt.scatter(xx, <span class="op">-</span>np.sort(np.log10(p)), s <span class="op">=</span> <span class="dv">8</span>, label<span class="op">=</span>label, color<span class="op">=</span>to_latent(color[<span class="st">"primary"</span>]))</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> BF:</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>        plt.axhline(y<span class="op">=-</span>np.log10(<span class="fl">0.05</span><span class="op">/</span>nn), linestyle<span class="op">=</span><span class="st">':'</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> CI:</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>        c95 <span class="op">=</span> np.zeros(nn)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>        c05 <span class="op">=</span> np.zeros(nn)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">1</span>, nn <span class="op">+</span> <span class="dv">1</span>):</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>            c95[i <span class="op">-</span> <span class="dv">1</span>] <span class="op">=</span> beta.ppf(<span class="fl">0.95</span>, i, nn <span class="op">-</span> i <span class="op">+</span> <span class="dv">1</span>) <span class="op">*</span> maxp</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>            c05[i <span class="op">-</span> <span class="dv">1</span>] <span class="op">=</span> beta.ppf(<span class="fl">0.05</span>, i, nn <span class="op">-</span> i <span class="op">+</span> <span class="dv">1</span>) <span class="op">*</span> maxp</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>        plt.plot(xx, <span class="op">-</span>np.log10(c95))</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>        plt.plot(xx, <span class="op">-</span>np.log10(c05))</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> qq_plot_unif(S_results, GWAS, save_dir, colors<span class="op">=</span><span class="va">False</span>, BF<span class="op">=</span><span class="va">False</span>, CI<span class="op">=</span><span class="va">False</span>, title_tag <span class="op">=</span> <span class="st">""</span>, plot_filename<span class="op">=</span><span class="st">"default"</span>):</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">6</span>))</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.axhline(-np.log10(0.05), color='red', linestyle='-', label='FDR = 0.05')</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.axhline(-np.log10(0.10), color='orange', linestyle='--', label='FDR = 0.10')</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.axhline(-np.log10(0.25), color='yellow', linestyle='-.', label='FDR = 0.25')</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> S_result <span class="kw">in</span> S_results.keys():</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">not</span> colors:</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>            qqunif_maxp(S_results[S_result].pvalue, label<span class="op">=</span><span class="ss">f'</span><span class="sc">{</span>S_result<span class="sc">}</span><span class="ss">'</span>, BF<span class="op">=</span>BF, CI<span class="op">=</span>CI)</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>            qqunif_maxp(S_results[S_result].pvalue, label<span class="op">=</span><span class="ss">f'</span><span class="sc">{</span>S_result<span class="sc">}</span><span class="ss">'</span>, BF<span class="op">=</span>BF, CI<span class="op">=</span>CI, color<span class="op">=</span>colors[S_result])</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">"Expected -log10(p)"</span>)</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">"Observed -log10(p)"</span>)</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f"QQ-unif plot for gwas: </span><span class="sc">{</span>GWAS<span class="sc">}</span><span class="ss">, "</span><span class="op">+</span>title_tag)</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> plot_filename <span class="op">==</span> <span class="st">"default"</span>:</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>        plt.savefig(os.path.join(save_dir,<span class="ss">f'</span><span class="sc">{</span>GWAS<span class="sc">}</span><span class="ss">_qqplot.png'</span>))</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>        plt.savefig(os.path.join(save_dir,plot_filename))</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> hist_plot(S_results, GWAS, model, save_dir):</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">6</span>))</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>    plt.hist(S_results.pvalue, bins<span class="op">=</span><span class="dv">100</span>)</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">"Expected -log10(p)"</span>)</span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">"Observed -log10(p)"</span>)</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f"QQ-unif plot for </span><span class="sc">{</span>GWAS<span class="sc">}</span><span class="ss">,</span><span class="sc">{</span>model<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>    plt.savefig(os.path.join(save_dir,<span class="ss">f'</span><span class="sc">{</span>GWAS<span class="sc">}</span><span class="ss">_</span><span class="sc">{</span>model<span class="sc">}</span><span class="ss">_histogram.png'</span>))</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> scatter_plots(S_results):</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Assumes equal rows in all dataframes</span></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> S_result <span class="kw">in</span> S_results.keys():</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>        dict_for_plot <span class="op">=</span> {}</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model <span class="kw">in</span> S_results[S_result].keys():</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>            sorted_cur_df <span class="op">=</span> S_results[S_result][model].sort_values(by<span class="op">=</span>[<span class="st">"gene"</span>])</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>            dict_for_plot[model] <span class="op">=</span> <span class="bu">list</span>(sorted_cur_df[<span class="st">"pvalue"</span>])</span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>        df_for_plot <span class="op">=</span> pd.DataFrame(dict_for_plot)</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>        pd.plotting.scatter_matrix(df_for_plot, figsize<span class="op">=</span>(<span class="bu">len</span>(S_results[S_result].keys())<span class="op">*</span><span class="dv">5</span>,<span class="bu">len</span>(S_results[S_result].keys())<span class="op">*</span><span class="dv">5</span>), diagonal<span class="op">=</span><span class="st">'kde'</span>)</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> create_subset_df(list_of_dfs, subset_col<span class="op">=</span><span class="st">"gene"</span>, join<span class="op">=</span><span class="st">"inner"</span>):</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>    subset_df <span class="op">=</span> list_of_dfs[<span class="dv">0</span>].copy()</span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> df <span class="kw">in</span> list_of_dfs[<span class="dv">1</span>:]:</span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>        subset_df <span class="op">=</span> subset_df.merge(df, how<span class="op">=</span>join,on<span class="op">=</span>subset_col)</span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> subset_df</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> print_dims(results_dict):</span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> gwas_type <span class="kw">in</span> results_dict.keys():</span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model_type <span class="kw">in</span> results_dict[gwas_type].keys():</span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>gwas_type<span class="sc">}</span><span class="ss"> </span><span class="sc">{</span>model_type<span class="sc">}</span><span class="ss"> : </span><span class="sc">{</span>results_dict[gwas_type][model_type]<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> upset_plot(loaded_results, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">""</span>], exclude_substrings<span class="op">=</span>[<span class="st">""</span>], percent<span class="op">=</span><span class="va">True</span>, colors<span class="op">=</span><span class="va">False</span>):</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a>    contents <span class="op">=</span> {}</span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> gwas_type <span class="kw">in</span> loaded_results.keys():</span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model_type <span class="kw">in</span> loaded_results[gwas_type].keys():</span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Check if substrings are present in keywords</span></span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>            all_substrings_present <span class="op">=</span> <span class="va">True</span></span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> substring <span class="kw">in</span> substrings:</span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> substring <span class="kw">not</span> <span class="kw">in</span> (model_type<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>gwas_type):</span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a>                    all_substrings_present <span class="op">=</span> <span class="va">False</span></span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">not</span> all_substrings_present:</span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a>                <span class="cf">continue</span></span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-109"><a href="#cb3-109" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Check if substrings for exclusion are present in keywords</span></span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a>            exclude_substring <span class="op">=</span> <span class="va">False</span></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> exclude_sub <span class="kw">in</span> exclude_substrings:</span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> exclude_sub <span class="op">==</span> <span class="st">""</span>:</span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">continue</span></span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> exclude_sub <span class="kw">in</span> (model_type<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>gwas_type):</span>
<span id="cb3-115"><a href="#cb3-115" aria-hidden="true" tabindex="-1"></a>                    exclude_substring <span class="op">=</span> <span class="va">True</span></span>
<span id="cb3-116"><a href="#cb3-116" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> exclude_substring:</span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a>                <span class="cf">continue</span>        </span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-119"><a href="#cb3-119" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Filter by pvalue</span></span>
<span id="cb3-120"><a href="#cb3-120" aria-hidden="true" tabindex="-1"></a>            cur_df <span class="op">=</span> loaded_results[gwas_type][model_type]</span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a>            cur_df_filt <span class="op">=</span> cur_df.loc[cur_df[<span class="st">"pvalue"</span>]<span class="op">&lt;</span>pval_thresh]</span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a>            cur_gene_list <span class="op">=</span> <span class="bu">list</span>(cur_df_filt[<span class="st">"gene"</span>])</span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a>            contents[model_type<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>gwas_type] <span class="op">=</span> cur_gene_list</span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a>    contents_for_plot <span class="op">=</span> upsetplot.from_contents(contents)</span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> percent:</span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">not</span> colors:</span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a>                u <span class="op">=</span> upsetplot.UpSet(contents_for_plot, orientation<span class="op">=</span><span class="st">'horizontal'</span>, </span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a>                        sort_by<span class="op">=</span><span class="st">'cardinality'</span>,</span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a>                        show_percentages<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb3-136"><a href="#cb3-136" aria-hidden="true" tabindex="-1"></a>                u <span class="op">=</span> upsetplot.UpSet(contents_for_plot, orientation<span class="op">=</span><span class="st">'horizontal'</span>, </span>
<span id="cb3-137"><a href="#cb3-137" aria-hidden="true" tabindex="-1"></a>                        sort_by<span class="op">=</span><span class="st">'cardinality'</span>,</span>
<span id="cb3-138"><a href="#cb3-138" aria-hidden="true" tabindex="-1"></a>                        show_percentages<span class="op">=</span><span class="va">True</span></span>
<span id="cb3-139"><a href="#cb3-139" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> twas_type <span class="kw">in</span> contents.keys():</span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a>                    twas_type_wo_gwas <span class="op">=</span> <span class="st">"_"</span>.join(twas_type.split(<span class="st">"_"</span>)[<span class="dv">0</span>:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a>                    u.style_categories(twas_type, shading_facecolor<span class="op">=</span>to_latent(colors[twas_type_wo_gwas][<span class="st">"primary"</span>]))</span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">not</span> colors:</span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a>                u <span class="op">=</span> upsetplot.UpSet(contents_for_plot, orientation<span class="op">=</span><span class="st">'horizontal'</span>, </span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a>                        sort_by<span class="op">=</span><span class="st">'cardinality'</span>,</span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a>                        show_counts<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a>                u <span class="op">=</span> upsetplot.UpSet(contents_for_plot, orientation<span class="op">=</span><span class="st">'horizontal'</span>, </span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a>                        sort_by<span class="op">=</span><span class="st">'cardinality'</span>,</span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a>                        show_counts<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb3-154"><a href="#cb3-154" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb3-155"><a href="#cb3-155" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> twas_type <span class="kw">in</span> contents.keys():</span>
<span id="cb3-156"><a href="#cb3-156" aria-hidden="true" tabindex="-1"></a>                    twas_type_wo_gwas <span class="op">=</span> <span class="st">"_"</span>.join(twas_type.split(<span class="st">"_"</span>)[<span class="dv">0</span>:<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb3-157"><a href="#cb3-157" aria-hidden="true" tabindex="-1"></a>                    u.style_categories(twas_type, shading_facecolor<span class="op">=</span>to_latent(colors[twas_type_wo_gwas][<span class="st">"primary"</span>]))</span>
<span id="cb3-158"><a href="#cb3-158" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb3-159"><a href="#cb3-159" aria-hidden="true" tabindex="-1"></a>        u.plot()</span>
<span id="cb3-160"><a href="#cb3-160" aria-hidden="true" tabindex="-1"></a>        plt.suptitle(<span class="st">"Gene overlap, pvalue &lt; "</span><span class="op">+</span><span class="bu">str</span>(pval_thresh))</span>
<span id="cb3-161"><a href="#cb3-161" aria-hidden="true" tabindex="-1"></a>        plt.show()</span>
<span id="cb3-162"><a href="#cb3-162" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb3-163"><a href="#cb3-163" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(e)</span>
<span id="cb3-164"><a href="#cb3-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-165"><a href="#cb3-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-166"><a href="#cb3-166" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> scatter_plot(first_df, second_df, ax, x_lab<span class="op">=</span><span class="st">"x"</span>, y_lab<span class="op">=</span><span class="st">"y"</span>, sort_by<span class="op">=</span><span class="st">"gene"</span>, bf<span class="op">=</span><span class="va">True</span>, colors<span class="op">=</span><span class="va">False</span>):</span>
<span id="cb3-167"><a href="#cb3-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-168"><a href="#cb3-168" aria-hidden="true" tabindex="-1"></a>    common_genes <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(first_df[<span class="st">"gene"</span>]).intersection(<span class="bu">set</span>(second_df[<span class="st">"gene"</span>])))</span>
<span id="cb3-169"><a href="#cb3-169" aria-hidden="true" tabindex="-1"></a>    first_df_subset <span class="op">=</span> first_df[first_df[<span class="st">"gene"</span>].isin(common_genes)].sort_values(by<span class="op">=</span>[sort_by])</span>
<span id="cb3-170"><a href="#cb3-170" aria-hidden="true" tabindex="-1"></a>    second_df_subset <span class="op">=</span> second_df[second_df[<span class="st">"gene"</span>].isin(common_genes)].sort_values(by<span class="op">=</span>[sort_by])</span>
<span id="cb3-171"><a href="#cb3-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-172"><a href="#cb3-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-173"><a href="#cb3-173" aria-hidden="true" tabindex="-1"></a>    nan_filt <span class="op">=</span> np.logical_not(np.logical_or(np.isnan(<span class="op">-</span>np.log10(<span class="bu">list</span>(first_df_subset[<span class="st">"pvalue"</span>]))), np.isnan(<span class="op">-</span>np.log10(<span class="bu">list</span>(second_df_subset[<span class="st">"pvalue"</span>])))))</span>
<span id="cb3-174"><a href="#cb3-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-175"><a href="#cb3-175" aria-hidden="true" tabindex="-1"></a>    r,p <span class="op">=</span> stats.pearsonr(x<span class="op">=-</span>np.log10(first_df_subset[nan_filt][<span class="st">"pvalue"</span>]),y<span class="op">=</span> <span class="op">-</span>np.log10(second_df_subset[nan_filt][<span class="st">"pvalue"</span>]))</span>
<span id="cb3-176"><a href="#cb3-176" aria-hidden="true" tabindex="-1"></a>    n <span class="op">=</span> <span class="bu">len</span>(first_df_subset[nan_filt][<span class="st">"pvalue"</span>])</span>
<span id="cb3-177"><a href="#cb3-177" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-178"><a href="#cb3-178" aria-hidden="true" tabindex="-1"></a>    ax.scatter(<span class="op">-</span>np.log10(first_df_subset[nan_filt][<span class="st">"pvalue"</span>]), <span class="op">-</span>np.log10(second_df_subset[nan_filt][<span class="st">"pvalue"</span>]), s<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb3-179"><a href="#cb3-179" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> bf:</span>
<span id="cb3-180"><a href="#cb3-180" aria-hidden="true" tabindex="-1"></a>        ax.axhline(y<span class="op">=-</span>np.log10(<span class="fl">0.05</span><span class="op">/</span>n), linestyle<span class="op">=</span><span class="st">':'</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb3-181"><a href="#cb3-181" aria-hidden="true" tabindex="-1"></a>        ax.axvline(x<span class="op">=-</span>np.log10(<span class="fl">0.05</span><span class="op">/</span>n), linestyle<span class="op">=</span><span class="st">':'</span>, color<span class="op">=</span><span class="st">'grey'</span>)</span>
<span id="cb3-182"><a href="#cb3-182" aria-hidden="true" tabindex="-1"></a>    ax.text(<span class="fl">.05</span>, <span class="fl">.8</span>, <span class="st">"Pearson's r =</span><span class="sc">{:.2f}</span><span class="st">"</span>.<span class="bu">format</span>(r), transform<span class="op">=</span>ax.transAxes)</span>
<span id="cb3-183"><a href="#cb3-183" aria-hidden="true" tabindex="-1"></a>    ax.text(<span class="fl">.05</span>, <span class="fl">.7</span>, <span class="ss">f"n =</span><span class="sc">{</span>n<span class="sc">}</span><span class="ss">"</span>, transform<span class="op">=</span>ax.transAxes)</span>
<span id="cb3-184"><a href="#cb3-184" aria-hidden="true" tabindex="-1"></a>    ax.set_xlabel(x_lab, fontsize<span class="op">=</span><span class="dv">12</span>, weight <span class="op">=</span> <span class="st">'bold'</span>)</span>
<span id="cb3-185"><a href="#cb3-185" aria-hidden="true" tabindex="-1"></a>    ax.set_ylabel(y_lab, fontsize<span class="op">=</span><span class="dv">12</span>, weight <span class="op">=</span> <span class="st">'bold'</span>)</span>
<span id="cb3-186"><a href="#cb3-186" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">not</span> colors:</span>
<span id="cb3-187"><a href="#cb3-187" aria-hidden="true" tabindex="-1"></a>        <span class="cf">pass</span></span>
<span id="cb3-188"><a href="#cb3-188" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb3-189"><a href="#cb3-189" aria-hidden="true" tabindex="-1"></a>        ax.xaxis.label.set_color(to_latent(colors[<span class="st">"_"</span>.join(x_lab.split(<span class="st">"_"</span>)[<span class="dv">0</span>:<span class="op">-</span><span class="dv">1</span>])][<span class="st">"primary"</span>], alpha<span class="op">=</span><span class="fl">1.0</span>))</span>
<span id="cb3-190"><a href="#cb3-190" aria-hidden="true" tabindex="-1"></a>        ax.yaxis.label.set_color(to_latent(colors[<span class="st">"_"</span>.join(y_lab.split(<span class="st">"_"</span>)[<span class="dv">0</span>:<span class="op">-</span><span class="dv">1</span>])][<span class="st">"primary"</span>], alpha<span class="op">=</span><span class="fl">1.0</span>))</span>
<span id="cb3-191"><a href="#cb3-191" aria-hidden="true" tabindex="-1"></a>    ax.axline((<span class="dv">0</span>, <span class="dv">0</span>), slope<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb3-192"><a href="#cb3-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-193"><a href="#cb3-193" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> scatter_plot_all(results_dict, plot_title<span class="op">=</span> <span class="st">"-log10 pvalue for pair-wise intersected genes"</span>, colors<span class="op">=</span><span class="va">False</span>):</span>
<span id="cb3-194"><a href="#cb3-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-195"><a href="#cb3-195" aria-hidden="true" tabindex="-1"></a>    results_count <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb3-196"><a href="#cb3-196" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> gwas_type <span class="kw">in</span> results_dict.keys():</span>
<span id="cb3-197"><a href="#cb3-197" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model_type <span class="kw">in</span> results_dict[gwas_type].keys():</span>
<span id="cb3-198"><a href="#cb3-198" aria-hidden="true" tabindex="-1"></a>            results_count <span class="op">+=</span> <span class="dv">1</span></span>
<span id="cb3-199"><a href="#cb3-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-200"><a href="#cb3-200" aria-hidden="true" tabindex="-1"></a>    fig <span class="op">=</span> plt.figure(figsize<span class="op">=</span>(<span class="dv">4</span><span class="op">*</span>results_count, <span class="dv">4</span><span class="op">*</span>results_count), layout<span class="op">=</span><span class="st">"constrained"</span>)</span>
<span id="cb3-201"><a href="#cb3-201" aria-hidden="true" tabindex="-1"></a>    fig.suptitle(plot_title, fontsize<span class="op">=</span><span class="dv">16</span>)</span>
<span id="cb3-202"><a href="#cb3-202" aria-hidden="true" tabindex="-1"></a>    spec <span class="op">=</span> fig.add_gridspec(ncols<span class="op">=</span>results_count, nrows<span class="op">=</span>results_count)</span>
<span id="cb3-203"><a href="#cb3-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-204"><a href="#cb3-204" aria-hidden="true" tabindex="-1"></a>    ind1 <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb3-205"><a href="#cb3-205" aria-hidden="true" tabindex="-1"></a>    ind2 <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb3-206"><a href="#cb3-206" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> gwas_type1 <span class="kw">in</span> results_dict.keys():</span>
<span id="cb3-207"><a href="#cb3-207" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> model_type1 <span class="kw">in</span> results_dict[gwas_type].keys():</span>
<span id="cb3-208"><a href="#cb3-208" aria-hidden="true" tabindex="-1"></a>            ind2 <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb3-209"><a href="#cb3-209" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> gwas_type2 <span class="kw">in</span> results_dict.keys():</span>
<span id="cb3-210"><a href="#cb3-210" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> model_type2 <span class="kw">in</span> results_dict[gwas_type].keys():</span>
<span id="cb3-211"><a href="#cb3-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-212"><a href="#cb3-212" aria-hidden="true" tabindex="-1"></a>                    ax <span class="op">=</span> fig.add_subplot(spec[ind1, ind2])</span>
<span id="cb3-213"><a href="#cb3-213" aria-hidden="true" tabindex="-1"></a>                    scatter_plot(results_dict[gwas_type1][model_type1], results_dict[gwas_type2][model_type2], ax,</span>
<span id="cb3-214"><a href="#cb3-214" aria-hidden="true" tabindex="-1"></a>                                 x_lab<span class="op">=</span>model_type1<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>gwas_type1, y_lab<span class="op">=</span>model_type2<span class="op">+</span><span class="st">"_"</span><span class="op">+</span>gwas_type2, colors<span class="op">=</span>colors)</span>
<span id="cb3-215"><a href="#cb3-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-216"><a href="#cb3-216" aria-hidden="true" tabindex="-1"></a>                    ind2 <span class="op">+=</span> <span class="dv">1</span></span>
<span id="cb3-217"><a href="#cb3-217" aria-hidden="true" tabindex="-1"></a>            ind2 <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb3-218"><a href="#cb3-218" aria-hidden="true" tabindex="-1"></a>            ind1 <span class="op">+=</span> <span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plots" class="level2">
<h2 class="anchored" data-anchor-id="plots">Plots</h2>
<div class="cell" data-execution_count="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># loaded_results = load_results(input_files)</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># print_dims(loaded_results)</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># upset_plot(loaded_results, pval_thresh=0.05)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="generate-histograms" class="level3">
<h3 class="anchored" data-anchor-id="generate-histograms">Generate Histograms</h3>
<div class="cell" data-execution_count="8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for gwas in loaded_results.keys():</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     for model in loaded_results[gwas].keys():</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#         pass</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>        <span class="co"># hist_plot(loaded_results[gwas][model], gwas, model, "/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="generate-qqplots" class="level3">
<h3 class="anchored" data-anchor-id="generate-qqplots">Generate qqplots</h3>
</section>
<section id="investigating-immune-related-genes" class="level3">
<h3 class="anchored" data-anchor-id="investigating-immune-related-genes">Investigating immune-related genes</h3>
<p>The gene list here is from the IRIS database of immune-related genes: https://pubmed.ncbi.nlm.nih.gov/15780753/ and downloaded from: https://www.innatedb.com/moleculeSearch.do</p>
<div class="cell" data-execution_count="9">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in immune gene list</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>path_to_immune_genes <span class="op">=</span> <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/InnateDB_genes.csv"</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>immune_genes <span class="op">=</span> pd.read_csv(path_to_immune_genes, sep<span class="op">=</span><span class="st">","</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>immune_genes.head()</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Also look at common Flu-NI genes</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co"># flu_genes = loaded_results['T2D']['Enpact_Flu']</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ni_genes = loaded_results['T2D']['Enpact_NI']</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># list_of_dfs = [flu_genes, ni_genes]</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co"># flu_ni_subset = create_subset_df([flu_genes, ni_genes], subset_col="gene")</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co"># print(flu_genes.shape, ni_genes.shape)</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co"># flu_ni_subset.shape</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="9">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">id</th>
<th data-quarto-table-cell-role="th">species</th>
<th data-quarto-table-cell-role="th">taxonId</th>
<th data-quarto-table-cell-role="th">ensembl</th>
<th data-quarto-table-cell-role="th">entrez</th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">fullname</th>
<th data-quarto-table-cell-role="th">synonym</th>
<th data-quarto-table-cell-role="th">signature</th>
<th data-quarto-table-cell-role="th">chromStart</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">goFunctions</th>
<th data-quarto-table-cell-role="th">goLocalizations</th>
<th data-quarto-table-cell-role="th">cerebralLocalization</th>
<th data-quarto-table-cell-role="th">nrIntxsValidated</th>
<th data-quarto-table-cell-role="th">nrIntxsPredicted</th>
<th data-quarto-table-cell-role="th">transcripts</th>
<th data-quarto-table-cell-role="th">humanOrthologs</th>
<th data-quarto-table-cell-role="th">mouseOrthologs</th>
<th data-quarto-table-cell-role="th">bovineOrthologs</th>
<th data-quarto-table-cell-role="th">lastModified</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>285</td>
<td>Homo sapiens</td>
<td>9606</td>
<td>ENSG00000158270</td>
<td>81035</td>
<td>COLEC12</td>
<td>collectin sub-family member 12</td>
<td>NaN</td>
<td>COLEC12|IDBG-285|ENSG00000158270|81035</td>
<td>319361</td>
<td>...</td>
<td>GO:0005044: scavenger receptor activity#GO:000...</td>
<td>GO:0005044: scavenger receptor activity,GO:000...</td>
<td>Plasma membrane</td>
<td>2</td>
<td>0</td>
<td>228706</td>
<td>NaN</td>
<td>Colec12|IDBG-128247|ENSMUSG00000036103|140792</td>
<td>COLEC12|IDBG-644123|ENSBTAG00000007705|504741</td>
<td>2017-02-20</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>885</td>
<td>Homo sapiens</td>
<td>9606</td>
<td>ENSG00000177663</td>
<td>23765</td>
<td>IL17RA</td>
<td>interleukin 17 receptor A</td>
<td>CANDF5|CD217|CDw217|hIL-17R|IL-17RA|IL17R</td>
<td>IL17RA|IDBG-885|ENSG00000177663|23765</td>
<td>17084954</td>
<td>...</td>
<td>GO:0005515: protein binding#GO:0030368: interl...</td>
<td>GO:0005515: protein binding,GO:0030368: interl...</td>
<td>Extracellular</td>
<td>21</td>
<td>14</td>
<td>887|810276</td>
<td>NaN</td>
<td>Il17ra|IDBG-183749|ENSMUSG00000002897|16172</td>
<td>IL17RA|IDBG-641259|ENSBTAG00000011067|525040</td>
<td>2017-02-20</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>1072</td>
<td>Homo sapiens</td>
<td>9606</td>
<td>ENSG00000154721</td>
<td>58494</td>
<td>JAM2</td>
<td>junctional adhesion molecule 2</td>
<td>NaN</td>
<td>JAM2|IDBG-1072|ENSG00000154721|58494</td>
<td>25639272</td>
<td>...</td>
<td>GO:0005515: protein binding#GO:0046982: protei...</td>
<td>GO:0005515: protein binding,GO:0046982: protei...</td>
<td>Cell surface</td>
<td>11</td>
<td>0</td>
<td>1074|227998|480678|586152</td>
<td>NaN</td>
<td>Jam2|IDBG-172706|ENSMUSG00000053062|67374</td>
<td>JAM2|IDBG-630318|ENSBTAG00000000603|538846</td>
<td>2017-02-20</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>1100</td>
<td>Homo sapiens</td>
<td>9606</td>
<td>ENSG00000142192</td>
<td>351</td>
<td>APP</td>
<td>amyloid beta (A4) precursor protein</td>
<td>AAA|ABETA|ABPP|AD1|APPI|CTFgamma|CVAP|PN-II|PN2</td>
<td>APP|IDBG-1100|ENSG00000142192|351</td>
<td>25880550</td>
<td>...</td>
<td>GO:0003677: DNA binding#GO:0004867: serine-typ...</td>
<td>GO:0003677: DNA binding,GO:0004867: serine-typ...</td>
<td>Nucleus</td>
<td>2628</td>
<td>31</td>
<td>1102|1104|1106|1108|1110|1112|361147|361148|36...</td>
<td>NaN</td>
<td>App|IDBG-172954|ENSMUSG00000022892|11820</td>
<td>APP|IDBG-630293|ENSBTAG00000017753|280722</td>
<td>2017-02-20</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>1597</td>
<td>Homo sapiens</td>
<td>9606</td>
<td>ENSG00000156299</td>
<td>102724932,7074</td>
<td>TIAM1</td>
<td>T-cell lymphoma invasion and metastasis 1</td>
<td>NaN</td>
<td>TIAM1|IDBG-1597|ENSG00000156299|102724932,7074</td>
<td>31118416</td>
<td>...</td>
<td>GO:0005057: receptor signaling protein activit...</td>
<td>GO:0005057: receptor signaling protein activit...</td>
<td>Cell surface</td>
<td>41</td>
<td>5</td>
<td>1599|361857|361858|597056</td>
<td>NaN</td>
<td>Tiam1|IDBG-174466|ENSMUSG00000002489|21844</td>
<td>BT.74139|IDBG-630064|ENSBTAG00000017839|</td>
<td>2017-02-20</td>
</tr>
</tbody>
</table>

<p>5 rows  26 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="10">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>create_gene_subsets_SPrediXcan_results(immune_genes,<span class="st">"immune"</span>, input_files)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># create_gene_subsets_SPrediXcan_results(flu_ni_subset,"flu_ni", input_files, gene_table_col="gene")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="11">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reload with filtered results</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># loaded_results = load_results(input_files, tag="immune")</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># print_dims(loaded_results)</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="12">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># print(loaded_results['T2D']['Enpact_Flu'].shape)</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># print(loaded_results['T2D']['Enpact_Flu_immune'].shape)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># print(loaded_results['T2D']['GTeX_WB_immune'].shape)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="how-does-the-new-normalization-with-isoform-quantification-compare-with-the-old-in-flu-condition-immune-gwas" class="level3">
<h3 class="anchored" data-anchor-id="how-does-the-new-normalization-with-isoform-quantification-compare-with-the-old-in-flu-condition-immune-gwas">How does the new normalization (with isoform quantification) compare with the old in Flu Condition, Immune GWAS?</h3>
<div class="cell" data-execution_count="13">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># input_files_compare_normalization = {</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     "AE": {</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#         "Flu_sourcekallisto":"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt",</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#         "Flu_gene_length":"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormTRUE_beforeTRUE_scale1/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     }</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># loaded_results_compare_normalization = load_results(input_files_compare_normalization, tag="immune")</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co"># gwas ="AE"</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co"># qq_plot_unif(loaded_results_compare_normalization[gwas], gwas, "/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots",</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                 plot_filename=f"compare_normalization_qqplot.png", BF=True, title_tag="EnPACT gene length vs. isoform length normalization")</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># upset_plot(loaded_results_compare_normalization, pval_thresh=1.0, exclude_substrings=["immune"])</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co"># upset_plot(loaded_results_compare_normalization, pval_thresh=1.0, substrings=["immune"])</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="co"># upset_plot(loaded_results_compare_normalization, pval_thresh=0.05, exclude_substrings=["immune"])</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="co"># upset_plot(loaded_results_compare_normalization, pval_thresh=0.05, substrings=["immune"])</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="co"># scatter_plot_all(load_results(input_files_compare_normalization))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="how-does-enpact-compare-between-flu-ni-in-allergy-eczema-gwas" class="level3">
<h3 class="anchored" data-anchor-id="how-does-enpact-compare-between-flu-ni-in-allergy-eczema-gwas">How does EnPACT compare between Flu, NI in Allergy Eczema GWAS?</h3>
<div class="cell" data-execution_count="45">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>input_files_compare_fluni <span class="op">=</span> {</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"AE"</span>: {</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_fluni <span class="op">=</span> load_results(input_files_compare_fluni, tag<span class="op">=</span><span class="st">"immune"</span>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>gwas <span class="op">=</span><span class="st">"AE"</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>color_pal_compare_fluni <span class="op">=</span> {</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},                                              </span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_fluni[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_fluni_</span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>, title_tag<span class="op">=</span><span class="st">"EnPACT Flu vs. NI"</span>, colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(load_results(input_files_compare_fluni), colors<span class="op">=</span>color_pal_compare_fluni, </span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(loaded_results_compare_fluni, colors<span class="op">=</span>color_pal_compare_fluni,</span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-5.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-6.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-13-output-7.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="how-does-enpact-compare-between-flu-ni-in-t2d-gwas" class="level3">
<h3 class="anchored" data-anchor-id="how-does-enpact-compare-between-flu-ni-in-t2d-gwas">How does EnPACT compare between Flu, NI in T2D GWAS?</h3>
<div class="cell" data-execution_count="46">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>input_files_compare_fluni <span class="op">=</span> {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T2D"</span>: {</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/T2D/T2D.txt"</span>,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/T2D/T2D.txt"</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_fluni <span class="op">=</span> load_results(input_files_compare_fluni, tag<span class="op">=</span><span class="st">"immune"</span>)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>gwas <span class="op">=</span><span class="st">"T2D"</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>color_pal_compare_fluni <span class="op">=</span> {</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},                                              </span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_fluni[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_fluni_</span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>, title_tag<span class="op">=</span><span class="st">"EnPACT Flu vs. NI"</span>, colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(load_results(input_files_compare_fluni), colors<span class="op">=</span>color_pal_compare_fluni, </span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(loaded_results_compare_fluni, colors<span class="op">=</span>color_pal_compare_fluni,</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_fluni, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_fluni)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-5.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-6.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-14-output-7.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="how-does-enpact-compare-with-gtex" class="level3">
<h3 class="anchored" data-anchor-id="how-does-enpact-compare-with-gtex">How does EnPACT compare with GTEx?</h3>
<div class="cell" data-execution_count="69">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>input_files_compare_gtex <span class="op">=</span> {</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"AE"</span>: {</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"GTeX_WB"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/GTex_WB/AE_SPrediXcan_results_GTex_WB_trimmed.txt"</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>gwas <span class="op">=</span> <span class="st">"AE"</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_gtex <span class="op">=</span> load_results(input_files_compare_gtex, tag<span class="op">=</span><span class="st">"immune"</span>)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>color_pal_compare_gtex <span class="op">=</span> {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"GTeX_WB"</span>: {<span class="st">"primary"</span>: color_pal[<span class="st">"models"</span>][<span class="st">"GTeX_WB"</span>]},</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"GTeX_WB_immune"</span>: {<span class="st">"primary"</span>: color_pal[<span class="st">"models"</span>][<span class="st">"GTeX_WB"</span>],</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},                                              </span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_gtex[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_gtex_</span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>, title_tag<span class="op">=</span><span class="st">"EnPACT Flu vs. GTeX WB"</span>, colors<span class="op">=</span>color_pal_compare_gtex)</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(load_results(input_files_compare_gtex), colors<span class="op">=</span>color_pal_compare_gtex, </span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. GTeX WB in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(loaded_results_compare_gtex, colors<span class="op">=</span>color_pal_compare_gtex,</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. GTeX WB in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_gtex, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_gtex)</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_gtex, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_gtex)</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_gtex, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_gtex)</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_gtex, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], colors<span class="op">=</span>color_pal_compare_gtex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                                               id
Flu_sourcekallisto_AE GTeX_WB_AE                 
True                  True        ENSG00000174123
                      True        ENSG00000134987
                      True        ENSG00000197712
                      True        ENSG00000115604
                      True        ENSG00000172057
...                                           ...
False                 True        ENSG00000250889
                      True        ENSG00000237372
                      True        ENSG00000272255
                      True        ENSG00000273759
                      True        ENSG00000253214

[19601 rows x 1 columns]
                                                             id
Flu_sourcekallisto_immune_AE GTeX_WB_immune_AE                 
True                         True               ENSG00000174123
                             True               ENSG00000115604
                             True               ENSG00000115602
                             False              ENSG00000123384
                             True               ENSG00000174130
...                                                         ...
False                        True               ENSG00000129450
                             True               ENSG00000205220
                             True               ENSG00000121933
                             True               ENSG00000221995
                             True               ENSG00000173486

[1445 rows x 1 columns]
                                               id
Flu_sourcekallisto_AE GTeX_WB_AE                 
True                  True        ENSG00000174123
                      True        ENSG00000134987
                      True        ENSG00000197712
                      False       ENSG00000115604
                      True        ENSG00000172057
...                                           ...
False                 True        ENSG00000140455
                      True        ENSG00000244509
                      True        ENSG00000135722
                      True        ENSG00000122376
                      True        ENSG00000184574

[2866 rows x 1 columns]
                                                             id
Flu_sourcekallisto_immune_AE GTeX_WB_immune_AE                 
True                         True               ENSG00000174123
                             False              ENSG00000115604
                             True               ENSG00000115602
                             False              ENSG00000123384
                             True               ENSG00000174130
...                                                         ...
False                        True               ENSG00000213949
                             True               ENSG00000229314
                             True               ENSG00000168811
                             True               ENSG00000197885
                             True               ENSG00000177575

[292 rows x 1 columns]</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-5.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-6.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-7.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-15-output-8.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="how-does-standard-predixcan-flu-compare-with-standard-predixcan-ni-in-allergy-eczema-gwas" class="level3">
<h3 class="anchored" data-anchor-id="how-does-standard-predixcan-flu-compare-with-standard-predixcan-ni-in-allergy-eczema-gwas">How does standard PrediXcan Flu compare with standard PrediXcan NI in Allergy Eczema GWAS?</h3>
<div class="cell" data-execution_count="74">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>input_files_compare_standard <span class="op">=</span> {</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"AE"</span>: {</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/Flu_AE_Farreira_filtered.txt"</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/NI_AE_Farreira_filtered.txt"</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>gwas <span class="op">=</span> <span class="st">"AE"</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>color_pal_compare_standard <span class="op">=</span> {</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_standard"</span>:  {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_standard"</span>:  {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_standard_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_standard_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},                                          </span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_standard <span class="op">=</span> load_results(input_files_compare_standard, tag<span class="op">=</span>[<span class="st">"immune"</span>])</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_standard[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_standard_Flu_NI_</span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>, title_tag<span class="op">=</span><span class="st">"Standard Flu vs. Standard NI"</span>, </span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>                colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(load_results(input_files_compare_standard), colors<span class="op">=</span>color_pal_compare_standard, </span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for Standard Flu vs. Standard NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(loaded_results_compare_standard, colors<span class="op">=</span>color_pal_compare_standard,</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for Standard Flu vs. Standard NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-16-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-16-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-16-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-16-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-16-output-5.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-16-output-6.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="how-does-enpact-compare-with-standard-predixcan-in-flu-condition" class="level3">
<h3 class="anchored" data-anchor-id="how-does-enpact-compare-with-standard-predixcan-in-flu-condition">How does EnPACT compare with Standard PrediXcan in Flu condition?</h3>
<div class="cell" data-execution_count="54">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>input_files_compare_standard <span class="op">=</span> {</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"AE"</span>: {</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/Flu_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Flu_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/Flu_AE_Farreira_filtered.txt"</span>,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_sourcekallisto"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/Con_EnPACT/models/NI_ws4_ns_3_7_lnormFALSE_beforeFALSE_scale1_sourcekallisto/intermediates/TWAS/Allergy_Eczema/Allergy_Eczema.txt"</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"NI_standard"</span>:<span class="st">"/beagle3/haky/users/saideep/projects/aracena_modeling/SPrediXcan/standard_predictDB_aracena/NI_AE_Farreira_filtered.txt"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>gwas <span class="op">=</span> <span class="st">"AE"</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>color_pal_compare_standard <span class="op">=</span> {</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_standard"</span>:  {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_sourcekallisto"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)}, </span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_standard"</span>:  {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>)},</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Flu_standard_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"Flu"</span>], </span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_sourcekallisto_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"EnPACT"</span>], </span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">"NI_standard_immune"</span>: {<span class="st">"primary"</span>: mx.lerp(color_pal[<span class="st">"models"</span>][<span class="st">"Standard"</span>], </span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>                                              color_pal[<span class="st">"conditions"</span>][<span class="st">"NI"</span>], </span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>                                              <span class="fl">0.5</span>),</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"secondary"</span>: color_pal[<span class="st">"subsets"</span>][<span class="st">"immune"</span>]},                                          </span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_standard_Flu <span class="op">=</span> load_results(input_files_compare_standard, tag<span class="op">=</span>[<span class="st">"immune"</span>], </span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>                                                   model_excludes_string<span class="op">=</span><span class="st">"NI"</span>)</span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_standard_Flu[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_standard_Flu_</span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>, title_tag<span class="op">=</span><span class="st">"EnPACT Flu vs. Standard"</span>, </span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>                colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(load_results(input_files_compare_standard, model_excludes_string<span class="op">=</span><span class="st">"NI"</span>), colors<span class="op">=</span>color_pal_compare_standard, </span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. standard Flu in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(loaded_results_compare_standard_Flu, colors<span class="op">=</span>color_pal_compare_standard,</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT Flu vs. standard Flu in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_Flu, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>, <span class="st">"Flu"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_Flu, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"Flu"</span>], exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_Flu, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>, <span class="st">"Flu"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Contains exclusion substring
Contains exclusion substring
Contains exclusion substring
Contains exclusion substring
Flu_sourcekallisto_immune_AE Flu_sourcekallisto_immune
Flu_standard_immune_AE Flu_standard_immune
Flu_sourcekallisto_AE Flu_sourcekallisto
Flu_standard_AE Flu_standard
Flu_sourcekallisto_immune_AE Flu_sourcekallisto_immune
Flu_standard_immune_AE Flu_standard_immune</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-17-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-17-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-17-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-17-output-5.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-17-output-6.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-17-output-7.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="how-does-enpact-compare-with-standard-predixcan-in-ni-condition" class="level3">
<h3 class="anchored" data-anchor-id="how-does-enpact-compare-with-standard-predixcan-in-ni-condition">How does EnPACT compare with Standard PrediXcan in NI condition?</h3>
<div class="cell" data-execution_count="55">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_standard_NI <span class="op">=</span> load_results(input_files_compare_standard, tag<span class="op">=</span>[<span class="st">"immune"</span>], </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                                                   model_excludes_string<span class="op">=</span><span class="st">"Flu"</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_standard_NI[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_standard_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>, title_tag<span class="op">=</span><span class="st">"EnPACT NI vs. Standard NI"</span>,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>                colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(load_results(input_files_compare_standard, model_excludes_string<span class="op">=</span><span class="st">"Flu"</span>), colors<span class="op">=</span>color_pal_compare_standard, </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT NI vs. standard NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>scatter_plot_all(loaded_results_compare_standard_NI, colors<span class="op">=</span>color_pal_compare_standard,</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>                 plot_title<span class="op">=</span><span class="ss">f"-log10 pvalue for pair-wise intersected genes for EnPACT NI vs. standard NI in </span><span class="sc">{</span>gwas<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_NI, pval_thresh<span class="op">=</span><span class="fl">1.0</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>, <span class="st">"NI"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_NI, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"NI"</span>], exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_NI, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>, <span class="st">"NI"</span>], percent<span class="op">=</span><span class="va">False</span>, colors<span class="op">=</span>color_pal_compare_standard)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Contains exclusion substring
Contains exclusion substring
Contains exclusion substring
Contains exclusion substring
NI_sourcekallisto_immune_AE NI_sourcekallisto_immune
NI_standard_immune_AE NI_standard_immune
NI_sourcekallisto_AE NI_sourcekallisto
NI_standard_AE NI_standard
NI_sourcekallisto_immune_AE NI_sourcekallisto_immune
NI_standard_immune_AE NI_standard_immune</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-18-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-18-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-18-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-18-output-5.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-18-output-6.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-18-output-7.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="we-can-try-subsetting-to-genes-which-pass-the-predictdb-threshold-in-the-standard-case" class="level3">
<h3 class="anchored" data-anchor-id="we-can-try-subsetting-to-genes-which-pass-the-predictdb-threshold-in-the-standard-case">We can try subsetting to genes which pass the PredictDB threshold in the Standard case</h3>
<div class="cell" data-execution_count="32">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First look at the Flu case</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>standard_flu_genes <span class="op">=</span> loaded_results_compare_standard_Flu[<span class="st">"AE"</span>][<span class="st">"Flu_standard"</span>]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>create_gene_subsets_SPrediXcan_results(standard_flu_genes,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"standard_Flu"</span>, input_files_compare_standard,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                                        gene_table_col<span class="op">=</span><span class="st">"gene"</span>)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>standard_flu_immune_genes <span class="op">=</span> standard_flu_genes.loc[standard_flu_genes[<span class="st">'gene'</span>].isin(immune_genes[<span class="st">'ensembl'</span>])] </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>create_gene_subsets_SPrediXcan_results(standard_flu_immune_genes,</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"standard_Flu_immune"</span>, input_files_compare_standard,</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>                                        gene_table_col<span class="op">=</span><span class="st">"gene"</span>)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_standard_Flu <span class="op">=</span> load_results(input_files_compare_standard, tag<span class="op">=</span>[<span class="st">"standard_Flu"</span>,<span class="st">"standard_Flu_immune"</span>], </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>                                                   only_tag<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>                                                   model_excludes_string<span class="op">=</span><span class="st">"NI"</span>)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_standard_Flu[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_standard_Flu_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_Flu, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>, <span class="st">"standard_Flu"</span>], exclude_substrings<span class="op">=</span>[<span class="st">"NI"</span>], percent<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_Flu, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"standard_Flu"</span>], exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>,<span class="st">"NI"</span>], percent<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>subsetting
(1196, 12)
1196
subsetting
(1202, 12)
1202
subsetting
(1197, 12)
1197
subsetting
(248, 12)
248
subsetting
(119, 12)
119
subsetting
(121, 12)
121
subsetting
(120, 12)
120
subsetting
(24, 12)
24
Contains exclusion substring
Contains exclusion substring</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-19-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-19-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-19-output-4.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="33">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Next, the NI case</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>standard_ni_genes <span class="op">=</span> loaded_results_compare_standard_NI[<span class="st">"AE"</span>][<span class="st">"NI_standard"</span>]</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>create_gene_subsets_SPrediXcan_results(standard_ni_genes,</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"standard_NI"</span>, input_files_compare_standard,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>                                        gene_table_col<span class="op">=</span><span class="st">"gene"</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>standard_ni_immune_genes <span class="op">=</span> standard_ni_genes.loc[standard_ni_genes[<span class="st">'gene'</span>].isin(immune_genes[<span class="st">'ensembl'</span>])] </span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>create_gene_subsets_SPrediXcan_results(standard_ni_immune_genes,</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"standard_NI_immune"</span>, input_files_compare_standard,</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>                                        gene_table_col<span class="op">=</span><span class="st">"gene"</span>)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>loaded_results_compare_standard_NI <span class="op">=</span> load_results(input_files_compare_standard, tag<span class="op">=</span>[<span class="st">"standard_NI"</span>,<span class="st">"standard_NI_immune"</span>], </span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>                                                   only_tag<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>                                                   model_excludes_string<span class="op">=</span><span class="st">"Flu"</span>)</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>qq_plot_unif(loaded_results_compare_standard_NI[gwas], gwas, <span class="st">"/beagle3/haky/users/saideep/github_repos/Daily-Blog-Sai/posts/2023-12-06-analyze_SPrediXcan_results/plots"</span>,</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>                plot_filename<span class="op">=</span><span class="ss">f"compare_standard_NI_qqplot.png"</span>, BF<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_NI, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"immune"</span>, <span class="st">"standard_NI"</span>], exclude_substrings<span class="op">=</span>[<span class="st">"Flu"</span>], percent<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>upset_plot(loaded_results_compare_standard_NI, pval_thresh<span class="op">=</span><span class="fl">0.05</span>, substrings<span class="op">=</span>[<span class="st">"standard_NI"</span>], exclude_substrings<span class="op">=</span>[<span class="st">"immune"</span>,<span class="st">"Flu"</span>], percent<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>subsetting
(1094, 12)
1094
subsetting
(248, 12)
248
subsetting
(1095, 12)
1095
subsetting
(1102, 12)
1102
subsetting
(94, 12)
94
subsetting
(24, 12)
24
subsetting
(94, 12)
94
subsetting
(96, 12)
96
Contains exclusion substring
Contains exclusion substring</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-20-output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-20-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="3.1.1._analyze_SPrediXcan_results_files/figure-html/cell-20-output-4.png" class="img-fluid"></p>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>